---
title: "Data Analyses"
output: github_document
author: "Devon Birdseye"
date: "08/07/2020"
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r set seed}
set.seed(586)
```
```{r libraries}
library(readr)
library(tibble)
library(plyr)
library(EnvStats)
library(ggplot2)
library(reshape2)
library(cowplot)
library(patchwork)
library(pathview)
library(data.table)
library(AnnotationDbi)
library(dplyr)
library(tidyr)
library(ggridges)
```

```{r define paths}
#Path to input data
PathToData <- "Data/"
#Path to output plots
PathToPlots <- "Plots/"
```

#Import protein data
```{r import protein data}
#Import protein data
tmt <- read.csv(paste0(PathToData,"Leaf Paper Data 050520-processed.csv"), stringsAsFactors = F)
#re-define protein column to shortened name
tmt <- tmt[,c(1:181,183)]
colnames(tmt)[182] <- "Protein"
#make new column for gene accession without transcript id
tmt$Gene <- substr(tmt$Accession, start = 1, stop = 14)
#Remove polymorphic peptides
tmt.withPolymorphic <- tmt
tmt <- tmt[-(grep("only", tmt$Accession)),]
#Convert to numeric
for(i in 1:170){
  tmt[,i] <- as.numeric(tmt[,i])
}
```
#Subset protein experiments
```{r subset BxM seedling leaf from original experiment}
#BxM from Original Experiment (OG)
tmt.SL <- tmt[,c(171:183,12:20)]
#Remove proteins not detected in one or more samples
tmt.SL <- na.omit(tmt.SL)
#Calculate averages of bio reps
tmt.SL$B73avg <- rowMeans(tmt.SL[,c("SeedlingLeaf.B73.r1", "SeedlingLeaf.B73.r2", "SeedlingLeaf.B73.r3")])
tmt.SL$Mo17avg <- rowMeans(tmt.SL[,c("SeedlingLeaf.Mo17.r1", "SeedlingLeaf.Mo17.r2", "SeedlingLeaf.Mo17.r3")])
tmt.SL$HYBavg <- rowMeans(tmt.SL[,c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3")])
tmt.SL$MPavg <- rowMeans(tmt.SL[,c("B73avg", "Mo17avg")])
#calculate ratios
tmt.SL$HYB.MP <- tmt.SL$HYBavg/tmt.SL$MPavg
tmt.SL$l2.B73xMo17.MP <- log2(tmt.SL$HYB.MP)
tmt.SL$B73.Mo17 <- tmt.SL$B73avg/tmt.SL$Mo17avg
tmt.SL$l2.B73.Mo17 <-log2(tmt.SL$B73.Mo17)
```
```{r subset BxM from day vs. night experiment (day)}
#BxM from Day vs. night experiment (DN)
tmt.DN <- tmt[,c(171:183,62:70)]
#Remove proteins not detected in one or more samples
tmt.DN <- na.omit(tmt.DN)
#Calculate averages of bio reps
tmt.DN$B73avg <- rowMeans(tmt.DN[,c("Day.B73.r1", "Day.B73.r2", "Day.B73.r3")])
tmt.DN$Mo17avg <- rowMeans(tmt.DN[,c("Day.Mo17.r1", "Day.Mo17.r2", "Day.Mo17.r3")])
tmt.DN$HYBavg <- rowMeans(tmt.DN[,c("Day.BM.r1", "Day.BM.r2", "Day.BM.r3")])
tmt.DN$MPavg <- rowMeans(tmt.DN[,c("B73avg", "Mo17avg")])
#calculate ratios
tmt.DN$HYB.MP <- tmt.DN$HYBavg/tmt.DN$MPavg
tmt.DN$l2.B73xMo17.MP <- log2(tmt.DN$HYB.MP)
tmt.DN$B73.Mo17 <- tmt.DN$B73avg/tmt.DN$Mo17avg
tmt.DN$l2.B73.Mo17 <-log2(tmt.DN$B73.Mo17)
```
```{r subset BxM mature leaf blade from original experiment}
#BxM from Original Experiment (OG)
tmt.LB <- tmt[,c(171:183,22:30)]
#Remove proteins not detected in one or more samples
tmt.LB <- na.omit(tmt.LB)
#Calculate averages of bio reps
tmt.LB$B73avg <- rowMeans(tmt.LB[,c("MatureLeaf.B73.r1", "MatureLeaf.B73.r2", "MatureLeaf.B73.r3")])
tmt.LB$Mo17avg <- rowMeans(tmt.LB[,c("MatureLeaf.Mo17.r1", "MatureLeaf.Mo17.r2", "MatureLeaf.Mo17.r3")])
tmt.LB$HYBavg <- rowMeans(tmt.LB[,c("MatureLeaf.BM.r1", "MatureLeaf.BM.r2", "MatureLeaf.BM.r3")])
tmt.LB$MPavg <- rowMeans(tmt.LB[,c("B73avg", "Mo17avg")])
#calculate ratios
tmt.LB$HYB.MP <- tmt.LB$HYBavg/tmt.LB$MPavg
tmt.LB$l2.B73xMo17.MP <- log2(tmt.LB$HYB.MP)
tmt.LB$B73.Mo17 <- tmt.LB$B73avg/tmt.LB$Mo17avg
tmt.LB$l2.B73.Mo17 <-log2(tmt.LB$B73.Mo17)
```
```{r subset BxM from day vs. night experiment (both day and night)}
#BxM from Day vs. night experiment (DN.both)
tmt.DN.both <- tmt[,c(171:183,62:70,72:80)]
#Remove proteins not detected in one or more samples
tmt.DN.both <- na.omit(tmt.DN.both)
#Calculate averages of bio reps
tmt.DN.both$day.B73avg <- rowMeans(tmt.DN.both[,c("Day.B73.r1", "Day.B73.r2", "Day.B73.r3")])
tmt.DN.both$day.Mo17avg <- rowMeans(tmt.DN.both[,c("Day.Mo17.r1", "Day.Mo17.r2", "Day.Mo17.r3")])
tmt.DN.both$day.HYBavg <- rowMeans(tmt.DN.both[,c("Day.BM.r1", "Day.BM.r2", "Day.BM.r3")])
tmt.DN.both$day.MPavg <- rowMeans(tmt.DN.both[,c("day.B73avg", "day.Mo17avg")])

tmt.DN.both$night.B73avg <- rowMeans(tmt.DN.both[,c("Night.B73.r1", "Night.B73.r2", "Night.B73.r3")])
tmt.DN.both$night.Mo17avg <- rowMeans(tmt.DN.both[,c("Night.Mo17.r1", "Night.Mo17.r2", "Night.Mo17.r3")])
tmt.DN.both$night.HYBavg <- rowMeans(tmt.DN.both[,c("Night.BM.r1", "Night.BM.r2", "Night.BM.r3")])
tmt.DN.both$night.MPavg <- rowMeans(tmt.DN.both[,c("night.B73avg", "night.Mo17avg")])
#calculate ratios
tmt.DN.both$day.HYB.MP <- tmt.DN.both$day.HYBavg/tmt.DN.both$day.MPavg
tmt.DN.both$day.l2.B73xMo17.MP <- log2(tmt.DN.both$day.HYB.MP)
tmt.DN.both$day.B73.Mo17 <- tmt.DN.both$day.B73avg/tmt.DN.both$day.Mo17avg
tmt.DN.both$day.l2.B73.Mo17 <-log2(tmt.DN.both$day.B73.Mo17)

tmt.DN.both$night.HYB.MP <- tmt.DN.both$night.HYBavg/tmt.DN.both$night.MPavg
tmt.DN.both$night.l2.B73xMo17.MP <- log2(tmt.DN.both$night.HYB.MP)
tmt.DN.both$night.B73.Mo17 <- tmt.DN.both$night.B73avg/tmt.DN.both$night.Mo17avg
tmt.DN.both$night.l2.B73.Mo17 <-log2(tmt.DN.both$night.B73.Mo17)
```
```{r subset all hybrids from 6-hybrid experiment}
#Subset all hybrids from 6-hybrid experiment (6H)
tmt.6H <- tmt[,c(171:183,31:60)]
#Remove proteins not detected in one or more samples
tmt.6H <- na.omit(tmt.6H)
#calculate averages
tmt.6H$B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r1", "X6Hybrid.B73.r2", "X6Hybrid.B73.r3")])
tmt.6H$Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r1", "X6Hybrid.Mo17.r2", "X6Hybrid.Mo17.r3")])
tmt.6H$B73xMo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B73xMo17.r1", "X6Hybrid.B73xMo17.r2", "X6Hybrid.B73xMo17.r3")])
tmt.6H$Mo17xB73 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17xB73.r1", "X6Hybrid.Mo17xB73.r2", "X6Hybrid.Mo17xB73.r3")])
tmt.6H$B84 <- rowMeans(tmt.6H[,c("X6Hybrid.B84.r1", "X6Hybrid.B84.r2", "X6Hybrid.B84.r3")])
tmt.6H$B84xB73 <- rowMeans(tmt.6H[,c("X6Hybrid.B84xB73.r1", "X6Hybrid.B84xB73.r2", "X6Hybrid.B84xB73.r3")])
tmt.6H$B84xMo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B84xMo17.r1", "X6Hybrid.B84xMo17.r2", "X6Hybrid.B84xMo17.r3")])
tmt.6H$A682 <- rowMeans(tmt.6H[,c("X6Hybrid.A682.r1", "X6Hybrid.A682.r2", "X6Hybrid.A682.r3")])
tmt.6H$A682xB73 <- rowMeans(tmt.6H[,c("X6Hybrid.A682xB73.r1", "X6Hybrid.A682xB73.r2", "X6Hybrid.A682xB73.r3")])
tmt.6H$A682xMo17 <- rowMeans(tmt.6H[,c("X6Hybrid.A682xMo17.r1", "X6Hybrid.A682xMo17.r2", "X6Hybrid.A682xMo17.r3")])
#Calculate hyb/MP ratios
tmt.6H$B73xMo17.MP <- tmt.6H$B73xMo17/rowMeans(tmt.6H[,c("B73", "Mo17")])
tmt.6H$Mo17xB73.MP <- tmt.6H$Mo17xB73/rowMeans(tmt.6H[,c("B73", "Mo17")])
tmt.6H$B84xB73.MP <- tmt.6H$B84xB73/rowMeans(tmt.6H[,c("B84", "B73")])
tmt.6H$B84xMo17.MP <- tmt.6H$B84xMo17/rowMeans(tmt.6H[,c("B84", "Mo17")])
tmt.6H$A682xB73.MP <- tmt.6H$A682xB73/rowMeans(tmt.6H[,c("A682", "B73")])
tmt.6H$A682xMo17.MP <- tmt.6H$A682xMo17/rowMeans(tmt.6H[,c("A682", "Mo17")])
#Calculate parent/parent ratios
tmt.6H$B73.Mo17 <- tmt.6H$B73/tmt.6H$Mo17
tmt.6H$Mo17.B73 <- tmt.6H$Mo17/tmt.6H$B73
tmt.6H$B84.B73 <- tmt.6H$B84/tmt.6H$B73
tmt.6H$B84.Mo17 <- tmt.6H$B84/tmt.6H$Mo17
tmt.6H$A682.B73 <- tmt.6H$A682/tmt.6H$B73
tmt.6H$A682.Mo17 <- tmt.6H$A682/tmt.6H$Mo17
#Calculate log2 of all ratios
tmt.6H$l2.B73xMo17.MP <- log2(tmt.6H$B73xMo17.MP)
tmt.6H$l2.Mo17xB73.MP <- log2(tmt.6H$Mo17xB73.MP)
tmt.6H$l2.B84xB73.MP <- log2(tmt.6H$B84xB73.MP)
tmt.6H$l2.B84xMo17.MP <- log2(tmt.6H$B84xMo17.MP)
tmt.6H$l2.A682xB73.MP <- log2(tmt.6H$A682xB73.MP)
tmt.6H$l2.A682xMo17.MP <- log2(tmt.6H$A682xMo17.MP)
tmt.6H$l2.B73.Mo17 <- log2(tmt.6H$B73.Mo17)
tmt.6H$l2.Mo17.B73 <- log2(tmt.6H$Mo17.B73)
tmt.6H$l2.B84.B73 <- log2(tmt.6H$B84.B73)
tmt.6H$l2.B84.Mo17 <- log2(tmt.6H$B84.Mo17)
tmt.6H$l2.A682.B73 <- log2(tmt.6H$A682.B73)
tmt.6H$l2.A682.Mo17 <- log2(tmt.6H$A682.Mo17)
```
```{r subset all hybrids from RIL experiment}
#Subset all hybrids from RIL experiment (RIL)
tmt.RIL <- tmt[,c(171:183,81:170)]
#Remove proteins not detected in one or more samples
tmt.RIL <- na.omit(tmt.RIL)
#Calculate averages
tmt.RIL$BM.B73 <- rowMeans(tmt.RIL[,c("B73.1", "B73.3", "B73.4")])
tmt.RIL$BM.Mo17 <- rowMeans(tmt.RIL[,c("Mo17.1", "Mo17.3", "Mo17.4")])
tmt.RIL$BM.BM <- rowMeans(tmt.RIL[,c("BM.1", "BM.2", "BM.3", "BM.4")])
tmt.RIL$M0014.B73 <- rowMeans(tmt.RIL[,c("B73.1.1", "B73.2", "B73.3.1", "B73.4.1")])
tmt.RIL$M0014.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.1", "Mo17.2", "Mo17.3.1", "Mo17.4.1")])
tmt.RIL$M0014.14 <- rowMeans(tmt.RIL[,c("X14.1", "X14.2", "X14.3", "X14.4")])
tmt.RIL$M0014.14B <- rowMeans(tmt.RIL[,c("X14B.1", "X14B.2", "X14B.3", "X14B.4")])
tmt.RIL$M0014.14M <- rowMeans(tmt.RIL[,c("X14M.1", "X14M.2", "X14M.3", "X14M.4")])
tmt.RIL$M0016.B73 <- rowMeans(tmt.RIL[,c("B73.1.2", "B73.2.1", "B73.3.2", "B73.4.2")])
tmt.RIL$M0016.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.2", "Mo17.2.1", "Mo17.3.2", "Mo17.4.2")])
tmt.RIL$M0016.16 <- rowMeans(tmt.RIL[,c("X16.1", "X16.2", "X16.3", "X16.4")])
tmt.RIL$M0016.16B <- rowMeans(tmt.RIL[,c("X16B.1", "X16B.2", "X16B.3", "X16B.4")])
tmt.RIL$M0016.16M <- rowMeans(tmt.RIL[,c("X16M.1", "X16M.2", "X16M.3", "X16M.4")])
tmt.RIL$M0021.B73 <- rowMeans(tmt.RIL[,c("B73.1.3", "B73.2.2", "B73.3.3", "B73.4.3")])
tmt.RIL$M0021.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.3", "Mo17.2.2", "Mo17.3.3", "Mo17.4.3")])
tmt.RIL$M0021.21 <- rowMeans(tmt.RIL[,c("X21.1", "X21.2", "X21.3", "X21.4")])
tmt.RIL$M0021.21B <- rowMeans(tmt.RIL[,c("X21B.1", "X21B.2", "X21B.3", "X21B.4")])
tmt.RIL$M0021.21M <- rowMeans(tmt.RIL[,c("X21M.1", "X21M.2", "X21M.3", "X21M.4")])
tmt.RIL$M0317.B73 <- rowMeans(tmt.RIL[,c("B73.1.4", "B73.2.3", "B73.3.4", "B73.4.4")])
tmt.RIL$M0317.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.4", "Mo17.2.3", "Mo17.3.4", "Mo17.4.4")])
tmt.RIL$M0317.317 <- rowMeans(tmt.RIL[,c("X317.1", "X317.2", "X317.3", "X317.4")])
tmt.RIL$M0317.317B <- rowMeans(tmt.RIL[,c("X317B.1", "X317B.2", "X317B.3", "X317B.4")])
tmt.RIL$M0317.317M <- rowMeans(tmt.RIL[,c("X317M.1", "X317M.2", "X317M.3", "X317M.4")])
#Calculate hyb/mp ratios
tmt.RIL$BM.BM.MP <- tmt.RIL$BM.BM/rowMeans(tmt.RIL[,c("BM.B73", "BM.Mo17")], na.rm = T)
tmt.RIL$M0014.14B.MP <- tmt.RIL$M0014.14B/rowMeans(tmt.RIL[,c("M0014.14", "M0014.B73")], na.rm = T)
tmt.RIL$M0014.14M.MP <- tmt.RIL$M0014.14M/rowMeans(tmt.RIL[,c("M0014.14", "M0014.Mo17")], na.rm = T)
tmt.RIL$M0016.16B.MP <- tmt.RIL$M0016.16B/rowMeans(tmt.RIL[,c("M0016.16", "M0016.B73")], na.rm = T)
tmt.RIL$M0016.16M.MP <- tmt.RIL$M0016.16M/rowMeans(tmt.RIL[,c("M0016.16", "M0016.Mo17")], na.rm = T)
tmt.RIL$M0021.21B.MP <- tmt.RIL$M0021.21B/rowMeans(tmt.RIL[,c("M0021.21", "M0021.B73")], na.rm = T)
tmt.RIL$M0021.21M.MP <- tmt.RIL$M0021.21M/rowMeans(tmt.RIL[,c("M0021.21", "M0021.Mo17")], na.rm = T)
tmt.RIL$M0317.317B.MP <- tmt.RIL$M0317.317B/rowMeans(tmt.RIL[,c("M0317.317", "M0317.B73")], na.rm = T)
tmt.RIL$M0317.317M.MP <- tmt.RIL$M0317.317M/rowMeans(tmt.RIL[,c("M0317.317", "M0317.Mo17")], na.rm = T)
#Calculate parent/parent ratios
tmt.RIL$BM.B73.Mo17 <- tmt.RIL$BM.B73/tmt.RIL$BM.Mo17
tmt.RIL$M0014.14.B73 <- tmt.RIL$M0014.14/tmt.RIL$M0014.B73
tmt.RIL$M0014.14.Mo17 <- tmt.RIL$M0014.14/tmt.RIL$M0014.Mo17
tmt.RIL$M0016.16.B73 <- tmt.RIL$M0016.16/tmt.RIL$M0016.B73
tmt.RIL$M0016.16.Mo17 <- tmt.RIL$M0016.16/tmt.RIL$M0016.Mo17
tmt.RIL$M0021.21.B73 <- tmt.RIL$M0021.21/tmt.RIL$M0021.B73
tmt.RIL$M0021.21.Mo17 <- tmt.RIL$M0021.21/tmt.RIL$M0021.Mo17
tmt.RIL$M0317.317.B73 <- tmt.RIL$M0317.317/tmt.RIL$M0317.B73
tmt.RIL$M0317.317.Mo17 <- tmt.RIL$M0317.317/tmt.RIL$M0317.Mo17
#Calculate log2 of all ratios
tmt.RIL$l2.B73xMo17.MP <- log2(tmt.RIL$BM.BM.MP)
tmt.RIL$l2.M0014.14B.MP <- log2(tmt.RIL$M0014.14B.MP)
tmt.RIL$l2.M0014.14M.MP <- log2(tmt.RIL$M0014.14M.MP)
tmt.RIL$l2.M0016.16B.MP <- log2(tmt.RIL$M0016.16B.MP)
tmt.RIL$l2.M0016.16M.MP <- log2(tmt.RIL$M0016.16M.MP)
tmt.RIL$l2.M0021.21B.MP <- log2(tmt.RIL$M0021.21B.MP)
tmt.RIL$l2.M0021.21M.MP <- log2(tmt.RIL$M0021.21M.MP)
tmt.RIL$l2.M0317.317B.MP <- log2(tmt.RIL$M0317.317B.MP)
tmt.RIL$l2.M0317.317M.MP <- log2(tmt.RIL$M0317.317M.MP)
tmt.RIL$l2.B73.Mo17 <- log2(tmt.RIL$BM.B73.Mo17)
tmt.RIL$l2.M0014.14.B73 <- log2(tmt.RIL$M0014.14.B73)
tmt.RIL$l2.M0014.14.Mo17  <- log2(tmt.RIL$M0014.14.Mo17)
tmt.RIL$l2.M0016.16.B73 <- log2(tmt.RIL$M0016.16.B73)
tmt.RIL$l2.M0016.16.Mo17  <- log2(tmt.RIL$M0016.16.Mo17)
tmt.RIL$l2.M0021.21.B73 <- log2(tmt.RIL$M0021.21.B73)
tmt.RIL$l2.M0021.21.Mo17  <- log2(tmt.RIL$M0021.21.Mo17)
tmt.RIL$l2.M0317.317.B73 <- log2(tmt.RIL$M0317.317.B73)
tmt.RIL$l2.M0317.317.Mo17  <- log2(tmt.RIL$M0317.317.Mo17)
```
```{r subset acs mutant experiment}
#Subset acs mutant experiment (acs)
tmt.acs <- tmt[,c(171:183,1:10)]
#Remove proteins not detected in one or more samples
tmt.acs <- na.omit(tmt.acs)
#Calculate averages of bio reps - acs
tmt.acs$B73avg <- rowMeans(tmt.acs[,c("B73.acs.B73.r1", "B73.acs.B73.r2", "B73.acs.B73.r3", "B73.acs.B73.r4", "B73.acs.B73.r5")])
tmt.acs$ACSavg <- rowMeans(tmt.acs[,c("B73.acs.acs.r1", "B73.acs.acs.r2", "B73.acs.acs.r3", "B73.acs.acs.r4", "B73.acs.acs.r5")])
#calculate ratio
tmt.acs$ACS.B73 <- tmt.acs$ACSavg/tmt.acs$B73avg
#calculate log2
tmt.acs$l2.ACS.B73 <- log2(tmt.acs$ACS.B73)
#merge with BxM from original data (OG)
tmt.acs.OG <- merge.data.frame(tmt.acs, tmt.SL[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
#merge with BxM from 6-hybrid data (6H)
tmt.acs.6H <- merge.data.frame(tmt.acs, tmt.6H[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
#merge with BxM from RIL data (RIL)
tmt.acs.RIL <- merge.data.frame(tmt.acs, tmt.RIL[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
#merge with BxM from day vs. night data (DN)
tmt.acs.DN <- merge.data.frame(tmt.acs, tmt.DN[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
```

#Import RNA data
```{r import RNA data}
#Original experiment
cpm.OG.data <- read_tsv(paste0(PathToData, "cpm.tsv"))
#6-hybrid experiment
cpm.6H.data <- readRDS(paste0(PathToData, "rn20b.rds"))
cpm.6H.tm <- cpm.6H.data$tm
cpm.6H.th <- cpm.6H.data$th
#RIL experiment
cpm.RIL.data <- readRDS(paste0(PathToData, "rn20b2.rds"))
cpm.RIL.tm <- cpm.RIL.data$tm
cpm.RIL.th <- cpm.RIL.data$th
```


#Subset RNA experiments
```{r subset BxM seedling leaf from original experiment}
#Original experiment seedling leaf BxM
cpm.SL <- as.data.frame(cpm.OG.data[,c(1,178:186)])
colnames(cpm.SL) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "B73xMo17.1", "B73xMo17.2", "B73xMo17.3")
#Remove genes not detected in one or mor rep
cpm.SL[cpm.SL==0] <- NA
cpm.SL <- na.omit(cpm.SL)
#Calculate averages of bio reps
cpm.SL$B73 <- rowMeans(cpm.SL[,c("B73.1", "B73.2", "B73.3")])
cpm.SL$Mo17 <- rowMeans(cpm.SL[,c("Mo17.1", "Mo17.2", "Mo17.3")])
cpm.SL$B73xMo17 <- rowMeans(cpm.SL[,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")])
#calculate ratios
cpm.SL$B73xMo17.MP <- cpm.SL$B73xMo17/rowMeans(cpm.SL[,c("B73", "Mo17")])
cpm.SL$l2.B73xMo17.MP <- log2(cpm.SL$B73xMo17.MP)
cpm.SL$B73.Mo17 <- cpm.SL$B73/cpm.SL$Mo17
cpm.SL$l2.B73.Mo17 <-log2(cpm.SL$B73.Mo17)
##Remove genes that weren't detected as protein
cpm.SL.protein <- cpm.SL[cpm.SL$Gene%in%tmt.SL$Gene,]
```
```{r subset BxM leaf blade from original experiment}
#Original experiment seedling leaf BxM
cpm.LB <- as.data.frame(cpm.OG.data[,c(1,2,3,5,4,6,8,7,9,10)])
colnames(cpm.LB) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "B73xMo17.1", "B73xMo17.2", "B73xMo17.3")
#Remove genes not detected in one or more rep
cpm.LB[cpm.LB==0] <- NA
cpm.LB <- na.omit(cpm.LB)
#Calculate averages of bio reps
cpm.LB$B73 <- rowMeans(cpm.LB[,c("B73.1", "B73.2", "B73.3")])
cpm.LB$Mo17 <- rowMeans(cpm.LB[,c("Mo17.1", "Mo17.2", "Mo17.3")])
cpm.LB$B73xMo17 <- rowMeans(cpm.LB[,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")])
#calculate ratios
cpm.LB$B73xMo17.MP <- cpm.LB$B73xMo17/rowMeans(cpm.LB[,c("B73", "Mo17")])
cpm.LB$l2.B73xMo17.MP <- log2(cpm.LB$B73xMo17.MP)
cpm.LB$B73.Mo17 <- cpm.LB$B73/cpm.LB$Mo17
cpm.LB$l2.B73.Mo17 <-log2(cpm.LB$B73.Mo17)
##Remove genes that weren't detected as protein
cpm.LB.protein <- cpm.LB[cpm.LB$Gene%in%tmt.LB$Gene,]
```
```{r subset 6 hybrid RNA data}
#subset to leaf samples
cpm.6H.th.ID <- cpm.6H.th[cpm.6H.th$Replicate%in%c(2,3,4) & cpm.6H.th$Tissue=="leaf_V2" & cpm.6H.th$Treatment=="m",]
cpm.6H <- cpm.6H.tm[cpm.6H.tm$SampleID%in%cpm.6H.th.ID$SampleID,c(1,2,6)]
#reformat df
cpm.6H <- dcast(cpm.6H, gid ~ SampleID)
#Rename columns
colnames(cpm.6H) <- c("Gene", paste(cpm.6H.th.ID$Genotype, cpm.6H.th.ID$Replicate, sep = "."))
#Remove genes not detected in one or mor rep
cpm.6H[cpm.6H==0] <- NA
cpm.6H <- na.omit(cpm.6H)
#Calculate averages
cpm.6H$B73 <- rowMeans(cpm.6H[,c("B73.2", "B73.3", "B73.4")], na.rm = T)
cpm.6H$B84 <- rowMeans(cpm.6H[,c("B84.2", "B84.3", "B84.4")], na.rm = T)
cpm.6H$Mo17 <- rowMeans(cpm.6H[,c("Mo17.2", "Mo17.3", "Mo17.4")], na.rm = T)
cpm.6H$A682 <- rowMeans(cpm.6H[,c("A682.2", "A682.3", "A682.4")], na.rm = T)
cpm.6H$B73xMo17 <- rowMeans(cpm.6H[,c("B73xMo17.2", "B73xMo17.3", "B73xMo17.4")], na.rm = T)
cpm.6H$Mo17xB73 <- rowMeans(cpm.6H[,c("Mo17xB73.2", "Mo17xB73.3", "Mo17xB73.4")], na.rm = T)
cpm.6H$B84xB73 <- rowMeans(cpm.6H[,c("B84xB73.2", "B84xB73.3", "B84xB73.4")], na.rm = T)
cpm.6H$B84xMo17 <- rowMeans(cpm.6H[,c("B84xMo17.2", "B84xMo17.3", "B84xMo17.4")], na.rm = T)
cpm.6H$A682xB73 <- rowMeans(cpm.6H[,c("A682xB73.2", "A682xB73.3", "A682xB73.4")], na.rm = T)
cpm.6H$A682xMo17 <- rowMeans(cpm.6H[,c("A682xMo17.2", "A682xMo17.3", "A682xMo17.4")], na.rm = T)
#calculate ratio of hybrid to midparent
cpm.6H$B73xMo17.MP <- cpm.6H$B73xMo17/rowMeans(cpm.6H[,c("B73","Mo17")], na.rm = T)
cpm.6H$Mo17xB73.MP <- cpm.6H$Mo17xB73/rowMeans(cpm.6H[,c("B73","Mo17")], na.rm = T)
cpm.6H$B84xB73.MP <- cpm.6H$B84xB73/rowMeans(cpm.6H[,c("B84","B73")], na.rm = T)
cpm.6H$B84xMo17.MP <- cpm.6H$B84xMo17/rowMeans(cpm.6H[,c("B84","Mo17")], na.rm = T)
cpm.6H$A682xB73.MP <- cpm.6H$A682xB73/rowMeans(cpm.6H[,c("A682","B73")], na.rm = T)
cpm.6H$A682xMo17.MP <- cpm.6H$A682xMo17/rowMeans(cpm.6H[,c("A682","Mo17")], na.rm = T)
#Calculate ratio of parent to parent
cpm.6H$B73.Mo17 <- cpm.6H$B73/cpm.6H$Mo17
cpm.6H$Mo17.B73 <- cpm.6H$Mo17/cpm.6H$B73
cpm.6H$B84.B73 <- cpm.6H$B84/cpm.6H$B73
cpm.6H$B84.Mo17 <- cpm.6H$B84/cpm.6H$Mo17
cpm.6H$A682.B73 <- cpm.6H$A682/cpm.6H$B73
cpm.6H$A682.Mo17 <- cpm.6H$A682/cpm.6H$Mo17
#Change Inf
#calculate log2 of ratios
cpm.6H$l2.B73xMo17.MP <- log2(cpm.6H$B73xMo17.MP)
cpm.6H$l2.Mo17xB73.MP <- log2(cpm.6H$Mo17xB73.MP)
cpm.6H$l2.B84xB73.MP <- log2(cpm.6H$B84xB73.MP)
cpm.6H$l2.B84xMo17.MP <- log2(cpm.6H$B84xMo17.MP)
cpm.6H$l2.A682xB73.MP <- log2(cpm.6H$A682xB73.MP )
cpm.6H$l2.A682xMo17.MP <- log2(cpm.6H$A682xMo17.MP)
cpm.6H$l2.B73.Mo17 <- log2(cpm.6H$B73.Mo17)
cpm.6H$l2.Mo17.B73 <- log2(cpm.6H$Mo17.B73)
cpm.6H$l2.B84.B73 <- log2(cpm.6H$B84.B73)
cpm.6H$l2.B84.Mo17 <- log2(cpm.6H$B84.Mo17)
cpm.6H$l2.A682.B73 <- log2(cpm.6H$A682.B73)
cpm.6H$l2.A682.Mo17 <- log2(cpm.6H$A682.Mo17)
##Remove genes that weren't detected as protein
cpm.6H.protein <- cpm.6H[cpm.6H$Gene%in%tmt.6H$Gene,]
```
```{r subset RIL RNA data}
#reformat df
cpm.RIL <- dcast(cpm.RIL.tm[,c(1,2,6)], gid ~ SampleID)
#Remove genes not detected in one or mor rep
cpm.RIL[cpm.RIL==0] <- NA
cpm.RIL <- na.omit(cpm.RIL)
#Rename columns
colnames(cpm.RIL) <- c("Gene", paste(cpm.RIL.th$Genotype, cpm.RIL.th$Replicate, sep = "."))
#Calculate averages
cpm.RIL$B73 <- rowMeans(cpm.RIL[,c("B73.1", "B73.2", "B73.3", "B73.4")], na.rm = T)
cpm.RIL$Mo17 <- rowMeans(cpm.RIL[,c("Mo17.1", "Mo17.2", "Mo17.3", "Mo17.4")], na.rm = T)
cpm.RIL$B73xMo17 <- rowMeans(cpm.RIL[,c("BxM.1", "BxM.2", "BxM.3", "BxM.4")], na.rm = T)
cpm.RIL$M0014 <- rowMeans(cpm.RIL[,c("M0014.1", "M0014.2", "M0014.3", "M0014.4")], na.rm = T)
cpm.RIL$M0014xB73 <- rowMeans(cpm.RIL[,c("M0014xB73.1", "M0014xB73.2", "M0014xB73.3", "M0014xB73.4")], na.rm = T)
cpm.RIL$M0014xMo17 <- rowMeans(cpm.RIL[,c("M0014xMo17.1", "M0014xMo17.2", "M0014xMo17.3", "M0014xMo17.4")], na.rm = T)
cpm.RIL$M0016 <- rowMeans(cpm.RIL[,c("M0016.1", "M0016.2", "M0016.3", "M0016.4")], na.rm = T)
cpm.RIL$M0016xB73 <- rowMeans(cpm.RIL[,c("M0016xB73.1", "M0016xB73.2", "M0016xB73.3")], na.rm = T)
cpm.RIL$M0016xMo17 <- rowMeans(cpm.RIL[,c("M0016xMo17.1", "M0016xMo17.2", "M0016xMo17.3", "M0016xMo17.4")], na.rm = T)
cpm.RIL$M0021 <- rowMeans(cpm.RIL[,c("M0021.1", "M0021.2", "M0021.3", "M0021.4")], na.rm = T)
cpm.RIL$M0021xB73 <- rowMeans(cpm.RIL[,c("M0021xB73.1", "M0021xB73.2", "M0021xB73.3", "M0021xB73.4")], na.rm = T)
cpm.RIL$M0021xMo17 <- rowMeans(cpm.RIL[,c("M0021xMo17.1", "M0021xMo17.2", "M0021xMo17.3", "M0021xMo17.4")], na.rm = T)
cpm.RIL$M0317 <- rowMeans(cpm.RIL[,c("M0317.1", "M0317.2", "M0317.3", "M0317.4")], na.rm = T)
cpm.RIL$M0317xB73 <- rowMeans(cpm.RIL[,c("M0317xB73.1", "M0317xB73.2", "M0317xB73.3", "M0317xB73.4")], na.rm = T)
cpm.RIL$M0317xMo17 <- rowMeans(cpm.RIL[,c("M0317xMo17.1", "M0317xMo17.2", "M0317xMo17.3", "M0317xMo17.4")], na.rm = T)
#calculate ratio of hybrid to midparent
cpm.RIL$B73xMo17.MP <- cpm.RIL$B73xMo17/rowMeans(cpm.RIL[,c("B73","Mo17")], na.rm = T)
cpm.RIL$M0014xB73.MP <- cpm.RIL$M0014xB73/rowMeans(cpm.RIL[,c("M0014", "B73")], na.rm = T)
cpm.RIL$M0014xMo17.MP <- cpm.RIL$M0014xMo17/rowMeans(cpm.RIL[,c("M0014", "Mo17")], na.rm = T)
cpm.RIL$M0016xB73.MP <- cpm.RIL$M0016xB73/rowMeans(cpm.RIL[,c("M0016", "B73")], na.rm = T)
cpm.RIL$M0016xMo17.MP <- cpm.RIL$M0016xMo17/rowMeans(cpm.RIL[,c("M0016", "Mo17")], na.rm = T)
cpm.RIL$M0021xB73.MP <- cpm.RIL$M0021xB73/rowMeans(cpm.RIL[,c("M0021", "B73")], na.rm = T)
cpm.RIL$M0021xMo17.MP <- cpm.RIL$M0021xMo17/rowMeans(cpm.RIL[,c("M0021", "Mo17")], na.rm = T)
cpm.RIL$M0317xB73.MP <- cpm.RIL$M0317xB73/rowMeans(cpm.RIL[,c("M0317", "B73")], na.rm = T)
cpm.RIL$M0317xMo17.MP <- cpm.RIL$M0317xMo17/rowMeans(cpm.RIL[,c("M0317", "Mo17")], na.rm = T)
#Calculate ratio of parent to parent
cpm.RIL$B73.Mo17 <- cpm.RIL$B73/cpm.RIL$Mo17
cpm.RIL$M0014.B73 <- cpm.RIL$M0014/cpm.RIL$B73
cpm.RIL$M0014.Mo17 <- cpm.RIL$M0014/cpm.RIL$Mo17
cpm.RIL$M0016.B73 <- cpm.RIL$M0016/cpm.RIL$B73
cpm.RIL$M0016.Mo17 <- cpm.RIL$M0016/cpm.RIL$Mo17
cpm.RIL$M0021.B73 <- cpm.RIL$M0021/cpm.RIL$B73
cpm.RIL$M0021.Mo17 <- cpm.RIL$M0021/cpm.RIL$Mo17
cpm.RIL$M0317.B73 <- cpm.RIL$M0317/cpm.RIL$B73
cpm.RIL$M0317.Mo17 <- cpm.RIL$M0317/cpm.RIL$Mo17
#calculate ratio of hybrid to midparent
cpm.RIL$l2.B73xMo17.MP <- log2(cpm.RIL$B73xMo17.MP)
cpm.RIL$l2.M0014xB73.MP <- log2(cpm.RIL$M0014xB73.MP)
cpm.RIL$l2.M0014xMo17.MP <- log2(cpm.RIL$M0014xMo17.MP)
cpm.RIL$l2.M0016xB73.MP <- log2(cpm.RIL$M0016xB73.MP)
cpm.RIL$l2.M0016xMo17.MP <- log2(cpm.RIL$M0016xMo17.MP)
cpm.RIL$l2.M0021xB73.MP <- log2(cpm.RIL$M0021xB73.MP)
cpm.RIL$l2.M0021xMo17.MP <- log2(cpm.RIL$M0021xMo17.MP )
cpm.RIL$l2.M0317xB73.MP <- log2(cpm.RIL$M0317xB73.MP)
cpm.RIL$l2.M0317xMo17.MP <- log2(cpm.RIL$M0317xMo17.MP)
#Calculate ratio of parent to parent
cpm.RIL$l2.B73.Mo17 <- log2(cpm.RIL$B73.Mo17)
cpm.RIL$l2.M0014.B73 <- log2(cpm.RIL$M0014.B73)
cpm.RIL$l2.M0014.Mo17 <- log2(cpm.RIL$M0014.Mo17)
cpm.RIL$l2.M0016.B73 <- log2(cpm.RIL$M0016.B73)
cpm.RIL$l2.M0016.Mo17 <- log2(cpm.RIL$M0016.Mo17)
cpm.RIL$l2.M0021.B73 <- log2(cpm.RIL$M0021.B73)
cpm.RIL$l2.M0021.Mo17 <- log2(cpm.RIL$M0021.Mo17)
cpm.RIL$l2.M0317.B73 <- log2(cpm.RIL$M0317.B73)
cpm.RIL$l2.M0317.Mo17 <- log2(cpm.RIL$M0317.Mo17)
##Remove genes that weren't detected as protein
cpm.RIL.protein <- cpm.RIL
cpm.RIL.protein <- cpm.RIL.protein[cpm.RIL.protein$Gene%in%tmt.RIL$Gene,]
```

#Define GOIs
##Accession conversion
```{r mgdb}
#Downloaded from MaizeGDB 2020/05/18, 07:11pm
mgdb <- read.csv(paste0(PathToData,"gene_model_xref_v4.csv"), stringsAsFactors = F, header = F)
colnames(mgdb) <- mgdb[5,]
mgdb <- mgdb[-c(1:5),]
```
```{r organelleEntrez}
organelleEntrez <- tmt[grep("Zema|Zeam", tmt$Accession),]
organelleEntrez <- data.frame("Accession"=organelleEntrez$Accession, "Entrez"=NA)
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp003.1"] <- "845178"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp004.1"] <- "845232"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp008.1"] <- "845201"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp013.1"] <- "845225"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp014.1"] <- "845226"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp016.1"] <- "845236"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp017.1"] <- "845174"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp018.1"] <- "845173"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp021.1"] <- "845229"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp022.1"] <- "845196"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp023.1"] <- "845195"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp024.1"] <- "1466361"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp026.1"] <- "845238"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp027.1"] <- "845189"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp028.1"] <- "845190"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp029.1"] <- "845182"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp030.1"] <- "845171"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp032.1"] <- "845212"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp035.1"] <- "845175"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp036.1"] <- "845191"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp039.1"] <- "845204"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp040.1"] <- "845203"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp043.1"] <- "845198"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp044.1"] <- "845222"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp045.1"] <- "845233"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp046.1"] <- "845217"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp047.1"] <- "845176"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp048.1"] <- "845200"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp050.1"] <- "845211"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp051.1"] <- "845205"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp052.1"] <- "845192"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp053.1"] <- "845193"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp054.1"] <- "845224"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp055.1"] <- "845228"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp056.1"] <- "845223"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp057.1"] <- "845177"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp058.1"] <- "845241"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp059.1"] <- "845213"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp060.1"] <- "845214"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp061.1"] <- "845237"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp062.1"] <- "845218"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp063.1"] <- "845235"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp065.1"] <- "845216"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp067.1"] <- "845220"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp075.1"] <- "845181"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp082.1"] <- "845230"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp083.1"] <- "845185"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp084.1"] <- "845221"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp085.1"] <- "1466378"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp086.1"] <- "845183"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp087.1"] <- "1466379"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp088.1"] <- "845184"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp090.1"] <- "845188"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp091.1"] <- "845179"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp092.1"] <- "845187"

organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp008.1"] <- "4055864"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp024.1"] <- "4055860"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp032.1"] <- "4055888"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp033.1"] <- "4055938"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp036.1"] <- "4055894"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp050.1"] <- "4055956"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp051.1"] <- "4055881"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp057.1"] <- "4055973"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp081.1"] <- "4055890"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp092.1"] <- "4055837"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp097.1"] <- "4055857"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp098.1"] <- "4055896"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp113.1"] <- "4055802"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp119.1"] <- "4055928"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp129.1"] <- "4055878"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp136.1"] <- "4056004"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp159.1"] <- "4055794"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp167.1"] <- "4056007"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp179.1"] <- "4056006"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp185.1"] <- "4055892"
```
```{r Entrez conversion}
Entrez <- mgdb[,c(1,12)]
colnames(Entrez) <- c("Accession", "Entrez")
Entrez <- rbind(Entrez, organelleEntrez)
```
##Plastid-encoded
```{r PE}
#Plastid-encoded proteins
PE<-tmt[grep("ZemaCp", tmt$Accession),]
#define list of plastid-encoded photosynthesis-associated genes
PE.PS <- PE$Gene[grep("psb|psa|ndh|rbc|pet|atp|ccs|cem", PE$Protein)]
PE.PS.PSI <- PE$Gene[grep("psa", PE$Protein)]
PE.PS.PSII <- PE$Gene[grep("psb", PE$Protein)]
PE.PS.NADH <- PE$Gene[grep("ndh", PE$Protein)]
PE.PS.b6f <- PE$Gene[grep("pet", PE$Protein)]
PE.PS.ATP <- PE$Gene[grep("atp", PE$Protein)]
PE.PS.rubisco <- PE$Gene[grep("rbc", PE$Protein)]
PhAPGs <- c(PE.PS.PSI,PE.PS.PSII,PE.PS.NADH,PE.PS.b6f,PE.PS.ATP,PE.PS.rubisco)
#Plastid encoded ribosomal proteins (EnsemblPlants Annotation)
#get subset of PE encoded genes annotated as ribosomal proteins
PE.ribo <- PE$Gene[grep("rps|rpl", PE$Protein)]
#Plastid encoded RNA polymerases (EnsemblPlants Annotation)
PE.rpo <- PE$Gene[grep("rpo", PE$Protein)]
#Define "other" as everything left
PE.other <- PE$Gene[!(PE$Gene%in%c(PE.PS.rubisco, PhAPGs, PE.ribo, PE.rpo))]
PE <- PE$Gene
```

##PhANGs
###KEGG - zma00195
```{r zma00195 tibble}
#Retrieved from KEGG on 02/26/2020
KEGG.zma00195.tib <-tibble::tribble(
              ~V1,                                                                                    ~V2,
          845199L,                            "psbA; photosystem II protein D1 [KO:K02703] [EC:1.10.3.9]",
          845200L,                                      "psbB; photosystem II 47 kDa protein [KO:K02704]",
          845201L,                                      "psbC; photosystem II 44 kDa protein [KO:K02705]",
          845202L,                            "psbD; photosystem II protein D2 [KO:K02706] [EC:1.10.3.9]",
          845203L,                                           "psbE; photosystem II protein V [KO:K02707]",
          845204L,                                          "psbF; photosystem II protein VI [KO:K02708]",
          845205L,                                           "psbH; photosystem II protein H [KO:K02709]",
          845206L,                                           "psbI; photosystem II protein I [KO:K02710]",
          845207L,                                           "psbJ; photosystem II protein J [KO:K02711]",
          845208L,                                           "psbK; photosystem II protein K [KO:K02712]",
          845209L,                                           "psbL; photosystem II protein L [KO:K02713]",
          845210L,                                           "psbM; photosystem II protein M [KO:K02714]",
       100191684L,                                     "uncharacterized protein LOC100191684 [KO:K02716]",
       100272890L,                                             "uncharacterized LOC100272890 [KO:K02716]",
       100281199L,    "chloroplast oxygen-evolving complex/thylakoid lumenal 25.6kDa protein [KO:K02717]",
       107648855L,                                     "uncharacterized protein LOC107648855 [KO:K02717]",
       100273117L,                        "oxygen-evolving enhancer protein 2, chloroplastic [KO:K02717]",
          542279L,               "PSBQ1; oxygen-evolving enhancer protein 3-1, chloroplastic [KO:K08901]",
       100273120L,                    "oxygen evolving enhancer protein 3 containing protein [KO:K08901]",
       103647735L,                 "oxygen-evolving enhancer protein 3-1, chloroplastic-like [KO:K08901]",
       103627333L,          "photosynthetic NDH subunit of lumenal location 2, chloroplastic [KO:K08901]",
       103653672L,          "photosynthetic NDH subunit of lumenal location 2, chloroplastic [KO:K08901]",
       100281646L,                         "photosystem II 10 kDa polypeptide, chloroplastic [KO:K03541]",
          542126L,                                      "psbs1; photosystem II subunit PsbS1 [KO:K03542]",
         1466364L,                                           "psbT; photosystem II protein T [KO:K02718]",
       100283816L,                          "pco070877; uncharacterized protein LOC100283816 [KO:K02721]",
       100283835L,                           "cl5838_2; uncharacterized protein LOC100283835 [KO:K02721]",
       100280994L,                                "photosystem II core complex proteins psbY [KO:K02723]",
       100281349L,                                "photosystem II core complex proteins psbY [KO:K02723]",
         1466358L,                                           "psbZ; photosystem II protein Z [KO:K02724]",
       100282043L,                                             "photosystem II 11 kD protein [KO:K08902]",
       100281235L,                                             "photosystem II 11 kD protein [KO:K08902]",
       103636309L,         "photosystem II reaction center PSB28 protein, chloroplastic-like [KO:K08903]",
       100284876L,                                 "photosystem II reaction center W protein [KO:K08903]",
       103650780L,              "photosystem II reaction center PSB28 protein, chloroplastic [KO:K08903]",
       109945128L,              "photosystem II reaction center PSB28 protein, chloroplastic [KO:K08903]",
       103650783L,              "photosystem II reaction center PSB28 protein, chloroplastic [KO:K08903]",
          845195L,                     "psaA; photosystem I P700 chlorophyll a apoprotein A1 [KO:K02689]",
          845196L,                     "psaB; photosystem I P700 chlorophyll a apoprotein A2 [KO:K02690]",
         1466379L,                                          "psaC; photosystem I subunit VII [KO:K02691]",
       100191984L,                                 "photosystem I reaction center subunit II [KO:K02692]",
          541791L,                  "photosystem I reaction center subunit II, chloroplastic [KO:K02692]",
       100283327L,                               "photosystem I reaction center subunit IV A [KO:K02693]",
       100283085L,                                     "uncharacterized protein LOC100283085 [KO:K02693]",
       100282027L,                                "photosystem I reaction center subunit III [KO:K02694]",
       103625835L,                 "photosystem I reaction center subunit III, chloroplastic [KO:K02694]",
       100285458L,                                  "photosystem I reaction center subunit V [KO:K08905]",
       100284847L,                                  "photosystem I reaction center subunit V [KO:K08905]",
          541685L, "psah1; photosystem I reaction center subunit VI, chloroplastic precursor [KO:K02695]",
          845197L,                                         "psaI; photosystem I subunit VIII [KO:K02696]",
          845198L,                                           "psaJ; photosystem I subunit IX [KO:K02697]",
       100286020L,                                    "psa6; photosystem I reaction center 6 [KO:K02698]",
       100281679L,                                 "photosystem I reaction center subunit XI [KO:K02699]",
       100383080L,                            "umc1974; uncharacterized protein LOC100383080 [KO:K02699]",
          542605L,                   "photosystem I reaction center subunit N, chloroplastic [KO:K02701]",
       103640891L,              "photosystem I reaction center subunit N, chloroplastic-like [KO:K02701]",
       100283848L,                          "pco139896; uncharacterized protein LOC100283848 [KO:K14332]",
          845191L,                                                       "petA; cytochrome f [KO:K02634]",
       103643185L,                                     "uncharacterized protein LOC103643185 [KO:K02634]",
          845192L,                                                      "petB; cytochrome b6 [KO:K02635]",
       100273354L,                                 "ris2; iron-sulfur protein 2 [KO:K02636] [EC:7.1.1.6]",
       100273026L,                 "cytochrome b6-f complex iron-sulfur subunit [KO:K02636] [EC:7.1.1.6]",
          845193L,                                 "petD; cytochrome b6/f complex subunit IV [KO:K02637]",
          845194L,                                  "petG; cytochrome b6/f complex subunit V [KO:K02640]",
         1466360L,                               "petN; cytochrome b6/f complex subunit VIII [KO:K03689]",
       100192779L,                                     "uncharacterized protein LOC100192779 [KO:K02638]",
       103629356L,                                              "plastocyanin, chloroplastic [KO:K02638]",
       100191704L,                                     "uncharacterized protein LOC100191704 [KO:K02639]",
       103628935L,                                        "fdx5; ferredoxin-5, chloroplastic [KO:K02639]",
          541619L,                                        "fdx2; ferredoxin-2, chloroplastic [KO:K02639]",
       100283643L,                                                             "ferredoxin-3 [KO:K02639]",
          541612L,                              "pco072676(750); ferredoxin-6, chloroplastic [KO:K02639]",
       100382495L,                                      "putative ferredoxin-4 chloroplastic [KO:K02639]",
       100281226L,                                                             "ferredoxin-6 [KO:K02639]",
          542276L,                                        "fdx3; ferredoxin-3, chloroplastic [KO:K02639]",
       100284745L,                                                               "ferredoxin [KO:K02639]",
       101027099L,                                         "FDX1; adrenodoxin, mitochondrial [KO:K02639]",
       103653165L,                                                               "ferredoxin [KO:K02639]",
          541626L,                                        "L-FNRII; ferredoxin [KO:K02641] [EC:1.18.1.2]",
       100194249L,               "IDP808; uncharacterized protein LOC100194249 [KO:K02641] [EC:1.18.1.2]",
          542713L,                          "fnr1; ferredoxin NADP reductase 1 [KO:K02641] [EC:1.18.1.2]",
       100282643L,                   "ferredoxin--NADP reductase, leaf isozyme [KO:K02641] [EC:1.18.1.2]",
          542558L,                                         "L-FNRI; ferredoxin [KO:K02641] [EC:1.18.1.2]",
       103644035L,  "ferredoxin--NADP reductase, leaf isozyme 1, chloroplastic [KO:K02641] [EC:1.18.1.2]",
       100382419L,                                                            "cytochrome c6 [KO:K08906]",
          845171L,                                   "atpE; ATP synthase CF1 epsilon subunit [KO:K02114]",
          845170L,                 "atpB; ATP synthase CF1 beta subunit [KO:K02112] [EC:7.1.2.2 7.2.2.1]",
       100284505L,        "csu309(atpc); ATP synthase subunit gamma, chloroplastic precursor [KO:K02115]",
       103637166L,                                "ATP synthase subunit gamma, chloroplastic [KO:K02115]",
       103655982L,                           "ATP synthase subunit gamma, chloroplastic-like [KO:K02115]",
       103649250L,                           "ATP synthase subunit gamma, chloroplastic-like [KO:K02115]",
          845169L,                "atpA; ATP synthase CF1 alpha subunit [KO:K02111] [EC:7.1.2.2 7.2.2.1]",
       100281924L,                                                 "ATP synthase delta chain [KO:K02113]",
       103627748L,                                  "ATP synthase delta chain, chloroplastic [KO:K02113]",
       100282566L,                         "pco080036b; uncharacterized protein LOC100282566 [KO:K02109]",
          845172L,                                         "atpF; ATP synthase CF0 B subunit [KO:K02109]",
          845173L,                                         "atpH; ATP synthase CF0 C subunit [KO:K02110]",
          845174L,                                         "atpI; ATP synthase CF0 A subunit [KO:K02108]"
       )

#rename columns
colnames(KEGG.zma00195.tib) <- c("Entrez", "Description")
KEGG.zma00195.tib$Entrez <- as.character(KEGG.zma00195.tib$Entrez)
```
```{r zma00195 digenomic proteins}
KEGG.zma00195.tib.digenomic <- KEGG.zma00195.tib[c(1:65,86:98),]
KEGG.zma00195.tib.digenomic <- join(KEGG.zma00195.tib.digenomic, Entrez)
```







###KEGG - zma00196
```{r zma00196 tibble}
#Retrieved from KEGG on 02/26/2020
KEGG.zma00196.tib <- tibble::tribble(
                   ~V1,                                                                              ~V2,
            100281791L,                                 "chlorophyll a-b binding protein 6A [KO:K08907]",
            100283212L,                                 "chlorophyll a-b binding protein 6A [KO:K08907]",
            100273427L,      "Photosystem I chlorophyll a/b-binding protein 6 chloroplastic [KO:K08908]",
            100191715L,                                    "chlorophyll a-b binding protein [KO:K08908]",
            100282512L,                                    "chlorophyll a-b binding protein [KO:K08908]",
            100282214L,                                  "chlorophyll a-b binding protein 8 [KO:K08909]",
            100284001L,                       "gpm571; uncharacterized protein LOC100284001 [KO:K08910]",
            100280410L,                               "uncharacterized protein LOC100280410 [KO:K08910]",
            100281795L,                "csu1028(lhcb); uncharacterized protein LOC100281795 [KO:K08911]",
            100281248L,         "chlorophyll a-b binding protein 1, chloroplastic precursor [KO:K08912]",
            100193833L,              "lhcb7; chlorophyll a-b binding protein, chloroplastic [KO:K08912]",
            100282054L,                                  "chlorophyll a-b binding protein 2 [KO:K08912]",
               542530L, "lhcb3; chlorophyll a-b binding protein M9, chloroplastic precursor [KO:K08912]",
            100274453L,                  "light harvesting chlorophyll a/b binding protein1 [KO:K08912]",
            103651575L,                  "chlorophyll a-b binding protein 48, chloroplastic [KO:K08912]",
            103651577L,                    "chlorophyll a-b binding protein of LHCII type 1 [KO:K08912]",
            103651578L,                    "chlorophyll a-b binding protein of LHCII type 1 [KO:K08912]",
            100281879L,                                    "chlorophyll a-b binding protein [KO:K08913]",
            103643653L,                     "chlorophyll a-b binding protein, chloroplastic [KO:K08913]",
            100284154L,                  "chlorophyll a-b binding protein of LHCII type III [KO:K08914]",
            100216729L,                               "uncharacterized protein LOC100216729 [KO:K08915]",
               542478L,                                    "psb29; photosystem II subunit29 [KO:K08915]",
               542320L,         "Lhcb5-2; chlorophyll a/b-binding apoprotein CP26 precursor [KO:K08916]",
               542716L,          "lhcb5; light harvesting chlorophyll a/b binding protein 5 [KO:K08916]",
            100273752L,                               "uncharacterized protein LOC100273752 [KO:K08917]",
               542321L,          "lhcb6; light harvesting chlorophyll a/b binding protein 6 [KO:K08917]",
            109944964L,            "chlorophyll a-b binding protein CP24 10B, chloroplastic [KO:K08917]",
            113561367L,                               "chlorophyll a-b binding protein CP24 [KO:K08917]",
            100274085L,                  "si486069e02; uncharacterized protein LOC100274085 [KO:K14172]"
            )

#rename columns
colnames(KEGG.zma00196.tib) <- c("Entrez", "Description")
KEGG.zma00196.tib$Entrez <- as.character(KEGG.zma00196.tib$Entrez)
```
```{r zma00196 digenomic}
KEGG.zma00196.tib.digenomic <- join(KEGG.zma00196.tib, Entrez)
```


###CornCyc - Light Reactions
```{r CornCycLightRxns}
#Retrieved from CornCyc on 02/26/2020
CornCycLightRxns <- tibble::tribble(
                          ~Gene.ID,  ~Gene.Accession,       ~Gene.name,                               ~Reaction.id, ~Reaction.EC,                   ~Enzymatic.activity, ~Evidence,
                       "GDQC-8387", "Zm00001d010715", "Zm00001d010715",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-8546", "Zm00001d020002", "Zm00001d020002",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-8747", "Zm00001d000417", "Zm00001d000417",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9170", "Zm00001d039715", "Zm00001d039715",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9254", "Zm00001d011362", "Zm00001d011362",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9445", "Zm00001d008681", "Zm00001d008681",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9899", "Zm00001d043167", "Zm00001d043167",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-10138", "Zm00001d008625", "Zm00001d008625",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-10311", "Zm00001d041824", "Zm00001d041824",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-10950", "Zm00001d022464", "Zm00001d022464",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11782", "Zm00001d044217", "Zm00001d044217",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11860", "Zm00001d030638", "Zm00001d030638",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11887",  "GRMZM5G808939",  "GRMZM5G808939",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11985", "Zm00001d043299", "Zm00001d043299",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11996", "Zm00001d034153", "Zm00001d034153",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-12078", "Zm00001d000260", "Zm00001d000260",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-13214", "Zm00001d031484", "Zm00001d031484",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-14006", "Zm00001d039852", "Zm00001d039852",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-8316", "Zm00001d016134", "Zm00001d016134", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                       "GDQC-9168", "Zm00001d018162", "Zm00001d018162", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                      "GDQC-11523", "Zm00001d053432", "Zm00001d053432", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                      "GDQC-13682",  "GRMZM5G800780",  "GRMZM5G800780", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                       "GDQC-8046", "Zm00001d003767", "Zm00001d003767",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                       "GDQC-8673", "Zm00001d034283", "Zm00001d034283",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                       "GDQC-9216",  "GRMZM5G815553",  "GRMZM5G815553",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                      "GDQC-10703", "Zm00001d013146", "Zm00001d013146",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                      "GDQC-12161", "Zm00001d000272", "Zm00001d000272",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                       "GDQC-7635", "Zm00001d011833", "Zm00001d011833",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-7812", "Zm00001d023296", "Zm00001d023296",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-8192", "Zm00001d045575", "Zm00001d045575",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-8377", "Zm00001d051680", "Zm00001d051680",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-8916", "Zm00001d048452", "Zm00001d048452",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-10511", "Zm00001d005458", "Zm00001d005458",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-10877", "Zm00001d018794", "Zm00001d018794",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-10917", "Zm00001d034099", "Zm00001d034099",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-11276", "Zm00001d042049", "Zm00001d042049",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-11941", "Zm00001d039519", "Zm00001d039519",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-11960", "Zm00001d034345", "Zm00001d034345",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-12126", "Zm00001d051407", "Zm00001d051407",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-12735", "Zm00001d045574", "Zm00001d045574",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-13931", "Zm00001d036041", "Zm00001d036041",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-14028", "Zm00001d037231", "Zm00001d037231",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP"
                      )

```
```{r CornCycLightRxns digenomic}
CornCycLightRxns.digenomic <- CornCycLightRxns[CornCycLightRxns$Reaction.EC=="1.97.1.12"|CornCycLightRxns$Reaction.EC=="1.10.3.9",]
```




###NE Rubisco
```{r NE.Rubisco}
NE.Rubisco <- c("Zm00001d052595", "Zm00001d004894")
```

###Combine groups
```{r PhANGs}
Ph <- unique(na.omit(c(KEGG.zma00195.tib.digenomic$Accession, KEGG.zma00196.tib.digenomic$Accession, CornCycLightRxns.digenomic$Gene.Accession, NE.Rubisco)))

PhANGs <- Ph[grep("Zm", Ph)]
```

##Plastid background
```{r plastid}
#Import supplemental Table I from Friso et al. 2010
friso <- read.csv(paste0(PathToData,"Friso2010_SupTab1.csv"), stringsAsFactors = F)
colnames(friso) <- as.character(friso[1,])
friso <- friso[-1,]
friso <- friso[friso$`best gene model (selected)`=="yes",]
friso$Gene <- substr(friso$Accession, start = 1, stop = 13)
#convert to v4 accessions
friso <- merge.data.frame(friso, mgdb[,c(1,11)], by.x = "Gene", by.y = "v3_gene_model")
friso.v4 <- friso$v4_gene_model
#add previously called photosynthesis digenomic proteins
plastid <- unique(na.omit(c(friso.v4, Ph)))
```

##Nuclear-encoded plastid ribosome
```{r NE.PtRibo}
#grep plastid ribosomal proteins
ribosom <- tmt[(grep("ribosom", tmt$Protein)),]
ribosom.pt <- ribosom[(grep("30S|50S", ribosom$Protein)),]

# Read in Maize GAMER GO dataset, which lists maize V4 accessions with all associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim(paste0(PathToData, "maize.B73.AGPv4.aggregate.gaf"), sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$GOterm <- AnnotationDbi::Term(as.character(gamer$term_accession))
#merge gamer GO terms
ribosom.pt <- merge.data.frame(ribosom.pt[,c(176,183)], gamer[,c(2,18)], by.x = "Gene", by.y = "db_object_id", all.x = T, all.y = F)
#Convert duplicated rows into multiple columns for each GO term
ribosom.pt.w <- dcast(setDT(ribosom.pt), Accession~rowid(Accession, prefix="GOterm"), value.var="GOterm")
ribosom.pt <- merge.data.frame(ribosom.pt, ribosom.pt.w, by.x = "Accession", by.y = "Accession")
ribosom.pt <- distinct(ribosom.pt)
#Combine columns of each pathway into single column
ribosom.pt <- unite(ribosom.pt, "GOterm", GOterm1:GOterm9, sep = " // ")
ribosom.pt$GOterm <- gsub(ribosom.pt$GOterm, pattern = " // NA", replacement = "")
ribosom.pt <- distinct(ribosom.pt)

NE.PtRibo <- ribosom.pt$Gene[grep("chloropl|plastid", ribosom.pt$GOterm)]
```







##Cytosolic ribosome
```{r CytoRibo}
#Cytosolic ribosomal proteins (EnsemblPlants Annotation)
CytoRibo<-tmt[(grep("40S|60S", tmt$Protein)),]
#remove the mitochondrial and biogenesis proteins
CytoRibo<- CytoRibo$Gene[-(grep("mito|Mito|biogen|Biogen", CytoRibo$Protein))]
```
##Etyhlene Biosynthesis
```{r CornCycEtBio}
#Retrieved from CornCyc on 05/11/2020
CornCycEtBio <- tibble::tribble(
                      ~Gene.ID,  ~Gene.Accession,       ~Gene.name,       ~Reaction.id, ~Reaction.EC,                          ~Enzymatic.activity, ~Evidence,
                   "GDQC-8129", "Zm00001d024754", "Zm00001d024754", "S-ADENMETSYN-RXN",    "2.5.1.6",             "methionine adenosyltransferase", "EV-COMP",
                   "GDQC-9315", "Zm00001d009146", "Zm00001d009146", "S-ADENMETSYN-RXN",    "2.5.1.6",             "methionine adenosyltransferase", "EV-COMP",
                  "GDQC-10512", "Zm00001d008175", "Zm00001d008175", "S-ADENMETSYN-RXN",    "2.5.1.6",             "methionine adenosyltransferase", "EV-COMP",
                  "GDQC-11102", "Zm00001d010920", "Zm00001d010920", "S-ADENMETSYN-RXN",    "2.5.1.6",             "methionine adenosyltransferase", "EV-COMP",
                  "GDQC-14097", "Zm00001d018517", "Zm00001d018517", "S-ADENMETSYN-RXN",    "2.5.1.6",             "methionine adenosyltransferase", "EV-COMP",
                  "GDQC-14756", "Zm00001d040697", "Zm00001d040697", "S-ADENMETSYN-RXN",    "2.5.1.6",             "methionine adenosyltransferase", "EV-COMP",
                   "GDQC-9936", "Zm00001d026060", "Zm00001d026060",     "4.4.1.14-RXN",   "4.4.1.14", "1-aminocyclopropane-1-carboxylate synthase", "EV-COMP",
                  "GDQC-10643", "Zm00001d033862", "Zm00001d033862",     "4.4.1.14-RXN",   "4.4.1.14", "1-aminocyclopropane-1-carboxylate synthase", "EV-COMP",
                  "GDQC-11104", "Zm00001d045479", "Zm00001d045479",     "4.4.1.14-RXN",   "4.4.1.14", "1-aminocyclopropane-1-carboxylate synthase", "EV-COMP",
                  "GDQC-11257", "Zm00001d039487", "Zm00001d039487",     "4.4.1.14-RXN",   "4.4.1.14", "1-aminocyclopropane-1-carboxylate synthase", "EV-COMP",
                  "GDQC-12498", "Zm00001d002592", "Zm00001d002592",     "4.4.1.14-RXN",   "4.4.1.14", "1-aminocyclopropane-1-carboxylate synthase", "EV-COMP",
                   "GDQC-7692", "Zm00001d014572", "Zm00001d014572",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-7977", "Zm00001d024850", "Zm00001d024850",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-8083", "Zm00001d020686", "Zm00001d020686",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-8443", "Zm00001d042996", "Zm00001d042996",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-8648", "Zm00001d015860", "Zm00001d015860",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-9116", "Zm00001d011208", "Zm00001d011208",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-9304", "Zm00001d024853", "Zm00001d024853",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-9498", "Zm00001d024851", "Zm00001d024851",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-9503", "Zm00001d024843", "Zm00001d024843",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                   "GDQC-9749", "Zm00001d018211", "Zm00001d018211",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-11380", "Zm00001d004719", "Zm00001d004719",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-12095", "Zm00001d043503", "Zm00001d043503",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-12184", "Zm00001d024865", "Zm00001d024865",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13009", "Zm00001d005927", "Zm00001d005927",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13037", "Zm00001d024852", "Zm00001d024852",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13116", "Zm00001d000163", "Zm00001d000163",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13242", "Zm00001d052136", "Zm00001d052136",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13842", "Zm00001d048562", "Zm00001d048562",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13898", "Zm00001d046848", "Zm00001d046848",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-13947", "Zm00001d036955", "Zm00001d036955",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP",
                  "GDQC-14537", "Zm00001d004718", "Zm00001d004718",        "ETHYL-RXN",  "1.14.17.4",  "1-aminocyclopropane-1-carboxylate oxidase", "EV-COMP"
                  )

MAT <- CornCycEtBio$Gene.Accession[(CornCycEtBio$Enzymatic.activity=="methionine adenosyltransferase")]
ACS <- CornCycEtBio$Gene.Accession[(CornCycEtBio$Enzymatic.activity=="1-aminocyclopropane-1-carboxylate synthase")]
ACO <- CornCycEtBio$Gene.Accession[(CornCycEtBio$Enzymatic.activity=="1-aminocyclopropane-1-carboxylate oxidase")]
EtBio <- c(MAT,ACS,ACO)
```





##GOI list
```{r GOIs list}
GOIs <- list(PE.ribo, PhANGs, PhAPGs, plastid, NE.PtRibo, CytoRibo, EtBio, MAT, ACS, ACO)
names(GOIs) <- c("PE.ribo", "PhANGs", "PhAPGs", "plastid", "NE.PtRibo", "CytoRibo", "EtBio", "MAT", "ACS", "ACO")
saveRDS(GOIs, "GOIs")
GOIs <- readRDS("GOIs")
```



#Plant Height data
```{r}
PlantHeights.6H <- read.csv(paste0(PathToData, "6HPlantHeights.csv"), stringsAsFactors = F)
PlantHeights.RIL <- read.csv(paste0(PathToData, "RILPlantHeights.csv"), stringsAsFactors = F)

PlantHeights.6H$B73xMo17.MP <- PlantHeights.6H$B73xMo17/rowMeans(PlantHeights.6H[,c("B73", "Mo17")])
PlantHeights.6H$Mo17xB73.MP <- PlantHeights.6H$Mo17xB73/rowMeans(PlantHeights.6H[,c("B73", "Mo17")])
PlantHeights.6H$B84xB73.MP <- PlantHeights.6H$B84xB73/rowMeans(PlantHeights.6H[,c("B84", "B73")])
PlantHeights.6H$B84xMo17.MP <- PlantHeights.6H$B84xMo17/rowMeans(PlantHeights.6H[,c("B84", "Mo17")])
PlantHeights.6H$A682xB73.MP <- PlantHeights.6H$A682xB73/rowMeans(PlantHeights.6H[,c("A682", "B73")])
PlantHeights.6H$A682xMo17.MP <- PlantHeights.6H$A682xMo17/rowMeans(PlantHeights.6H[,c("A682", "Mo17")])
PlantHeights.6h.MP <- as.numeric(PlantHeights.6H[1,12:17])
PlantHeights.6h.MP.cob <- as.numeric(PlantHeights.6H[2,12:17])
PlantHeights.6H$B73xMo17.BxM <- PlantHeights.6H$B73xMo17/PlantHeights.6H$B73xMo17
PlantHeights.6H$Mo17xB73.BxM <- PlantHeights.6H$Mo17xB73/PlantHeights.6H$B73xMo17
PlantHeights.6H$B84xB73.BxM <- PlantHeights.6H$B84xB73/PlantHeights.6H$B73xMo17
PlantHeights.6H$B84xMo17.BxM <- PlantHeights.6H$B84xMo17/PlantHeights.6H$B73xMo17
PlantHeights.6H$A682xB73.BxM <- PlantHeights.6H$A682xB73/PlantHeights.6H$B73xMo17
PlantHeights.6H$A682xMo17.BxM <- PlantHeights.6H$A682xMo17/PlantHeights.6H$B73xMo17
PlantHeights.6h.BxM <- as.numeric(PlantHeights.6H[1,18:23])

PlantHeights.RIL$B73xMo17.MP <- PlantHeights.RIL$B73xMo17/rowMeans(PlantHeights.RIL[,c("B73", "Mo17")])
PlantHeights.RIL$M0014xB73.MP <- PlantHeights.RIL$M0014xB73/rowMeans(PlantHeights.RIL[,c("M0014", "B73")])
PlantHeights.RIL$M0014xMo17.MP <- PlantHeights.RIL$M0014xMo17/rowMeans(PlantHeights.RIL[,c("M0014", "Mo17")])
PlantHeights.RIL$M0016xB73.MP <- PlantHeights.RIL$M0016xB73/rowMeans(PlantHeights.RIL[,c("M0016", "B73")])
PlantHeights.RIL$M0016xMo17.MP <- PlantHeights.RIL$M0016xMo17/rowMeans(PlantHeights.RIL[,c("M0016", "Mo17")])
PlantHeights.RIL$M0021xB73.MP <- PlantHeights.RIL$M0021xB73/rowMeans(PlantHeights.RIL[,c("M0021", "B73")])
PlantHeights.RIL$M0021xMo17.MP <- PlantHeights.RIL$M0021xMo17/rowMeans(PlantHeights.RIL[,c("M0021", "Mo17")])
PlantHeights.RIL$M0317xB73.MP <- PlantHeights.RIL$M0317xB73/rowMeans(PlantHeights.RIL[,c("M0317", "B73")])
PlantHeights.RIL$M0317xMo17.MP <- PlantHeights.RIL$M0317xMo17/rowMeans(PlantHeights.RIL[,c("M0317", "Mo17")])
PlantHeights.ril.MP <- as.numeric(PlantHeights.RIL[1,16:24])
PlantHeights.RIL$B73xMo17.BxM <- PlantHeights.RIL$B73xMo17/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0014xB73.BxM <- PlantHeights.RIL$M0014xB73/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0014xMo17.BxM <- PlantHeights.RIL$M0014xMo17/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0016xB73.BxM <- PlantHeights.RIL$M0016xB73/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0016xMo17.BxM <- PlantHeights.RIL$M0016xMo17/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0021xB73.BxM <- PlantHeights.RIL$M0021xB73/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0021xMo17.BxM <- PlantHeights.RIL$M0021xMo17/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0317xB73.BxM <- PlantHeights.RIL$M0317xB73/PlantHeights.RIL$B73xMo17
PlantHeights.RIL$M0317xMo17.BxM <- PlantHeights.RIL$M0317xMo17/PlantHeights.RIL$B73xMo17
PlantHeights.ril.BxM <- as.numeric(PlantHeights.RIL[1,25:33])
```

#Define settings and functions for plotting
```{r CorFun}
CorFun <- function(x, xcols, y){
  for(i in 1:nrow(x)){
    v <- as.numeric(x[i,xcols])
    c <- cor(v, y, method = "pearson")
    x[i,"cor"]<- c
  }
  x
}
```
```{r get_legend}
get_legend<-function(myggplot){
  tmp <- ggplot_gtable(ggplot_build(myggplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}
```
```{r T-test function}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```
```{r plotting colors}
colors.p <- c("Other"="grey70", "Plastid"="black", "PE plastid ribosome"="#ef73ef", "NE plastid ribosome"="#a742da", "PhAPGs"="#7ae281", "PhANGs"="#009fa7", "Ethylene biosynthesis"="#fab5bd")

colors.v <- c("#000000","#004949","#009292","#ff6db6","#ffb6db",
 "#490092","#006ddb","#b66dff","#6db6ff","#b6dbff",
 "#920000","#924900","#db6d00","#24ff24","#ffff6d")

colors.e <- c("Other"="black","Methionine adenosyltransferase"="#24ff24", "ACC synthase"="#ff6db6", "ACC oxidase"="#920000")

colors.pe <- c("Plastid"="black", "PE plastid ribosome"="#ef73ef", "NE plastid ribosome"="#a742da", "PhAPGs"="#7ae281", "PhANGs"="#009fa7", "Ethylene biosynthesis"="#fab5bd")

colors.c <- c("Other"="black", "Cytosolic ribosome"="#ffa744", "NE plastid ribosome"="#ff0000", "PhANGs"="#3447ff")

colors.pc <- c("Other"="black", "PE plastid ribosome"="#ef73ef", "NE plastid ribosome"="#a742da", "PhAPGs"="#7ae281", "PhANGs"="#009fa7", "Cytosolic ribosome"="#fab5bd")
```
#Figure CV
```{r setup dataframe CV}
#Protein CV
##Calculate CV for BxM reps
tmt.SL$CV <- NA
for(i in 1:nrow(tmt.SL)){
  tmt.SL[i,"CV"] <- cv(as.numeric(tmt.SL[i,c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3")]))
}
tmt.6H$CV <- NA
for(i in 1:nrow(tmt.6H)){
  tmt.6H[i,"CV"] <- cv(as.numeric(tmt.6H[i,c("X6Hybrid.B73xMo17.r1", "X6Hybrid.B73xMo17.r2", "X6Hybrid.B73xMo17.r3")]))
}
tmt.RIL$CV <- NA
for(i in 1:nrow(tmt.RIL)){
  tmt.RIL[i,"CV"] <- cv(as.numeric(tmt.RIL[i,c("BM.1", "BM.2", "BM.3", "BM.4")]))
}
#RNA CV
##Calculate CV
cpm.SL.protein$CV <- NA
for(i in 1:nrow(cpm.SL.protein)){
  cpm.SL.protein[i,"CV"] <- cv(as.numeric(cpm.SL.protein[i,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")]))
}
cpm.6H.protein$CV <- NA
for(i in 1:nrow(cpm.6H.protein)){
  cpm.6H.protein[i,"CV"] <- cv(as.numeric(cpm.6H.protein[i,c("B73xMo17.2", "B73xMo17.3", "B73xMo17.4")]))
}
cpm.RIL.protein$CV <- NA
for(i in 1:nrow(cpm.RIL.protein)){
  cpm.RIL.protein[i,"CV"] <- cv(as.numeric(cpm.RIL.protein[i,c("BxM.1", "BxM.2", "BxM.3", "BxM.4")]))
}
#merge protein and RNA CVs
CV.SL.protein <- merge.data.frame(tmt.SL[,c("Gene", "CV")], cpm.SL.protein[,c("Gene", "CV")], by.x = "Gene", by.y = "Gene", all=F)
CV.6h.protein <- merge.data.frame(tmt.6H[,c("Gene", "CV")], cpm.6H.protein[,c("Gene", "CV")], by.x = "Gene", by.y = "Gene", all=F)
CV.ril.protein <- merge.data.frame(tmt.RIL[,c("Gene", "CV")], cpm.RIL.protein[,c("Gene", "CV")], by.x = "Gene", by.y = "Gene", all=F)
#Rename columns
colnames(CV.SL.protein) <- c("Gene", "Protein", "RNA")
colnames(CV.6h.protein) <- c("Gene", "Protein", "RNA")
colnames(CV.ril.protein) <- c("Gene", "Protein", "RNA")
#Define experiment
CV.SL.protein$Experiment <- "Original"
CV.6h.protein$Experiment <- "6-hybrid"
CV.ril.protein$Experiment <- "RIL"
#combine experiments into one dataframe
CV.protein <- rbind(CV.SL.protein,CV.6h.protein)
CV.protein <- rbind(CV.protein, CV.ril.protein)
CV.protein <- melt(CV.protein)
CV.protein <- na.omit(CV.protein)
```
```{r plot CV}
ggplot(CV.protein, mapping=aes(x=Experiment, y=value, fill=variable))+
  geom_violin(draw_quantiles = 0.5, )+
  scale_fill_manual(values=c("#af8dc3", "#7fbf7b"), name="")+
  theme_minimal()+
  ylab("Coefficient of variation (CV)")+
  theme(legend.position = c(0.09,0.88), legend.text = element_text(size=10), axis.text.x = element_text(size=10), axis.text.y = element_text(size=10), axis.title.x = element_text(size=12), axis.title.y = element_text(size=12))+
  theme(panel.grid.minor = element_blank())+
ggsave(paste0(PathToPlots, "CV.pdf"), width = 4.75, height = 4)
```
#Figure tmt.cpm.BxM.SL.plastid
```{r tmt.cpm.BxM.SL.plastid}
###Set limits
tmt.SL.lims <- tmt.SL
tmt.SL.lims$l2.B73xMo17.MP[tmt.SL.lims$l2.B73xMo17.MP <= -1] <- -1
tmt.SL.lims$l2.B73xMo17.MP[tmt.SL.lims$l2.B73xMo17.MP >= 1] <- 1
tmt.SL.lims$l2.B73.Mo17[tmt.SL.lims$l2.B73.Mo17 <= -1] <- -1
tmt.SL.lims$l2.B73.Mo17[tmt.SL.lims$l2.B73.Mo17 >= 1] <- 1
###Set GOIs
tmt.SL.lims.plastid <- tmt.SL.lims
tmt.SL.lims.plastid$Category <- "Other"
tmt.SL.lims.plastid$Order <- 1
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.SL.lims.plastid[(tmt.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6

###Re-order
tmt.SL.lims.plastid <- tmt.SL.lims.plastid[(order(tmt.SL.lims.plastid$Order)),]
###factor
tmt.SL.lims.plastid$Category <- factor(tmt.SL.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
###Scatterplot
tmt.SL.lims.plastid.scat <- ggplot(tmt.SL.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("A")+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(tmt.SL.lims.plastid.scat)
tmt.SL.lims.plastid.scat <- tmt.SL.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.SL.lims.plastid[,"l2.B73xMo17.MP"], color=tmt.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.SL.lims.plastid[,"l2.B73.Mo17"], color=tmt.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots tSLether
      A <- insert_xaxis_grob(tmt.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      A <- insert_yaxis_grob(A, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      ###Set limits
cpm.SL.lims <- cpm.SL.protein
cpm.SL.lims$l2.B73xMo17.MP[cpm.SL.lims$l2.B73xMo17.MP <= -1] <- -1
cpm.SL.lims$l2.B73xMo17.MP[cpm.SL.lims$l2.B73xMo17.MP >= 1] <- 1
cpm.SL.lims$l2.B73.Mo17[cpm.SL.lims$l2.B73.Mo17 <= -2] <- -2
cpm.SL.lims$l2.B73.Mo17[cpm.SL.lims$l2.B73.Mo17 >= 2] <- 2
###Set GOIs
cpm.SL.lims.plastid <- cpm.SL.lims
cpm.SL.lims.plastid$Category <- "Other"
cpm.SL.lims.plastid$Order <- 1
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
cpm.SL.lims.plastid[(cpm.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
cpm.SL.lims.plastid <- cpm.SL.lims.plastid[(order(cpm.SL.lims.plastid$Order)),]
###factor
cpm.SL.lims.plastid$Category <- factor(cpm.SL.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
###Scatterplot
cpm.SL.lims.plastid.scat <- ggplot(cpm.SL.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("B")+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(cpm.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(cpm.SL.lims.plastid[,"l2.B73xMo17.MP"], color=cpm.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.SL.lims.plastid[,"l2.B73.Mo17"], color=cpm.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots tSLether
      B <- insert_xaxis_grob(cpm.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      B <- insert_yaxis_grob(B, densityplotY, grid::unit(0.275, "null"), position = "right")

p.plotlist <- list(A, B, legend)
p.design <- "AB
             CC"
p.plotpanel<-patchwork::wrap_plots(A=p.plotlist[[1]], B=p.plotlist[[2]], C=p.plotlist[[3]],  nrow = 2, heights = c(3,1.2), design = p.design)
ggsave(plot=p.plotpanel, filename =paste0(PathToPlots, "tmt.cpm.BxM.SL.plastid.pdf"), dpi=300, width=4.75, height=3)
```
#Figure tmt.BxM.SL.plastid.all4
```{r setup dataframe tmt.BxM.SL.plastid.all4}
#copy
tmt.BxM.SL.OG <- tmt.SL
tmt.BxM.SL.6h <- tmt.6H
tmt.BxM.SL.ril <- tmt.RIL
tmt.BxM.SL.dn <- tmt.DN
#name plot panels
tmt.BxM.SL.OG$plot <- "A"
tmt.BxM.SL.6h$plot <- "B"
tmt.BxM.SL.ril$plot <- "C"
tmt.BxM.SL.dn$plot <- "D"
#trim
tmt.BxM.SL.OG <- tmt.BxM.SL.OG[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.SL.6h <- tmt.BxM.SL.6h[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.SL.ril <- tmt.BxM.SL.ril[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.SL.dn <- tmt.BxM.SL.dn[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
#bind
tmt.BxM.SL <- rbind(tmt.BxM.SL.OG,tmt.BxM.SL.6h)
tmt.BxM.SL <- rbind(tmt.BxM.SL,tmt.BxM.SL.ril)
tmt.BxM.SL <- rbind(tmt.BxM.SL,tmt.BxM.SL.dn)
```
```{r plot tmt.BxM.SL.plastid.all4}
 plot<-unique(tmt.BxM.SL[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.BxM.SL[,"plot"])
  for(i in 1:length(plot)){
    df<-tmt.BxM.SL[which(tmt.BxM.SL[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -.75] <- -.75
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= .75] <- .75

df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 <= -1] <- -1
df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims

df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
      xlim(c(-.75,.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 B73/Mo17")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -3.5, hjust = -0.485), axis.text.x = element_text(size = 9), axis.text.y = element_text(size = 9), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 3))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.B73xMo17.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.B73.Mo17"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
       des <- "AB
               CD
               EE"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], nrow = 3, heights = c(3,3,1.2), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "tmt.BxM.SL.plastid.all4.pdf"),
          dpi=300, width=4.75, height=5)  
```


```{r}
PlotAccession <- function(Accession){
#set limits
tmt.SL.lims <- tmt.SL
tmt.SL.lims$l2.B73xMo17.MP[tmt.SL.lims$l2.B73xMo17.MP <= -1] <- -1
tmt.SL.lims$l2.B73xMo17.MP[tmt.SL.lims$l2.B73xMo17.MP >= 1] <- 1
tmt.SL.lims$l2.B73.Mo17[tmt.SL.lims$l2.B73.Mo17 <= -1] <- -1
tmt.SL.lims$l2.B73.Mo17[tmt.SL.lims$l2.B73.Mo17 >= 1] <- 1
#Set GOIs
tmt.SL.lims$Category <- "Other"
tmt.SL.lims$Order <- 1
tmt.SL.lims[(tmt.SL.lims$Accession %in% Accession),"Category"] <- "Accession"
tmt.SL.lims[(tmt.SL.lims$Accession %in% Accession),"Order"] <- 2
#Re-order
tmt.SL.lims <- tmt.SL.lims[(order(tmt.SL.lims$Order)),]
#factor
tmt.SL.lims$Category <- factor(tmt.SL.lims$Category, levels = c("Other", "Accession"))

#scatter plot with marginal density curves
    tmt.SL.lims.scat <- ggplot(tmt.SL.lims, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=3.5)+
  scale_color_manual(values=c("black", "red"))+
      scale_shape_manual(values=c(1,19))+
    theme_classic()+
      xlim(c(-1,1))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 B73/Mo17")+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10),
        legend.position="none", axis.text.x = element_text(size = 15), axis.text.y = element_text(size = 15), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 3))
    tmt.SL.lims.scat
}
```



#Figure DayVsNight
```{r setup dataframe DayVsNight}
###Set limits
tmt.DN.both.lims <- tmt.DN.both
tmt.DN.both.lims$day.l2.B73xMo17.MP[tmt.DN.both.lims$day.l2.B73xMo17.MP <= -0.75] <- -0.75
tmt.DN.both.lims$day.l2.B73xMo17.MP[tmt.DN.both.lims$day.l2.B73xMo17.MP >= 0.75] <- 0.75
tmt.DN.both.lims$day.l2.B73.Mo17[tmt.DN.both.lims$day.l2.B73.Mo17 <= -0.75] <- -0.75
tmt.DN.both.lims$day.l2.B73.Mo17[tmt.DN.both.lims$day.l2.B73.Mo17 >= 0.75] <- 0.75

tmt.DN.both.lims$night.l2.B73xMo17.MP[tmt.DN.both.lims$night.l2.B73xMo17.MP <= -0.75] <- -0.75
tmt.DN.both.lims$night.l2.B73xMo17.MP[tmt.DN.both.lims$night.l2.B73xMo17.MP >= 0.75] <- 0.75
tmt.DN.both.lims$night.l2.B73.Mo17[tmt.DN.both.lims$night.l2.B73.Mo17 <= -0.75] <- -0.75
tmt.DN.both.lims$night.l2.B73.Mo17[tmt.DN.both.lims$night.l2.B73.Mo17 >= 0.75] <- 0.75
###Set GOIs
tmt.DN.both.lims.plastid <- tmt.DN.both.lims
tmt.DN.both.lims.plastid$Category <- "Other"
tmt.DN.both.lims.plastid$Order <- 1
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.DN.both.lims.plastid[(tmt.DN.both.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
tmt.DN.both.lims.plastid <- tmt.DN.both.lims.plastid[(order(tmt.DN.both.lims.plastid$Order)),]
###remove "other"
tmt.DN.both.lims.plastid <- tmt.DN.both.lims.plastid[!tmt.DN.both.lims.plastid$Category=="Other",]
###factor
tmt.DN.both.lims.plastid$Category <- factor(tmt.DN.both.lims.plastid$Category, levels = c("Plastid", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
```
```{r plot DayVsNight}
###Scatterplot
tmt.DN.both.BM.MP.lims.plastid.scat <- ggplot(tmt.DN.both.lims.plastid, mapping=aes(x=day.l2.B73xMo17.MP, y=night.l2.B73xMo17.MP, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
 theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          xlim(c(-0.75,0.75))+
          ylim(c(-0.75,0.75))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          geom_abline(slope = 1, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485))+
          guides(col=guide_legend(ncol=2))+
    xlab("log2 day BxM/MP")+
    ylab("log2 night BxM/MP")+
    ggtitle("A")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
dn.plastid.legend <- get_legend(tmt.DN.both.BM.MP.lims.plastid.scat)
tmt.DN.both.BM.MP.lims.plastid.scat <- tmt.DN.both.BM.MP.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.DN.both.BM.MP.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.DN.both.lims.plastid[,"day.l2.B73xMo17.MP"], color=tmt.DN.both.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.DN.both.BM.MP.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.DN.both.lims.plastid[,"night.l2.B73xMo17.MP"], color=tmt.DN.both.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      tmt.DN.both.BM.MP.lims.plastid.scat <- insert_xaxis_grob(tmt.DN.both.BM.MP.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      tmt.DN.both.BM.MP.lims.plastid.scat <- insert_yaxis_grob(tmt.DN.both.BM.MP.lims.plastid.scat, densityplotY, grid::unit(0.275, "null"), position = "right")
      
tmt.DN.both.B73.Mo17.lims.plastid.scat <- ggplot(tmt.DN.both.lims.plastid, mapping=aes(x=day.l2.B73.Mo17, y=night.l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          xlim(c(-0.75,0.75))+
          ylim(c(-0.75,0.75))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          geom_abline(slope = 1, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485))+
          guides(col=guide_legend(nrow=3))+
    xlab("log2 day B73/Mo17")+
    ylab("log2 night B73/Mo17")+
    ggtitle("B")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(tmt.DN.both.B73.Mo17.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.DN.both.lims.plastid[,"day.l2.B73.Mo17"], color=tmt.DN.both.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.DN.both.B73.Mo17.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.DN.both.lims.plastid[,"night.l2.B73.Mo17"], color=tmt.DN.both.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      tmt.DN.both.B73.Mo17.lims.plastid.scat <- insert_xaxis_grob(tmt.DN.both.B73.Mo17.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      tmt.DN.both.B73.Mo17.lims.plastid.scat <- insert_yaxis_grob(tmt.DN.both.B73.Mo17.lims.plastid.scat, densityplotY, grid::unit(0.275, "null"), position = "right")

dn.plotlist <- list(tmt.DN.both.BM.MP.lims.plastid.scat, tmt.DN.both.B73.Mo17.lims.plastid.scat, dn.plastid.legend)
dn.design <- "AB
              CC"
dn.plotpanel<-patchwork::wrap_plots(A=dn.plotlist[[1]], B=dn.plotlist[[2]], C=dn.plotlist[[3]],  nrow = 2, heights = c(3,1.2), design = dn.design)
ggsave(plot=dn.plotpanel, filename =paste0(PathToPlots, "DayVsNight.pdf"), dpi=300, width=4.75, height=3)
```
#Figure cpm.BxM.SL.plastid.all4
```{r setup dataframe cpm.BxM.SL.plastid.all4}
cpm.BxM.SL.OG <- cpm.SL.protein
cpm.BxM.SL.6h <- cpm.6H.protein
cpm.BxM.SL.ril <- cpm.RIL.protein

cpm.BxM.SL.OG$plot <- "A"
cpm.BxM.SL.6h$plot <- "B"
cpm.BxM.SL.ril$plot <- "C"

cpm.BxM.SL.OG <- cpm.BxM.SL.OG[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.BxM.SL.6h <- cpm.BxM.SL.6h[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.BxM.SL.ril <- cpm.BxM.SL.ril[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]

cpm.BxM.SL <- rbind(cpm.BxM.SL.OG,cpm.BxM.SL.6h)
cpm.BxM.SL <- rbind(cpm.BxM.SL,cpm.BxM.SL.ril)
```
```{r plot cpm.BxM.SL.plastid.all4}
plot<-unique(cpm.BxM.SL[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(cpm.BxM.SL[,"plot"])
  for(i in 1:length(plot)){
    df<-cpm.BxM.SL[which(cpm.BxM.SL[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -1] <- -1
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= 1] <- 1

df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 <= -2] <- -2
df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 >= 2] <- 2
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4

###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid", "PhANGs", "NE plastid ribosome"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
      xlim(c(-1,1))+
      ylim(c(-2,2))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 B73/Mo17")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -3.5, hjust = -0.485), axis.text.x = element_text(size = 9), axis.text.y = element_text(size = 9), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 1))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.B73xMo17.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.B73.Mo17"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
        des <- "AB
                C#
                EE"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], E=plotList[[4]], nrow = 3, heights = c(3,3,1.2), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "cpm.BxM.SL.plastid.all4.pdf"),
          dpi=300, width=4.75, height=5)  
```
#Figure tmt.6H.plastid
```{r setup dataframe tmt.6H.plastid}
#subset hybrids
tmt.6h.b73xmo17 <- tmt.6H[,c("Gene","l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.6h.mo17xb73 <- tmt.6H[,c("Gene", "l2.Mo17xB73.MP", "l2.Mo17.B73" )]
tmt.6h.b84xb73 <- tmt.6H[,c("Gene", "l2.B84xB73.MP", "l2.B84.B73")]
tmt.6h.b84xmo17 <- tmt.6H[,c("Gene", "l2.B84xMo17.MP", "l2.B84.B73")]
tmt.6h.a682xb73 <- tmt.6H[,c("Gene", "l2.A682xB73.MP", "l2.A682.B73")]
tmt.6h.a682xmo17<- tmt.6H[,c("Gene", "l2.A682xMo17.MP", "l2.A682.Mo17")]
#rename columns
colnames(tmt.6h.b73xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.mo17xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.b84xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.b84xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.a682xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.a682xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
#name plot panels
tmt.6h.b73xmo17$plot <- "A"
tmt.6h.mo17xb73$plot <- "B"
tmt.6h.b84xb73$plot <- "C"
tmt.6h.a682xmo17$plot <- "D"
tmt.6h.a682xb73$plot <- "E"
tmt.6h.b84xmo17$plot <- "F"
#combine
tmt.6h.all6 <- rbind(tmt.6h.b73xmo17, tmt.6h.mo17xb73, tmt.6h.b84xb73, tmt.6h.a682xmo17, tmt.6h.a682xb73, tmt.6h.b84xmo17)
```
```{r plot tmt.6H.plastid}
  plot<-unique(tmt.6h.all6[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.6h.all6[,"plot"])
  for(i in 1:length(plot)){
    df<-tmt.6h.all6[which(tmt.6h.all6[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP <= -0.75] <- -0.75
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP >= 0.75] <- 0.75

df.lims$l2.parent.parent[df.lims$l2.parent.parent <= -1] <- -1
df.lims$l2.parent.parent[df.lims$l2.parent.parent >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.HYB.MP, y=l2.parent.parent, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
      xlim(c(-0.75,0.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -3.5, hjust = -0.485), axis.text.x = element_text(size = 9), axis.text.y = element_text(size = 9), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 3))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
    if(i==1){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B73xMo17/MP")+ ylab("log2 B73/Mo17")
    }
    else if(i==2){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 Mo17xB73/MP")+ ylab("log2 Mo17/B73")
    }
    else if(i==3){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xB73/MP")+ ylab("log2 B84/B73")
    }
    else if(i==4){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xMo17/MP")+ ylab("log2 A682/Mo17")
    }
    else if(i==5){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xB73/MP")+ ylab("log2 A682/B73")
    }
    else if(i==6){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xMo17/MP")+ ylab("log2 B84/Mo17")
    }
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.HYB.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.parent.parent"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  des <- "AB
          CD
          EG
          HH"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], G=plotList[[6]], H=plotList[[7]], nrow = 4, heights = c(3,3,3,1.5), design = des)
  
ggsave(plot=r, filename =paste0(PathToPlots, "tmt.6H.plastid.pdf"),
          dpi=300, width=4.75, height=6.5)  
```
#Figure cpm.6H.plastid
```{r setup dataframe cpm.6H.plastid}
#subset to different hybrids
cpm.6H.bm <- cpm.6H.protein[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.6H.mb <- cpm.6H.protein[,c("Gene", "l2.Mo17xB73.MP", "l2.Mo17.B73")]
cpm.6H.a682xb <- cpm.6H.protein[,c("Gene", "l2.A682xB73.MP", "l2.A682.B73")]
cpm.6H.a682xm <- cpm.6H.protein[,c("Gene", "l2.A682xMo17.MP", "l2.A682.Mo17")]
cpm.6H.b84xb <- cpm.6H.protein[,c("Gene", "l2.B84xB73.MP", "l2.B84.B73")]
cpm.6H.b84xm <- cpm.6H.protein[,c("Gene", "l2.B84xB73.MP", "l2.B84.B73")]
#rename columns to match
colnames(cpm.6H.bm)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6H.mb)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6H.a682xb)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6H.a682xm)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6H.b84xb)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6H.b84xm)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
#name plots
cpm.6H.bm$plot <- "A"
cpm.6H.mb$plot <- "B"
cpm.6H.b84xb$plot <- "C"
cpm.6H.a682xm$plot <- "D"
cpm.6H.a682xb$plot <- "E"
cpm.6H.b84xm$plot <- "F"
#rbind
cpm.6H.plot <- rbind(cpm.6H.bm,cpm.6H.mb,cpm.6H.b84xb,cpm.6H.a682xm,cpm.6H.a682xb,cpm.6H.b84xm)
```
```{r plot cpm.6H.plastid}
 plot<-unique(cpm.6H.plot[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(cpm.6H.plot[,"plot"])
  for(i in 1:length(plot)){
    df<-cpm.6H.plot[which(cpm.6H.plot[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP <= -2] <- -2
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP >= 2] <- 2

df.lims$l2.parent.parent[df.lims$l2.parent.parent <= -2] <- -2
df.lims$l2.parent.parent[df.lims$l2.parent.parent >= 2] <- 2
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid", "PhANGs", "NE plastid ribosome"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.HYB.MP, y=l2.parent.parent, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
      xlim(c(-2,2))+
      ylim(c(-2,2))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -3.5, hjust = -0.485), axis.text.x = element_text(size = 9), axis.text.y = element_text(size = 9), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 1))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
    if(i==1){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B73xMo17/MP")+ ylab("log2 B73/Mo17")
    }
    else if(i==2){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 Mo17xB73/MP")+ ylab("log2 Mo17/B73")
    }
    else if(i==3){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xB73/MP")+ ylab("log2 B84/B73")
    }
    else if(i==4){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xMo17/MP")+ ylab("log2 A682/Mo17")
    }
    else if(i==5){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xB73/MP")+ ylab("log2 A682/B73")
    }
    else if(i==6){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xMo17/MP")+ ylab("log2 B84/Mo17")
    }
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.HYB.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.parent.parent"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  des <- "AB
          CD
          EG
          HH"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], G=plotList[[6]], H=plotList[[7]], nrow = 4, heights = c(3,3,3,1.5), design = des)
  
ggsave(plot=r, filename =paste0(PathToPlots, "cpm.6H.plastid.pdf"),
          dpi=300, width=4.75, height=6.5)  
```
#Figure tmt.PlastidRiboCor.Hyb2BxM
```{r setup dataframe tmt.PlastidRiboCor.Hyb2BxM}
#subset RIL hybrids
tmt.RIL.Hyb2BxM <- tmt.RIL[,c(6,104:126)]
tmt.RIL.Hyb2BxM$BM.BM.BM <- tmt.RIL.Hyb2BxM$BM.BM/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0014.14B.BM <- tmt.RIL.Hyb2BxM$M0014.14B/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0014.14M.BM <- tmt.RIL.Hyb2BxM$M0014.14M/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0016.16B.BM <- tmt.RIL.Hyb2BxM$M0016.16B/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0016.16M.BM <- tmt.RIL.Hyb2BxM$M0016.16M/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0021.21B.BM <- tmt.RIL.Hyb2BxM$M0021.21B/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0021.21M.BM <- tmt.RIL.Hyb2BxM$M0021.21M/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0317.317B.BM <- tmt.RIL.Hyb2BxM$M0317.317B/tmt.RIL.Hyb2BxM$BM.BM
tmt.RIL.Hyb2BxM$M0317.317M.BM <- tmt.RIL.Hyb2BxM$M0317.317M/tmt.RIL.Hyb2BxM$BM.BM
#subset 6-hybrid
tmt.6H.Hyb2BxM <- tmt.6H[,c(6,44:53)]
tmt.6H.Hyb2BxM$B73xMo17.BM <- tmt.6H.Hyb2BxM$B73xMo17/tmt.6H.Hyb2BxM$B73xMo17
tmt.6H.Hyb2BxM$Mo17xB73.BM <- tmt.6H.Hyb2BxM$Mo17xB73/tmt.6H.Hyb2BxM$B73xMo17
tmt.6H.Hyb2BxM$B84xB73.BM <- tmt.6H.Hyb2BxM$B84xB73/tmt.6H.Hyb2BxM$B73xMo17
tmt.6H.Hyb2BxM$B84xMo17.BM <- tmt.6H.Hyb2BxM$B84xMo17/tmt.6H.Hyb2BxM$B73xMo17
tmt.6H.Hyb2BxM$A682xB73.BM <- tmt.6H.Hyb2BxM$A682xB73/tmt.6H.Hyb2BxM$B73xMo17
tmt.6H.Hyb2BxM$A682xMo17.BM <- tmt.6H.Hyb2BxM$A682xMo17/tmt.6H.Hyb2BxM$B73xMo17
#combine RIL and 6 hybrids
tmt.RIL.6H.Hyb2BxM <- merge.data.frame(tmt.RIL.Hyb2BxM[,c(1,25:33)], tmt.6H.Hyb2BxM[,c(1,12:17)], by.x = "Accession", by.y = "Accession", all = F)
#Make gene column to call genes of interest
tmt.RIL.6H.Hyb2BxM$Gene <- substr(tmt.RIL.6H.Hyb2BxM$Accession, start = 1, stop = 14)
#trim to plastid ribosomal proteins only
tmt.PlastidRiboCor.Hyb2BxM <- tmt.RIL.6H.Hyb2BxM[tmt.RIL.6H.Hyb2BxM$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo),]
#Remove gene column
tmt.PlastidRiboCor.Hyb2BxM <- tmt.PlastidRiboCor.Hyb2BxM[,-17]
#Re-name columns to identify experiment
colnames(tmt.PlastidRiboCor.Hyb2BxM)[2:16] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17", "6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
n <- nrow(tmt.PlastidRiboCor.Hyb2BxM)
#melt
tmt.PlastidRiboCor.Hyb2BxM <- melt(tmt.PlastidRiboCor.Hyb2BxM)
#Add hyb/BxM plant height ratios
tmt.PlastidRiboCor.Hyb2BxM.height <- c(PlantHeights.ril.BxM, PlantHeights.6h.BxM)
tmt.PlastidRiboCor.Hyb2BxM$height <- c(rep(tmt.PlastidRiboCor.Hyb2BxM.height[1], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[2], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[3], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[4], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[5], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[6], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[7], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[8], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[9], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[10], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[11], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[12], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[13], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[14], n), rep(tmt.PlastidRiboCor.Hyb2BxM.height[15], n))
#Remove missing values
tmt.PlastidRiboCor.Hyb2BxM <- na.omit(tmt.PlastidRiboCor.Hyb2BxM)
#Calcualte linear model
tmt.PlastidRiboCor.Hyb2BxM.split <- split.data.frame(tmt.PlastidRiboCor.Hyb2BxM, tmt.PlastidRiboCor.Hyb2BxM$variable)
tmt.PlastidRiboCor.Hyb2BxM.split.avg <- data.frame("hybrid"=unique(tmt.PlastidRiboCor.Hyb2BxM$variable), "value"=NA, "height"=tmt.PlastidRiboCor.Hyb2BxM.height)
for(i in 1:15){
  tmt.PlastidRiboCor.Hyb2BxM.split.avg[i,"value"] <- mean(tmt.PlastidRiboCor.Hyb2BxM.split[[i]][,3])
}
tmt.PlastidRiboCor.Hyb2BxM.split.avg.lm <- lm(value~height, data = tmt.PlastidRiboCor.Hyb2BxM.split.avg)
tmt.PlastidRiboCor.Hyb2BxM.split.avg.lm.rsquared <- summary(tmt.PlastidRiboCor.Hyb2BxM.split.avg.lm)$r.squared
```
```{r plot tmt.PlastidRiboCor.Hyb2BxM}
tmt.PlastidRiboCor.Hyb2BxM.plot <- ggplot(data=tmt.PlastidRiboCor.Hyb2BxM, mapping = aes(x=height, y=value, fill=variable))+
  geom_violin(width=0.02, alpha=0.7, position = "identity")+
  scale_fill_manual(values = colors.v, name="Hybrid")+
  theme_minimal()+
  xlab("Hybrid/BxM plant height")+
  ylab("Hybrid/BxM plastid ribosomal protein abundance")+
  ggtitle(bquote(R^2 ~ "=" ~ .(tmt.PlastidRiboCor.Hyb2BxM.split.avg.lm.rsquared)))+
  theme(text = element_text(size = 10), plot.title = element_text(size=10, vjust = -8))+
  geom_abline(slope = tmt.PlastidRiboCor.Hyb2BxM.split.avg.lm$coefficients[2], intercept = tmt.PlastidRiboCor.Hyb2BxM.split.avg.lm$coefficients[1], linetype="dashed")+
  stat_summary(fun.data=mean_cl_normal, show.legend = F)
ggsave(plot = tmt.PlastidRiboCor.Hyb2BxM.plot, paste0(PathToPlots, "tmt.PlastidRiboCor.Hyb2BxM.pdf"), width = 4.75, height = 4.5)
```
#Figure tmt.PlastidRiboCor.Hyb2MP
```{r setup dataframe tmt.PlastidRiboCor.Hyb2MP}
#subset RIL hybrids
tmt.RIL.Hyb2MP <- tmt.RIL[,c(6,127:135)]
#subset 6-hybrid
tmt.6H.Hyb2MP <- tmt.6H[,c(6,54:59)]
#combine RIL and 6 hybrids
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL.Hyb2MP, tmt.6H.Hyb2MP, by.x = "Accession", by.y = "Accession", all.x = T, all.y = F)
#Make gene column to call genes of interest
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#trim to plastid ribosomal proteins only
tmt.PlastidRiboCor.Hyb2MP <- tmt.RIL.6H.Hyb2MP[tmt.RIL.6H.Hyb2MP$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo),]
#Remove gene column
tmt.PlastidRiboCor.Hyb2MP <- tmt.PlastidRiboCor.Hyb2MP[,-17]
#Rename columns
colnames(tmt.PlastidRiboCor.Hyb2MP)[2:16] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17", "6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
#get number of proteins
n <- nrow(tmt.PlastidRiboCor.Hyb2MP)
#melt
tmt.PlastidRiboCor.Hyb2MP <- melt(tmt.PlastidRiboCor.Hyb2MP)
#Add hybrid/midparent plant height
PlantHeights.ril.6h.MP <- c(PlantHeights.ril.MP, PlantHeights.6h.MP)
tmt.PlastidRiboCor.Hyb2MP$height <- c(rep(PlantHeights.ril.6h.MP[1], n), rep(PlantHeights.ril.6h.MP[2], n), rep(PlantHeights.ril.6h.MP[3], n), rep(PlantHeights.ril.6h.MP[4], n), rep(PlantHeights.ril.6h.MP[5], n), rep(PlantHeights.ril.6h.MP[6], n), rep(PlantHeights.ril.6h.MP[7], n), rep(PlantHeights.ril.6h.MP[8], n), rep(PlantHeights.ril.6h.MP[9], n), rep(PlantHeights.ril.6h.MP[10], n), rep(PlantHeights.ril.6h.MP[11], n), rep(PlantHeights.ril.6h.MP[12], n), rep(PlantHeights.ril.6h.MP[13], n), rep(PlantHeights.ril.6h.MP[14], n), rep(PlantHeights.ril.6h.MP[15], n))
#remove missing values
tmt.PlastidRiboCor.Hyb2MP <- na.omit(tmt.PlastidRiboCor.Hyb2MP)
#calculate averages and linear model
tmt.PlastidRiboCor.Hyb2MP.split <- split.data.frame(tmt.PlastidRiboCor.Hyb2MP, tmt.PlastidRiboCor.Hyb2MP$variable)
tmt.PlastidRiboCor.Hyb2MP.split.avg <- data.frame("hybrid"=unique(tmt.PlastidRiboCor.Hyb2MP$variable), "value"=NA, "height"=unique(tmt.PlastidRiboCor.Hyb2MP$height))
for(i in 1:15){
  tmt.PlastidRiboCor.Hyb2MP.split.avg[i,"value"] <- mean(tmt.PlastidRiboCor.Hyb2MP.split[[i]][,3])
}
tmt.PlastidRiboCor.Hyb2MP.split.avg.lm <- lm(value~height, data = tmt.PlastidRiboCor.Hyb2MP.split.avg)
tmt.PlastidRiboCor.Hyb2MP.split.avg.lm.rsquared <- summary(tmt.PlastidRiboCor.Hyb2MP.split.avg.lm)$r.squared
```
```{r plot tmt.PlastidRiboCor.Hyb2MP}
tmt.PlastidRiboCor.Hyb2MP.plot <- ggplot(data=tmt.PlastidRiboCor.Hyb2MP, mapping = aes(x=height, y=value, fill=variable))+
  geom_violin(width=0.02, alpha=0.7, position = "identity")+
  scale_fill_manual(values = colors.v, name="Hybrid")+
  theme_minimal()+
  xlab("Hybrid/mid-parent plant height")+
  ylab("Hybrid/mid-parent plastid ribosomal protein abundance")+
  ggtitle(bquote(R^2 ~ "=" ~ .(tmt.PlastidRiboCor.Hyb2MP.split.avg.lm.rsquared)))+
  theme(text = element_text(size = 10), plot.title = element_text(size=10, vjust = -8))+
  geom_abline(slope = tmt.PlastidRiboCor.Hyb2MP.split.avg.lm$coefficients[2], intercept = tmt.PlastidRiboCor.Hyb2MP.split.avg.lm$coefficients[1], linetype="dashed")+
  stat_summary(fun.data=mean_cl_normal, show.legend = F)
ggsave(plot = tmt.PlastidRiboCor.Hyb2MP.plot, paste0(PathToPlots, "tmt.PlastidRiboCor.Hyb2MP.pdf"), width = 4.75, height = 4.5)
```


#Figure cpm.PlastidRiboCor.hyb2MP
```{r setup dataframe cpm.PlastidRiboCor.hyb2MP}
#combine RIL and 6 hybrids
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL.protein[,c(1,76:84)], cpm.6H.protein[,c(1,42:47)], by.x = "Gene", by.y = "Gene", all = T)
#limit to plastid ribosomal proteins
cpm.PlastidRiboCor.hyb2MP <- cpm.RIL.6H.Hyb2MP[cpm.RIL.6H.Hyb2MP$Gene %in% GOIs$NE.PtRibo,]
#re-name columns
colnames(cpm.PlastidRiboCor.hyb2MP)[2:16] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17", "6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
#get number of proteins
n <- nrow(cpm.PlastidRiboCor.hyb2MP)
#melt
cpm.PlastidRiboCor.hyb2MP <- melt(cpm.PlastidRiboCor.hyb2MP)
#add hybrid/mid-parent plant height
PlantHeights.ril.6h.MP <- c(PlantHeights.ril.MP, PlantHeights.6h.MP)
cpm.PlastidRiboCor.hyb2MP$height <- c(rep(PlantHeights.ril.6h.MP[1], n), rep(PlantHeights.ril.6h.MP[2], n), rep(PlantHeights.ril.6h.MP[3], n), rep(PlantHeights.ril.6h.MP[4], n), rep(PlantHeights.ril.6h.MP[5], n), rep(PlantHeights.ril.6h.MP[6], n), rep(PlantHeights.ril.6h.MP[7], n), rep(PlantHeights.ril.6h.MP[8], n), rep(PlantHeights.ril.6h.MP[9], n), rep(PlantHeights.ril.6h.MP[10], n), rep(PlantHeights.ril.6h.MP[11], n), rep(PlantHeights.ril.6h.MP[12], n), rep(PlantHeights.ril.6h.MP[13], n), rep(PlantHeights.ril.6h.MP[14], n), rep(PlantHeights.ril.6h.MP[15], n))
#remove missing values
cpm.PlastidRiboCor.hyb2MP <- na.omit(cpm.PlastidRiboCor.hyb2MP)
#calculate averages and linear model
cpm.PlastidRiboCor.hyb2MP.split <- split.data.frame(cpm.PlastidRiboCor.hyb2MP, cpm.PlastidRiboCor.hyb2MP$variable)
cpm.PlastidRiboCor.hyb2MP.split.avg <- data.frame("hybrid"=unique(cpm.PlastidRiboCor.hyb2MP$variable), "value"=NA, "height"=unique(cpm.PlastidRiboCor.hyb2MP$height))
for(i in 1:15){
  cpm.PlastidRiboCor.hyb2MP.split.avg[i,"value"] <- mean(cpm.PlastidRiboCor.hyb2MP.split[[i]][,3])
}
cpm.PlastidRiboCor.hyb2MP.split.avg.lm <- lm(value~height, data = cpm.PlastidRiboCor.hyb2MP.split.avg)
cpm.PlastidRiboCor.hyb2MP.split.avg.lm.rsquared <- summary(cpm.PlastidRiboCor.hyb2MP.split.avg.lm)$r.squared
```
```{r plot cpm.PlastidRiboCor.hyb2MP}
cpm.PlastidRiboCor.hyb2MP.plot <- ggplot(data=cpm.PlastidRiboCor.hyb2MP, mapping = aes(x=height, y=value, fill=variable))+
  geom_violin(width=0.02, alpha=0.7, position = "identity")+
  scale_fill_manual(values = colors.v, name="Hybrid")+
  theme_minimal()+
  xlab("Hybrid/mid-parent plant height")+
  ylab("Hybrid/mid-parent plastid ribosomal mRNA abundance")+
  labs(subtitle = bquote(R^2 ~ "=" ~ .(cpm.PlastidRiboCor.hyb2MP.split.avg.lm.rsquared)), title = "A")+
  theme(text = element_text(size = 10), plot.title = element_text(size = 14, hjust = -0.1))+
  geom_abline(slope = cpm.PlastidRiboCor.hyb2MP.split.avg.lm$coefficients[2], intercept = cpm.PlastidRiboCor.hyb2MP.split.avg.lm$coefficients[1], linetype="dashed")+
  stat_summary(fun.data=mean_cl_normal, show.legend = F)
ggsave(plot=cpm.PlastidRiboCor.hyb2MP.plot, filename =paste0(PathToPlots, "cpm.PlastidRiboCor.hyb2MP.pdf"), dpi=300, width = 4.75, height = 4.5) 
```

#Figure tmt.cpm.RIL.6H.Hyb2MP.cor


```{r setup dataframe tmt.cpm.RIL.6H.Hyb2MP.cor}
#combine RIL and 6 hybrids RNA
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL[,c(1,76:84)], cpm.6H[,c(1,42:47)], by.x = "Gene", by.y = "Gene", all = T)
#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[,c(6,127:135)], tmt.6H[,c(6,54:59)], by.x = "Accession", by.y = "Accession", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Accession <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
colnames(tmt.RIL.6H.Hyb2MP)[1] <- "Gene"
#run correlation function
cpm.RIL.6H.Hyb2MP.cor <- CorFun(cpm.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#merge RIL and 6 hybrids
tmt.cpm.RIL.6H.Hyb2MP.cor <- merge.data.frame(tmt.RIL.6H.Hyb2MP.cor[,c(1,17)], cpm.RIL.6H.Hyb2MP.cor[,c(1,17)], by.x = "Gene", by.y = "Gene", all=T)
#Set GOIs
tmt.cpm.RIL.6H.Hyb2MP.cor$Category <- "Other"
tmt.cpm.RIL.6H.Hyb2MP.cor$Order <- 1
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 4
#Re-order
tmt.cpm.RIL.6H.Hyb2MP.cor <- tmt.cpm.RIL.6H.Hyb2MP.cor[(order(tmt.cpm.RIL.6H.Hyb2MP.cor$Order)),]
#factor
tmt.cpm.RIL.6H.Hyb2MP.cor$Category <- factor(tmt.cpm.RIL.6H.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PhANGs"))
```

```{r plot tmt.cpm.RIL.6H.Hyb2MP.cor}
tmt.cpm.RIL.6H.Hyb2MP.cor.plot <- ggplot(tmt.cpm.RIL.6H.Hyb2MP.cor, mapping=aes(x=cor.x, y=cor.y, color=Category, shape=Category))+
    geom_point(size=2.5)+
    scale_colour_manual(values = colors.c)+
  scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
      xlim(-1,1)+
      ylim(-1,1)+
      geom_vline(xintercept=0, linetype="dashed", color = "gray20", size=0.7)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray20", size=0.7)+
      geom_vline(xintercept=0.5, linetype="dashed", color = "gray40", size=0.5)+
      geom_hline(yintercept=0.5, linetype="dashed", color = "gray40", size=0.5)+
      geom_vline(xintercept= -0.5, linetype="dashed", color = "gray40", size=0.5)+
      geom_hline(yintercept= -0.5, linetype="dashed", color = "gray40", size=0.5)+
      xlab("Protein correlation")+
      ylab("RNA correlation")+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      ggtitle("B")+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size = 14, hjust = -0.2))+
          guides(col=guide_legend(nrow=2))

###Add density curves
    densityplotX<-axis_canvas(tmt.cpm.RIL.6H.Hyb2MP.cor.plot, axis="x")+
      geom_density(aes(tmt.cpm.RIL.6H.Hyb2MP.cor[,"cor.x"], color=tmt.cpm.RIL.6H.Hyb2MP.cor[,"Category"]), size=1)+
    scale_color_manual(values=colors.c, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray20", size=0.7)+
      geom_vline(xintercept=0.5, linetype="dashed", color = "gray40", size=0.5)+
      geom_vline(xintercept= -0.5, linetype="dashed", color = "gray40", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      xlim(-1,1)+
      coord_cartesian(expand = F)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.cpm.RIL.6H.Hyb2MP.cor.plot, axis="y", coord_flip = T)+
      geom_density(aes(tmt.cpm.RIL.6H.Hyb2MP.cor[,"cor.y"], color=tmt.cpm.RIL.6H.Hyb2MP.cor[,"Category"]), size=1)+
      scale_color_manual(values=colors.c, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray20", size=0.7)+
      geom_vline(xintercept=0.5, linetype="dashed", color = "gray40", size=0.5)+
      geom_vline(xintercept= -0.5, linetype="dashed", color = "gray40", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      xlim(-1,1)+
      coord_flip(expand = F)
    #put two plots together
      tmt.cpm.RIL.6H.Hyb2MP.cor.plot <- insert_xaxis_grob(tmt.cpm.RIL.6H.Hyb2MP.cor.plot, densityplotX, grid::unit(0.275, "null"), position = "top")
      tmt.cpm.RIL.6H.Hyb2MP.cor.plot <- insert_yaxis_grob(tmt.cpm.RIL.6H.Hyb2MP.cor.plot, densityplotY, grid::unit(0.275, "null"), position = "right")
ggsave(plot=tmt.cpm.RIL.6H.Hyb2MP.cor.plot, filename =paste0(PathToPlots, "tmt.cpm.RIL.6H.Hyb2MP.cor.pdf"),
          dpi=300, width = 4.75, height = 6)  
```
#Figure Cor Densities
```{r}
tmt.RIL.Hyb2MP <- tmt.RIL[,c(6,127:135,9)]
tmt.6H.Hyb2MP <-tmt.6H[,c(6,54:59, 9)]
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[,c(6,127:135)], tmt.6H[,c(6,54:59)], by.x = "Accession", by.y = "Accession", all = F)

tmt.RIL.Hyb2MP.cor <- CorFun(tmt.RIL.Hyb2MP, 2:10, PlantHeights.ril.MP)
tmt.6H.Hyb2MP.cor <- CorFun(tmt.6H.Hyb2MP, 2:7, PlantHeights.6h.MP)
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
tmt.RIL.6H.Hyb2MP.cor$Gene <- substr(tmt.RIL.6H.Hyb2MP.cor$Accession, start=0, stop=14)
```
```{r}
cpm.RIL.Hyb2MP <- cpm.RIL[,c(1,76:84)]
cpm.6H.Hyb2MP <- cpm.6H[,c(1,42:47)]
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL[,c(1,76:84)], cpm.6H[,c(1,42:47)], by.x = "Gene", by.y = "Gene", all = T)

cpm.RIL.Hyb2MP.cor <- CorFun(cpm.RIL.Hyb2MP, 2:10, PlantHeights.ril.MP)
cpm.6H.Hyb2MP.cor <- CorFun(cpm.6H.Hyb2MP, 2:7, PlantHeights.6h.MP)
cpm.RIL.6H.Hyb2MP.cor <- CorFun(cpm.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
```


```{r}
#6H
tmt.6H.Hyb2MP.cor$Category <- "Other"
tmt.6H.Hyb2MP.cor$Order <- 1
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.6H.Hyb2MP.cor[(tmt.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
tmt.6H.Hyb2MP.cor <- tmt.6H.Hyb2MP.cor[(order(tmt.6H.Hyb2MP.cor$Order)),]
tmt.6H.Hyb2MP.cor$Category <- factor(tmt.6H.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
tmt.6H.Hyb2MP.cor.plot <- ggplot(tmt.6H.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)+
  ggtitle("C")
ggsave(filename =paste0(PathToPlots, "tmt.6H.Hyb2MP.cor.densities.pdf"), dpi=300, width=4.75, height=3.5)

#RIL
tmt.RIL.Hyb2MP.cor$Category <- "Other"
tmt.RIL.Hyb2MP.cor$Order <- 1
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.RIL.Hyb2MP.cor[(tmt.RIL.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
tmt.RIL.Hyb2MP.cor <- tmt.RIL.Hyb2MP.cor[(order(tmt.RIL.Hyb2MP.cor$Order)),]
tmt.RIL.Hyb2MP.cor$Category <- factor(tmt.RIL.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
tmt.RIL.Hyb2MP.cor.plot <- ggplot(tmt.RIL.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)+
  ggtitle("B")
ggsave(filename =paste0(PathToPlots, "tmt.RIL.Hyb2MP.cor.densities.pdf"), dpi=300, width=4.75, height=3.5)

#RIL+6H
tmt.RIL.6H.Hyb2MP.cor$Category <- "Other"
tmt.RIL.6H.Hyb2MP.cor$Order <- 1
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.RIL.6H.Hyb2MP.cor[(tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
tmt.RIL.6H.Hyb2MP.cor <- tmt.RIL.6H.Hyb2MP.cor[(order(tmt.RIL.6H.Hyb2MP.cor$Order)),]
tmt.RIL.6H.Hyb2MP.cor$Category <- factor(tmt.RIL.6H.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
tmt.RIL.6H.Hyb2MP.cor.plot <- ggplot(tmt.RIL.6H.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)+
  ggtitle("A")
ggsave(filename =paste0(PathToPlots, "tmt.RIL.6H.Hyb2MP.cor.densities.pdf"), dpi=300, width=4.75, height=3.5)

        des <- "A
                B
                C"
tmt.RIL.6H.Hyb2MP.cor.densities.panel<-patchwork::wrap_plots(A=tmt.RIL.6H.Hyb2MP.cor.plot, B=tmt.RIL.Hyb2MP.cor.plot, C=tmt.6H.Hyb2MP.cor.plot,  nrow = 3, heights = c(1,1,1), design = des)
ggsave(plot=tmt.RIL.6H.Hyb2MP.cor.densities.panel, filename =paste0(PathToPlots, "tmt.RIL.6H.Hyb2MP.cor.densities.panel.pdf"), dpi=300, width=4.75, height=7)
```

```{r}
#6H
cpm.6H.Hyb2MP.cor$Category <- "Other"
cpm.6H.Hyb2MP.cor$Order <- 1
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
cpm.6H.Hyb2MP.cor[(cpm.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
cpm.6H.Hyb2MP.cor <- cpm.6H.Hyb2MP.cor[(order(cpm.6H.Hyb2MP.cor$Order)),]
cpm.6H.Hyb2MP.cor$Category <- factor(cpm.6H.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
cpm.6H.Hyb2MP.cor.plot <- ggplot(cpm.6H.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)
  ggtitle("C")
ggsave(filename =paste0(PathToPlots, "cpm.6H.Hyb2MP.cor.densities.pdf"), dpi=300, width=4.75, height=3.5)
```

```{r}
#RIL
cpm.RIL.Hyb2MP.cor$Category <- "Other"
cpm.RIL.Hyb2MP.cor$Order <- 1
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
cpm.RIL.Hyb2MP.cor[(cpm.RIL.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
cpm.RIL.Hyb2MP.cor <- cpm.RIL.Hyb2MP.cor[(order(cpm.RIL.Hyb2MP.cor$Order)),]
cpm.RIL.Hyb2MP.cor$Category <- factor(cpm.RIL.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
cpm.RIL.Hyb2MP.cor.plot <- ggplot(cpm.RIL.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)+
  ggtitle("B")
ggsave(filename =paste0(PathToPlots, "cpm.RIL.Hyb2MP.cor.densities.pdf"), dpi=300, width=4.75, height=3.5)

#RIL+6H
cpm.RIL.6H.Hyb2MP.cor$Category <- "Other"
cpm.RIL.6H.Hyb2MP.cor$Order <- 1
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
cpm.RIL.6H.Hyb2MP.cor[(cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
cpm.RIL.6H.Hyb2MP.cor <- cpm.RIL.6H.Hyb2MP.cor[(order(cpm.RIL.6H.Hyb2MP.cor$Order)),]
cpm.RIL.6H.Hyb2MP.cor$Category <- factor(cpm.RIL.6H.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
cpm.RIL.6H.Hyb2MP.cor.plot <- ggplot(cpm.RIL.6H.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)+
  ggtitle("A")
ggsave(filename =paste0(PathToPlots, "cpm.RIL.6H.Hyb2MP.cor.densities.pdf"), dpi=300, width=4.75, height=3.5)

        des <- "A
                B
                C"
cpm.RIL.6H.Hyb2MP.cor.densities.panel<-patchwork::wrap_plots(A=cpm.RIL.6H.Hyb2MP.cor.plot, B=cpm.RIL.Hyb2MP.cor.plot, C=cpm.6H.Hyb2MP.cor.plot,  nrow = 3, heights = c(1,1,1), design = des)
ggsave(plot=cpm.RIL.6H.Hyb2MP.cor.densities.panel, filename =paste0(PathToPlots, "cpm.RIL.6H.Hyb2MP.cor.densities.panel.pdf"), dpi=300, width=4.75, height=7)
```
##Figure S8
```{r}
  des <- "AB"
  S8<-patchwork::wrap_plots(A=cpm.PlastidRiboCor.hyb2MP.plot, B=tmt.cpm.RIL.6H.Hyb2MP.cor.plot, nrow = 1, widths = c(1,1), design = des)
  ggsave(plot=S8, filename =paste0(PathToPlots, "S8.pdf"),
          dpi=300, width=10, height=5)
```
#Figure tmt.SL.EtBio
```{r setup dataframe tmt.SL.EtBio}
###Set limits
tmt.SL.lims <- tmt.SL
tmt.SL.lims$l2.B73xMo17.MP[tmt.SL.lims$l2.B73xMo17.MP <= -1] <- -1
tmt.SL.lims$l2.B73xMo17.MP[tmt.SL.lims$l2.B73xMo17.MP >= 1] <- 1
tmt.SL.lims$l2.B73.Mo17[tmt.SL.lims$l2.B73.Mo17 <= -1] <- -1
tmt.SL.lims$l2.B73.Mo17[tmt.SL.lims$l2.B73.Mo17 >= 1] <- 1
###Set GOIs
tmt.SL.lims.etbio <- tmt.SL.lims
tmt.SL.lims.etbio$Category <- "Other"
tmt.SL.lims.etbio$Order <- 1
tmt.SL.lims.etbio[(tmt.SL.lims.etbio$Gene %in% GOIs$MAT),"Category"] <- "Methionine adenosyltransferase"
tmt.SL.lims.etbio[(tmt.SL.lims.etbio$Gene %in% GOIs$MAT),"Order"] <- 2
tmt.SL.lims.etbio[(tmt.SL.lims.etbio$Gene %in% GOIs$ACS),"Category"] <- "ACC synthase"
tmt.SL.lims.etbio[(tmt.SL.lims.etbio$Gene %in% GOIs$ACS),"Order"] <- 3
tmt.SL.lims.etbio[(tmt.SL.lims.etbio$Gene %in% GOIs$ACO),"Category"] <- "ACC oxidase"
tmt.SL.lims.etbio[(tmt.SL.lims.etbio$Gene %in% GOIs$ACO),"Order"] <- 4
###Re-order
tmt.SL.lims.etbio <- tmt.SL.lims.etbio[(order(tmt.SL.lims.etbio$Order)),]
###factor
tmt.SL.lims.etbio$Category <- factor(tmt.SL.lims.etbio$Category, levels = c("Other", "Methionine adenosyltransferase", "ACC synthase", "ACC oxidase"))
```
```{r setup dataframe cpm.SL.EtBio}
###Set limits
cpm.SL.lims <- cpm.SL.protein
cpm.SL.lims$l2.B73xMo17.MP[cpm.SL.lims$l2.B73xMo17.MP <= -2] <- -2
cpm.SL.lims$l2.B73xMo17.MP[cpm.SL.lims$l2.B73xMo17.MP >= 2] <- 2
cpm.SL.lims$l2.B73.Mo17[cpm.SL.lims$l2.B73.Mo17 <= -2] <- -2
cpm.SL.lims$l2.B73.Mo17[cpm.SL.lims$l2.B73.Mo17 >= 2] <- 2
###Set GOIs
cpm.SL.lims.etbio <- cpm.SL.lims
cpm.SL.lims.etbio$Category <- "Other"
cpm.SL.lims.etbio$Order <- 1
cpm.SL.lims.etbio[(cpm.SL.lims.etbio$Gene %in% GOIs$MAT),"Category"] <- "Methionine adenosyltransferase"
cpm.SL.lims.etbio[(cpm.SL.lims.etbio$Gene %in% GOIs$MAT),"Order"] <- 2
cpm.SL.lims.etbio[(cpm.SL.lims.etbio$Gene %in% GOIs$ACS),"Category"] <- "ACC synthase"
cpm.SL.lims.etbio[(cpm.SL.lims.etbio$Gene %in% GOIs$ACS),"Order"] <- 3
cpm.SL.lims.etbio[(cpm.SL.lims.etbio$Gene %in% GOIs$ACO),"Category"] <- "ACC oxidase"
cpm.SL.lims.etbio[(cpm.SL.lims.etbio$Gene %in% GOIs$ACO),"Order"] <- 4
###Re-order
cpm.SL.lims.etbio <- cpm.SL.lims.etbio[(order(cpm.SL.lims.etbio$Order)),]
###factor
cpm.SL.lims.etbio$Category <- factor(cpm.SL.lims.etbio$Category, levels = c("Other", "Methionine adenosyltransferase", "ACC synthase", "ACC oxidase"))
```
```{r plot tmt.cpm.SL.EtBio}
#Scatterplot - tmt
tmt.SL.lims.etbio.scat <- ggplot(tmt.SL.lims.etbio, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.e, name="Category")+
  scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          ggtitle("A")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"),  plot.title = element_text(vjust = -3.5, hjust = -0.485))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.SL.lims.etbio.scat.legend <- get_legend(tmt.SL.lims.etbio.scat)
tmt.SL.lims.etbio.scat <- tmt.SL.lims.etbio.scat+theme(legend.position = "none")
#Add density curves
    densityplotX<-axis_canvas(tmt.SL.lims.etbio.scat, axis="x")+
      geom_density(aes(tmt.SL.lims.etbio[,"l2.B73xMo17.MP"], color=tmt.SL.lims.etbio[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.e, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.SL.lims.etbio.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.SL.lims.etbio[,"l2.B73.Mo17"], color=tmt.SL.lims.etbio[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.e, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))+
      coord_flip()
    #put two plots together
      tmt.SL.lims.etbio.scat <- insert_xaxis_grob(tmt.SL.lims.etbio.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      tmt.SL.lims.etbio.scat <- insert_yaxis_grob(tmt.SL.lims.etbio.scat, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      
#Scatterplot - cpm
cpm.SL.lims.etbio.scat <- ggplot(cpm.SL.lims.etbio, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.e, name="Category")+
  scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          ggtitle("B")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"),  plot.title = element_text(vjust = -3.5, hjust = -0.485))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
#Add density curves
    densityplotX<-axis_canvas(cpm.SL.lims.etbio.scat, axis="x")+
      geom_density(aes(cpm.SL.lims.etbio[,"l2.B73xMo17.MP"], color=cpm.SL.lims.etbio[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.e, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.SL.lims.etbio.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.SL.lims.etbio[,"l2.B73.Mo17"], color=cpm.SL.lims.etbio[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.e, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))+
      coord_flip()
    #put two plots together
      cpm.SL.lims.etbio.scat <- insert_xaxis_grob(cpm.SL.lims.etbio.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      cpm.SL.lims.etbio.scat <- insert_yaxis_grob(cpm.SL.lims.etbio.scat, densityplotY, grid::unit(0.275, "null"), position = "right")
      
#Combine plots into panel
      des <- "AB
              CC"
tmt.cpm.SL.lims.etbio.scat<-patchwork::wrap_plots(A=tmt.SL.lims.etbio.scat, B=cpm.SL.lims.etbio.scat, C=tmt.SL.lims.etbio.scat.legend,  nrow = 2, heights = c(3,1.2), design = des)
ggsave(plot=tmt.cpm.SL.lims.etbio.scat, filename =paste0(PathToPlots, "tmt.cpm.SL.etbio.pdf"), dpi=300, width=4.75, height=3)
```
#Figure tmt.SL.EtBio.vol
```{r setup dataframe tmt.SL.EtBio}
###Set GOIs
tmt.SL.etbio <- tmt.SL
tmt.SL.etbio$l2.B73xMo17.MP[tmt.SL.etbio$l2.B73xMo17.MP <= -1] <- -1
tmt.SL.etbio$l2.B73xMo17.MP[tmt.SL.etbio$l2.B73xMo17.MP >= 1] <- 1
tmt.SL.etbio$MP1 <- rowMeans(tmt.SL.etbio[,c("SeedlingLeaf.B73.r1","SeedlingLeaf.Mo17.r1")])
tmt.SL.etbio$MP2 <- rowMeans(tmt.SL.etbio[,c("SeedlingLeaf.B73.r2","SeedlingLeaf.Mo17.r2")])
tmt.SL.etbio$MP3 <- rowMeans(tmt.SL.etbio[,c("SeedlingLeaf.B73.r3","SeedlingLeaf.Mo17.r3")])
tmt.SL.etbio$ttest <- ttestFun(tmt.SL.etbio, 27:29, 16:18)
tmt.SL.etbio$l10.ttest <- -(log10(tmt.SL.etbio$ttest))
tmt.SL.etbio$Category <- "Other"
tmt.SL.etbio$Order <- 1
tmt.SL.etbio[(tmt.SL.etbio$Gene %in% GOIs$MAT),"Category"] <- "Methionine adenosyltransferase"
tmt.SL.etbio[(tmt.SL.etbio$Gene %in% GOIs$MAT),"Order"] <- 2
tmt.SL.etbio[(tmt.SL.etbio$Gene %in% GOIs$ACS),"Category"] <- "ACC synthase"
tmt.SL.etbio[(tmt.SL.etbio$Gene %in% GOIs$ACS),"Order"] <- 3
tmt.SL.etbio[(tmt.SL.etbio$Gene %in% GOIs$ACO),"Category"] <- "ACC oxidase"
tmt.SL.etbio[(tmt.SL.etbio$Gene %in% GOIs$ACO),"Order"] <- 4
###Re-order
tmt.SL.etbio <- tmt.SL.etbio[(order(tmt.SL.etbio$Order)),]
###factor
tmt.SL.etbio$Category <- factor(tmt.SL.etbio$Category, levels = c("Other", "Methionine adenosyltransferase", "ACC synthase", "ACC oxidase"))
```
```{r setup dataframe cpm.SL.EtBio}
###Set GOIs
cpm.SL.etbio <- cpm.SL
cpm.SL.etbio$l2.B73xMo17.MP[cpm.SL.etbio$l2.B73xMo17.MP <= -2.5] <- -2.5
cpm.SL.etbio$l2.B73xMo17.MP[cpm.SL.etbio$l2.B73xMo17.MP >= 2.5] <- 2.5
cpm.SL.etbio$MP1 <- rowMeans(cpm.SL.etbio[,c("B73.1","Mo17.1")])
cpm.SL.etbio$MP2 <- rowMeans(cpm.SL.etbio[,c("B73.2","Mo17.2")])
cpm.SL.etbio$MP3 <- rowMeans(cpm.SL.etbio[,c("B73.3","Mo17.3")])
cpm.SL.etbio$ttest <- ttestFun(cpm.SL.etbio, 18:20, 8:10)
cpm.SL.etbio$l10.ttest <- -(log10(cpm.SL.etbio$ttest))
cpm.SL.etbio$Category <- "Other"
cpm.SL.etbio$Order <- 1
cpm.SL.etbio[(cpm.SL.etbio$Gene %in% GOIs$MAT),"Category"] <- "Methionine adenosyltransferase"
cpm.SL.etbio[(cpm.SL.etbio$Gene %in% GOIs$MAT),"Order"] <- 2
cpm.SL.etbio[(cpm.SL.etbio$Gene %in% GOIs$ACS),"Category"] <- "ACC synthase"
cpm.SL.etbio[(cpm.SL.etbio$Gene %in% GOIs$ACS),"Order"] <- 3
cpm.SL.etbio[(cpm.SL.etbio$Gene %in% GOIs$ACO),"Category"] <- "ACC oxidase"
cpm.SL.etbio[(cpm.SL.etbio$Gene %in% GOIs$ACO),"Order"] <- 4
###Re-order
cpm.SL.etbio <- cpm.SL.etbio[(order(cpm.SL.etbio$Order)),]
###factor
cpm.SL.etbio$Category <- factor(cpm.SL.etbio$Category, levels = c("Other", "Methionine adenosyltransferase", "ACC synthase", "ACC oxidase"))
```
```{r plot tmt.cpm.SL.EtBio}
#Scatterplot - tmt
tmt.SL.etbio.vol <- ggplot(tmt.SL.etbio, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=2)+
  scale_color_manual(values=colors.e, name="Category")+
  scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
          ggtitle("A")+
          geom_hline(yintercept=1.30103, linetype="dashed", color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"),  plot.title = element_text(vjust = -0.14, hjust = -0.075))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.SL.etbio.vol.legend <- get_legend(tmt.SL.etbio.vol)
tmt.SL.etbio.vol <- tmt.SL.etbio.vol+theme(legend.position = "none")
      
      
#volterplot - cpm
cpm.SL.etbio.vol <- ggplot(cpm.SL.etbio, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=2)+
  scale_color_manual(values=colors.e, name="Category")+
  scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
          ggtitle("B")+
          geom_hline(yintercept=1.30103, linetype="dashed", color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"),  plot.title = element_text(vjust = -0.14, hjust = -0.075))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
      
#Combine plots into panel
      des <- "AB
              CC"
tmt.cpm.SL.etbio.vol<-patchwork::wrap_plots(A=tmt.SL.etbio.vol, B=cpm.SL.etbio.vol, C=tmt.SL.etbio.vol.legend,  nrow = 2, heights = c(3,1.2), design = des)
ggsave(plot=tmt.cpm.SL.etbio.vol, filename =paste0(PathToPlots, "tmt.cpm.SL.etbio.volcano.pdf"), dpi=300, width=4.75, height=3)
```
```{r}
colnames(tmt.RIL)
```
```{r}
###Set GOIs
tmt.RIL.GOIs <- tmt.RIL
tmt.RIL.GOIs$l2.B73xMo17.MP[tmt.RIL.GOIs$l2.B73xMo17.MP <= -1] <- -1
tmt.RIL.GOIs$l2.B73xMo17.MP[tmt.RIL.GOIs$l2.B73xMo17.MP >= 1] <- 1
tmt.RIL.GOIs$MP1 <- rowMeans(tmt.RIL.GOIs[,c("B73.1","Mo17.1")])
tmt.RIL.GOIs$MP2 <- rowMeans(tmt.RIL.GOIs[,c("B73.3","Mo17.3")])
tmt.RIL.GOIs$MP3 <- rowMeans(tmt.RIL.GOIs[,c("B73.4","Mo17.4")])
tmt.RIL.GOIs$ttest <- ttestFun(tmt.RIL.GOIs, 159:161, 16:18)
tmt.RIL.GOIs$l10.ttest <- -(log10(tmt.RIL.GOIs$ttest))
tmt.RIL.GOIs$Category <- "Other"
tmt.RIL.GOIs$Order <- 1
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$PhANGs),"Order"] <- 3
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$PhAPGs),"Order"] <- 4
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$NE.PtRibo),"Order"] <- 5
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.RIL.GOIs[(tmt.RIL.GOIs$Gene %in% GOIs$PE.ribo),"Order"] <- 6
###Re-order
tmt.RIL.GOIs <- tmt.RIL.GOIs[(order(tmt.RIL.GOIs$Order)),]
###factor
tmt.RIL.GOIs$Category <- factor(tmt.RIL.GOIs$Category, levels = c("Other", "Plastid", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
#Scatterplot - tmt
tmt.RIL.GOIs.vol <- ggplot(tmt.RIL.GOIs, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=3)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
          geom_hline(yintercept=1.30103, linetype="dashed", size=1.5, color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed",size=1.5, color = "gray50")+
          theme(axis.text=element_text(size=15), legend.text=element_text(size=15),axis.title = element_text(size=15),
                legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -0.14, hjust = -0.075))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.RIL.GOIs.vol.legend <- get_legend(tmt.RIL.GOIs.vol)
tmt.RIL.GOIs.vol <- tmt.RIL.GOIs.vol+ theme(legend.position = "none")
ggsave(plot = tmt.RIL.GOIs.vol, filename = "tmt.RIL.GOIs.vol.png", height=4, width=4)
```
```{r}
###Set GOIs
tmt.SL.GOIs <- tmt.SL
tmt.SL.GOIs$l2.B73xMo17.MP[tmt.SL.GOIs$l2.B73xMo17.MP <= -1] <- -1
tmt.SL.GOIs$l2.B73xMo17.MP[tmt.SL.GOIs$l2.B73xMo17.MP >= 1] <- 1
tmt.SL.GOIs$MP1 <- rowMeans(tmt.SL.GOIs[,c("SeedlingLeaf.B73.r1","SeedlingLeaf.Mo17.r1")])
tmt.SL.GOIs$MP2 <- rowMeans(tmt.SL.GOIs[,c("SeedlingLeaf.B73.r2","SeedlingLeaf.Mo17.r2")])
tmt.SL.GOIs$MP3 <- rowMeans(tmt.SL.GOIs[,c("SeedlingLeaf.B73.r3","SeedlingLeaf.Mo17.r3")])
tmt.SL.GOIs$ttest <- ttestFun(tmt.SL.GOIs, 27:29, 16:18)
tmt.SL.GOIs$l10.ttest <- -(log10(tmt.SL.GOIs$ttest))
tmt.SL.GOIs$Category <- "Other"
tmt.SL.GOIs$Order <- 1
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$PhANGs),"Order"] <- 3
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$PhAPGs),"Order"] <- 4
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$NE.PtRibo),"Order"] <- 5
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.SL.GOIs[(tmt.SL.GOIs$Gene %in% GOIs$PE.ribo),"Order"] <- 6
###Re-order
tmt.SL.GOIs <- tmt.SL.GOIs[(order(tmt.SL.GOIs$Order)),]
###factor
tmt.SL.GOIs$Category <- factor(tmt.SL.GOIs$Category, levels = c("Other", "Plastid", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
#Scatterplot - tmt
tmt.SL.GOIs.vol <- ggplot(tmt.SL.GOIs, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=3)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
          geom_hline(yintercept=1.30103, linetype="dashed", size=1.5,color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed",size=1.5, color = "gray50")+
          theme(axis.text=element_text(size=15), legend.text=element_text(size=15),axis.title = element_text(size=15),
                legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -0.14, hjust = -0.075))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.SL.GOIs.vol.legend <- get_legend(tmt.SL.GOIs.vol)
tmt.SL.GOIs.vol <- tmt.SL.GOIs.vol+ theme(legend.position = "none")
ggsave(plot = tmt.SL.GOIs.vol, filename = "tmt.SL.GOIs.vol.png", height=4, width=4)
```

```{r}
###Set GOIs
tmt.6H.GOIs <- tmt.6H
tmt.6H.GOIs$l2.B73xMo17.MP[tmt.6H.GOIs$l2.B73xMo17.MP <= -1] <- -1
tmt.6H.GOIs$l2.B73xMo17.MP[tmt.6H.GOIs$l2.B73xMo17.MP >= 1] <- 1
tmt.6H.GOIs$MP1 <- rowMeans(tmt.6H.GOIs[,c("X6Hybrid.B73.r1","X6Hybrid.Mo17.r1")])
tmt.6H.GOIs$MP2 <- rowMeans(tmt.6H.GOIs[,c("X6Hybrid.B73.r2","X6Hybrid.Mo17.r2")])
tmt.6H.GOIs$MP3 <- rowMeans(tmt.6H.GOIs[,c("X6Hybrid.B73.r3","X6Hybrid.Mo17.r3")])
tmt.6H.GOIs$ttest <- ttestFun(tmt.6H.GOIs, 74:76, c(14,24,34))
tmt.6H.GOIs$l10.ttest <- -(log10(tmt.6H.GOIs$ttest))
tmt.6H.GOIs$Category <- "Other"
tmt.6H.GOIs$Order <- 1
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$PhANGs),"Order"] <- 3
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$PhAPGs),"Order"] <- 4
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$NE.PtRibo),"Order"] <- 5
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.6H.GOIs[(tmt.6H.GOIs$Gene %in% GOIs$PE.ribo),"Order"] <- 6
###Re-order
tmt.6H.GOIs <- tmt.6H.GOIs[(order(tmt.6H.GOIs$Order)),]
###factor
tmt.6H.GOIs$Category <- factor(tmt.6H.GOIs$Category, levels = c("Other", "Plastid", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
#Scatterplot - tmt
tmt.6H.GOIs.vol <- ggplot(tmt.6H.GOIs, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=3)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
          geom_hline(yintercept=1.30103, linetype="dashed", color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=15), legend.text=element_text(size=15),axis.title = element_text(size=15),
                legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -0.14, hjust = -0.075))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.6H.GOIs.vol.legend <- get_legend(tmt.6H.GOIs.vol)
tmt.6H.GOIs.vol <- tmt.6H.GOIs.vol+ theme(legend.position = "none")
ggsave(plot = tmt.6H.GOIs.vol, filename = "tmt.6H.GOIs.vol.png", height=4, width=4)
```

```{r}
###Set GOIs
tmt.DN.GOIs <- tmt.DN
tmt.DN.GOIs$l2.B73xMo17.MP[tmt.DN.GOIs$l2.B73xMo17.MP <= -1] <- -1
tmt.DN.GOIs$l2.B73xMo17.MP[tmt.DN.GOIs$l2.B73xMo17.MP >= 1] <- 1
tmt.DN.GOIs$MP1 <- rowMeans(tmt.DN.GOIs[,c("Day.B73.r1","Day.Mo17.r1")])
tmt.DN.GOIs$MP2 <- rowMeans(tmt.DN.GOIs[,c("Day.B73.r2","Day.Mo17.r2")])
tmt.DN.GOIs$MP3 <- rowMeans(tmt.DN.GOIs[,c("Day.B73.r3","Day.Mo17.r3")])
tmt.DN.GOIs$ttest <- ttestFun(tmt.DN.GOIs, 27:29, 16:18)
tmt.DN.GOIs$l10.ttest <- -(log10(tmt.DN.GOIs$ttest))
tmt.DN.GOIs$Category <- "Other"
tmt.DN.GOIs$Order <- 1
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$PhANGs),"Order"] <- 3
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$PhAPGs),"Order"] <- 4
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$NE.PtRibo),"Order"] <- 5
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.DN.GOIs[(tmt.DN.GOIs$Gene %in% GOIs$PE.ribo),"Order"] <- 6
###Re-order
tmt.DN.GOIs <- tmt.DN.GOIs[(order(tmt.DN.GOIs$Order)),]
###factor
tmt.DN.GOIs$Category <- factor(tmt.DN.GOIs$Category, levels = c("Other", "Plastid", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
#Scatterplot - tmt
tmt.DN.GOIs.vol <- ggplot(tmt.DN.GOIs, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=3)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
          geom_hline(yintercept=1.30103, linetype="dashed", color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=15), legend.text=element_text(size=15), axis.title = element_text(size=15),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(vjust = -0.14, hjust = -0.075))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.DN.GOIs.vol.legend <- get_legend(tmt.DN.GOIs.vol)
tmt.DN.GOIs.vol <- tmt.DN.GOIs.vol+ theme(legend.position = "none")
ggsave(plot=tmt.DN.GOIs.vol.legend, "tmt.DN.GOIs.vol.legend.png")
ggsave(plot = tmt.DN.GOIs.vol, filename = "tmt.DN.GOIs.vol.png", height=4, width=4)
```



#Figure tmt.acs.all4.plastid
```{r setup dataframe tmt.acs.all4.plastid}
panel.acs.tmt.OG <- tmt.acs.OG
panel.acs.tmt.6h <- tmt.acs.6H
panel.acs.tmt.ril <- tmt.acs.RIL
panel.acs.tmt.dn <- tmt.acs.DN

panel.acs.tmt.OG$plot <- "A"
panel.acs.tmt.6h$plot <- "B"
panel.acs.tmt.ril$plot <- "C"
panel.acs.tmt.dn$plot <- "D"

panel.acs.tmt.OG <- panel.acs.tmt.OG[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]
panel.acs.tmt.6h <- panel.acs.tmt.6h[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]
panel.acs.tmt.ril <- panel.acs.tmt.ril[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]
panel.acs.tmt.dn <- panel.acs.tmt.dn[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]

panel.acs.tmt <- rbind(panel.acs.tmt.OG,panel.acs.tmt.6h)
panel.acs.tmt <- rbind(panel.acs.tmt,panel.acs.tmt.ril)
panel.acs.tmt <- rbind(panel.acs.tmt,panel.acs.tmt.dn)
```
```{r plot tmt.acs.all4.plastid}
  plot<-unique(panel.acs.tmt[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(panel.acs.tmt[,"plot"])
  for(i in 1:length(plot)){
    df<-panel.acs.tmt[which(panel.acs.tmt[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -0.75] <- -0.75
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= 0.75] <- 0.75
df.lims$l2.ACS.B73[df.lims$l2.ACS.B73 <= -1] <- -1
df.lims$l2.ACS.B73[df.lims$l2.ACS.B73 >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$EtBio),"Category"] <- "Ethylene biosynthesis"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$EtBio),"Order"] <- 7
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs", "Ethylene biosynthesis"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.ACS.B73, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.pe, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
      xlim(c(-0.75,0.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 acs/B73")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(vjust = -3.5, hjust = -0.485), axis.text.x = element_text(size = 9), axis.text.y = element_text(size = 9), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 3))

legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.B73xMo17.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.pe, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.ACS.B73"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.pe, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  des <- "AB
          CD
          EE"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], nrow = 3, heights = c(3,3,1.2), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "tmt.acs.all4.plastid.pdf"),
          dpi=300, width=4.75, height=5)
```
#Figure tmt.BxM.zma00592
```{r setup dataframe tmt.BxM.zma00592}
#merge with mdb to get entrez 
tmt.BxM.6H.entrez <- merge.data.frame(tmt.6H, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
tmt.BxM.6H.entrez.nodups <- tmt.BxM.6H.entrez[!is.na(tmt.BxM.6H.entrez$Entrez),]
tmt.BxM.6H.entrez.nodups <- tmt.BxM.6H.entrez.nodups[!tmt.BxM.6H.entrez.nodups$Entrez=="",]
tmt.BxM.6H.entrez.nodups <- tmt.BxM.6H.entrez.nodups[!duplicated(tmt.BxM.6H.entrez.nodups$Entrez),]
#make new dataframe with just l2(BM/MP) values and Entrez as rownames
tmt.BxM.6H.entrez.nodups.trim <- data.frame("l2.B73xMo17.MP"=tmt.BxM.6H.entrez.nodups$l2.B73xMo17.MP)
rownames(tmt.BxM.6H.entrez.nodups.trim) <- tmt.BxM.6H.entrez.nodups$Entrez
```
```{r plot tmt.BxM.6H.zma00592}
pathview(gene.data = tmt.BxM.6H.entrez.nodups.trim, pathway.id = "zma00260", species = "zma",  bins = 20, limit = 0.5, out.suffix="tmt.BxM")
```
```{r}
pathview(gene.data = tmt.BxM.6H.entrez.nodups.trim, pathway.id = "zma01230", species="zma", out.suffix = "tmt.BxM", keys.align = "y", kegg.native = F, match.data = F, multi.state = T, same.layer = T, cex=0.125, afactor=3, map.null = T, expand.node = T)
```

#Figure tmt.acs.zma00592
```{r setup dataframe tmt.acs.zma00592}
#merge with mdb to get entrez 
tmt.acs.entrez <- merge.data.frame(tmt.acs, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
tmt.acs.entrez.nodups <- tmt.acs.entrez[!is.na(tmt.acs.entrez$Entrez),]
tmt.acs.entrez.nodups <- tmt.acs.entrez.nodups[!tmt.acs.entrez.nodups$Entrez=="",]
tmt.acs.entrez.nodups <- tmt.acs.entrez.nodups[!duplicated(tmt.acs.entrez.nodups$Entrez),]
#make new dataframe with just l2(acs/B73) values and Entrez as rownames
tmt.acs.entrez.nodups.trim <- data.frame("l2.ACS.B73"=tmt.acs.entrez.nodups$l2.ACS.B73)
rownames(tmt.acs.entrez.nodups.trim) <- tmt.acs.entrez.nodups$Entrez
```
```{r plot tmt.acs.zma00592}
pathview(gene.data = tmt.acs.entrez.nodups.trim, pathway.id = "zma00592", species = "zma",  bins = 20, limit = 0.5, out.suffix="tmt.acs.zma00592")
```

#GOI correlations
```{r GOIcor protein}
#make dataframe
GOIs.cor <- data.frame("Group"=c("PE plastid ribosome", "PhANGs", "PhAPGs", "plastid", "NE plastid ribosome", "Cytosolic ribosome", "Ethylene biosynthesis", "Methionine adenosyltransferase", "ACC synthase", "ACC oxidase"), "protein.Rsquared.RILs"=NA, "protein.Rsquared.6Hybrid"=NA, "protein.Rsquared.RILs.6Hybrid"=NA, "RNA.Rsquared.RILs"=NA, "RNA.Rsquared.6Hybrid"=NA, "RNA.Rsquared.RILs.6Hybrid"=NA)
tmt.RIL.Hyb2MP <- tmt.RIL[,c(6,127:135)]
tmt.6H.Hyb2MP <- tmt.6H[,c(6,54:59)]
#combine RIL and 6 hybrids
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL.Hyb2MP, tmt.6H.Hyb2MP, by.x = "Accession", by.y = "Accession", all=F)
#Rename columns
colnames(tmt.RIL.Hyb2MP)[2:10] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17")
colnames(tmt.6H.Hyb2MP)[2:7] <- c("6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
colnames(tmt.RIL.6H.Hyb2MP)[2:16] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17", "6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
#Make gene column to call genes of interest
tmt.RIL.Hyb2MP$Gene <- substr(tmt.RIL.Hyb2MP$Accession, start = 1, stop = 14)
tmt.6H.Hyb2MP$Gene <- substr(tmt.6H.Hyb2MP$Accession, start = 1, stop = 14)
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)

for(i in 1:length(GOIs)){
#trim to GOI
tmt.RIL.Hyb2MP.cor <- tmt.RIL.Hyb2MP[tmt.RIL.Hyb2MP$Gene %in% GOIs[[i]],]
tmt.6H.Hyb2MP.cor <- tmt.6H.Hyb2MP[tmt.6H.Hyb2MP$Gene %in% GOIs[[i]],]
tmt.RIL.6H.Hyb2MP.cor <- tmt.RIL.6H.Hyb2MP[tmt.RIL.6H.Hyb2MP$Gene %in% GOIs[[i]],]
#Remove gene column
tmt.RIL.Hyb2MP.cor <- tmt.RIL.Hyb2MP.cor[,-11]
tmt.6H.Hyb2MP.cor <- tmt.6H.Hyb2MP.cor[,-8]
tmt.RIL.6H.Hyb2MP.cor <- tmt.RIL.6H.Hyb2MP.cor[,-17]
#get number of proteins
n.ril <- nrow(tmt.RIL.Hyb2MP.cor)
n.6h <- nrow(tmt.6H.Hyb2MP.cor)
n.ril.6h <- nrow(tmt.RIL.6H.Hyb2MP.cor)
#melt
tmt.RIL.Hyb2MP.cor <- melt(tmt.RIL.Hyb2MP.cor)
tmt.6H.Hyb2MP.cor <- melt(tmt.6H.Hyb2MP.cor)
tmt.RIL.6H.Hyb2MP.cor <- melt(tmt.RIL.6H.Hyb2MP.cor)

#Add hybrid/midparent plant height
tmt.RIL.Hyb2MP.cor$height <- c(rep(PlantHeights.ril.MP[1], n.ril), rep(PlantHeights.ril.MP[2], n.ril), rep(PlantHeights.ril.MP[3], n.ril), rep(PlantHeights.ril.MP[4], n.ril), rep(PlantHeights.ril.MP[5], n.ril), rep(PlantHeights.ril.MP[6], n.ril), rep(PlantHeights.ril.MP[7], n.ril), rep(PlantHeights.ril.MP[8], n.ril), rep(PlantHeights.ril.MP[9], n.ril))

tmt.6H.Hyb2MP.cor$height <- c(rep(PlantHeights.6h.MP[1], n.6h), rep(PlantHeights.6h.MP[2], n.6h), rep(PlantHeights.6h.MP[3], n.6h), rep(PlantHeights.6h.MP[4], n.6h), rep(PlantHeights.6h.MP[5], n.6h), rep(PlantHeights.6h.MP[6], n.6h))

tmt.RIL.6H.Hyb2MP.cor$height <- c(rep(PlantHeights.ril.MP[1], n.ril.6h), rep(PlantHeights.ril.MP[2], n.ril.6h), rep(PlantHeights.ril.MP[3], n.ril.6h), rep(PlantHeights.ril.MP[4], n.ril.6h), rep(PlantHeights.ril.MP[5], n.ril.6h), rep(PlantHeights.ril.MP[6], n.ril.6h), rep(PlantHeights.ril.MP[7], n.ril.6h), rep(PlantHeights.ril.MP[8], n.ril.6h), rep(PlantHeights.ril.MP[9], n.ril.6h), rep(PlantHeights.6h.MP[1], n.ril.6h), rep(PlantHeights.6h.MP[2], n.ril.6h), rep(PlantHeights.6h.MP[3], n.ril.6h), rep(PlantHeights.6h.MP[4], n.ril.6h), rep(PlantHeights.6h.MP[5], n.ril.6h), rep(PlantHeights.6h.MP[6], n.ril.6h))
#remove missing values
tmt.RIL.Hyb2MP.cor <- na.omit(tmt.RIL.Hyb2MP.cor)
tmt.6H.Hyb2MP.cor <- na.omit(tmt.6H.Hyb2MP.cor)
tmt.RIL.6H.Hyb2MP.cor <- na.omit(tmt.RIL.6H.Hyb2MP.cor)
#calculate averages and linear model
##split
tmt.RIL.Hyb2MP.cor.split <- split.data.frame(tmt.RIL.Hyb2MP.cor, tmt.RIL.Hyb2MP.cor$variable)
tmt.6H.Hyb2MP.cor.split <- split.data.frame(tmt.6H.Hyb2MP.cor, tmt.6H.Hyb2MP.cor$variable)
tmt.RIL.6H.Hyb2MP.cor.split <- split.data.frame(tmt.RIL.6H.Hyb2MP.cor, tmt.RIL.6H.Hyb2MP.cor$variable)
##make new df
tmt.RIL.Hyb2MP.cor.split.avg <- data.frame("hybrid"=unique(tmt.RIL.Hyb2MP.cor$variable), "value"=NA, "height"=PlantHeights.ril.MP)
tmt.6H.Hyb2MP.cor.split.avg <- data.frame("hybrid"=unique(tmt.6H.Hyb2MP.cor$variable), "value"=NA, "height"=PlantHeights.6h.MP)
tmt.RIL.6H.Hyb2MP.cor.split.avg <- data.frame("hybrid"=unique(tmt.RIL.6H.Hyb2MP.cor$variable), "value"=NA, "height"=PlantHeights.ril.6h.MP)
##take averages and populate new df
for(j in 1:9){
  tmt.RIL.Hyb2MP.cor.split.avg[j,"value"] <- mean(tmt.RIL.Hyb2MP.cor.split[[j]][,3])
}
for(j in 1:6){
  tmt.6H.Hyb2MP.cor.split.avg[j,"value"] <- mean(tmt.6H.Hyb2MP.cor.split[[j]][,3])
}
for(j in 1:15){
  tmt.RIL.6H.Hyb2MP.cor.split.avg[j,"value"] <- mean(tmt.RIL.6H.Hyb2MP.cor.split[[j]][,3])
}
##calculate Rsquared
tmt.RIL.Hyb2MP.cor.split.avg.lm <- lm(value~height, data = tmt.RIL.Hyb2MP.cor.split.avg)
tmt.RIL.Hyb2MP.cor.split.avg.lm.rsquared <- summary(tmt.RIL.Hyb2MP.cor.split.avg.lm)$r.squared

tmt.6H.Hyb2MP.cor.split.avg.lm <- lm(value~height, data = tmt.6H.Hyb2MP.cor.split.avg)
tmt.6H.Hyb2MP.cor.split.avg.lm.rsquared <- summary(tmt.6H.Hyb2MP.cor.split.avg.lm)$r.squared

tmt.RIL.6H.Hyb2MP.cor.split.avg.lm <- lm(value~height, data = tmt.RIL.6H.Hyb2MP.cor.split.avg)
tmt.RIL.6H.Hyb2MP.cor.split.avg.lm.rsquared <- summary(tmt.RIL.6H.Hyb2MP.cor.split.avg.lm)$r.squared
#populate GOIs.cor with Rsquared values
GOIs.cor[i, "protein.Rsquared.RILs"] <- tmt.RIL.Hyb2MP.cor.split.avg.lm.rsquared
GOIs.cor[i, "protein.Rsquared.6Hybrid"] <- tmt.6H.Hyb2MP.cor.split.avg.lm.rsquared
GOIs.cor[i, "protein.Rsquared.RILs.6Hybrid"] <- tmt.RIL.6H.Hyb2MP.cor.split.avg.lm.rsquared
}
```

```{r GOIcor RNA}
#subset to hyb/mp values
cpm.RIL.Hyb2MP <- cpm.RIL[,c(1,76:84)]
cpm.6H.Hyb2MP <- cpm.6H[,c(1,42:47)]
#combine RIL and 6 hybrids
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL.Hyb2MP, cpm.6H.Hyb2MP, by.x = "Gene", by.y = "Gene", all = F)
for(i in c(2,4:10)){
#limit to plastid ribosomal proteins
cpm.RIL.Hyb2MP.cor<- cpm.RIL.Hyb2MP[cpm.RIL.Hyb2MP$Gene %in% GOIs[[i]],]
cpm.6H.Hyb2MP.cor <- cpm.6H.Hyb2MP[cpm.6H.Hyb2MP$Gene %in% GOIs[[i]],]
cpm.RIL.6H.Hyb2MP.cor <- cpm.RIL.6H.Hyb2MP[cpm.RIL.6H.Hyb2MP$Gene %in% GOIs[[i]],]
#re-name columns
colnames(cpm.RIL.Hyb2MP.cor)[2:10] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17")
colnames(cpm.6H.Hyb2MP.cor)[2:7] <- c("6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
colnames(cpm.RIL.6H.Hyb2MP.cor)[2:16] <- c("RIL RIL.B73xMo17", "RIL RIL14xB73", "RIL RIL14xMo17", "RIL RIL16xB73", "RIL RIL16xMo17", "RIL RIL21xB73", "RIL RIL21xMo17", "RIL RIL317xB73", "RIL RIL317xMo17", "6H B73xMo17", "6H Mo17xB73", "6H B84xB73", "6H B84xMo17", "6H A682xB73", "6H A682xMo17")
#get number of proteins
n.ril <- nrow(cpm.RIL.Hyb2MP.cor)
n.6h <- nrow(cpm.6H.Hyb2MP.cor)
n.ril.6h <- nrow(cpm.RIL.6H.Hyb2MP.cor)
#melt
cpm.RIL.Hyb2MP.cor <- melt(cpm.RIL.Hyb2MP.cor)
cpm.6H.Hyb2MP.cor <- melt(cpm.6H.Hyb2MP.cor)
cpm.RIL.6H.Hyb2MP.cor <- melt(cpm.RIL.6H.Hyb2MP.cor)
#add hybrid/mid-parent plant height
cpm.RIL.Hyb2MP.cor$height <- c(rep(PlantHeights.ril.MP[1], n.ril), rep(PlantHeights.ril.MP[2], n.ril), rep(PlantHeights.ril.MP[3], n.ril), rep(PlantHeights.ril.MP[4], n.ril), rep(PlantHeights.ril.MP[5], n.ril), rep(PlantHeights.ril.MP[6], n.ril), rep(PlantHeights.ril.MP[7], n.ril), rep(PlantHeights.ril.MP[8], n.ril), rep(PlantHeights.ril.MP[9], n.ril))

cpm.6H.Hyb2MP.cor$height <- c(rep(PlantHeights.6h.MP[1], n.6h), rep(PlantHeights.6h.MP[2], n.6h), rep(PlantHeights.6h.MP[3], n.6h), rep(PlantHeights.6h.MP[4], n.6h), rep(PlantHeights.6h.MP[5], n.6h), rep(PlantHeights.6h.MP[6], n.6h))

cpm.RIL.6H.Hyb2MP.cor$height <- c(rep(PlantHeights.ril.MP[1], n.ril.6h), rep(PlantHeights.ril.MP[2], n.ril.6h), rep(PlantHeights.ril.MP[3], n.ril.6h), rep(PlantHeights.ril.MP[4], n.ril.6h), rep(PlantHeights.ril.MP[5], n.ril.6h), rep(PlantHeights.ril.MP[6], n.ril.6h), rep(PlantHeights.ril.MP[7], n.ril.6h), rep(PlantHeights.ril.MP[8], n.ril.6h), rep(PlantHeights.ril.MP[9], n.ril.6h), rep(PlantHeights.6h.MP[1], n.ril.6h), rep(PlantHeights.6h.MP[2], n.ril.6h), rep(PlantHeights.6h.MP[3], n.ril.6h), rep(PlantHeights.6h.MP[4], n.ril.6h), rep(PlantHeights.6h.MP[5], n.ril.6h), rep(PlantHeights.6h.MP[6], n.ril.6h))
#calculate averages and linear model
##split
cpm.RIL.Hyb2MP.cor.split <- split.data.frame(cpm.RIL.Hyb2MP.cor, cpm.RIL.Hyb2MP.cor$variable)
cpm.6H.Hyb2MP.cor.split <- split.data.frame(cpm.6H.Hyb2MP.cor, cpm.6H.Hyb2MP.cor$variable)
cpm.RIL.6H.Hyb2MP.cor.split <- split.data.frame(cpm.RIL.6H.Hyb2MP.cor, cpm.RIL.6H.Hyb2MP.cor$variable)
##make new df
cpm.RIL.Hyb2MP.cor.split.avg <- data.frame("hybrid"=unique(cpm.RIL.Hyb2MP.cor$variable), "value"=NA, "height"=unique(cpm.RIL.Hyb2MP.cor$height))
cpm.6H.Hyb2MP.cor.split.avg <- data.frame("hybrid"=unique(cpm.6H.Hyb2MP.cor$variable), "value"=NA, "height"=unique(cpm.6H.Hyb2MP.cor$height))
cpm.RIL.6H.Hyb2MP.cor.split.avg <- data.frame("hybrid"=unique(cpm.RIL.6H.Hyb2MP.cor$variable), "value"=NA, "height"=unique(cpm.RIL.6H.Hyb2MP.cor$height))
##take averages and populate new df
for(k in 1:9){
  cpm.RIL.Hyb2MP.cor.split.avg[k,"value"] <- mean(cpm.RIL.Hyb2MP.cor.split[[k]][,3])
}
for(k in 1:6){
  cpm.6H.Hyb2MP.cor.split.avg[k,"value"] <- mean(cpm.6H.Hyb2MP.cor.split[[k]][,3])
}
for(k in 1:15){
  cpm.RIL.6H.Hyb2MP.cor.split.avg[k,"value"] <- mean(cpm.RIL.6H.Hyb2MP.cor.split[[k]][,3])
}
##calculate Rsquared
cpm.RIL.Hyb2MP.cor.split.avg.lm <- lm(value~height, data = cpm.RIL.Hyb2MP.cor.split.avg)
cpm.RIL.Hyb2MP.cor.split.avg.lm.rsquared <- summary(cpm.RIL.Hyb2MP.cor.split.avg.lm)$r.squared

cpm.6H.Hyb2MP.cor.split.avg.lm <- lm(value~height, data = cpm.6H.Hyb2MP.cor.split.avg)
cpm.6H.Hyb2MP.cor.split.avg.lm.rsquared <- summary(cpm.6H.Hyb2MP.cor.split.avg.lm)$r.squared

cpm.RIL.6H.Hyb2MP.cor.split.avg.lm <- lm(value~height, data = cpm.RIL.6H.Hyb2MP.cor.split.avg)
cpm.RIL.6H.Hyb2MP.cor.split.avg.lm.rsquared <- summary(cpm.RIL.6H.Hyb2MP.cor.split.avg.lm)$r.squared
#populate GOIs.cor with Rsquared values
GOIs.cor[i, "RNA.Rsquared.RILs"] <- cpm.RIL.Hyb2MP.cor.split.avg.lm.rsquared
GOIs.cor[i, "RNA.Rsquared.6Hybrid"] <- cpm.6H.Hyb2MP.cor.split.avg.lm.rsquared
GOIs.cor[i, "RNA.Rsquared.RILs.6Hybrid"] <- cpm.RIL.6H.Hyb2MP.cor.split.avg.lm.rsquared
}

#save
write.csv(GOIs.cor, "GOIs.cor.csv", row.names = F)
```

#Figure tmt.cpm.BxM.LB.plastid
```{r tmt.cpm.BxM.LB.plastid}
###Set limits
tmt.LB.lims <- tmt.LB
tmt.LB.lims$l2.B73xMo17.MP[tmt.LB.lims$l2.B73xMo17.MP <= -1] <- -1
tmt.LB.lims$l2.B73xMo17.MP[tmt.LB.lims$l2.B73xMo17.MP >= 1] <- 1
tmt.LB.lims$l2.B73.Mo17[tmt.LB.lims$l2.B73.Mo17 <= -1] <- -1
tmt.LB.lims$l2.B73.Mo17[tmt.LB.lims$l2.B73.Mo17 >= 1] <- 1
###Set GOIs
tmt.LB.lims.plastid <- tmt.LB.lims
tmt.LB.lims.plastid$Category <- "Other"
tmt.LB.lims.plastid$Order <- 1
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.LB.lims.plastid[(tmt.LB.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
tmt.LB.lims.plastid <- tmt.LB.lims.plastid[(order(tmt.LB.lims.plastid$Order)),]
###factor
tmt.LB.lims.plastid$Category <- factor(tmt.LB.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))

###Scatterplot
tmt.LB.lims.plastid.scat <- ggplot(tmt.LB.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
 # theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("A")+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(tmt.LB.lims.plastid.scat)
tmt.LB.lims.plastid.scat <- tmt.LB.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.LB.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.LB.lims.plastid[,"l2.B73xMo17.MP"], color=tmt.LB.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.LB.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.LB.lims.plastid[,"l2.B73.Mo17"], color=tmt.LB.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      A <- insert_xaxis_grob(tmt.LB.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      A <- insert_yaxis_grob(A, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      ###Set limits
cpm.LB.lims <- cpm.LB.protein
cpm.LB.lims$l2.B73xMo17.MP[cpm.LB.lims$l2.B73xMo17.MP <= -1] <- -1
cpm.LB.lims$l2.B73xMo17.MP[cpm.LB.lims$l2.B73xMo17.MP >= 1] <- 1
cpm.LB.lims$l2.B73.Mo17[cpm.LB.lims$l2.B73.Mo17 <= -2] <- -2
cpm.LB.lims$l2.B73.Mo17[cpm.LB.lims$l2.B73.Mo17 >= 2] <- 2
###Set GOIs
cpm.LB.lims.plastid <- cpm.LB.lims
cpm.LB.lims.plastid$Category <- "Other"
cpm.LB.lims.plastid$Order <- 1
cpm.LB.lims.plastid[(cpm.LB.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
cpm.LB.lims.plastid[(cpm.LB.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
cpm.LB.lims.plastid[(cpm.LB.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.LB.lims.plastid[(cpm.LB.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.LB.lims.plastid[(cpm.LB.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.LB.lims.plastid[(cpm.LB.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5

###Re-order
cpm.LB.lims.plastid <- cpm.LB.lims.plastid[(order(cpm.LB.lims.plastid$Order)),]
###remove "other"
#cpm.LB.lims.plastid <- cpm.LB.lims.plastid[!cpm.LB.lims.plastid$Category=="Other",]
###factor
cpm.LB.lims.plastid$Category <- factor(cpm.LB.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome", "PhANGs"))
###Scatterplot
cpm.LB.lims.plastid.scat <- ggplot(cpm.LB.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("B")+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(cpm.LB.lims.plastid.scat, axis="x")+
      geom_density(aes(cpm.LB.lims.plastid[,"l2.B73xMo17.MP"], color=cpm.LB.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.LB.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.LB.lims.plastid[,"l2.B73.Mo17"], color=cpm.LB.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      B <- insert_xaxis_grob(cpm.LB.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      B <- insert_yaxis_grob(B, densityplotY, grid::unit(0.275, "null"), position = "right")

FigLB.plotlist <- list(A, B, legend)
FigLB.design <- "AB
                 CC"
FigLB.plotpanel<-patchwork::wrap_plots(A=FigLB.plotlist[[1]], B=FigLB.plotlist[[2]], C=FigLB.plotlist[[3]],  nrow = 2, heights = c(3,1.2), design = FigLB.design)
ggsave(plot=FigLB.plotpanel, filename =paste0(PathToPlots, "tmt.cpm.BxM.LB.plastid.pdf"), dpi=300, width=4.75, height=3)
```
#Figure ProteinVsRNA
```{r ProteinVsRNA}
ProteinVsRNA <- merge.data.frame(tmt.SL[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], cpm.SL[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Gene", by.y = "Gene", all=F)
colnames(ProteinVsRNA) <- c("Gene", "tmt.l2.B73xMo17.MP", "tmt.l2.B73.Mo17", "cpm.l2.B73xMo17.MP", "cpm.l2.B73.Mo17")
###Set limits
ProteinVsRNA.lims <- ProteinVsRNA
ProteinVsRNA.lims$tmt.l2.B73xMo17.MP[ProteinVsRNA.lims$tmt.l2.B73xMo17.MP <= -1] <- -1
ProteinVsRNA.lims$tmt.l2.B73xMo17.MP[ProteinVsRNA.lims$tmt.l2.B73xMo17.MP >= 1] <- 1
ProteinVsRNA.lims$cpm.l2.B73xMo17.MP[ProteinVsRNA.lims$cpm.l2.B73xMo17.MP <= -2] <- -2
ProteinVsRNA.lims$cpm.l2.B73xMo17.MP[ProteinVsRNA.lims$cpm.l2.B73xMo17.MP >= 2] <- 2
ProteinVsRNA.lims$tmt.l2.B73.Mo17[ProteinVsRNA.lims$tmt.l2.B73.Mo17 <= -2] <- -2
ProteinVsRNA.lims$tmt.l2.B73.Mo17[ProteinVsRNA.lims$tmt.l2.B73.Mo17 >= 2] <- 2
ProteinVsRNA.lims$cpm.l2.B73.Mo17[ProteinVsRNA.lims$cpm.l2.B73.Mo17 <= -3] <- -3
ProteinVsRNA.lims$cpm.l2.B73.Mo17[ProteinVsRNA.lims$cpm.l2.B73.Mo17 >= 3] <- 3
###Set GOIs
ProteinVsRNA.lims.plastid <- ProteinVsRNA.lims
ProteinVsRNA.lims.plastid$Category <- "Other"
ProteinVsRNA.lims.plastid$Order <- 1
ProteinVsRNA.lims.plastid[(ProteinVsRNA.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
ProteinVsRNA.lims.plastid[(ProteinVsRNA.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
ProteinVsRNA.lims.plastid[(ProteinVsRNA.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
ProteinVsRNA.lims.plastid[(ProteinVsRNA.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
ProteinVsRNA.lims.plastid[(ProteinVsRNA.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
ProteinVsRNA.lims.plastid[(ProteinVsRNA.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5

###Re-order
ProteinVsRNA.lims.plastid <- ProteinVsRNA.lims.plastid[(order(ProteinVsRNA.lims.plastid$Order)),]
###factor
ProteinVsRNA.lims.plastid$Category <- factor(ProteinVsRNA.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome","PhANGs"))
###Scatterplot
ProteinVsRNA.lims.plastid.scat <- ggplot(ProteinVsRNA.lims.plastid, mapping=aes(x=tmt.l2.B73xMo17.MP, y=cpm.l2.B73xMo17.MP, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=2.5)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(ncol=2))+
    xlab("log2 BxM/MP protein")+
    ylab("log2 BxM/MP RNA")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(ProteinVsRNA.lims.plastid.scat, axis="x")+
      geom_density(aes(ProteinVsRNA.lims.plastid[,"tmt.l2.B73xMo17.MP"], color=ProteinVsRNA.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(ProteinVsRNA.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(ProteinVsRNA.lims.plastid[,"cpm.l2.B73xMo17.MP"], color=ProteinVsRNA.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(ProteinVsRNA.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
  ggsave(plot=q, filename = paste0(PathToPlots, "ProteinVsRNA.Hyb2MP.pdf"),  dpi=300, width=4.75, height = 5, scale = 1, units = "in")
  
  
  ProteinVsRNA.lims.plastid.scat2 <- ggplot(ProteinVsRNA.lims.plastid, mapping=aes(x=tmt.l2.B73.Mo17, y=cpm.l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=2.5)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
          xlim(c(-2,2))+
          ylim(c(-3,3))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(ncol=2))+
    xlab("log2 B73/Mo17 protein")+
    ylab("log2 B73/Mo17 RNA")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(ProteinVsRNA.lims.plastid.scat2, axis="x")+
      geom_density(aes(ProteinVsRNA.lims.plastid[,"tmt.l2.B73.Mo17"], color=ProteinVsRNA.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(ProteinVsRNA.lims.plastid.scat2, axis="y", coord_flip = T)+
      geom_density(aes(ProteinVsRNA.lims.plastid[,"cpm.l2.B73.Mo17"], color=ProteinVsRNA.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(ProteinVsRNA.lims.plastid.scat2, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
  ggsave(plot=q, filename = paste0(PathToPlots, "ProteinVsRNA.Parent2Parent.pdf"),  dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


#Figure LeafBladeVsSeedlingLeaf
```{r setup dataframe LeafBladeVsSeedlingLeaf}
tmt.LB.SL <- merge.data.frame(tmt.LB[,c("Accession", "l2.B73xMo17.MP", "l2.B73.Mo17")], tmt.SL[,c("Accession", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Accession", by.y = "Accession", all=F)
tmt.LB.SL$Gene <- substr(tmt.LB.SL$Accession, start = 1, stop = 14)
cpm.LB.SL <- merge.data.frame(cpm.LB[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], cpm.SL[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Gene", by.y = "Gene", all = F)
```
```{r plot LeafBladeVsSeedlingLeaf.Hyb2MP}
###Set limits
tmt.LB.SL.lims <- tmt.LB.SL
tmt.LB.SL.lims$l2.B73xMo17.MP.x[tmt.LB.SL.lims$l2.B73xMo17.MP.x <= -1] <- -1
tmt.LB.SL.lims$l2.B73xMo17.MP.x[tmt.LB.SL.lims$l2.B73xMo17.MP.x >= 1] <- 1
tmt.LB.SL.lims$l2.B73xMo17.MP.y[tmt.LB.SL.lims$l2.B73xMo17.MP.y <= -1] <- -1
tmt.LB.SL.lims$l2.B73xMo17.MP.y[tmt.LB.SL.lims$l2.B73xMo17.MP.y >= 1] <- 1
###Set GOIs
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims
tmt.LB.SL.lims.plastid$Category <- "Other"
tmt.LB.SL.lims.plastid$Order <- 1
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6

###Re-order
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims.plastid[(order(tmt.LB.SL.lims.plastid$Order)),]
###factor
tmt.LB.SL.lims.plastid$Category <- factor(tmt.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
###Scatterplot
tmt.LB.SL.lims.plastid.scat <- ggplot(tmt.LB.SL.lims.plastid, mapping=aes(x=l2.B73xMo17.MP.x, y=l2.B73xMo17.MP.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("A")+
    xlab("log2 BxM/MP LB")+
    ylab("log2 BxM/MP SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(tmt.LB.SL.lims.plastid.scat)
tmt.LB.SL.lims.plastid.scat <- tmt.LB.SL.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73xMo17.MP.x"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73xMo17.MP.y"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      A <- insert_xaxis_grob(tmt.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      A <- insert_yaxis_grob(A, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      
      ###Set limits
cpm.LB.SL.lims <- cpm.LB.SL
cpm.LB.SL.lims$l2.B73xMo17.MP.x[cpm.LB.SL.lims$l2.B73xMo17.MP.x <= -2] <- -2
cpm.LB.SL.lims$l2.B73xMo17.MP.x[cpm.LB.SL.lims$l2.B73xMo17.MP.x >= 2] <- 2
cpm.LB.SL.lims$l2.B73xMo17.MP.y[cpm.LB.SL.lims$l2.B73xMo17.MP.y <= -2] <- -2
cpm.LB.SL.lims$l2.B73xMo17.MP.y[cpm.LB.SL.lims$l2.B73xMo17.MP.y >= 2] <- 2
###Set GOIs
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims
cpm.LB.SL.lims.plastid$Category <- "Other"
cpm.LB.SL.lims.plastid$Order <- 1
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
###Re-order
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims.plastid[(order(cpm.LB.SL.lims.plastid$Order)),]
###factor
cpm.LB.SL.lims.plastid$Category <- factor(cpm.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome", "PhANGs"))
###Scatterplot
cpm.LB.SL.lims.plastid.scat <- ggplot(cpm.LB.SL.lims.plastid, mapping=aes(x=l2.B73xMo17.MP.x, y=l2.B73xMo17.MP.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
          xlim(c(-2,2))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("B")+
    xlab("log2 BxM/MP LB")+
    ylab("log2 BxM/MP SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73xMo17.MP.x"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73xMo17.MP.y"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      B <- insert_xaxis_grob(cpm.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      B <- insert_yaxis_grob(B, densityplotY, grid::unit(0.275, "null"), position = "right")
###Set limits
tmt.LB.SL.lims <- tmt.LB.SL
tmt.LB.SL.lims$l2.B73.Mo17.x[tmt.LB.SL.lims$l2.B73.Mo17.x <= -1] <- -1
tmt.LB.SL.lims$l2.B73.Mo17.x[tmt.LB.SL.lims$l2.B73.Mo17.x >= 1] <- 1
tmt.LB.SL.lims$l2.B73.Mo17.y[tmt.LB.SL.lims$l2.B73.Mo17.y <= -1] <- -1
tmt.LB.SL.lims$l2.B73.Mo17.y[tmt.LB.SL.lims$l2.B73.Mo17.y >= 1] <- 1
###Set GOIs
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims
tmt.LB.SL.lims.plastid$Category <- "Other"
tmt.LB.SL.lims.plastid$Order <- 1
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6

###Re-order
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims.plastid[(order(tmt.LB.SL.lims.plastid$Order)),]
###factor
tmt.LB.SL.lims.plastid$Category <- factor(tmt.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
###Scatterplot
tmt.LB.SL.lims.plastid.scat <- ggplot(tmt.LB.SL.lims.plastid, mapping=aes(x=l2.B73.Mo17.x, y=l2.B73.Mo17.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("C")+
    xlab("log2 B73/Mo17 LB")+
    ylab("log2 B73/Mo17 SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(tmt.LB.SL.lims.plastid.scat)
tmt.LB.SL.lims.plastid.scat <- tmt.LB.SL.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73.Mo17.x"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73.Mo17.y"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      C <- insert_xaxis_grob(tmt.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      C <- insert_yaxis_grob(C, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      
      ###Set limits
cpm.LB.SL.lims <- cpm.LB.SL
cpm.LB.SL.lims$l2.B73.Mo17.x[cpm.LB.SL.lims$l2.B73.Mo17.x <= -2] <- -2
cpm.LB.SL.lims$l2.B73.Mo17.x[cpm.LB.SL.lims$l2.B73.Mo17.x >= 2] <- 2
cpm.LB.SL.lims$l2.B73.Mo17.y[cpm.LB.SL.lims$l2.B73.Mo17.y <= -2] <- -2
cpm.LB.SL.lims$l2.B73.Mo17.y[cpm.LB.SL.lims$l2.B73.Mo17.y >= 2] <- 2
###Set GOIs
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims
cpm.LB.SL.lims.plastid$Category <- "Other"
cpm.LB.SL.lims.plastid$Order <- 1
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
###Re-order
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims.plastid[(order(cpm.LB.SL.lims.plastid$Order)),]
###factor
cpm.LB.SL.lims.plastid$Category <- factor(cpm.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid", "NE plastid ribosome", "PhANGs"))
###Scatterplot
cpm.LB.SL.lims.plastid.scat <- ggplot(cpm.LB.SL.lims.plastid, mapping=aes(x=l2.B73.Mo17.x, y=l2.B73.Mo17.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
          xlim(c(-2,2))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(vjust = -3.5, hjust = -0.485), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=2))+
    ggtitle("D")+
    xlab("log2 B73/Mo17 LB")+
    ylab("log2 B73/Mo17 SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73.Mo17.x"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73.Mo17.y"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      D <- insert_xaxis_grob(cpm.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      D <- insert_yaxis_grob(D, densityplotY, grid::unit(0.275, "null"), position = "right")


FigLB.SL.design <- "AB
                    CD
                    EE"
FigLB.SL.plotpanel<-patchwork::wrap_plots(A=A, B=B, C=C, D=D, E=legend,  nrow = 3, heights = c(3,3,1.2), design = FigLB.SL.design)
ggsave(plot=FigLB.SL.plotpanel, filename =paste0(PathToPlots, "LeafBladeVsSeedlingLeaf.Hyb2MP.Par2Par.pdf"), dpi=300, width=4.75, height=5)
```







#Figure cpm.6H.RIL.cor
```{r}
ParentToParent.6h <- abs(cpm.6H[,60:65])
ParentToParent.6h$Gene <- cpm.6H$Gene
ParentToParent.ril <- abs(cpm.RIL[,85:93])
ParentToParent.ril$Gene <- cpm.RIL$Gene
```
```{r}
cpm.6H.HybToMP.cor <- merge.data.frame(cpm.6H.HybToMP.cor, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all.x = T, all.y = F)
cpm.6H.HybToMP.cor <- cpm.6H.HybToMP.cor[order(-cpm.6H.HybToMP.cor$cor),]
write.csv(cpm.6H.HybToMP.cor, "cpm.6H.HybToMP.CobWeight.cor.csv", row.names = F)
```

```{r setup dataframe cpm.6H.RIL.cor}
#RNA correlations
cpm.6H.HybToMP.cor <- CorFun(cpm.6H, 42:47, PlantHeights.6h.MP.cob)
cpm.6H.ParentToParent.cor <- CorFun(ParentToParent.6h, 1:6, PlantHeights.6h.MP.cob)
#Combine RNA correlations
cpm.cor <- merge.data.frame(cpm.6H.HybToMP.cor[,c("Gene", "cor")], cpm.6H.ParentToParent.cor[,c("Gene", "cor")], by.x="Gene", by.y="Gene", all=T)
colnames(cpm.cor) <- c("Gene", "Cor.HtoMP", "Cor.PtoP")
#Define GOIs
cpm.cor$GOI <- "Other"
cpm.cor$Order <- 1
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$PhANGs] <- "PhANGs"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$PhANGs] <- 2
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$PhAPGs] <- "PhAPGs"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$PhAPGs] <- 3
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$NE.PtRibo] <- "NE.PtRibo"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$NE.PtRibo] <- 4
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$PE.ribo] <- "PE.ribo"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$PE.ribo] <- 5
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$CytoRibo] <- "CytoRibo"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$CytoRibo] <- 6
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$MAT] <- "MAT"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$MAT] <- 7
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$ACS] <- "ACS"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$ACS] <- 8
cpm.cor$GOI[cpm.cor$Gene%in%GOIs$ACO] <- "ACO"
cpm.cor$Order[cpm.cor$Gene%in%GOIs$ACO] <- 9
cpm.cor <- cpm.cor[order(cpm.cor$Order),]
cpm.cor$GOI <- factor(cpm.cor$GOI, levels = c("Other", "PhANGs", "PhAPGs", "NE.PtRibo", "PE.ribo", "CytoRibo", "MAT", "ACS", "ACO"))
```
```{r}
ggplot(cpm.cor[cpm.cor$Order<7,], aes(x = Cor.HtoMP, color=GOI))+
  geom_density(size=3)+
  theme_bw()+
  geom_vline(xintercept = 0, linetype="dashed", color = "gray60", size=0.5)+
  xlab("Hyb/MP Pearson Cor")+
  scale_color_manual(values = c("Other"="black","PhANGs"="green", "PhAPGs"="blue", "NE.PtRibo"= "pink", "PE.ribo"= "red", "CytoRibo"= "purple", "MAT"= "yellow", "ACS"= "orange", "ACO"="brown"))+
  ggtitle("6 hybrids RNA correlations")
ggsave("cpm.6H.Hyb2MPcor.CobWeight.png", height = 5, width = 7)
```

```{r plot cpm.6H.RIL.cor}
p <- ggplot(cpm.cor, aes(x = Cor.HtoMP, y = Cor.PtoP, color=GOI))+
  geom_point(size=3)+
  theme_bw()+
  coord_equal(expand = F)+
  geom_hline(yintercept = 0, linetype="dashed", color = "gray60", size=0.5)+
  geom_vline(xintercept = 0, linetype="dashed", color = "gray60", size=0.5)+
  ylab("parent/parent cor")+
  xlab("hybrid/midparent cor")+
  scale_color_manual(values = c("Other"="black","PhANGs"="green", "PhAPGs"="blue", "NE.PtRibo"= "pink", "PE.ribo"= "red", "CytoRibo"= "purple", "MAT"= "yellow", "ACS"= "orange", "ACO"="brown"))+
  ggtitle("6 hybrids RNA correlations")
###Add density curves
    densityplotX<-axis_canvas(p, axis="x")+
      geom_density(aes(cpm.cor[cpm.cor$Order<7,"Cor.HtoMP"], color=cpm.cor[cpm.cor$Order<7,"GOI"]), size=0.75)+
    scale_color_manual(values=c("Other"="black","PhANGs"="green", "PhAPGs"="blue", "NE.PtRibo"= "pink", "PE.ribo"= "red", "CytoRibo"= "purple", "MAT"= "yellow", "ACS"= "orange", "ACO"="brown"), name="GOI")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(p, axis="y", coord_flip = T)+
      geom_density(aes(cpm.cor[cpm.cor$Order<7,"Cor.PtoP"], color=cpm.cor[cpm.cor$Order<7,"GOI"]), size=0.75)+
      scale_color_manual(values=c("Other"="black","PhANGs"="green", "PhAPGs"="blue", "NE.PtRibo"= "pink", "PE.ribo"= "red", "CytoRibo"= "purple", "MAT"= "yellow", "ACS"= "orange", "ACO"="brown"), name="GOI")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(p, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
ggsave(plot=q, filename = paste0(PathToPlots,"cpm.6H.cor.png"))
```
```{r}
colnames(tmt.RIL)
```

#Figure tmt.6H.RIL.cor
```{r setup dataframe tmt.6H.RIL.cor}
#Protein correlations
tmt.6H.cor <- CorFun(tmt.6H, 66:71, PlantHeights.6h.MP)
tmt.RIL.cor <- CorFun(tmt.RIL, 145:153, PlantHeights.ril.MP)
#Combine protein correlations
tmt.cor <- merge.data.frame(tmt.6H.cor[,c("Accession", "cor")], tmt.RIL.cor[,c("Accession", "cor")], by.x="Accession", by.y="Accession", all=T)
colnames(tmt.cor) <- c("Accession", "Cor.6h", "Cor.RIL")
#Define GOIs
tmt.cor$Gene <- substr(tmt.cor$Accession, start = 1, stop = 14)
tmt.cor$GOI <- "Other"
tmt.cor$Order <- 1
tmt.cor$GOI[tmt.cor$Gene%in%GOIs$CytoRibo] <- "Cytosolic ribosome"
tmt.cor$Order[tmt.cor$Gene%in%GOIs$CytoRibo] <- 2
tmt.cor$GOI[tmt.cor$Gene%in%GOIs$PhANGs] <- "PhANGs"
tmt.cor$Order[tmt.cor$Gene%in%GOIs$PhANGs] <- 3
tmt.cor$GOI[tmt.cor$Gene%in%GOIs$PhAPGs] <- "PhAPGs"
tmt.cor$Order[tmt.cor$Gene%in%GOIs$PhAPGs] <- 4
tmt.cor$GOI[tmt.cor$Gene%in%GOIs$NE.PtRibo] <- "NE plastid ribosome"
tmt.cor$Order[tmt.cor$Gene%in%GOIs$NE.PtRibo] <- 5
tmt.cor$GOI[tmt.cor$Gene%in%GOIs$PE.ribo] <- "PE plastid ribosome"
tmt.cor$Order[tmt.cor$Gene%in%GOIs$PE.ribo] <- 6
tmt.cor <- tmt.cor[order(tmt.cor$Order),]
tmt.cor$GOI <- factor(tmt.cor$GOI, levels = c("Other", "Cytosolic ribosome", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
```
```{r plot tmt.6H.RIL.cor}
p <- ggplot(tmt.cor, aes(x = Cor.6h, y = Cor.RIL, color=GOI))+
  geom_point(size=3)+
  theme_bw()+
  coord_equal(expand = F)+
  geom_hline(yintercept = 0, linetype="dashed", color = "gray60", size=0.5)+
  geom_vline(xintercept = 0, linetype="dashed", color = "gray60", size=0.5)+
  scale_color_manual(values = colors.pc)+
  ggtitle("6 hybrids vs RIL hybrids Protein correlations")
###Add density curves
    densityplotX<-axis_canvas(p, axis="x")+
      geom_density(aes(tmt.cor$Cor.6h, color=tmt.cor$GOI), size=0.75)+
    scale_color_manual(values=colors.pc, name="GOI")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(p, axis="y", coord_flip = T)+
      geom_density(aes(tmt.cor$Cor.RIL, color=tmt.cor$GOI), size=0.75)+
      scale_color_manual(values=colors.pc, name="GOI")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(p, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
ggsave(plot=q, paste0(PathToPlots, "tmt.6H.RIL.cor.png"))
```


#Cor between parent/parent and hyb/mp
```{r}
tmt.6H.Hyb2MP.Par2Par <- tmt.6H[,c(6,66:77)]
tmt.RIL.Hyb2MP.Par2Par <- tmt.RIL[,c(6,145:162)]

for(i in 2:ncol(tmt.6H.Hyb2MP.Par2Par)){
  tmt.6H.Hyb2MP.Par2Par[,i] <- abs(tmt.6H.Hyb2MP.Par2Par[,i])
}

for(i in 2:ncol(tmt.RIL.Hyb2MP.Par2Par)){
  tmt.RIL.Hyb2MP.Par2Par[,i] <- abs(tmt.RIL.Hyb2MP.Par2Par[,i])
}

tmt.6H.RIL.Hyb2MP.Par2Par <- merge.data.frame(tmt.6H.Hyb2MP.Par2Par, tmt.RIL.Hyb2MP.Par2Par, by.x = "Accession", by.y = "Accession", all=F)

tmt.6H.RIL.Hyb2MP.Par2Par <- tmt.6H.RIL.Hyb2MP.Par2Par[,c(1:7,14:22,8:13,23:31)]
```
```{r CorFun2}
CorFun2 <- function(x, col1, col2){
  x$cor <- NA
  for(i in 1:nrow(x)){
    v1 <- as.numeric(x[i,col1])
    v2 <- as.numeric(x[i,col2])
    c <- cor(v1, v2, method = "pearson")
    x[i,"cor"]<- c
  }
  x}
```
```{r}
tmt.6H.RIL.Hyb2MP.Par2Par.cor <- CorFun2(tmt.6H.RIL.Hyb2MP.Par2Par, 2:16, 17:31)
```
```{r}
tmt.6H.RIL.Hyb2MP.Par2Par.cor$Gene <- substr(tmt.6H.RIL.Hyb2MP.Par2Par.cor$Accession, start=0, stop=14)
mean(tmt.6H.RIL.Hyb2MP.Par2Par.cor$cor[tmt.6H.RIL.Hyb2MP.Par2Par.cor$Gene %in% c(GOIs$NE.PtRibo, GOIs$PE.ribo)])
mean(tmt.6H.RIL.Hyb2MP.Par2Par.cor$cor)
```
```{r}
tmt.cpm.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL.6H.Hyb2MP, cpm.RIL.6H.Hyb2MP, by.x = "Gene", by.y = "Gene", all=F)
tmt.cpm.RIL.6H.Hyb2MP <- na.omit(tmt.cpm.RIL.6H.Hyb2MP)
tmt.cpm.RIL.6H.Hyb2MP.cor2 <- CorFun2(tmt.cpm.RIL.6H.Hyb2MP, 2:16, 17:31)
mean(tmt.cpm.RIL.6H.Hyb2MP.cor2$cor[tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhANGs])
```

```{r}
tmt.cpm.RIL.6H.Hyb2MP.cor2$Category <- "Other"
tmt.cpm.RIL.6H.Hyb2MP.cor2$Order <- 1
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhAPGs),"Order"] <- 6
tmt.cpm.RIL.6H.Hyb2MP.cor2 <- tmt.cpm.RIL.6H.Hyb2MP.cor2[(order(tmt.cpm.RIL.6H.Hyb2MP.cor2$Order)),]
tmt.cpm.RIL.6H.Hyb2MP.cor2$Category <- factor(tmt.cpm.RIL.6H.Hyb2MP.cor2$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
ggplot(tmt.cpm.RIL.6H.Hyb2MP.cor2, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=12), plot.title = element_text(hjust = -.6))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)
ggsave(filename =paste0(PathToPlots, "tmt.cpm.RIL.6H.Hyb2MP.cor2.pdf"), dpi=300, width=4.75, height=3.5)
```


