---
title: "Data Analyses"
output: github_document
author: "Devon Birdseye"
date: "08/07/2020"
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r set seed}
set.seed(586)
```
```{r libraries}
library(plyr)
library(EnvStats)
library(ggplot2)
library(reshape2)
library(cowplot)
library(patchwork)
library(pathview)
library(data.table)
library(AnnotationDbi)
library(dplyr)
library(tidyr)
library(ggridges)
```
```{r define paths}
#Path to input data
PathToData <- "Data/"
#Path to output plots
PathToPlots <- "Plots/"
```


#Import protein data
```{r import protein data}
#Import protein data
tmt <- read.csv(paste0(PathToData,"SupplementalDataSet5.csv"), stringsAsFactors = F)
#re-define protein column to shortened name
tmt <- tmt[,c(1:181,183)]
colnames(tmt)[182] <- "Protein"
#make new column for gene accession without transcript id
tmt$Gene <- substr(tmt$Accession, start = 1, stop = 14)
#Remove polymorphic peptides
tmt.withPolymorphic <- tmt
tmt <- tmt[-(grep("only", tmt$Accession)),]
#Convert to numeric
for(i in 1:170){
  tmt[,i] <- as.numeric(tmt[,i])
}
```
#Subset protein experiments
```{r subset BxM seedling leaf from original experiment}
#BxM from Original Experiment (OG)
tmt.SL <- tmt[,c(171:183,12:20)]
#Remove proteins not detected in one or more samples
tmt.SL <- na.omit(tmt.SL)
#Calculate averages of bio reps
tmt.SL$B73avg <- rowMeans(tmt.SL[,c("SeedlingLeaf.B73.r1", "SeedlingLeaf.B73.r2", "SeedlingLeaf.B73.r3")])
tmt.SL$Mo17avg <- rowMeans(tmt.SL[,c("SeedlingLeaf.Mo17.r1", "SeedlingLeaf.Mo17.r2", "SeedlingLeaf.Mo17.r3")])
tmt.SL$HYBavg <- rowMeans(tmt.SL[,c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3")])
tmt.SL$MPavg <- rowMeans(tmt.SL[,c("B73avg", "Mo17avg")])
#calculate ratios
tmt.SL$HYB.MP <- tmt.SL$HYBavg/tmt.SL$MPavg
tmt.SL$l2.B73xMo17.MP <- log2(tmt.SL$HYB.MP)
tmt.SL$B73.Mo17 <- tmt.SL$B73avg/tmt.SL$Mo17avg
tmt.SL$l2.B73.Mo17 <-log2(tmt.SL$B73.Mo17)
```
```{r subset BxM from day vs. night experiment (day)}
#BxM from Day vs. night experiment (DN)
tmt.DN <- tmt[,c(171:183,62:70)]
#Remove proteins not detected in one or more samples
tmt.DN <- na.omit(tmt.DN)
#Calculate averages of bio reps
tmt.DN$B73avg <- rowMeans(tmt.DN[,c("Day.B73.r1", "Day.B73.r2", "Day.B73.r3")])
tmt.DN$Mo17avg <- rowMeans(tmt.DN[,c("Day.Mo17.r1", "Day.Mo17.r2", "Day.Mo17.r3")])
tmt.DN$HYBavg <- rowMeans(tmt.DN[,c("Day.BM.r1", "Day.BM.r2", "Day.BM.r3")])
tmt.DN$MPavg <- rowMeans(tmt.DN[,c("B73avg", "Mo17avg")])
#calculate ratios
tmt.DN$HYB.MP <- tmt.DN$HYBavg/tmt.DN$MPavg
tmt.DN$l2.B73xMo17.MP <- log2(tmt.DN$HYB.MP)
tmt.DN$B73.Mo17 <- tmt.DN$B73avg/tmt.DN$Mo17avg
tmt.DN$l2.B73.Mo17 <-log2(tmt.DN$B73.Mo17)
```
```{r subset BxM mature leaf blade from original experiment}
#BxM from Original Experiment (OG)
tmt.LB <- tmt[,c(171:183,22:30)]
#Remove proteins not detected in one or more samples
tmt.LB <- na.omit(tmt.LB)
#Calculate averages of bio reps
tmt.LB$B73avg <- rowMeans(tmt.LB[,c("MatureLeaf.B73.r1", "MatureLeaf.B73.r2", "MatureLeaf.B73.r3")])
tmt.LB$Mo17avg <- rowMeans(tmt.LB[,c("MatureLeaf.Mo17.r1", "MatureLeaf.Mo17.r2", "MatureLeaf.Mo17.r3")])
tmt.LB$HYBavg <- rowMeans(tmt.LB[,c("MatureLeaf.BM.r1", "MatureLeaf.BM.r2", "MatureLeaf.BM.r3")])
tmt.LB$MPavg <- rowMeans(tmt.LB[,c("B73avg", "Mo17avg")])
#calculate ratios
tmt.LB$HYB.MP <- tmt.LB$HYBavg/tmt.LB$MPavg
tmt.LB$l2.B73xMo17.MP <- log2(tmt.LB$HYB.MP)
tmt.LB$B73.Mo17 <- tmt.LB$B73avg/tmt.LB$Mo17avg
tmt.LB$l2.B73.Mo17 <-log2(tmt.LB$B73.Mo17)
```
```{r subset BxM from day vs. night experiment (both day and night)}
#BxM from Day vs. night experiment (DN.both)
tmt.DN.both <- tmt[,c(171:183,62:70,72:80)]
#Remove proteins not detected in one or more samples
tmt.DN.both <- na.omit(tmt.DN.both)
#Calculate averages of bio reps
tmt.DN.both$day.B73avg <- rowMeans(tmt.DN.both[,c("Day.B73.r1", "Day.B73.r2", "Day.B73.r3")])
tmt.DN.both$day.Mo17avg <- rowMeans(tmt.DN.both[,c("Day.Mo17.r1", "Day.Mo17.r2", "Day.Mo17.r3")])
tmt.DN.both$day.HYBavg <- rowMeans(tmt.DN.both[,c("Day.BM.r1", "Day.BM.r2", "Day.BM.r3")])
tmt.DN.both$day.MPavg <- rowMeans(tmt.DN.both[,c("day.B73avg", "day.Mo17avg")])

tmt.DN.both$night.B73avg <- rowMeans(tmt.DN.both[,c("Night.B73.r1", "Night.B73.r2", "Night.B73.r3")])
tmt.DN.both$night.Mo17avg <- rowMeans(tmt.DN.both[,c("Night.Mo17.r1", "Night.Mo17.r2", "Night.Mo17.r3")])
tmt.DN.both$night.HYBavg <- rowMeans(tmt.DN.both[,c("Night.BM.r1", "Night.BM.r2", "Night.BM.r3")])
tmt.DN.both$night.MPavg <- rowMeans(tmt.DN.both[,c("night.B73avg", "night.Mo17avg")])
#calculate ratios
tmt.DN.both$day.HYB.MP <- tmt.DN.both$day.HYBavg/tmt.DN.both$day.MPavg
tmt.DN.both$day.l2.B73xMo17.MP <- log2(tmt.DN.both$day.HYB.MP)
tmt.DN.both$day.B73.Mo17 <- tmt.DN.both$day.B73avg/tmt.DN.both$day.Mo17avg
tmt.DN.both$day.l2.B73.Mo17 <-log2(tmt.DN.both$day.B73.Mo17)

tmt.DN.both$night.HYB.MP <- tmt.DN.both$night.HYBavg/tmt.DN.both$night.MPavg
tmt.DN.both$night.l2.B73xMo17.MP <- log2(tmt.DN.both$night.HYB.MP)
tmt.DN.both$night.B73.Mo17 <- tmt.DN.both$night.B73avg/tmt.DN.both$night.Mo17avg
tmt.DN.both$night.l2.B73.Mo17 <-log2(tmt.DN.both$night.B73.Mo17)
```
```{r subset all hybrids from 6-hybrid experiment}
#Subset all hybrids from 6-hybrid experiment (6H)
tmt.6H <- tmt[,c(171:183,31:60)]
#Remove proteins not detected in one or more samples
tmt.6H <- na.omit(tmt.6H)
#calculate averages
tmt.6H$B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r1", "X6Hybrid.B73.r2", "X6Hybrid.B73.r3")])
tmt.6H$Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r1", "X6Hybrid.Mo17.r2", "X6Hybrid.Mo17.r3")])
tmt.6H$B73xMo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B73xMo17.r1", "X6Hybrid.B73xMo17.r2", "X6Hybrid.B73xMo17.r3")])
tmt.6H$Mo17xB73 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17xB73.r1", "X6Hybrid.Mo17xB73.r2", "X6Hybrid.Mo17xB73.r3")])
tmt.6H$B84 <- rowMeans(tmt.6H[,c("X6Hybrid.B84.r1", "X6Hybrid.B84.r2", "X6Hybrid.B84.r3")])
tmt.6H$B84xB73 <- rowMeans(tmt.6H[,c("X6Hybrid.B84xB73.r1", "X6Hybrid.B84xB73.r2", "X6Hybrid.B84xB73.r3")])
tmt.6H$B84xMo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B84xMo17.r1", "X6Hybrid.B84xMo17.r2", "X6Hybrid.B84xMo17.r3")])
tmt.6H$A682 <- rowMeans(tmt.6H[,c("X6Hybrid.A682.r1", "X6Hybrid.A682.r2", "X6Hybrid.A682.r3")])
tmt.6H$A682xB73 <- rowMeans(tmt.6H[,c("X6Hybrid.A682xB73.r1", "X6Hybrid.A682xB73.r2", "X6Hybrid.A682xB73.r3")])
tmt.6H$A682xMo17 <- rowMeans(tmt.6H[,c("X6Hybrid.A682xMo17.r1", "X6Hybrid.A682xMo17.r2", "X6Hybrid.A682xMo17.r3")])
#Calculate hyb/MP ratios
tmt.6H$B73xMo17.MP <- tmt.6H$B73xMo17/rowMeans(tmt.6H[,c("B73", "Mo17")])
tmt.6H$Mo17xB73.MP <- tmt.6H$Mo17xB73/rowMeans(tmt.6H[,c("B73", "Mo17")])
tmt.6H$B84xB73.MP <- tmt.6H$B84xB73/rowMeans(tmt.6H[,c("B84", "B73")])
tmt.6H$B84xMo17.MP <- tmt.6H$B84xMo17/rowMeans(tmt.6H[,c("B84", "Mo17")])
tmt.6H$A682xB73.MP <- tmt.6H$A682xB73/rowMeans(tmt.6H[,c("A682", "B73")])
tmt.6H$A682xMo17.MP <- tmt.6H$A682xMo17/rowMeans(tmt.6H[,c("A682", "Mo17")])
#Calculate parent/parent ratios
tmt.6H$B73.Mo17 <- tmt.6H$B73/tmt.6H$Mo17
tmt.6H$Mo17.B73 <- tmt.6H$Mo17/tmt.6H$B73
tmt.6H$B84.B73 <- tmt.6H$B84/tmt.6H$B73
tmt.6H$B84.Mo17 <- tmt.6H$B84/tmt.6H$Mo17
tmt.6H$A682.B73 <- tmt.6H$A682/tmt.6H$B73
tmt.6H$A682.Mo17 <- tmt.6H$A682/tmt.6H$Mo17
#Calculate log2 of all ratios
tmt.6H$l2.B73xMo17.MP <- log2(tmt.6H$B73xMo17.MP)
tmt.6H$l2.Mo17xB73.MP <- log2(tmt.6H$Mo17xB73.MP)
tmt.6H$l2.B84xB73.MP <- log2(tmt.6H$B84xB73.MP)
tmt.6H$l2.B84xMo17.MP <- log2(tmt.6H$B84xMo17.MP)
tmt.6H$l2.A682xB73.MP <- log2(tmt.6H$A682xB73.MP)
tmt.6H$l2.A682xMo17.MP <- log2(tmt.6H$A682xMo17.MP)
tmt.6H$l2.B73.Mo17 <- log2(tmt.6H$B73.Mo17)
tmt.6H$l2.Mo17.B73 <- log2(tmt.6H$Mo17.B73)
tmt.6H$l2.B84.B73 <- log2(tmt.6H$B84.B73)
tmt.6H$l2.B84.Mo17 <- log2(tmt.6H$B84.Mo17)
tmt.6H$l2.A682.B73 <- log2(tmt.6H$A682.B73)
tmt.6H$l2.A682.Mo17 <- log2(tmt.6H$A682.Mo17)
```
```{r subset all hybrids from RIL experiment}
#Subset all hybrids from RIL experiment (RIL)
tmt.RIL <- tmt[,c(171:183,81:170)]
#Remove proteins not detected in one or more samples
tmt.RIL <- na.omit(tmt.RIL)
#Calculate averages
tmt.RIL$BM.B73 <- rowMeans(tmt.RIL[,c("B73.1", "B73.3", "B73.4")])
tmt.RIL$BM.Mo17 <- rowMeans(tmt.RIL[,c("Mo17.1", "Mo17.3", "Mo17.4")])
tmt.RIL$BM.BM <- rowMeans(tmt.RIL[,c("BM.1", "BM.2", "BM.3", "BM.4")])
tmt.RIL$M0014.B73 <- rowMeans(tmt.RIL[,c("B73.1.1", "B73.2", "B73.3.1", "B73.4.1")])
tmt.RIL$M0014.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.1", "Mo17.2", "Mo17.3.1", "Mo17.4.1")])
tmt.RIL$M0014.14 <- rowMeans(tmt.RIL[,c("X14.1", "X14.2", "X14.3", "X14.4")])
tmt.RIL$M0014.14B <- rowMeans(tmt.RIL[,c("X14B.1", "X14B.2", "X14B.3", "X14B.4")])
tmt.RIL$M0014.14M <- rowMeans(tmt.RIL[,c("X14M.1", "X14M.2", "X14M.3", "X14M.4")])
tmt.RIL$M0016.B73 <- rowMeans(tmt.RIL[,c("B73.1.2", "B73.2.1", "B73.3.2", "B73.4.2")])
tmt.RIL$M0016.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.2", "Mo17.2.1", "Mo17.3.2", "Mo17.4.2")])
tmt.RIL$M0016.16 <- rowMeans(tmt.RIL[,c("X16.1", "X16.2", "X16.3", "X16.4")])
tmt.RIL$M0016.16B <- rowMeans(tmt.RIL[,c("X16B.1", "X16B.2", "X16B.3", "X16B.4")])
tmt.RIL$M0016.16M <- rowMeans(tmt.RIL[,c("X16M.1", "X16M.2", "X16M.3", "X16M.4")])
tmt.RIL$M0021.B73 <- rowMeans(tmt.RIL[,c("B73.1.3", "B73.2.2", "B73.3.3", "B73.4.3")])
tmt.RIL$M0021.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.3", "Mo17.2.2", "Mo17.3.3", "Mo17.4.3")])
tmt.RIL$M0021.21 <- rowMeans(tmt.RIL[,c("X21.1", "X21.2", "X21.3", "X21.4")])
tmt.RIL$M0021.21B <- rowMeans(tmt.RIL[,c("X21B.1", "X21B.2", "X21B.3", "X21B.4")])
tmt.RIL$M0021.21M <- rowMeans(tmt.RIL[,c("X21M.1", "X21M.2", "X21M.3", "X21M.4")])
tmt.RIL$M0317.B73 <- rowMeans(tmt.RIL[,c("B73.1.4", "B73.2.3", "B73.3.4", "B73.4.4")])
tmt.RIL$M0317.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.4", "Mo17.2.3", "Mo17.3.4", "Mo17.4.4")])
tmt.RIL$M0317.317 <- rowMeans(tmt.RIL[,c("X317.1", "X317.2", "X317.3", "X317.4")])
tmt.RIL$M0317.317B <- rowMeans(tmt.RIL[,c("X317B.1", "X317B.2", "X317B.3", "X317B.4")])
tmt.RIL$M0317.317M <- rowMeans(tmt.RIL[,c("X317M.1", "X317M.2", "X317M.3", "X317M.4")])
#Calculate hyb/mp ratios
tmt.RIL$BM.BM.MP <- tmt.RIL$BM.BM/rowMeans(tmt.RIL[,c("BM.B73", "BM.Mo17")], na.rm = T)
tmt.RIL$M0014.14B.MP <- tmt.RIL$M0014.14B/rowMeans(tmt.RIL[,c("M0014.14", "M0014.B73")], na.rm = T)
tmt.RIL$M0014.14M.MP <- tmt.RIL$M0014.14M/rowMeans(tmt.RIL[,c("M0014.14", "M0014.Mo17")], na.rm = T)
tmt.RIL$M0016.16B.MP <- tmt.RIL$M0016.16B/rowMeans(tmt.RIL[,c("M0016.16", "M0016.B73")], na.rm = T)
tmt.RIL$M0016.16M.MP <- tmt.RIL$M0016.16M/rowMeans(tmt.RIL[,c("M0016.16", "M0016.Mo17")], na.rm = T)
tmt.RIL$M0021.21B.MP <- tmt.RIL$M0021.21B/rowMeans(tmt.RIL[,c("M0021.21", "M0021.B73")], na.rm = T)
tmt.RIL$M0021.21M.MP <- tmt.RIL$M0021.21M/rowMeans(tmt.RIL[,c("M0021.21", "M0021.Mo17")], na.rm = T)
tmt.RIL$M0317.317B.MP <- tmt.RIL$M0317.317B/rowMeans(tmt.RIL[,c("M0317.317", "M0317.B73")], na.rm = T)
tmt.RIL$M0317.317M.MP <- tmt.RIL$M0317.317M/rowMeans(tmt.RIL[,c("M0317.317", "M0317.Mo17")], na.rm = T)
#Calculate parent/parent ratios
tmt.RIL$BM.B73.Mo17 <- tmt.RIL$BM.B73/tmt.RIL$BM.Mo17
tmt.RIL$M0014.14.B73 <- tmt.RIL$M0014.14/tmt.RIL$M0014.B73
tmt.RIL$M0014.14.Mo17 <- tmt.RIL$M0014.14/tmt.RIL$M0014.Mo17
tmt.RIL$M0016.16.B73 <- tmt.RIL$M0016.16/tmt.RIL$M0016.B73
tmt.RIL$M0016.16.Mo17 <- tmt.RIL$M0016.16/tmt.RIL$M0016.Mo17
tmt.RIL$M0021.21.B73 <- tmt.RIL$M0021.21/tmt.RIL$M0021.B73
tmt.RIL$M0021.21.Mo17 <- tmt.RIL$M0021.21/tmt.RIL$M0021.Mo17
tmt.RIL$M0317.317.B73 <- tmt.RIL$M0317.317/tmt.RIL$M0317.B73
tmt.RIL$M0317.317.Mo17 <- tmt.RIL$M0317.317/tmt.RIL$M0317.Mo17
#Calculate log2 of all ratios
tmt.RIL$l2.B73xMo17.MP <- log2(tmt.RIL$BM.BM.MP)
tmt.RIL$l2.M0014.14B.MP <- log2(tmt.RIL$M0014.14B.MP)
tmt.RIL$l2.M0014.14M.MP <- log2(tmt.RIL$M0014.14M.MP)
tmt.RIL$l2.M0016.16B.MP <- log2(tmt.RIL$M0016.16B.MP)
tmt.RIL$l2.M0016.16M.MP <- log2(tmt.RIL$M0016.16M.MP)
tmt.RIL$l2.M0021.21B.MP <- log2(tmt.RIL$M0021.21B.MP)
tmt.RIL$l2.M0021.21M.MP <- log2(tmt.RIL$M0021.21M.MP)
tmt.RIL$l2.M0317.317B.MP <- log2(tmt.RIL$M0317.317B.MP)
tmt.RIL$l2.M0317.317M.MP <- log2(tmt.RIL$M0317.317M.MP)
tmt.RIL$l2.B73.Mo17 <- log2(tmt.RIL$BM.B73.Mo17)
tmt.RIL$l2.M0014.14.B73 <- log2(tmt.RIL$M0014.14.B73)
tmt.RIL$l2.M0014.14.Mo17  <- log2(tmt.RIL$M0014.14.Mo17)
tmt.RIL$l2.M0016.16.B73 <- log2(tmt.RIL$M0016.16.B73)
tmt.RIL$l2.M0016.16.Mo17  <- log2(tmt.RIL$M0016.16.Mo17)
tmt.RIL$l2.M0021.21.B73 <- log2(tmt.RIL$M0021.21.B73)
tmt.RIL$l2.M0021.21.Mo17  <- log2(tmt.RIL$M0021.21.Mo17)
tmt.RIL$l2.M0317.317.B73 <- log2(tmt.RIL$M0317.317.B73)
tmt.RIL$l2.M0317.317.Mo17  <- log2(tmt.RIL$M0317.317.Mo17)
```
```{r subset acs mutant experiment}
#Subset acs mutant experiment (acs)
tmt.acs <- tmt[,c(171:183,1:10)]
#Remove proteins not detected in one or more samples
tmt.acs <- na.omit(tmt.acs)
#Calculate averages of bio reps - acs
tmt.acs$B73avg <- rowMeans(tmt.acs[,c("B73.acs.B73.r1", "B73.acs.B73.r2", "B73.acs.B73.r3", "B73.acs.B73.r4", "B73.acs.B73.r5")])
tmt.acs$ACSavg <- rowMeans(tmt.acs[,c("B73.acs.acs.r1", "B73.acs.acs.r2", "B73.acs.acs.r3", "B73.acs.acs.r4", "B73.acs.acs.r5")])
#calculate ratio
tmt.acs$ACS.B73 <- tmt.acs$ACSavg/tmt.acs$B73avg
#calculate log2
tmt.acs$l2.ACS.B73 <- log2(tmt.acs$ACS.B73)
#merge with BxM from original SL data (OG)
tmt.acs.OG <- merge.data.frame(tmt.acs, tmt.SL[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
#merge with BxM from original LB data
tmt.acs.LB <- merge.data.frame(tmt.acs, tmt.LB[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
#merge with BxM from 6-hybrid data (6H)
tmt.acs.6H <- merge.data.frame(tmt.acs, tmt.6H[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
#merge with BxM from RIL data (RIL)
tmt.acs.RIL <- merge.data.frame(tmt.acs, tmt.RIL[,c("Accession", "l2.B73xMo17.MP")], by.x = "Accession", by.y = "Accession", all=F)
```

#Import RNA data
```{r import RNA data}
#Original experiment
cpm.OG.data <- read.table(paste0(PathToData, "cpm.tsv"), sep = '\t', header = TRUE)
#6-hybrid experiment
cpm.6H <- read.csv(paste0(PathToData, "SupplementalDataSet6.csv"), stringsAsFactors = F)
#RIL experiment
cpm.RIL <- read.csv(paste0(PathToData, "SupplementalDataSet7.csv"), stringsAsFactors = F)
```



#Subset RNA experiments
```{r subset BxM seedling leaf from original experiment}
#Original experiment seedling leaf BxM
cpm.SL <- as.data.frame(cpm.OG.data[,c("gid","BR193","BR194","BR195","BR196","BR197","BR198","BR199", "BR200","BR201")])
colnames(cpm.SL) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "B73xMo17.1", "B73xMo17.2", "B73xMo17.3")
#Remove genes not detected in one or mor rep
cpm.SL[cpm.SL==0] <- NA
cpm.SL <- na.omit(cpm.SL)
#Calculate averages of bio reps
cpm.SL$B73 <- rowMeans(cpm.SL[,c("B73.1", "B73.2", "B73.3")])
cpm.SL$Mo17 <- rowMeans(cpm.SL[,c("Mo17.1", "Mo17.2", "Mo17.3")])
cpm.SL$B73xMo17 <- rowMeans(cpm.SL[,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")])
#calculate ratios
cpm.SL$B73xMo17.MP <- cpm.SL$B73xMo17/rowMeans(cpm.SL[,c("B73", "Mo17")])
cpm.SL$l2.B73xMo17.MP <- log2(cpm.SL$B73xMo17.MP)
cpm.SL$B73.Mo17 <- cpm.SL$B73/cpm.SL$Mo17
cpm.SL$l2.B73.Mo17 <-log2(cpm.SL$B73.Mo17)
##Remove genes that weren't detected as protein
cpm.SL.protein <- cpm.SL[cpm.SL$Gene%in%tmt.SL$Gene,]
```
```{r subset BxM leaf blade from original experiment}
#Original experiment seedling leaf BxM
cpm.LB <- as.data.frame(cpm.OG.data[,c("gid","BR001", "BR002", "BR004", "BR003", "BR005", "BR007", "BR006", "BR008", "BR009")])
colnames(cpm.LB) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "B73xMo17.1", "B73xMo17.2", "B73xMo17.3")
#Remove genes not detected in one or more rep
cpm.LB[cpm.LB==0] <- NA
cpm.LB <- na.omit(cpm.LB)
#Calculate averages of bio reps
cpm.LB$B73 <- rowMeans(cpm.LB[,c("B73.1", "B73.2", "B73.3")])
cpm.LB$Mo17 <- rowMeans(cpm.LB[,c("Mo17.1", "Mo17.2", "Mo17.3")])
cpm.LB$B73xMo17 <- rowMeans(cpm.LB[,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")])
#calculate ratios
cpm.LB$B73xMo17.MP <- cpm.LB$B73xMo17/rowMeans(cpm.LB[,c("B73", "Mo17")])
cpm.LB$l2.B73xMo17.MP <- log2(cpm.LB$B73xMo17.MP)
cpm.LB$B73.Mo17 <- cpm.LB$B73/cpm.LB$Mo17
cpm.LB$l2.B73.Mo17 <-log2(cpm.LB$B73.Mo17)
##Remove genes that weren't detected as protein
cpm.LB.protein <- cpm.LB[cpm.LB$Gene%in%tmt.LB$Gene,]
```
```{r subset 6 hybrid RNA data}
#Remove genes not detected in one or mor rep
cpm.6H[cpm.6H==0] <- NA
cpm.6H <- na.omit(cpm.6H)
#Calculate averages
cpm.6H$B73 <- rowMeans(cpm.6H[,c("B73.2", "B73.3", "B73.4")], na.rm = T)
cpm.6H$B84 <- rowMeans(cpm.6H[,c("B84.2", "B84.3", "B84.4")], na.rm = T)
cpm.6H$Mo17 <- rowMeans(cpm.6H[,c("Mo17.2", "Mo17.3", "Mo17.4")], na.rm = T)
cpm.6H$A682 <- rowMeans(cpm.6H[,c("A682.2", "A682.3", "A682.4")], na.rm = T)
cpm.6H$B73xMo17 <- rowMeans(cpm.6H[,c("B73xMo17.2", "B73xMo17.3", "B73xMo17.4")], na.rm = T)
cpm.6H$Mo17xB73 <- rowMeans(cpm.6H[,c("Mo17xB73.2", "Mo17xB73.3", "Mo17xB73.4")], na.rm = T)
cpm.6H$B84xB73 <- rowMeans(cpm.6H[,c("B84xB73.2", "B84xB73.3", "B84xB73.4")], na.rm = T)
cpm.6H$B84xMo17 <- rowMeans(cpm.6H[,c("B84xMo17.2", "B84xMo17.3", "B84xMo17.4")], na.rm = T)
cpm.6H$A682xB73 <- rowMeans(cpm.6H[,c("A682xB73.2", "A682xB73.3", "A682xB73.4")], na.rm = T)
cpm.6H$A682xMo17 <- rowMeans(cpm.6H[,c("A682xMo17.2", "A682xMo17.3", "A682xMo17.4")], na.rm = T)
#calculate ratio of hybrid to midparent
cpm.6H$B73xMo17.MP <- cpm.6H$B73xMo17/rowMeans(cpm.6H[,c("B73","Mo17")], na.rm = T)
cpm.6H$Mo17xB73.MP <- cpm.6H$Mo17xB73/rowMeans(cpm.6H[,c("B73","Mo17")], na.rm = T)
cpm.6H$B84xB73.MP <- cpm.6H$B84xB73/rowMeans(cpm.6H[,c("B84","B73")], na.rm = T)
cpm.6H$B84xMo17.MP <- cpm.6H$B84xMo17/rowMeans(cpm.6H[,c("B84","Mo17")], na.rm = T)
cpm.6H$A682xB73.MP <- cpm.6H$A682xB73/rowMeans(cpm.6H[,c("A682","B73")], na.rm = T)
cpm.6H$A682xMo17.MP <- cpm.6H$A682xMo17/rowMeans(cpm.6H[,c("A682","Mo17")], na.rm = T)
#Calculate ratio of parent to parent
cpm.6H$B73.Mo17 <- cpm.6H$B73/cpm.6H$Mo17
cpm.6H$Mo17.B73 <- cpm.6H$Mo17/cpm.6H$B73
cpm.6H$B84.B73 <- cpm.6H$B84/cpm.6H$B73
cpm.6H$B84.Mo17 <- cpm.6H$B84/cpm.6H$Mo17
cpm.6H$A682.B73 <- cpm.6H$A682/cpm.6H$B73
cpm.6H$A682.Mo17 <- cpm.6H$A682/cpm.6H$Mo17
#Change Inf
#calculate log2 of ratios
cpm.6H$l2.B73xMo17.MP <- log2(cpm.6H$B73xMo17.MP)
cpm.6H$l2.Mo17xB73.MP <- log2(cpm.6H$Mo17xB73.MP)
cpm.6H$l2.B84xB73.MP <- log2(cpm.6H$B84xB73.MP)
cpm.6H$l2.B84xMo17.MP <- log2(cpm.6H$B84xMo17.MP)
cpm.6H$l2.A682xB73.MP <- log2(cpm.6H$A682xB73.MP )
cpm.6H$l2.A682xMo17.MP <- log2(cpm.6H$A682xMo17.MP)
cpm.6H$l2.B73.Mo17 <- log2(cpm.6H$B73.Mo17)
cpm.6H$l2.Mo17.B73 <- log2(cpm.6H$Mo17.B73)
cpm.6H$l2.B84.B73 <- log2(cpm.6H$B84.B73)
cpm.6H$l2.B84.Mo17 <- log2(cpm.6H$B84.Mo17)
cpm.6H$l2.A682.B73 <- log2(cpm.6H$A682.B73)
cpm.6H$l2.A682.Mo17 <- log2(cpm.6H$A682.Mo17)
##Remove genes that weren't detected as protein
cpm.6H.protein <- cpm.6H[cpm.6H$Gene%in%tmt.6H$Gene,]
```
```{r subset RIL RNA data}
#Remove genes not detected in one or mor rep
cpm.RIL[cpm.RIL==0] <- NA
cpm.RIL <- na.omit(cpm.RIL)
#Calculate averages
cpm.RIL$B73 <- rowMeans(cpm.RIL[,c("B73.1", "B73.2", "B73.3", "B73.4")], na.rm = T)
cpm.RIL$Mo17 <- rowMeans(cpm.RIL[,c("Mo17.1", "Mo17.2", "Mo17.3", "Mo17.4")], na.rm = T)
cpm.RIL$B73xMo17 <- rowMeans(cpm.RIL[,c("BxM.1", "BxM.2", "BxM.3", "BxM.4")], na.rm = T)
cpm.RIL$M0014 <- rowMeans(cpm.RIL[,c("M0014.1", "M0014.2", "M0014.3", "M0014.4")], na.rm = T)
cpm.RIL$M0014xB73 <- rowMeans(cpm.RIL[,c("M0014xB73.1", "M0014xB73.2", "M0014xB73.3", "M0014xB73.4")], na.rm = T)
cpm.RIL$M0014xMo17 <- rowMeans(cpm.RIL[,c("M0014xMo17.1", "M0014xMo17.2", "M0014xMo17.3", "M0014xMo17.4")], na.rm = T)
cpm.RIL$M0016 <- rowMeans(cpm.RIL[,c("M0016.1", "M0016.2", "M0016.3", "M0016.4")], na.rm = T)
cpm.RIL$M0016xB73 <- rowMeans(cpm.RIL[,c("M0016xB73.1", "M0016xB73.2", "M0016xB73.3")], na.rm = T)
cpm.RIL$M0016xMo17 <- rowMeans(cpm.RIL[,c("M0016xMo17.1", "M0016xMo17.2", "M0016xMo17.3", "M0016xMo17.4")], na.rm = T)
cpm.RIL$M0021 <- rowMeans(cpm.RIL[,c("M0021.1", "M0021.2", "M0021.3", "M0021.4")], na.rm = T)
cpm.RIL$M0021xB73 <- rowMeans(cpm.RIL[,c("M0021xB73.1", "M0021xB73.2", "M0021xB73.3", "M0021xB73.4")], na.rm = T)
cpm.RIL$M0021xMo17 <- rowMeans(cpm.RIL[,c("M0021xMo17.1", "M0021xMo17.2", "M0021xMo17.3", "M0021xMo17.4")], na.rm = T)
cpm.RIL$M0317 <- rowMeans(cpm.RIL[,c("M0317.1", "M0317.2", "M0317.3", "M0317.4")], na.rm = T)
cpm.RIL$M0317xB73 <- rowMeans(cpm.RIL[,c("M0317xB73.1", "M0317xB73.2", "M0317xB73.3", "M0317xB73.4")], na.rm = T)
cpm.RIL$M0317xMo17 <- rowMeans(cpm.RIL[,c("M0317xMo17.1", "M0317xMo17.2", "M0317xMo17.3", "M0317xMo17.4")], na.rm = T)
#calculate ratio of hybrid to midparent
cpm.RIL$B73xMo17.MP <- cpm.RIL$B73xMo17/rowMeans(cpm.RIL[,c("B73","Mo17")], na.rm = T)
cpm.RIL$M0014xB73.MP <- cpm.RIL$M0014xB73/rowMeans(cpm.RIL[,c("M0014", "B73")], na.rm = T)
cpm.RIL$M0014xMo17.MP <- cpm.RIL$M0014xMo17/rowMeans(cpm.RIL[,c("M0014", "Mo17")], na.rm = T)
cpm.RIL$M0016xB73.MP <- cpm.RIL$M0016xB73/rowMeans(cpm.RIL[,c("M0016", "B73")], na.rm = T)
cpm.RIL$M0016xMo17.MP <- cpm.RIL$M0016xMo17/rowMeans(cpm.RIL[,c("M0016", "Mo17")], na.rm = T)
cpm.RIL$M0021xB73.MP <- cpm.RIL$M0021xB73/rowMeans(cpm.RIL[,c("M0021", "B73")], na.rm = T)
cpm.RIL$M0021xMo17.MP <- cpm.RIL$M0021xMo17/rowMeans(cpm.RIL[,c("M0021", "Mo17")], na.rm = T)
cpm.RIL$M0317xB73.MP <- cpm.RIL$M0317xB73/rowMeans(cpm.RIL[,c("M0317", "B73")], na.rm = T)
cpm.RIL$M0317xMo17.MP <- cpm.RIL$M0317xMo17/rowMeans(cpm.RIL[,c("M0317", "Mo17")], na.rm = T)
#Calculate ratio of parent to parent
cpm.RIL$B73.Mo17 <- cpm.RIL$B73/cpm.RIL$Mo17
cpm.RIL$M0014.B73 <- cpm.RIL$M0014/cpm.RIL$B73
cpm.RIL$M0014.Mo17 <- cpm.RIL$M0014/cpm.RIL$Mo17
cpm.RIL$M0016.B73 <- cpm.RIL$M0016/cpm.RIL$B73
cpm.RIL$M0016.Mo17 <- cpm.RIL$M0016/cpm.RIL$Mo17
cpm.RIL$M0021.B73 <- cpm.RIL$M0021/cpm.RIL$B73
cpm.RIL$M0021.Mo17 <- cpm.RIL$M0021/cpm.RIL$Mo17
cpm.RIL$M0317.B73 <- cpm.RIL$M0317/cpm.RIL$B73
cpm.RIL$M0317.Mo17 <- cpm.RIL$M0317/cpm.RIL$Mo17
#calculate ratio of hybrid to midparent
cpm.RIL$l2.B73xMo17.MP <- log2(cpm.RIL$B73xMo17.MP)
cpm.RIL$l2.M0014xB73.MP <- log2(cpm.RIL$M0014xB73.MP)
cpm.RIL$l2.M0014xMo17.MP <- log2(cpm.RIL$M0014xMo17.MP)
cpm.RIL$l2.M0016xB73.MP <- log2(cpm.RIL$M0016xB73.MP)
cpm.RIL$l2.M0016xMo17.MP <- log2(cpm.RIL$M0016xMo17.MP)
cpm.RIL$l2.M0021xB73.MP <- log2(cpm.RIL$M0021xB73.MP)
cpm.RIL$l2.M0021xMo17.MP <- log2(cpm.RIL$M0021xMo17.MP )
cpm.RIL$l2.M0317xB73.MP <- log2(cpm.RIL$M0317xB73.MP)
cpm.RIL$l2.M0317xMo17.MP <- log2(cpm.RIL$M0317xMo17.MP)
#Calculate ratio of parent to parent
cpm.RIL$l2.B73.Mo17 <- log2(cpm.RIL$B73.Mo17)
cpm.RIL$l2.M0014.B73 <- log2(cpm.RIL$M0014.B73)
cpm.RIL$l2.M0014.Mo17 <- log2(cpm.RIL$M0014.Mo17)
cpm.RIL$l2.M0016.B73 <- log2(cpm.RIL$M0016.B73)
cpm.RIL$l2.M0016.Mo17 <- log2(cpm.RIL$M0016.Mo17)
cpm.RIL$l2.M0021.B73 <- log2(cpm.RIL$M0021.B73)
cpm.RIL$l2.M0021.Mo17 <- log2(cpm.RIL$M0021.Mo17)
cpm.RIL$l2.M0317.B73 <- log2(cpm.RIL$M0317.B73)
cpm.RIL$l2.M0317.Mo17 <- log2(cpm.RIL$M0317.Mo17)
##Remove genes that weren't detected as protein
cpm.RIL.protein <- cpm.RIL
cpm.RIL.protein <- cpm.RIL.protein[cpm.RIL.protein$Gene%in%tmt.RIL$Gene,]
```
#Plant Height data
```{r PlantHeights}
PlantHeights <- read.csv(paste0(PathToData, "SupplementalTable1.csv"), stringsAsFactors = F)
#Make vector of 6H ratios
PlantHeights.6h.MP <- PlantHeights$Hyb.MP[PlantHeights$Experiment=="6H"]
#Make vector of RIL ratios
PlantHeights.ril.MP <- PlantHeights$Hyb.MP[PlantHeights$Experiment=="RIL"]
```
```{r HighParent}
PlantHeights$HPheight <- NA
for(i in 1:nrow(PlantHeights)){
  if(PlantHeights[i,"MomHeight"] > PlantHeights[i,"DadHeight"]){
    PlantHeights[i,"HPheight"] <- PlantHeights[i, "MomHeight"]
  }
  else{PlantHeights[i,"HPheight"] <- PlantHeights[i, "DadHeight"]}
}

PlantHeights$Hyb.HP <- PlantHeights$HybridHeight/PlantHeights$HPheight

#Make vector of 6H ratios
PlantHeights.6h.HP <- PlantHeights$Hyb.HP[PlantHeights$Experiment=="6H"]
#Make vector of RIL ratios
PlantHeights.ril.HP <- PlantHeights$Hyb.HP[PlantHeights$Experiment=="RIL"]
```




#Define GOIs
##Accession conversion
```{r mgdb}
#Downloaded from MaizeGDB 2020/05/18, 07:11pm
mgdb <- read.csv(paste0(PathToData,"mgdb.csv"), stringsAsFactors = F, header = F)
colnames(mgdb) <- mgdb[5,]
mgdb <- mgdb[-c(1:5),]
```
```{r organelleEntrez}
organelleEntrez <- tmt[grep("Zema|Zeam", tmt$Accession),]
organelleEntrez <- data.frame("Accession"=organelleEntrez$Accession, "Entrez"=NA)
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp003.1"] <- "845178"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp004.1"] <- "845232"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp008.1"] <- "845201"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp013.1"] <- "845225"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp014.1"] <- "845226"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp016.1"] <- "845236"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp017.1"] <- "845174"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp018.1"] <- "845173"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp021.1"] <- "845229"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp022.1"] <- "845196"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp023.1"] <- "845195"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp024.1"] <- "1466361"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp026.1"] <- "845238"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp027.1"] <- "845189"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp028.1"] <- "845190"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp029.1"] <- "845182"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp030.1"] <- "845171"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp032.1"] <- "845212"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp035.1"] <- "845175"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp036.1"] <- "845191"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp039.1"] <- "845204"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp040.1"] <- "845203"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp043.1"] <- "845198"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp044.1"] <- "845222"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp045.1"] <- "845233"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp046.1"] <- "845217"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp047.1"] <- "845176"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp048.1"] <- "845200"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp050.1"] <- "845211"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp051.1"] <- "845205"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp052.1"] <- "845192"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp053.1"] <- "845193"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp054.1"] <- "845224"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp055.1"] <- "845228"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp056.1"] <- "845223"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp057.1"] <- "845177"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp058.1"] <- "845241"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp059.1"] <- "845213"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp060.1"] <- "845214"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp061.1"] <- "845237"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp062.1"] <- "845218"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp063.1"] <- "845235"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp065.1"] <- "845216"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp067.1"] <- "845220"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp075.1"] <- "845181"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp082.1"] <- "845230"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp083.1"] <- "845185"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp084.1"] <- "845221"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp085.1"] <- "1466378"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp086.1"] <- "845183"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp087.1"] <- "1466379"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp088.1"] <- "845184"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp090.1"] <- "845188"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp091.1"] <- "845179"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp092.1"] <- "845187"

organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp008.1"] <- "4055864"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp024.1"] <- "4055860"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp032.1"] <- "4055888"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp033.1"] <- "4055938"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp036.1"] <- "4055894"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp050.1"] <- "4055956"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp051.1"] <- "4055881"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp057.1"] <- "4055973"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp081.1"] <- "4055890"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp092.1"] <- "4055837"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp097.1"] <- "4055857"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp098.1"] <- "4055896"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp113.1"] <- "4055802"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp119.1"] <- "4055928"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp129.1"] <- "4055878"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp136.1"] <- "4056004"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp159.1"] <- "4055794"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp167.1"] <- "4056007"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp179.1"] <- "4056006"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp185.1"] <- "4055892"
```
```{r Entrez conversion}
Entrez <- mgdb[,c(1,12)]
colnames(Entrez) <- c("Accession", "Entrez")
Entrez <- rbind(Entrez, organelleEntrez)
```
##Plastid-encoded
```{r PE}
#Plastid-encoded (PE) proteins
PE<-tmt[grep("ZemaCp", tmt$Accession),]
#define vector of PE photosynthesis-associated proteins
PEI <- PE$Gene[grep("psa", PE$Protein)]
PEII <- PE$Gene[grep("psb", PE$Protein)]
PE.NADH <- PE$Gene[grep("ndh", PE$Protein)]
PE.b6f <- PE$Gene[grep("pet", PE$Protein)]
PE.ATP <- PE$Gene[grep("atp", PE$Protein)]
PE.rubisco <- PE$Gene[grep("rbc", PE$Protein)]
PhAPGs <- c(PEI,PEII,PE.NADH,PE.b6f,PE.ATP,PE.rubisco)
#define vector of PE ribosomal proteins
PE.ribo <- PE$Gene[grep("rps|rpl", PE$Protein)]
```

##PhANGs
###KEGG - zma00195
```{r KEGG.zma00195.digenomic}
#Retrieved from KEGG on 02/26/2020
KEGG.zma00195 <- read.csv(paste0(PathToData, "KEGGzma00195.csv"), stringsAsFactors = F)
#Get v4 accessions
KEGG.zma00195.digenomic <- join(KEGG.zma00195, Entrez)
```
###KEGG - zma00196
```{r KEGG.zma00196}
#Retrieved from KEGG on 02/26/2020
KEGG.zma00196 <- read.csv(paste0(PathToData, "KEGGzma00196.csv"), stringsAsFactors = F)
#Get v4 accessions
KEGG.zma00196 <- join(KEGG.zma00196, Entrez)
```

###CornCyc - Light Reactions
```{r CornCycLightRxns.digenomic}
#Retrieved from CornCyc on 02/26/2020
CornCycLightRxns.digenomic <- read.csv(paste0(PathToData, "CornCycLightRxns.csv"), stringsAsFactors = F)
```

###NE Rubisco
```{r NE.Rubisco}
#Defined by MaizeGDB
NE.Rubisco <- c("Zm00001d052595", "Zm00001d004894")
```

###Combine groups
```{r PhANGs}
PhA <- unique(na.omit(c(KEGG.zma00195.digenomic$Accession, KEGG.zma00196$Accession, CornCycLightRxns.digenomic$Gene.Accession, NE.Rubisco)))
#Trim to nuclear-encoded
PhANGs <- PhA[grep("Zm", PhA)]
```


##Plastid background
```{r plastid}
#Import supplemental Table I from Friso et al. 2010
friso <- read.csv(paste0(PathToData,"Friso2010_SupTab1.csv"), stringsAsFactors = F)
colnames(friso) <- as.character(friso[1,])
friso <- friso[-1,]
friso <- friso[friso$`best gene model (selected)`=="yes",]
friso$Gene <- substr(friso$Accession, start = 1, stop = 13)
#convert to v4 accessions
friso <- merge.data.frame(friso, mgdb[,c(1,11)], by.x = "Gene", by.y = "v3_gene_model")
friso.v4 <- friso$v4_gene_model
#add previously called photosynthesis digenomic proteins
plastid <- as.character(unique(na.omit(c(friso.v4, PE, PhANGs))))
```

##Nuclear-encoded plastid ribosome
```{r NE.PtRibo}
#grep plastid ribosomal proteins from NCBI protein name column
ribosom <- tmt[(grep("ribosom", tmt$Protein)),]
ribosom.pt <- ribosom[(grep("30S|50S", ribosom$Protein)),]
# Read in Maize GAMER GO dataset, which lists maize V4 accessions with all associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim(paste0(PathToData, "maize.B73.AGPv4.aggregate.gaf"), sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$GOterm <- AnnotationDbi::Term(as.character(gamer$term_accession))
#merge gamer GO terms
ribosom.pt <- merge.data.frame(ribosom.pt[,c(176,183)], gamer[,c(2,18)], by.x = "Gene", by.y = "db_object_id", all.x = T, all.y = F)
#Convert duplicated rows into multiple columns for each GO term
ribosom.pt.w <- dcast(setDT(ribosom.pt), Accession~rowid(Accession, prefix="GOterm"), value.var="GOterm")
ribosom.pt <- merge.data.frame(ribosom.pt, ribosom.pt.w, by.x = "Accession", by.y = "Accession")
ribosom.pt <- distinct(ribosom.pt)
#Combine columns of each pathway into single column
ribosom.pt <- unite(ribosom.pt, "GOterm", GOterm1:GOterm9, sep = " // ")
ribosom.pt$GOterm <- gsub(ribosom.pt$GOterm, pattern = " // NA", replacement = "")
ribosom.pt <- distinct(ribosom.pt)
#Define NE plastid ribosomal proteins as those having chloroplast or plastid cellular compartment GO terms
NE.PtRibo <- ribosom.pt$Gene[grep("chloropl|plastid", ribosom.pt$GOterm)]
```


##Cytosolic ribosome
```{r CytoRibo}
#Cytosolic ribosomal proteins (NCBI Annotation)
CytoRibo<-tmt[(grep("40S|60S", tmt$Protein)),]
#remove the mitochondrial and biogenesis proteins
CytoRibo<- CytoRibo$Gene[-(grep("mito|Mito|biogen|Biogen", CytoRibo$Protein))]
```
##Etyhlene Biosynthesis
```{r CornCycEtBio}
#Retrieved from CornCyc on 05/11/2020
CornCycEtBio <- read.csv(paste0(PathToData, "CornCycEtBio.csv"), stringsAsFactors = F)
EtBio <- CornCycEtBio$Gene.Accession
SAMS <- CornCycEtBio$Gene.Accession[CornCycEtBio$Reaction.EC=="2.5.1.6"]
ACS <- CornCycEtBio$Gene.Accession[CornCycEtBio$Reaction.EC=="4.4.1.14"]
ACO <- CornCycEtBio$Gene.Accession[CornCycEtBio$Reaction.EC=="1.14.17.4"]
```
```{r acs2 and acs6}
acs2 <- "Zm00001d002592"
acs6 <- "Zm00001d033862"
```
```{r homologs}
AtACO1hom <- "Zm00001d036955"
AtACO2hom <- "Zm00001d020686"
AtSAM1AtSAM2hom <- "Zm00001d010920"
```

##GOI list
```{r make GOIs list}
GOIs <- list(PE.ribo, PhANGs, PhAPGs, plastid, NE.PtRibo, CytoRibo, EtBio, acs2, acs6, AtACO1hom, AtACO2hom, AtSAM1AtSAM2hom, SAMS, ACS, ACO)
names(GOIs) <- c("PE.ribo", "PhANGs", "PhAPGs", "plastid", "NE.PtRibo", "CytoRibo", "EtBio", "acs2",  "acs6", "AtACO1hom", "AtACO2hom", "AtSAM1AtSAM2hom", "SAMS", "ACS", "ACO")
saveRDS(GOIs, "GOIs")
```
```{r read GOI list}
GOIs <- readRDS("GOIs")
```
#Define settings and functions for plotting
```{r CorFun}
#using dataframe and vector as input
CorFun <- function(x, xcols, y){
  for(i in 1:nrow(x)){
    v <- as.numeric(x[i,xcols])
    c <- cor(v, y, method = "pearson")
    x[i,"cor"]<- c
  }
  x
}
```
```{r CorFun2}
#Using two column sets from the same dataframe as input
CorFun2 <- function(x, col1, col2){
  x$cor <- NA
  for(i in 1:nrow(x)){
    v1 <- as.numeric(x[i,col1])
    v2 <- as.numeric(x[i,col2])
    c <- cor(v1, v2, method = "pearson")
    x[i,"cor"]<- c
  }
  x}
```
```{r get_legend}
get_legend<-function(myggplot){
  tmp <- ggplot_gtable(ggplot_build(myggplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}
```
```{r T-test function}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```
```{r plotting colors}
colors.p <- c("Other"="grey70", "Plastid-localized"="black", "PE plastid ribosome"="#ef73ef", "NE plastid ribosome"="#a742da", "PhAPGs"="#7ae281", "PhANGs"="#009fa7", "Ethylene biosynthesis"="#fab5bd")

colors.v <- c("#000000","#004949","#009292","#ff6db6","#ffb6db",
 "#490092","#006ddb","#b66dff","#6db6ff","#b6dbff",
 "#920000","#924900","#db6d00","#24ff24","#ffff6d")

colors.e <- c("Other"="black","Methionine adenosyltransferase"="#24ff24", "ACC synthase"="#ff6db6", "ACC oxidase"="#920000")

colors.ea <- c("Other"="grey", "SAMS"="black", "ACS"="black", "ACO"="black", "ZmACS2"="#24ff24", "ZmACS6"="#ff6db6", "AtACO1 homolog"="#006ddb", "AtACO2 homolog"="#b66dff", "AtSAM1/AtSAM2 homolog"="#db6d00")

colors.pe <- c("Plastid-localized"="black", "PE plastid ribosome"="#ef73ef", "NE plastid ribosome"="#a742da", "PhAPGs"="#7ae281", "PhANGs"="#009fa7", "Ethylene biosynthesis"="#fab5bd")

colors.c <- c("Other"="black", "Cytosolic ribosome"="#ffa744", "NE plastid ribosome"="#ff0000", "PhANGs"="#3447ff")

colors.pc <- c("Other"="black", "PE plastid ribosome"="#ef73ef", "NE plastid ribosome"="#a742da", "PhAPGs"="#7ae281", "PhANGs"="#009fa7", "Cytosolic ribosome"="#fab5bd")
```
#Figure 1
```{r setup dataframe tmt.BxM.SL.plastid.all4}
#copy
tmt.BxM.SL.OG <- tmt.SL
tmt.BxM.SL.lb <- tmt.LB
tmt.BxM.SL.6h <- tmt.6H
tmt.BxM.SL.ril <- tmt.RIL
#name plot panels
tmt.BxM.SL.OG$plot <- "A"
tmt.BxM.SL.lb$plot <- "B"
tmt.BxM.SL.6h$plot <- "C"
tmt.BxM.SL.ril$plot <- "D"
#trim
tmt.BxM.SL.OG <- tmt.BxM.SL.OG[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.SL.lb <- tmt.BxM.SL.lb[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.SL.6h <- tmt.BxM.SL.6h[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.SL.ril <- tmt.BxM.SL.ril[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
#bind
tmt.BxM.SL <- rbind(tmt.BxM.SL.OG,tmt.BxM.SL.lb)
tmt.BxM.SL <- rbind(tmt.BxM.SL,tmt.BxM.SL.6h)
tmt.BxM.SL <- rbind(tmt.BxM.SL,tmt.BxM.SL.ril)
```
```{r plot tmt.BxM.SL.plastid.all4}
 plot<-unique(tmt.BxM.SL[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.BxM.SL[,"plot"])
  for(i in 1:length(plot)){
    df<-tmt.BxM.SL[which(tmt.BxM.SL[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -.75] <- -.75
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= .75] <- .75

df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 <= -1] <- -1
df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims

df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
      xlim(c(-.75,.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 B73/Mo17")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text =element_text(size=10), legend.text = element_text(size=10), axis.title = element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3, hjust = -0.3), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 2))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.B73xMo17.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.B73.Mo17"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
       des <- "AB
               CD
               EE"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], nrow = 3, heights = c(3,3,1.2), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "Figure1.pdf"),
          dpi=300, width=6.5, height=6.84)  
```




#Figure 2
```{r tmt.cpm.RIL.6H.Hyb2MP.cor.densities}
#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[,c(6,127:135)], tmt.6H[,c(6,54:59)], by.x = "Accession", by.y = "Accession", all = F)
#combine RIL and 6 hybrids RNA
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL[,c(1,76:84)], cpm.6H[,c(1,42:47)], by.x = "Gene", by.y = "Gene", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#run correlation function
cpm.RIL.6H.Hyb2MP.cor <- CorFun(cpm.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#Combine data for plotting
tmt.RIL.6H.Hyb2MP.cor$data <- "Protein"
cpm.RIL.6H.Hyb2MP.cor$data <- "RNA"
tmt.cpm.RIL.6H.Hyb2MP.cor <- rbind(tmt.RIL.6H.Hyb2MP.cor[,c("Gene", "cor", "data")], cpm.RIL.6H.Hyb2MP.cor[,c("Gene", "cor", "data")])
#Set GOIs
tmt.cpm.RIL.6H.Hyb2MP.cor$Category <- "Other"
tmt.cpm.RIL.6H.Hyb2MP.cor$Order <- 1
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.cpm.RIL.6H.Hyb2MP.cor[(tmt.cpm.RIL.6H.Hyb2MP.cor$Gene %in% GOIs$PhAPGs),"Order"] <- 6
tmt.cpm.RIL.6H.Hyb2MP.cor <- tmt.cpm.RIL.6H.Hyb2MP.cor[(order(tmt.cpm.RIL.6H.Hyb2MP.cor$Order)),]
tmt.cpm.RIL.6H.Hyb2MP.cor$Category <- factor(tmt.cpm.RIL.6H.Hyb2MP.cor$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
tmt.cpm.RIL.6H.Hyb2MP.cor.plot <- ggplot(tmt.cpm.RIL.6H.Hyb2MP.cor, mapping=aes(y=Category, x=cor, fill=Category,  linetype=data))+
  geom_density_ridges(alpha=0.7) +
  scale_fill_manual(values=colors.pc, guide = 'none')+
  scale_linetype_manual(values = c("RNA"="dotted", "Protein"="solid"), name="")+
  theme_ridges() +
  theme(axis.text =element_text(size=10), legend.text = element_text(size=10), axis.title = element_text(size=10), legend.position = "bottom", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=10), legend.justification = c(0.5,1))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)
tmt.cpm.RIL.6H.Hyb2MP.cor.plot
ggsave(filename =paste0(PathToPlots, "Figure2.pdf"), dpi=300, width=3.25, height=2.74)
```
#Figure 3
```{r setup dataframe tmt.acs.all4.plastid}
panel.acs.tmt.OG <- tmt.acs.OG
panel.acs.tmt.LB <- tmt.acs.LB
panel.acs.tmt.6h <- tmt.acs.6H
panel.acs.tmt.ril <- tmt.acs.RIL

panel.acs.tmt.OG$plot <- "A"
panel.acs.tmt.LB$plot <- "B"
panel.acs.tmt.6h$plot <- "C"
panel.acs.tmt.ril$plot <- "D"


panel.acs.tmt.OG <- panel.acs.tmt.OG[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]
panel.acs.tmt.LB <- panel.acs.tmt.LB[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]
panel.acs.tmt.6h <- panel.acs.tmt.6h[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]
panel.acs.tmt.ril <- panel.acs.tmt.ril[,c("plot", "Gene", "l2.ACS.B73", "l2.B73xMo17.MP")]


panel.acs.tmt <- rbind(panel.acs.tmt.OG,panel.acs.tmt.LB)
panel.acs.tmt <- rbind(panel.acs.tmt,panel.acs.tmt.6h)
panel.acs.tmt <- rbind(panel.acs.tmt,panel.acs.tmt.ril)
```
```{r plot tmt.acs.all4.plastid}
  plot<-unique(panel.acs.tmt[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(panel.acs.tmt[,"plot"])
  for(i in 1:length(plot)){
    df<-panel.acs.tmt[which(panel.acs.tmt[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -0.75] <- -0.75
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= 0.75] <- 0.75
df.lims$l2.ACS.B73[df.lims$l2.ACS.B73 <= -1] <- -1
df.lims$l2.ACS.B73[df.lims$l2.ACS.B73 >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$EtBio),"Category"] <- "Ethylene biosynthesis"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$EtBio),"Order"] <- 7
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid-localized", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs", "Ethylene biosynthesis"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.ACS.B73, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.pe, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
      xlim(c(-0.75,0.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 acs/B73")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3, hjust = -0.3),  legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 2))

legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.B73xMo17.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.pe, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.ACS.B73"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.pe, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  des <- "AB
          CD
          EE"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], nrow = 3, heights = c(3,3,1.2), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "Figure3.pdf"),
          dpi=300, width=6.5, height=6.84)
```
#Figure S1
```{r setup dataframe CV}
#Protein CV
##Calculate CV for BxM reps
tmt.SL$CV <- NA
for(i in 1:nrow(tmt.SL)){
  tmt.SL[i,"CV"] <- cv(as.numeric(tmt.SL[i,c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3")]))
}
tmt.6H$CV <- NA
for(i in 1:nrow(tmt.6H)){
  tmt.6H[i,"CV"] <- cv(as.numeric(tmt.6H[i,c("X6Hybrid.B73xMo17.r1", "X6Hybrid.B73xMo17.r2", "X6Hybrid.B73xMo17.r3")]))
}
tmt.RIL$CV <- NA
for(i in 1:nrow(tmt.RIL)){
  tmt.RIL[i,"CV"] <- cv(as.numeric(tmt.RIL[i,c("BM.1", "BM.2", "BM.3", "BM.4")]))
}
#RNA CV
##Calculate CV
cpm.SL.protein$CV <- NA
for(i in 1:nrow(cpm.SL.protein)){
  cpm.SL.protein[i,"CV"] <- cv(as.numeric(cpm.SL.protein[i,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")]))
}
cpm.6H.protein$CV <- NA
for(i in 1:nrow(cpm.6H.protein)){
  cpm.6H.protein[i,"CV"] <- cv(as.numeric(cpm.6H.protein[i,c("B73xMo17.2", "B73xMo17.3", "B73xMo17.4")]))
}
cpm.RIL.protein$CV <- NA
for(i in 1:nrow(cpm.RIL.protein)){
  cpm.RIL.protein[i,"CV"] <- cv(as.numeric(cpm.RIL.protein[i,c("BxM.1", "BxM.2", "BxM.3", "BxM.4")]))
}
#merge protein and RNA CVs
CV.SL.protein <- merge.data.frame(tmt.SL[,c("Gene", "CV")], cpm.SL.protein[,c("Gene", "CV")], by.x = "Gene", by.y = "Gene", all=F)
CV.6h.protein <- merge.data.frame(tmt.6H[,c("Gene", "CV")], cpm.6H.protein[,c("Gene", "CV")], by.x = "Gene", by.y = "Gene", all=F)
CV.ril.protein <- merge.data.frame(tmt.RIL[,c("Gene", "CV")], cpm.RIL.protein[,c("Gene", "CV")], by.x = "Gene", by.y = "Gene", all=F)
#Rename columns
colnames(CV.SL.protein) <- c("Gene", "Protein", "RNA")
colnames(CV.6h.protein) <- c("Gene", "Protein", "RNA")
colnames(CV.ril.protein) <- c("Gene", "Protein", "RNA")
#Define experiment
CV.SL.protein$Experiment <- "Original"
CV.6h.protein$Experiment <- "6-hybrid"
CV.ril.protein$Experiment <- "RIL"
#combine experiments into one dataframe
CV.protein <- rbind(CV.SL.protein,CV.6h.protein)
CV.protein <- rbind(CV.protein, CV.ril.protein)
CV.protein <- melt(CV.protein)
CV.protein <- na.omit(CV.protein)
```
```{r plot CV}
ggplot(CV.protein, mapping=aes(x=Experiment, y=value, fill=variable))+
  geom_violin(draw_quantiles = 0.5, )+
  scale_fill_manual(values=c("#af8dc3", "#7fbf7b"), name="")+
  theme_minimal()+
  ylab("Coefficient of variation (CV)")+
  theme(legend.position = c(0.09,0.88), axis.text =element_text(size=10), legend.text = element_text(size=10), axis.title = element_text(size=10))+
  theme(panel.grid.minor = element_blank())+
ggsave(paste0(PathToPlots, "FigureS1.pdf"), width = 3.25, height = 2.74)
```
#Figure S2

```{r setup dataframe tmt.BxM.HP.SL.plastid.all4}
#copy
tmt.BxM.HP.SL.OG <- tmt.SL
tmt.BxM.HP.SL.lb <- tmt.LB
tmt.BxM.HP.SL.6h <- tmt.6H
tmt.BxM.HP.SL.ril <- tmt.RIL
#name plot panels
tmt.BxM.HP.SL.OG$plot <- "A"
tmt.BxM.HP.SL.lb$plot <- "B"
tmt.BxM.HP.SL.6h$plot <- "C"
tmt.BxM.HP.SL.ril$plot <- "D"
#trim
tmt.BxM.HP.SL.OG <- tmt.BxM.HP.SL.OG[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.HP.SL.lb <- tmt.BxM.HP.SL.lb[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.HP.SL.6h <- tmt.BxM.HP.SL.6h[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.BxM.HP.SL.ril <- tmt.BxM.HP.SL.ril[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
#bind
tmt.BxM.HP.SL <- rbind(tmt.BxM.HP.SL.OG,tmt.BxM.HP.SL.lb)
tmt.BxM.HP.SL <- rbind(tmt.BxM.HP.SL,tmt.BxM.HP.SL.6h)
tmt.BxM.HP.SL <- rbind(tmt.BxM.HP.SL,tmt.BxM.HP.SL.ril)
```

```{r plot tmt.BxM.HP.SL.plastid.all4}
 plot<-unique(tmt.BxM.HP.SL[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.BxM.HP.SL[,"plot"])
  for(i in 1:length(plot)){
    df<-tmt.BxM.HP.SL[which(tmt.BxM.HP.SL[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -.75] <- -.75
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= .75] <- .75

df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 <= -1] <- -1
df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims

df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs"))
    #scatter plot with marginal density curves
    df.lims.plastid.den <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, color=Category))+
   geom_density(size=1)+
      xlim(-.75,.75)+
    scale_color_manual(values=colors.p)+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      ggtitle(plotLab[i])+
      theme(legend.position = "bottom", legend.title = element_blank(), legend.text = element_text(size=10), plot.title = element_text(size=10))+
      guides(col=guide_legend(nrow = 3))
    legend <- get_legend(df.lims.plastid.den)
    df.lims.plastid.den <- df.lims.plastid.den+theme(legend.position = "none")

    if(i==1){
      plotList<-list()
      plotList[[1]]<-df.lims.plastid.den
    }
    else{
      plotList <- c(plotList, list(df.lims.plastid.den))
    }
  }
      xlab <- ggplot(mapping = aes())+
        ggtitle("log2 BxM/High-parent")+
        theme(plot.title = element_text(size = 10, hjust = 0.5))
    xlab <- get_title(xlab)
    
    xscale <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, color=Category))+
   geom_density(size=1)+
      xlim(-.75,.75)+
      coord_cartesian(expand = F)+
      xlab("log2 BxM/High-parent")+
      theme(axis.text =element_text(size=10), axis.title = element_text(size=10),)
xscale <- get_x_axis(xscale)
    
  plotList <- c(plotList, list(xscale),list(xlab), list(legend))
       des <- "A
               B
               C
               D
               E
               G
               H"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], G=plotList[[6]], H=plotList[[7]], nrow=7, heights = c(3,3,3,3,.5,.75,1.5), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "FigureS2.pdf"),
          dpi=300, width=3.25, height=7.31)  
```


#Figure S3
```{r setup dataframe cpm.BxM.SL.plastid.all4}
cpm.BxM.SL.OG <- cpm.SL.protein
cpm.BxM.LB.OG <- cpm.LB.protein
cpm.BxM.SL.6h <- cpm.6H.protein
cpm.BxM.SL.ril <- cpm.RIL.protein

cpm.BxM.SL.OG$plot <- "A"
cpm.BxM.LB.OG$plot <- "B"
cpm.BxM.SL.6h$plot <- "C"
cpm.BxM.SL.ril$plot <- "D"

cpm.BxM.SL.OG <- cpm.BxM.SL.OG[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.BxM.LB.OG <- cpm.BxM.LB.OG[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.BxM.SL.6h <- cpm.BxM.SL.6h[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.BxM.SL.ril <- cpm.BxM.SL.ril[,c("plot", "Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")]

cpm.BxM.SL <- rbind(cpm.BxM.SL.OG,cpm.BxM.LB.OG)
cpm.BxM.SL <- rbind(cpm.BxM.SL,cpm.BxM.SL.6h)
cpm.BxM.SL <- rbind(cpm.BxM.SL,cpm.BxM.SL.ril)
```
```{r plot cpm.BxM.SL.plastid.all4}
plot<-unique(cpm.BxM.SL[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(cpm.BxM.SL[,"plot"])
  for(i in 1:length(plot)){
    df<-cpm.BxM.SL[which(cpm.BxM.SL[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP <= -1] <- -1
df.lims$l2.B73xMo17.MP[df.lims$l2.B73xMo17.MP >= 1] <- 1

df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 <= -2] <- -2
df.lims$l2.B73.Mo17[df.lims$l2.B73.Mo17 >= 2] <- 2
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.B73xMo17.MP, y=l2.B73.Mo17, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19))+
    theme_classic()+
      xlim(c(-1,1))+
      ylim(c(-2,2))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      xlab("log2 BxM/MP")+
      ylab("log2 B73/Mo17")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3., hjust = -0.3), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 1))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.B73xMo17.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.B73.Mo17"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
        des <- "AB
                CD
                EE"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], nrow = 3, heights = c(3,3,1.2), design = des)
  ggsave(plot=r, filename =paste0(PathToPlots, "FigureS3.pdf"),
          dpi=300, width=6.5, height=6.84)  
```

#Figure S4
```{r setup dataframe tmt.6H.plastid}
#subset hybrids
tmt.6h.a682xb73 <- tmt.6H[,c("Gene", "l2.A682xB73.MP", "l2.A682.B73")]
tmt.6h.b84xmo17 <- tmt.6H[,c("Gene", "l2.B84xMo17.MP", "l2.B84.B73")]
tmt.6h.mo17xb73 <- tmt.6H[,c("Gene", "l2.Mo17xB73.MP", "l2.Mo17.B73" )]
tmt.6h.b73xmo17 <- tmt.6H[,c("Gene","l2.B73xMo17.MP", "l2.B73.Mo17")]
tmt.6h.b84xb73 <- tmt.6H[,c("Gene", "l2.B84xB73.MP", "l2.B84.B73")]
tmt.6h.a682xmo17<- tmt.6H[,c("Gene", "l2.A682xMo17.MP", "l2.A682.Mo17")]
#rename columns
colnames(tmt.6h.a682xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.b84xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.mo17xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.b73xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.b84xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(tmt.6h.a682xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
#name plot panels
tmt.6h.a682xb73$plot <- "A"
tmt.6h.b84xmo17$plot <- "B"
tmt.6h.mo17xb73$plot <- "C"
tmt.6h.b73xmo17$plot <- "D"
tmt.6h.b84xb73$plot <- "E"
tmt.6h.a682xmo17$plot <- "F"
#combine
tmt.6h.all6 <- rbind(tmt.6h.a682xb73,tmt.6h.b84xmo17,tmt.6h.mo17xb73,tmt.6h.b73xmo17,tmt.6h.b84xb73,tmt.6h.a682xmo17)
```
```{r plot tmt.6H.plastid}
  plot<-unique(tmt.6h.all6[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.6h.all6[,"plot"])
  for(i in 1:length(plot)){
    df<-tmt.6h.all6[which(tmt.6h.all6[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP <= -0.75] <- -0.75
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP >= 0.75] <- 0.75

df.lims$l2.parent.parent[df.lims$l2.parent.parent <= -1] <- -1
df.lims$l2.parent.parent[df.lims$l2.parent.parent >= 1] <- 1
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 5
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid-localized", "PhANGs", "NE plastid ribosome", "PE plastid ribosome", "PhAPGs"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.HYB.MP, y=l2.parent.parent, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
      xlim(c(-0.75,0.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),axis.title = element_text(size = 10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3.5, hjust = -0.485), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 3))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
    if(i==1){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xB73/MP")+ ylab("log2 A682/B73")
    }
    else if(i==2){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xMo17/MP")+ ylab("log2 B84/Mo17")
    }
    else if(i==3){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 Mo17xB73/MP")+ ylab("log2 Mo17/B73")
    }
    else if(i==4){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B73xMo17/MP")+ ylab("log2 B73/Mo17")
    }
    else if(i==5){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xB73/MP")+ ylab("log2 B84/B73")
    }
    else if(i==6){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xMo17/MP")+ ylab("log2 A682/Mo17")
    }
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.HYB.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.parent.parent"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  des <- "AB
          CD
          EG
          HH"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], G=plotList[[6]], H=plotList[[7]], nrow = 4, heights = c(3,3,3,1.5), design = des)
  
ggsave(plot=r, filename =paste0(PathToPlots, "FigureS4.pdf"),
          dpi=300, width=6.5, height=7.5)  
```

#Figure S5
```{r setup dataframe cpm.6H.plastid}
#subset hybrids
cpm.6h.a682xb73 <- cpm.6H[,c("Gene", "l2.A682xB73.MP", "l2.A682.B73")]
cpm.6h.b84xmo17 <- cpm.6H[,c("Gene", "l2.B84xMo17.MP", "l2.B84.B73")]
cpm.6h.mo17xb73 <- cpm.6H[,c("Gene", "l2.Mo17xB73.MP", "l2.Mo17.B73" )]
cpm.6h.b73xmo17 <- cpm.6H[,c("Gene","l2.B73xMo17.MP", "l2.B73.Mo17")]
cpm.6h.b84xb73 <- cpm.6H[,c("Gene", "l2.B84xB73.MP", "l2.B84.B73")]
cpm.6h.a682xmo17<- cpm.6H[,c("Gene", "l2.A682xMo17.MP", "l2.A682.Mo17")]
#rename columns
colnames(cpm.6h.a682xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6h.b84xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6h.mo17xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6h.b73xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6h.b84xb73)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
colnames(cpm.6h.a682xmo17)[2:3] <- c("l2.HYB.MP", "l2.parent.parent")
#name plot panels
cpm.6h.a682xb73$plot <- "A"
cpm.6h.b84xmo17$plot <- "B"
cpm.6h.mo17xb73$plot <- "C"
cpm.6h.b73xmo17$plot <- "D"
cpm.6h.b84xb73$plot <- "E"
cpm.6h.a682xmo17$plot <- "F"
#combine
cpm.6h.all6 <- rbind(cpm.6h.a682xb73,cpm.6h.b84xmo17,cpm.6h.mo17xb73,cpm.6h.b73xmo17,cpm.6h.b84xb73,cpm.6h.a682xmo17)
```
```{r plot cpm.6H.plastid}
  plot<-unique(cpm.6h.all6[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(cpm.6h.all6[,"plot"])
  for(i in 1:length(plot)){
    df<-cpm.6h.all6[which(cpm.6h.all6[,"plot"]==plot[i]),]
    ###Set limits
df.lims <- df
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP <= -2] <- -2
df.lims$l2.HYB.MP[df.lims$l2.HYB.MP >= 2] <- 2

df.lims$l2.parent.parent[df.lims$l2.parent.parent <= -2] <- -2
df.lims$l2.parent.parent[df.lims$l2.parent.parent >= 2] <- 2
###Set GOIs
df.lims.plastid <- df.lims
df.lims.plastid$Category <- "Other"
df.lims.plastid$Order <- 1
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 3
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
df.lims.plastid[(df.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 4
###Re-order
df.lims.plastid <- df.lims.plastid[(order(df.lims.plastid$Order)),]
###remove "other"
df.lims.plastid <- df.lims.plastid[!df.lims.plastid$Category=="Other",]
###factor
df.lims.plastid$Category <- factor(df.lims.plastid$Category, levels = c("Plastid-localized", "PhANGs", "NE plastid ribosome"))
    #scatter plot with marginal density curves
    df.lims.plastid.scat <- ggplot(df.lims.plastid, mapping=aes(x=l2.HYB.MP, y=l2.parent.parent, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
      xlim(c(-2,2))+
      ylim(c(-2,2))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size = 10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3.5, hjust = -0.485), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(nrow = 1))
legend <- get_legend(df.lims.plastid.scat)
    df.lims.plastid.scat <- df.lims.plastid.scat+theme(legend.position = "none")
    if(i==1){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xB73/MP")+ ylab("log2 A682/B73")
    }
    else if(i==2){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xMo17/MP")+ ylab("log2 B84/Mo17")
    }
    else if(i==3){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 Mo17xB73/MP")+ ylab("log2 Mo17/B73")
    }
    else if(i==4){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B73xMo17/MP")+ ylab("log2 B73/Mo17")
    }
    else if(i==5){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 B84xB73/MP")+ ylab("log2 B84/B73")
    }
    else if(i==6){
      df.lims.plastid.scat <- df.lims.plastid.scat+ xlab("log2 A682xMo17/MP")+ ylab("log2 A682/Mo17")
    }
###Add density curves
    densityplotX<-axis_canvas(df.lims.plastid.scat, axis="x")+
      geom_density(aes(df.lims.plastid[,"l2.HYB.MP"], color=df.lims.plastid[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(df.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(df.lims.plastid[,"l2.parent.parent"], color=df.lims.plastid[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q <- insert_xaxis_grob(df.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList <- c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  des <- "AB
          CD
          EG
          HH"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], G=plotList[[6]], H=plotList[[7]], nrow = 4, heights = c(3,3,3,1.5), design = des)
  
ggsave(plot=r, filename =paste0(PathToPlots, "FigureS5.pdf"),
          dpi=300, width=6.5, height=7.5)  
```

#Figure S6
```{r setup dataframe LeafBladeVsSeedlingLeaf}
tmt.LB.SL <- merge.data.frame(tmt.LB[,c("Accession", "l2.B73xMo17.MP", "l2.B73.Mo17")], tmt.SL[,c("Accession", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Accession", by.y = "Accession", all=F)
tmt.LB.SL$Gene <- substr(tmt.LB.SL$Accession, start = 1, stop = 14)
cpm.LB.SL <- merge.data.frame(cpm.LB.protein[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], cpm.SL.protein[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Gene", by.y = "Gene", all = F)
```
```{r plot LeafBladeVsSeedlingLeaf.Hyb2MP}
###Set limits
tmt.LB.SL.lims <- tmt.LB.SL
tmt.LB.SL.lims$l2.B73xMo17.MP.x[tmt.LB.SL.lims$l2.B73xMo17.MP.x <= -1] <- -1
tmt.LB.SL.lims$l2.B73xMo17.MP.x[tmt.LB.SL.lims$l2.B73xMo17.MP.x >= 1] <- 1
tmt.LB.SL.lims$l2.B73xMo17.MP.y[tmt.LB.SL.lims$l2.B73xMo17.MP.y <= -1] <- -1
tmt.LB.SL.lims$l2.B73xMo17.MP.y[tmt.LB.SL.lims$l2.B73xMo17.MP.y >= 1] <- 1
###Set GOIs
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims
tmt.LB.SL.lims.plastid$Category <- "Other"
tmt.LB.SL.lims.plastid$Order <- 1
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6

###Re-order
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims.plastid[(order(tmt.LB.SL.lims.plastid$Order)),]
###factor
tmt.LB.SL.lims.plastid$Category <- factor(tmt.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
###Scatterplot
tmt.LB.SL.lims.plastid.scat <- ggplot(tmt.LB.SL.lims.plastid, mapping=aes(x=l2.B73xMo17.MP.x, y=l2.B73xMo17.MP.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.3), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    ggtitle("A")+
    xlab("log2 BxM/MP LB")+
    ylab("log2 BxM/MP SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
tmt.LB.SL.lims.plastid.scat <- tmt.LB.SL.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73xMo17.MP.x"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73xMo17.MP.y"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      A <- insert_xaxis_grob(tmt.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      A <- insert_yaxis_grob(A, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      
      ###Set limits
cpm.LB.SL.lims <- cpm.LB.SL
cpm.LB.SL.lims$l2.B73xMo17.MP.x[cpm.LB.SL.lims$l2.B73xMo17.MP.x <= -2] <- -2
cpm.LB.SL.lims$l2.B73xMo17.MP.x[cpm.LB.SL.lims$l2.B73xMo17.MP.x >= 2] <- 2
cpm.LB.SL.lims$l2.B73xMo17.MP.y[cpm.LB.SL.lims$l2.B73xMo17.MP.y <= -2] <- -2
cpm.LB.SL.lims$l2.B73xMo17.MP.y[cpm.LB.SL.lims$l2.B73xMo17.MP.y >= 2] <- 2
###Set GOIs
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims
cpm.LB.SL.lims.plastid$Category <- "Other"
cpm.LB.SL.lims.plastid$Order <- 1
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
###Re-order
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims.plastid[(order(cpm.LB.SL.lims.plastid$Order)),]
###factor
cpm.LB.SL.lims.plastid$Category <- factor(cpm.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome", "PhANGs"))
###Scatterplot
cpm.LB.SL.lims.plastid.scat <- ggplot(cpm.LB.SL.lims.plastid, mapping=aes(x=l2.B73xMo17.MP.x, y=l2.B73xMo17.MP.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
          xlim(c(-2,2))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.3), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    ggtitle("B")+
    xlab("log2 BxM/MP LB")+
    ylab("log2 BxM/MP SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73xMo17.MP.x"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73xMo17.MP.y"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      B <- insert_xaxis_grob(cpm.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      B <- insert_yaxis_grob(B, densityplotY, grid::unit(0.275, "null"), position = "right")
###Set limits
tmt.LB.SL.lims <- tmt.LB.SL
tmt.LB.SL.lims$l2.B73.Mo17.x[tmt.LB.SL.lims$l2.B73.Mo17.x <= -1] <- -1
tmt.LB.SL.lims$l2.B73.Mo17.x[tmt.LB.SL.lims$l2.B73.Mo17.x >= 1] <- 1
tmt.LB.SL.lims$l2.B73.Mo17.y[tmt.LB.SL.lims$l2.B73.Mo17.y <= -1] <- -1
tmt.LB.SL.lims$l2.B73.Mo17.y[tmt.LB.SL.lims$l2.B73.Mo17.y >= 1] <- 1
###Set GOIs
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims
tmt.LB.SL.lims.plastid$Category <- "Other"
tmt.LB.SL.lims.plastid$Order <- 1
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.LB.SL.lims.plastid[(tmt.LB.SL.lims.plastid$Gene %in% GOIs$PhAPGs),"Order"] <- 6

###Re-order
tmt.LB.SL.lims.plastid <- tmt.LB.SL.lims.plastid[(order(tmt.LB.SL.lims.plastid$Order)),]
###factor
tmt.LB.SL.lims.plastid$Category <- factor(tmt.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
###Scatterplot
tmt.LB.SL.lims.plastid.scat <- ggplot(tmt.LB.SL.lims.plastid, mapping=aes(x=l2.B73.Mo17.x, y=l2.B73.Mo17.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.3), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=3))+
    ggtitle("C")+
    xlab("log2 B73/Mo17 LB")+
    ylab("log2 B73/Mo17 SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(tmt.LB.SL.lims.plastid.scat)
tmt.LB.SL.lims.plastid.scat <- tmt.LB.SL.lims.plastid.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73.Mo17.x"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.LB.SL.lims.plastid[,"l2.B73.Mo17.y"], color=tmt.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      C <- insert_xaxis_grob(tmt.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      C <- insert_yaxis_grob(C, densityplotY, grid::unit(0.275, "null"), position = "right")
      
      
      ###Set limits
cpm.LB.SL.lims <- cpm.LB.SL
cpm.LB.SL.lims$l2.B73.Mo17.x[cpm.LB.SL.lims$l2.B73.Mo17.x <= -2] <- -2
cpm.LB.SL.lims$l2.B73.Mo17.x[cpm.LB.SL.lims$l2.B73.Mo17.x >= 2] <- 2
cpm.LB.SL.lims$l2.B73.Mo17.y[cpm.LB.SL.lims$l2.B73.Mo17.y <= -2] <- -2
cpm.LB.SL.lims$l2.B73.Mo17.y[cpm.LB.SL.lims$l2.B73.Mo17.y >= 2] <- 2
###Set GOIs
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims
cpm.LB.SL.lims.plastid$Category <- "Other"
cpm.LB.SL.lims.plastid$Order <- 1
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$plastid),"Order"] <- 2
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
cpm.LB.SL.lims.plastid[(cpm.LB.SL.lims.plastid$Gene %in% GOIs$PhANGs),"Order"] <- 5
###Re-order
cpm.LB.SL.lims.plastid <- cpm.LB.SL.lims.plastid[(order(cpm.LB.SL.lims.plastid$Order)),]
###factor
cpm.LB.SL.lims.plastid$Category <- factor(cpm.LB.SL.lims.plastid$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome", "PhANGs"))
###Scatterplot
cpm.LB.SL.lims.plastid.scat <- ggplot(cpm.LB.SL.lims.plastid, mapping=aes(x=l2.B73.Mo17.x, y=l2.B73.Mo17.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p, name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
    theme_classic()+
          xlim(c(-2,2))+
          ylim(c(-2,2))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size = 10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.3), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    ggtitle("D")+
    xlab("log2 B73/Mo17 LB")+
    ylab("log2 B73/Mo17 SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
###Add density curves
    densityplotX<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="x")+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73.Mo17.x"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p, name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.LB.SL.lims.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.LB.SL.lims.plastid[,"l2.B73.Mo17.y"], color=cpm.LB.SL.lims.plastid[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p, name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      D <- insert_xaxis_grob(cpm.LB.SL.lims.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      D <- insert_yaxis_grob(D, densityplotY, grid::unit(0.275, "null"), position = "right")


FigLB.SL.design <- "AB
                    CD
                    EE"
FigLB.SL.plotpanel<-patchwork::wrap_plots(A=A, B=B, C=C, D=D, E=legend,  nrow = 3, heights = c(3,3,1.2), design = FigLB.SL.design)
ggsave(plot=FigLB.SL.plotpanel, filename =paste0(PathToPlots, "FigureS6.pdf"), dpi=300, width=6.5, height=6.84)
```




#Figure S7
```{r setup dataframe tmt.SL.acs.vol}
###Set GOIs
tmt.SL.acs <- tmt.SL
tmt.SL.acs$l2.B73xMo17.MP[tmt.SL.acs$l2.B73xMo17.MP <= -1] <- -1
tmt.SL.acs$l2.B73xMo17.MP[tmt.SL.acs$l2.B73xMo17.MP >= 1] <- 1
tmt.SL.acs$MP1 <- rowMeans(tmt.SL.acs[,c("SeedlingLeaf.B73.r1","SeedlingLeaf.Mo17.r1")])
tmt.SL.acs$MP2 <- rowMeans(tmt.SL.acs[,c("SeedlingLeaf.B73.r2","SeedlingLeaf.Mo17.r2")])
tmt.SL.acs$MP3 <- rowMeans(tmt.SL.acs[,c("SeedlingLeaf.B73.r3","SeedlingLeaf.Mo17.r3")])
tmt.SL.acs$ttest <- ttestFun(tmt.SL.acs, c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3"), c("MP1", "MP2", "MP3"))
tmt.SL.acs$l10.ttest <- -(log10(tmt.SL.acs$ttest))
tmt.SL.acs$Category <- "Other"
tmt.SL.acs$Order <- 1
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$SAMS),"Category"] <- "SAMS"
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$SAMS),"Order"] <- 2
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$ACS),"Category"] <- "ACS"
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$ACS),"Order"] <- 3
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$ACO),"Category"] <- "ACO"
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$ACO),"Order"] <- 4
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$acs2),"Category"] <- "ZmACS2"
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$acs2),"Order"] <- 5
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$acs6),"Category"] <- "ZmACS6"
tmt.SL.acs[(tmt.SL.acs$Gene %in% GOIs$acs6),"Order"] <- 6
tmt.SL.acs[(tmt.SL.acs$Gene %in% AtACO1hom),"Category"] <- "AtACO1 homolog"
tmt.SL.acs[(tmt.SL.acs$Gene %in% AtACO1hom),"Order"] <- 7
tmt.SL.acs[(tmt.SL.acs$Gene %in% AtACO2hom),"Category"] <- "AtACO2 homolog"
tmt.SL.acs[(tmt.SL.acs$Gene %in% AtACO2hom),"Order"] <- 8
tmt.SL.acs[(tmt.SL.acs$Gene %in% AtSAM1AtSAM2hom),"Category"] <- "AtSAM1/AtSAM2 homolog"
tmt.SL.acs[(tmt.SL.acs$Gene %in% AtSAM1AtSAM2hom),"Order"] <- 9

###Re-order
tmt.SL.acs <- tmt.SL.acs[(order(tmt.SL.acs$Order)),]
###factor
tmt.SL.acs$Category <- factor(tmt.SL.acs$Category, levels = c("Other", "SAMS", "ACS", "ACO", "ZmACS2", "ZmACS6","AtACO1 homolog", "AtACO2 homolog", "AtSAM1/AtSAM2 homolog"))
```

```{r setup dataframe cpm.SL.acs.vol}
###Set GOIs
cpm.SL.acs <- cpm.SL
cpm.SL.acs$l2.B73xMo17.MP[cpm.SL.acs$l2.B73xMo17.MP <= -2.5] <- -2.5
cpm.SL.acs$l2.B73xMo17.MP[cpm.SL.acs$l2.B73xMo17.MP >= 2.5] <- 2.5
cpm.SL.acs$MP1 <- rowMeans(cpm.SL.acs[,c("B73.1","Mo17.1")])
cpm.SL.acs$MP2 <- rowMeans(cpm.SL.acs[,c("B73.2","Mo17.2")])
cpm.SL.acs$MP3 <- rowMeans(cpm.SL.acs[,c("B73.3","Mo17.3")])
cpm.SL.acs$ttest <- ttestFun(cpm.SL.acs, c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3"), c("MP1", "MP2", "MP3"))
cpm.SL.acs$l10.ttest <- -(log10(cpm.SL.acs$ttest))
cpm.SL.acs$Category <- "Other"
cpm.SL.acs$Order <- 1
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$SAMS),"Category"] <- "SAMS"
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$SAMS),"Order"] <- 2
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$ACS),"Category"] <- "ACS"
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$ACS),"Order"] <- 3
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$ACO),"Category"] <- "ACO"
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$ACO),"Order"] <- 4
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$acs2),"Category"] <- "ZmACS2"
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$acs2),"Order"] <- 5
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$acs6),"Category"] <- "ZmACS6"
cpm.SL.acs[(cpm.SL.acs$Gene %in% GOIs$acs6),"Order"] <- 6
cpm.SL.acs[(cpm.SL.acs$Gene %in% AtACO1hom),"Category"] <- "AtACO1 homolog"
cpm.SL.acs[(cpm.SL.acs$Gene %in% AtACO1hom),"Order"] <- 7
cpm.SL.acs[(cpm.SL.acs$Gene %in% AtACO2hom),"Category"] <- "AtACO2 homolog"
cpm.SL.acs[(cpm.SL.acs$Gene %in% AtACO2hom),"Order"] <- 8
cpm.SL.acs[(cpm.SL.acs$Gene %in% AtSAM1AtSAM2hom),"Category"] <- "AtSAM1/AtSAM2 homolog"
cpm.SL.acs[(cpm.SL.acs$Gene %in% AtSAM1AtSAM2hom),"Order"] <- 9

###Re-order
cpm.SL.acs <- cpm.SL.acs[(order(cpm.SL.acs$Order)),]
###factor
cpm.SL.acs$Category <- factor(cpm.SL.acs$Category, levels = c("Other", "SAMS", "ACS", "ACO", "ZmACS2", "ZmACS6","AtACO1 homolog", "AtACO2 homolog", "AtSAM1/AtSAM2 homolog"))
```
```{r plot tmt.cpm.SL.acs.vol}
#Scatterplot - tmt
tmt.SL.acs.vol <- ggplot(tmt.SL.acs, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=2)+
  scale_color_manual(values=colors.ea, name="Category")+
  scale_shape_manual(values=c("Other"= 1, "SAMS"= 22, "ACS"= 23, "ACO"= 24,"ZmACS2"= 18,"ZmACS6"= 18,"AtACO1 homolog"=17,"AtACO2 homolog"=17, "AtSAM1/AtSAM2 homolog"=15))+
    theme_classic()+
          ggtitle("A")+
          geom_hline(yintercept=1.30103, linetype="dashed", color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10), axis.title = element_text(size = 10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(size=10, hjust = -.05))+
          guides(col=guide_legend(nrow=1))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
      
#volterplot - cpm
cpm.SL.acs.vol <- ggplot(cpm.SL.acs, mapping=aes(x=l2.B73xMo17.MP, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=2)+
  scale_color_manual(values=colors.ea, name="Category")+
  scale_shape_manual(values=c("Other"= 1, "SAMS"= 22, "ACS"= 23, "ACO"= 24,"ZmACS2"= 18,"ZmACS6"= 18,"AtACO1 homolog"=17,"AtACO2 homolog"=17, "AtSAM1/AtSAM2 homolog"=15))+
    theme_classic()+
          ggtitle("B")+
          geom_hline(yintercept=1.30103, linetype="dashed", color = "gray50")+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),axis.title = element_text(size = 10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(size=10, hjust = -.05))+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("-log10(p-value)")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
cpm.SL.acs.vol.legend <- get_legend(cpm.SL.acs.vol)
cpm.SL.acs.vol <- cpm.SL.acs.vol+theme(legend.position = "none")

#Wrap into panel
design <- "AB
           CC"
plotpanel<-patchwork::wrap_plots(A=tmt.SL.acs.vol, B=cpm.SL.acs.vol, C=cpm.SL.acs.vol.legend,  nrow = 2, heights = c(3,1), design = design)
ggsave(plot=plotpanel, filename =paste0(PathToPlots, "FigureS7.pdf"), dpi=300, width=6.5, height=4.1)
```






#Figure S8
```{r setup dataframe tmt.BxM.zma00592}
#merge with mdb to get entrez 
tmt.BxM.6H.entrez <- merge.data.frame(tmt.6H, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
tmt.BxM.6H.entrez.nodups <- tmt.BxM.6H.entrez[!is.na(tmt.BxM.6H.entrez$Entrez),]
tmt.BxM.6H.entrez.nodups <- tmt.BxM.6H.entrez.nodups[!tmt.BxM.6H.entrez.nodups$Entrez=="",]
tmt.BxM.6H.entrez.nodups <- tmt.BxM.6H.entrez.nodups[!duplicated(tmt.BxM.6H.entrez.nodups$Entrez),]
#make new dataframe with just l2(BM/MP) values and Entrez as rownames
tmt.BxM.6H.entrez.nodups.trim <- data.frame("l2.B73xMo17.MP"=tmt.BxM.6H.entrez.nodups$l2.B73xMo17.MP)
rownames(tmt.BxM.6H.entrez.nodups.trim) <- tmt.BxM.6H.entrez.nodups$Entrez
```
```{r plot tmt.BxM.6H.zma00592}
tmt.BxM.6H.entrez.nodups.trim.pv <- pathview(gene.data = tmt.BxM.6H.entrez.nodups.trim, pathway.id = "zma00592", species = "zma",  bins = 20, limit = 0.5, out.suffix="tmt.BxM")
```

```{r setup dataframe tmt.acs.zma00592}
#merge with mdb to get entrez 
tmt.acs.entrez <- merge.data.frame(tmt.acs, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
tmt.acs.entrez.nodups <- tmt.acs.entrez[!is.na(tmt.acs.entrez$Entrez),]
tmt.acs.entrez.nodups <- tmt.acs.entrez.nodups[!tmt.acs.entrez.nodups$Entrez=="",]
tmt.acs.entrez.nodups <- tmt.acs.entrez.nodups[!duplicated(tmt.acs.entrez.nodups$Entrez),]
#make new dataframe with just l2(acs/B73) values and Entrez as rownames
tmt.acs.entrez.nodups.trim <- data.frame("l2.ACS.B73"=tmt.acs.entrez.nodups$l2.ACS.B73)
rownames(tmt.acs.entrez.nodups.trim) <- tmt.acs.entrez.nodups$Entrez
```
```{r plot tmt.acs.zma00592}
tmt.acs.entrez.nodups.trim.pv <- pathview(gene.data = tmt.acs.entrez.nodups.trim, pathway.id = "zma00592", species = "zma",  bins = 20, limit = 0.5, out.suffix="tmt.acs")
```


#Figure S9
```{r setup dataframe tmt.cpm.RIL.6H.Hyb2MP.cor2}
#Subset hyb/MP values from protein data
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[,c(6,127:135)], tmt.6H[,c(6,54:59)], by.x = "Accession", by.y = "Accession", all = F)
#Make new gene column to merge with RNA data
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)

#Subset hyb/MP values from RNA data
cpm.RIL.Hyb2MP <- cpm.RIL[,c(1,76:84)]
cpm.6H.Hyb2MP <- cpm.6H[,c(1,42:47)]
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL[,c(1,76:84)], cpm.6H[,c(1,42:47)], by.x = "Gene", by.y = "Gene", all = F)
#Combine tmt and cpm
tmt.cpm.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL.6H.Hyb2MP, cpm.RIL.6H.Hyb2MP, by.x = "Gene", by.y = "Gene", all=F)
#Run correlation analysis between protein hyb/mp and RNA hyb/mp values
tmt.cpm.RIL.6H.Hyb2MP.cor2 <- CorFun2(tmt.cpm.RIL.6H.Hyb2MP, 3:17, 18:32)
```

```{r plot tmt.cpm.RIL.6H.Hyb2MP.cor2}
tmt.cpm.RIL.6H.Hyb2MP.cor2$Category <- "Other"
tmt.cpm.RIL.6H.Hyb2MP.cor2$Order <- 1
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$CytoRibo),"Order"] <- 2
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$NE.PtRibo),"Order"] <- 3
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PE.ribo),"Order"] <- 4
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhANGs),"Order"] <- 5
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
tmt.cpm.RIL.6H.Hyb2MP.cor2[(tmt.cpm.RIL.6H.Hyb2MP.cor2$Gene %in% GOIs$PhAPGs),"Order"] <- 6
tmt.cpm.RIL.6H.Hyb2MP.cor2 <- tmt.cpm.RIL.6H.Hyb2MP.cor2[(order(tmt.cpm.RIL.6H.Hyb2MP.cor2$Order)),]
tmt.cpm.RIL.6H.Hyb2MP.cor2$Category <- factor(tmt.cpm.RIL.6H.Hyb2MP.cor2$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome", "PE plastid ribosome", "PhANGs", "PhAPGs"))
ggplot(tmt.cpm.RIL.6H.Hyb2MP.cor2, mapping=aes(y=Category, x=cor, fill=Category))+
  geom_density_ridges(alpha=0.8) +
  theme_ridges() +
  theme(legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=10), axis.text = element_text(size=10))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)+
  scale_fill_manual(values=colors.pc)
ggsave(filename =paste0(PathToPlots, "FigureS9.pdf"), dpi=300, width=3.25, height=2)
```


#Table1
```{r setup dataframe tmt.cpm.RIL.6H.Hyb2MP.cor}
#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[,c(6,127:135)], tmt.6H[,c(6,54:59)], by.x = "Accession", by.y = "Accession", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#run correlation function
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#Set GOIs
table1 <- tmt.RIL.6H.Hyb2MP.cor[,c(17,1,18)]
colnames(table1) <- c("Gene", "Accession", "Pearson")
table1$Group <- "Other"
table1$Group[table1$Gene %in% GOIs$plastid] <- "Plastid"
table1$Group[table1$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
table1$Group[table1$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
table1$Group[table1$Gene %in% GOIs$PhANGs] <- "PhANGs"
table1$Group[table1$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
table1$Group[table1$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
table1$Group[table1$Gene %in% GOIs$EtBio] <- "Ethylene biosynthesis"
table1$Group[table1$Gene %in% GOIs$acs2] <- "acs2"
table1$Group[table1$Gene %in% GOIs$acs6] <- "acs6"
#Re-order
table1 <- table1[(order(-table1$Pearson)),]
#Trim to top 1.5%
table1 <- table1[1:(.015*nrow(table1)),]
#Cut off Gene column
table1 <- table1[,c("Accession", "Pearson", "Group")]
write.csv(table1, "Table1.csv", row.names = F)
```
#Supplemental Table 2
```{r}
SupTab2a <- (tmt.BxM.6H.entrez.nodups.trim.pv.df[,c("labels", "all.mapped")])
SupTab2a <- separate(SupTab2a, 'all.mapped', paste("all.mapped", 1:4, sep="."), sep=",", extra="drop")
SupTab2a <- melt(SupTab2a, id.vars="labels")
SupTab2a <- SupTab2a[,c("labels", "value")]
SupTab2a$value[SupTab2a$value==""] <- NA
SupTab2a <- na.omit(SupTab2a)
colnames(SupTab2a) <- c("labels", "Gene.entrez")

SupTab2b <- (tmt.acs.entrez.nodups.trim.pv.df[,c("labels", "all.mapped")])
SupTab2b <- separate(SupTab2b, 'all.mapped', paste("all.mapped", 1:4, sep="."), sep=",", extra="drop")
SupTab2b <- melt(SupTab2b, id.vars="labels")
SupTab2b <- SupTab2b[,c("labels", "value")]
SupTab2b$value[SupTab2b$value==""] <- NA
SupTab2b <- na.omit(SupTab2b)
colnames(SupTab2b) <- c("labels", "Gene.entrez")

SupTab2 <- merge.data.frame(SupTab2a, SupTab2b, by.x = "labels", by.y = "labels", all=T)
SupTab2 <- unique(SupTab2[,c(1,3)])
colnames(SupTab2) <- c("labels", "Gene.entrez")
SupTab2 <- merge.data.frame(SupTab2, mgdb[,c("v4_gene_model", "Entrez")],by.x="Gene.entrez", by.y="Entrez", all.x = T, all.y = F)
SupTab2 <- merge.data.frame(SupTab2, tmt.6H[,c("Gene", "l2.B73xMo17.MP")], by.x = "v4_gene_model", by.y = "Gene", all.x = T, ally.=F)
SupTab2 <- merge.data.frame(SupTab2, tmt.acs[,c("Gene", "l2.ACS.B73")], by.x = "v4_gene_model", by.y = "Gene", all.x = T, ally.=F)
write.csv(SupTab2, "SupplementalTable2.csv", row.names = F)
```




#Supplemental Data Set 1
```{r supDS1 protein}
SupDS1a <- Reduce(function(x,y) merge(x,y,by="Accession",all=TRUE) ,list(tmt.SL[,c("Accession", "HYB.MP")], tmt.LB[,c("Accession", "HYB.MP")], tmt.6H[,c("Accession", "B73xMo17.MP")], tmt.RIL[,c("Accession", "BM.BM.MP")]))

colnames(SupDS1a) <- c("Accession", "Protein_SeedlingLeaf_BxM.MP", "Protein_LeafBlade_BxM.MP", "Protein_6Hybrids_BxM.MP", "Protein_RILhybrids_BxM.MP")
```
```{r supDS1 rna}
SupDS1b <- Reduce(function(x,y) merge(x,y,by="Gene",all=TRUE) ,list(cpm.SL[,c("Gene", "B73xMo17.MP")], cpm.LB[,c("Gene", "B73xMo17.MP")], cpm.6H[,c("Gene", "B73xMo17.MP")], cpm.RIL[,c("Gene", "B73xMo17.MP")]))

colnames(SupDS1b) <- c("Gene", "RNA_SeedlingLeaf_BxM.MP", "RNA_LeafBlade_BxM.MP", "RNA_6Hybrids_BxM.MP", "RNA_RILhybrids_BxM.MP")
```
```{r supDS1 combine protein and rna}
SupDS1a$Gene <- substr(SupDS1a$Accession, start = 1, stop = 14)
SupDS1 <- merge.data.frame(SupDS1a, SupDS1b, by.x = "Gene", by.y = "Gene", all = T)
SupDS1 <- SupDS1[order(-SupDS1$Protein_SeedlingLeaf_BxM.MP),]
```
```{r supDS1 add category column}
SupDS1$Category <- "Other"
SupDS1$Category[SupDS1$Gene %in% GOIs$plastid] <- "Plastid"
SupDS1$Category[SupDS1$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
SupDS1$Category[SupDS1$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
SupDS1$Category[SupDS1$Gene %in% GOIs$PhANGs] <- "PhANGs"
SupDS1$Category[SupDS1$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
SupDS1$Category[SupDS1$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
SupDS1$Category[SupDS1$Gene %in% GOIs$SAMS] <- "Ethylene biosynthesis - SAM synthase"
SupDS1$Category[SupDS1$Gene %in% GOIs$ACS] <- "Ethylene biosynthesis - ACC synthase"
SupDS1$Category[SupDS1$Gene %in% GOIs$ACO] <- "Ethylene biosynthesis - ACC oxidase"
SupDS1$Category[SupDS1$Gene %in% GOIs$acs2] <- "Ethylene biosynthesis - ACS2"
SupDS1$Category[SupDS1$Gene %in% GOIs$acs6] <- "Ethylene biosynthesis - ACS6"
SupDS1$Category[SupDS1$Gene %in% GOIs$AtACO1hom] <- "Ethylene biosynthesis - AtACO1 homolog"
SupDS1$Category[SupDS1$Gene %in% GOIs$AtACO2hom] <- "Ethylene biosynthesis - AtACO2 homolog"
SupDS1$Category[SupDS1$Gene %in% GOIs$AtSAM1AtSAM2hom] <- "Ethylene biosynthesis - AtSAM1/2 homolog"

write.csv(SupDS1[,c(1,2,11,3:10)], "SupplementalDataSet1.csv", row.names = F)
```

#Supplemental Data Set 3
```{r supplemental table 3}
#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[,c(6,127:135)], tmt.6H[,c(6,54:59)], by.x = "Accession", by.y = "Accession", all = F)
#combine RIL and 6 hybrids RNA
cpm.RIL.6H.Hyb2MP <- merge.data.frame(cpm.RIL[,c(1,76:84)], cpm.6H[,c(1,42:47)], by.x = "Gene", by.y = "Gene", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#run correlation function
cpm.RIL.6H.Hyb2MP.cor <- CorFun(cpm.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#merge RIL and 6 hybrids
tmt.cpm.RIL.6H.Hyb2MP.cor <- merge.data.frame(tmt.RIL.6H.Hyb2MP.cor[,c(1,17,18)], cpm.RIL.6H.Hyb2MP.cor[,c(1,17)], by.x = "Gene", by.y = "Gene", all=T)
#Set GOIs
SupTDS3 <- tmt.cpm.RIL.6H.Hyb2MP.cor
colnames(SupTDS3) <- c("Gene", "Accession", "Protein_PearsonCor", "RNA_PearsonCor")
SupTDS3$Category <- "Other"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$plastid] <- "Plastid"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$PhANGs] <- "PhANGs"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$SAMS] <- "Ethylene biosynthesis - SAM synthase"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$ACS] <- "Ethylene biosynthesis - ACC synthase"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$ACO] <- "Ethylene biosynthesis - ACC oxidase"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$acs2] <- "Ethylene biosynthesis - ACS2"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$acs6] <- "Ethylene biosynthesis - ACS6"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$AtACO1hom] <- "Ethylene biosynthesis - AtACO1 homolog"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$AtACO2hom] <- "Ethylene biosynthesis - AtACO2 homolog"
SupTDS3$Category[SupTDS3$Gene %in% GOIs$AtSAM1AtSAM2hom] <- "Ethylene biosynthesis - AtSAM1/2 homolog"
#Re-order
SupTDS3 <- SupTDS3[(order(-SupTDS3$Protein_PearsonCor)),]
write.csv(SupTDS3[,c(1,2,5,3,4)], "SupplementalDataSet3.csv", row.names = F)
```


#Supplemental Data Set 4
```{r write GOI dataframe}
GOI.df <- as.data.frame(unlist(GOIs, use.names = F))
colnames(GOI.df) <- "Accession"
GOI.df$Accession <- as.character(GOI.df$Accession)
GOI.df$Group <- NA
GOI.df$Source <- NA
GOI.df[(GOI.df$Accession %in% GOIs$plastid),"Group"] <- "Plastid-localized"
GOI.df[(GOI.df$Accession %in% GOIs$plastid),"Source"] <- "Friso et al. (2010)"
GOI.df[(GOI.df$Accession %in% GOIs$PhANGs),"Group"] <- "PhANGs"
GOI.df[(GOI.df$Accession %in% KEGG.zma00195.digenomic$Accession),"Source"] <- "KEGG zma00195"
GOI.df[(GOI.df$Accession %in% KEGG.zma00196$Accession),"Source"] <- "KEGG zma00196"
GOI.df[(GOI.df$Accession %in% CornCycLightRxns.digenomic$Gene.Accession),"Source"] <- "CornCyc Photosynthesis Light Reactions"
GOI.df[(GOI.df$Accession %in% NE.Rubisco),"Source"] <- "MaizeGDB"
GOI.df[(GOI.df$Accession %in% GOIs$PhAPGs),"Group"] <- "PhAPGs"
GOI.df[(GOI.df$Accession %in% GOIs$PhAPGs),"Source"] <- "NCBI"
GOI.df[(GOI.df$Accession %in% GOIs$NE.PtRibo),"Group"] <- "NE plastid ribosome"
GOI.df[(GOI.df$Accession %in% GOIs$NE.PtRibo),"Source"] <- "NCBI and maize-GAMER GO"
GOI.df[(GOI.df$Accession %in% GOIs$PE.ribo),"Group"] <- "PE plastid ribosome"
GOI.df[(GOI.df$Accession %in% GOIs$PE.ribo),"Source"] <- "NCBI"
GOI.df[(GOI.df$Accession %in% GOIs$CytoRibo),"Group"] <- "Cytosolic ribosome"
GOI.df[(GOI.df$Accession %in% GOIs$CytoRibo),"Source"] <- "NCBI"
GOI.df[(GOI.df$Accession %in% GOIs$SAMS),"Group"] <- "SAM synthase"
GOI.df[(GOI.df$Accession %in% GOIs$SAMS),"Source"] <- "CornCyc 2.5.1.6"
GOI.df[(GOI.df$Accession %in% GOIs$ACS),"Group"] <- "ACC synthase"
GOI.df[(GOI.df$Accession %in% GOIs$ACS),"Source"] <- "CornCyc 4.4.1.14"
GOI.df[(GOI.df$Accession %in% GOIs$ACO),"Group"] <- "ACC oxidase"
GOI.df[(GOI.df$Accession %in% GOIs$ACO),"Source"] <- "CornCyc 1.14.17.4"
GOI.df[(GOI.df$Accession %in% GOIs$acs2),"Group"] <- "ACS2"
GOI.df[(GOI.df$Accession %in% GOIs$acs2),"Source"] <- "MaizeGDB"
GOI.df[(GOI.df$Accession %in% GOIs$acs6),"Group"] <- "ACS6"
GOI.df[(GOI.df$Accession %in% GOIs$acs6),"Source"] <- "MaizeGDB"
GOI.df[(GOI.df$Accession %in% GOIs$AtACO1hom),"Group"] <- "AtACO1 homolog"
GOI.df[(GOI.df$Accession %in% GOIs$AtACO1hom),"Source"] <- "BLAST"
GOI.df[(GOI.df$Accession %in% GOIs$AtACO2hom),"Group"] <- "AtACO2 homolog"
GOI.df[(GOI.df$Accession %in% GOIs$AtACO2hom),"Source"] <- "BLAST"
GOI.df[(GOI.df$Accession %in% GOIs$AtSAM1AtSAM2hom),"Group"] <- "AtSAM1/2 homolog"
GOI.df[(GOI.df$Accession %in% GOIs$AtSAM1AtSAM2hom),"Source"] <- "BLAST"
GOI.df <- GOI.df[order(GOI.df$Group),]

write.csv(GOI.df, "SupplementalDataSet4.csv", row.names = F)
```







