---
title: "Data Analyses of protein and RNA expression"
output: github_document
author: "Devon Birdseye"
date: "08/07/2020"
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r set seed}
set.seed(586)
```
```{r libraries}
library(plyr)
library(ggplot2)
library(reshape2)
library(cowplot)
library(patchwork)
library(pathview)
library(data.table)
library(AnnotationDbi)
library(dplyr)
library(tidyr)
library(ggridges)
library(stringr)
library(BSDA)
```
```{r define paths}
#Path to input data
PathToData <- "Data/"
#Path to output plots
PathToPlots <- "Plots/"
#Path to output tables
PathToTables <- "Tables/"
#Path to output datasets
PathToDatasets <- "Datasets/"
```
```{r T-test function}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```
#Import protein data
```{r import protein data, warning=F}
#Import protein data
tmt <- read.csv(paste0(PathToData,"SupplementalTable9.csv"), stringsAsFactors = F)
#re-define protein column to shortened name
tmt <- tmt[,c(1:181,183)]
colnames(tmt)[182] <- "Protein"
#make new column for gene accession without transcript id
tmt$Gene <- substr(tmt$Accession, start = 1, stop = 14)
#Remove polymorphic peptides
tmt.withPolymorphic <- tmt
tmt <- tmt[-(grep("only", tmt$Accession)),]
#Convert to numeric
tmt[1:170] <- sapply(tmt[1:170],as.numeric)
```
#Subset protein experiments
```{r subset BxM seedling leaf}
#BxM from Original Experiment (OG)
tmt.SL <- select(tmt,c(Group:Gene, SeedlingLeaf.B73.r1:SeedlingLeaf.BM.r3))
#Remove proteins not detected in one or more samples
tmt.SL <- na.omit(tmt.SL)
#Calculate averages of bio reps
tmt.SL$B73 <- rowMeans(select(tmt.SL,contains("B73")))
tmt.SL$Mo17 <- rowMeans(select(tmt.SL,contains("Mo17")))
tmt.SL$B73xMo17 <- rowMeans(select(tmt.SL,contains("BM")))
tmt.SL$MP <- rowMeans(tmt.SL[,c("B73", "Mo17")])
#calculate ratios
tmt.SL$l2.B73xMo17.MP <- log2(tmt.SL$B73xMo17/tmt.SL$MP)
tmt.SL$l2.B73.Mo17 <-log2(tmt.SL$B73/tmt.SL$Mo17)
#Calculate hyb/MP significance
tmt.SL$MP1 <- rowMeans(tmt.SL[,c("SeedlingLeaf.B73.r1","SeedlingLeaf.Mo17.r1")])
tmt.SL$MP2 <- rowMeans(tmt.SL[,c("SeedlingLeaf.B73.r2","SeedlingLeaf.Mo17.r2")])
tmt.SL$MP3 <- rowMeans(tmt.SL[,c("SeedlingLeaf.B73.r3","SeedlingLeaf.Mo17.r3")])
tmt.SL$ttest.MP <- ttestFun(tmt.SL, c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3"), c("MP1", "MP2", "MP3"))
tmt.SL$l10.ttest.MP <- -(log10(tmt.SL$ttest.MP))
#Define high-parent(HP)
tmt.SL$HP1 <- pmax(tmt.SL$SeedlingLeaf.B73.r1, tmt.SL$SeedlingLeaf.Mo17.r1)
tmt.SL$HP2 <- pmax(tmt.SL$SeedlingLeaf.B73.r2, tmt.SL$SeedlingLeaf.Mo17.r2)
tmt.SL$HP3 <- pmax(tmt.SL$SeedlingLeaf.B73.r3, tmt.SL$SeedlingLeaf.Mo17.r3)
#Calculate hyb/HP
tmt.SL$HP <- rowMeans(tmt.SL[,c("HP1", "HP2", "HP3")])
tmt.SL$l2.B73xMo17.HP <- log2(tmt.SL$B73xMo17/tmt.SL$HP)
#Calculate hyb/HP significance
tmt.SL$ttest.HP <- ttestFun(tmt.SL, c("SeedlingLeaf.BM.r1", "SeedlingLeaf.BM.r2", "SeedlingLeaf.BM.r3"), c("HP1", "HP2", "HP3"))
tmt.SL$l10.ttest.HP <- -(log10(tmt.SL$ttest.HP))
#Calculate Par/Par significance
tmt.SL$ttest.P <- ttestFun(tmt.SL, c("SeedlingLeaf.B73.r1", "SeedlingLeaf.B73.r2", "SeedlingLeaf.B73.r3"), c("SeedlingLeaf.Mo17.r1", "SeedlingLeaf.Mo17.r2", "SeedlingLeaf.Mo17.r3"))
tmt.SL$l10.ttest.P <- -(log10(tmt.SL$ttest.P))
```
```{r subset BxM mature leaf blade}
#BxM from Original Experiment (OG)
tmt.LB <- select(tmt,c(Group:Gene, MatureLeaf.B73.r1:MatureLeaf.BM.r3))
#Remove proteins not detected in one or more samples
tmt.LB <- na.omit(tmt.LB)
#Calculate averages of bio reps
tmt.LB$B73 <- rowMeans(select(tmt.LB,contains("B73")))
tmt.LB$Mo17 <- rowMeans(select(tmt.LB,contains("Mo17")))
tmt.LB$B73xMo17 <- rowMeans(select(tmt.LB,contains("BM")))
tmt.LB$MP <- rowMeans(tmt.LB[,c("B73", "Mo17")])
#calculate ratios
tmt.LB$l2.B73xMo17.MP <- log2(tmt.LB$B73xMo17/tmt.LB$MP)
tmt.LB$l2.B73.Mo17 <-log2(tmt.LB$B73/tmt.LB$Mo17)
#Calculate hyb/MP significance
tmt.LB$MP1 <- rowMeans(tmt.LB[,c("MatureLeaf.B73.r1","MatureLeaf.Mo17.r1")])
tmt.LB$MP2 <- rowMeans(tmt.LB[,c("MatureLeaf.B73.r2","MatureLeaf.Mo17.r2")])
tmt.LB$MP3 <- rowMeans(tmt.LB[,c("MatureLeaf.B73.r3","MatureLeaf.Mo17.r3")])
tmt.LB$ttest.MP <- ttestFun(tmt.LB, c("MatureLeaf.BM.r1", "MatureLeaf.BM.r2", "MatureLeaf.BM.r3"), c("MP1", "MP2", "MP3"))
tmt.LB$l10.ttest.MP <- -(log10(tmt.LB$ttest.MP))
#Define high-parent(HP)
tmt.LB$HP1 <- pmax(tmt.LB$MatureLeaf.B73.r1, tmt.LB$MatureLeaf.Mo17.r1)
tmt.LB$HP2 <- pmax(tmt.LB$MatureLeaf.B73.r2, tmt.LB$MatureLeaf.Mo17.r2)
tmt.LB$HP3 <- pmax(tmt.LB$MatureLeaf.B73.r3, tmt.LB$MatureLeaf.Mo17.r3)
#Calculate hyb/HP
tmt.LB$HP <- rowMeans(tmt.LB[,c("HP1", "HP2", "HP3")])
tmt.LB$l2.B73xMo17.HP <- log2(tmt.LB$B73xMo17/tmt.LB$HP)
#Calculate hyb/HP significance
tmt.LB$ttest.HP <- ttestFun(tmt.LB, c("MatureLeaf.BM.r1", "MatureLeaf.BM.r2", "MatureLeaf.BM.r3"), c("HP1", "HP2", "HP3"))
tmt.LB$l10.ttest.HP <- -(log10(tmt.LB$ttest.HP))
#Calculate Par/Par significance
tmt.LB$ttest.P <- ttestFun(tmt.LB, c("MatureLeaf.B73.r1", "MatureLeaf.B73.r2", "MatureLeaf.B73.r3"), c("MatureLeaf.Mo17.r1", "MatureLeaf.Mo17.r2", "MatureLeaf.Mo17.r3"))
tmt.LB$l10.ttest.P <- -(log10(tmt.LB$ttest.P))
```
```{r subset 6-hybrid experiment}
#Subset all hybrids from 6-hybrid experiment (6H)
tmt.6H <- select(tmt,c(Group:Gene, contains("X6Hybrid")))
#Remove proteins not detected in one or more samples
tmt.6H <- na.omit(tmt.6H)
#calculate averages
tmt.6H$B73 <- rowMeans(select(tmt.6H,contains(".B73.")))
tmt.6H$Mo17 <- rowMeans(select(tmt.6H,contains(".Mo17.")))
tmt.6H$B73xMo17 <- rowMeans(select(tmt.6H,contains(".B73xMo17.")))
tmt.6H$Mo17xB73 <- rowMeans(select(tmt.6H,contains(".Mo17xB73.")))
tmt.6H$B84 <- rowMeans(select(tmt.6H,contains(".B84.")))
tmt.6H$B84xB73 <- rowMeans(select(tmt.6H,contains(".B84xB73.")))
tmt.6H$B84xMo17 <- rowMeans(select(tmt.6H,contains(".B84xMo17.")))
tmt.6H$A682 <- rowMeans(select(tmt.6H,contains(".A682.")))
tmt.6H$A682xB73 <- rowMeans(select(tmt.6H,contains(".A682xB73.")))
tmt.6H$A682xMo17 <- rowMeans(select(tmt.6H,contains(".A682xMo17.")))
#Calculate hyb/MP ratios
tmt.6H$l2.B73xMo17.MP <- log2(tmt.6H$B73xMo17/rowMeans(tmt.6H[,c("B73", "Mo17")]))
tmt.6H$l2.Mo17xB73.MP <- log2(tmt.6H$Mo17xB73/rowMeans(tmt.6H[,c("B73", "Mo17")]))
tmt.6H$l2.B84xB73.MP <- log2(tmt.6H$B84xB73/rowMeans(tmt.6H[,c("B84", "B73")]))
tmt.6H$l2.B84xMo17.MP <- log2(tmt.6H$B84xMo17/rowMeans(tmt.6H[,c("B84", "Mo17")]))
tmt.6H$l2.A682xB73.MP <- log2(tmt.6H$A682xB73/rowMeans(tmt.6H[,c("A682", "B73")]))
tmt.6H$l2.A682xMo17.MP <- log2(tmt.6H$A682xMo17/rowMeans(tmt.6H[,c("A682", "Mo17")]))
#Calculate parent/parent ratios
tmt.6H$l2.B73.Mo17 <- log2(tmt.6H$B73/tmt.6H$Mo17)
tmt.6H$l2.Mo17.B73 <- log2(tmt.6H$Mo17/tmt.6H$B73)
tmt.6H$l2.B84.B73 <- log2(tmt.6H$B84/tmt.6H$B73)
tmt.6H$l2.B84.Mo17 <- log2(tmt.6H$B84/tmt.6H$Mo17)
tmt.6H$l2.A682.B73 <- log2(tmt.6H$A682/tmt.6H$B73)
tmt.6H$l2.A682.Mo17 <- log2(tmt.6H$A682/tmt.6H$Mo17)
#Calculate hyb/MP pval
##Calculate MP reps
tmt.6H$MP1.B73Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r1","X6Hybrid.Mo17.r1")])
tmt.6H$MP2.B73Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r2","X6Hybrid.Mo17.r2")])
tmt.6H$MP3.B73Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r3","X6Hybrid.Mo17.r3")])

tmt.6H$MP1.B84B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r1","X6Hybrid.B84.r1")])
tmt.6H$MP2.B84B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r2","X6Hybrid.B84.r2")])
tmt.6H$MP3.B84B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r3","X6Hybrid.B84.r3")])

tmt.6H$MP1.B84Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r1","X6Hybrid.B84.r1")])
tmt.6H$MP2.B84Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r2","X6Hybrid.B84.r2")])
tmt.6H$MP3.B84Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r3","X6Hybrid.B84.r3")])

tmt.6H$MP1.A682B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r1","X6Hybrid.A682.r1")])
tmt.6H$MP2.A682B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r2","X6Hybrid.A682.r2")])
tmt.6H$MP3.A682B73 <- rowMeans(tmt.6H[,c("X6Hybrid.B73.r3","X6Hybrid.A682.r3")])

tmt.6H$MP1.A682Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r1","X6Hybrid.A682.r1")])
tmt.6H$MP2.A682Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r2","X6Hybrid.A682.r2")])
tmt.6H$MP3.A682Mo17 <- rowMeans(tmt.6H[,c("X6Hybrid.Mo17.r3","X6Hybrid.A682.r3")])
##Calculate pval
tmt.6H$ttest.B73xMo17.MP <- (ttestFun(tmt.6H, c("X6Hybrid.B73xMo17.r1", "X6Hybrid.B73xMo17.r2", "X6Hybrid.B73xMo17.r3"), c("MP1.B73Mo17", "MP2.B73Mo17", "MP3.B73Mo17")))
tmt.6H$ttest.Mo17xB73.MP <- (ttestFun(tmt.6H, c("X6Hybrid.Mo17xB73.r1", "X6Hybrid.Mo17xB73.r2", "X6Hybrid.Mo17xB73.r3"), c("MP1.B73Mo17", "MP2.B73Mo17", "MP3.B73Mo17")))
tmt.6H$ttest.B84xB73.MP <- (ttestFun(tmt.6H, c("X6Hybrid.B84xB73.r1", "X6Hybrid.B84xB73.r2", "X6Hybrid.B84xB73.r3"), c("MP1.B84B73", "MP2.B84B73", "MP3.B84B73")))
tmt.6H$ttest.B84xMo17.MP <- (ttestFun(tmt.6H, c("X6Hybrid.B84xMo17.r1", "X6Hybrid.B84xMo17.r2", "X6Hybrid.B84xMo17.r3"), c("MP1.B84Mo17", "MP2.B84Mo17", "MP3.B84Mo17")))
tmt.6H$ttest.A682xB73.MP <- (ttestFun(tmt.6H, c("X6Hybrid.A682xB73.r1", "X6Hybrid.A682xB73.r2", "X6Hybrid.A682xB73.r3"), c("MP1.A682B73", "MP2.A682B73", "MP3.A682B73")))
tmt.6H$ttest.A682xMo17.MP <- (ttestFun(tmt.6H, c("X6Hybrid.A682xMo17.r1", "X6Hybrid.A682xMo17.r2", "X6Hybrid.A682xMo17.r3"), c("MP1.A682Mo17", "MP2.A682Mo17", "MP3.A682Mo17")))
#Define high-parent(HP)
tmt.6H$HP.B73Mo17 <- pmax(tmt.6H$B73, tmt.6H$Mo17)
tmt.6H$HP.B84B73 <- pmax(tmt.6H$B73, tmt.6H$B84)
tmt.6H$HP.B84Mo17 <- pmax(tmt.6H$Mo17, tmt.6H$B84)
tmt.6H$HP.A682B73 <- pmax(tmt.6H$B73, tmt.6H$A682)
tmt.6H$HP.A682Mo17 <- pmax(tmt.6H$Mo17, tmt.6H$A682)
#Calculate Hyb/HP
tmt.6H$l2.B73xMo17.HP <- log2(tmt.6H$B73xMo17/tmt.6H$HP.B73Mo17)
tmt.6H$l2.Mo17xB73.HP <- log2(tmt.6H$Mo17xB73/tmt.6H$HP.B73Mo17)

tmt.6H$l2.B84xB73.HP <- log2(tmt.6H$B84xB73/tmt.6H$HP.B84B73)
tmt.6H$l2.B84xMo17.HP <- log2(tmt.6H$B84xMo17/tmt.6H$HP.B84Mo17)

tmt.6H$l2.A682xB73.HP <- log2(tmt.6H$A682xB73/tmt.6H$HP.A682B73)
tmt.6H$l2.A682xMo17.HP <- log2(tmt.6H$A682xMo17/tmt.6H$HP.A682Mo17)
```
```{r subset RIL experiment}
#Subset all hybrids from RIL experiment (RIL)
tmt.RIL <- select(tmt,c(Group:Gene, B73.1:X317M.4))
#Remove proteins not detected in one or more samples
tmt.RIL <- na.omit(tmt.RIL)
#Calculate averages
tmt.RIL$BM.B73 <- rowMeans(tmt.RIL[,c("B73.1", "B73.3", "B73.4")])
tmt.RIL$BM.Mo17 <- rowMeans(tmt.RIL[,c("Mo17.1", "Mo17.3", "Mo17.4")])
tmt.RIL$BM.BM <- rowMeans(tmt.RIL[,c("BM.1", "BM.2", "BM.3", "BM.4")])
tmt.RIL$M0014.B73 <- rowMeans(tmt.RIL[,c("B73.1.1", "B73.2", "B73.3.1", "B73.4.1")])
tmt.RIL$M0014.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.1", "Mo17.2", "Mo17.3.1", "Mo17.4.1")])
tmt.RIL$M0014.14 <- rowMeans(tmt.RIL[,c("X14.1", "X14.2", "X14.3", "X14.4")])
tmt.RIL$M0014.14B <- rowMeans(tmt.RIL[,c("X14B.1", "X14B.2", "X14B.3", "X14B.4")])
tmt.RIL$M0014.14M <- rowMeans(tmt.RIL[,c("X14M.1", "X14M.2", "X14M.3", "X14M.4")])
tmt.RIL$M0016.B73 <- rowMeans(tmt.RIL[,c("B73.1.2", "B73.2.1", "B73.3.2", "B73.4.2")])
tmt.RIL$M0016.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.2", "Mo17.2.1", "Mo17.3.2", "Mo17.4.2")])
tmt.RIL$M0016.16 <- rowMeans(tmt.RIL[,c("X16.1", "X16.2", "X16.3", "X16.4")])
tmt.RIL$M0016.16B <- rowMeans(tmt.RIL[,c("X16B.1", "X16B.2", "X16B.3", "X16B.4")])
tmt.RIL$M0016.16M <- rowMeans(tmt.RIL[,c("X16M.1", "X16M.2", "X16M.3", "X16M.4")])
tmt.RIL$M0021.B73 <- rowMeans(tmt.RIL[,c("B73.1.3", "B73.2.2", "B73.3.3", "B73.4.3")])
tmt.RIL$M0021.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.3", "Mo17.2.2", "Mo17.3.3", "Mo17.4.3")])
tmt.RIL$M0021.21 <- rowMeans(tmt.RIL[,c("X21.1", "X21.2", "X21.3", "X21.4")])
tmt.RIL$M0021.21B <- rowMeans(tmt.RIL[,c("X21B.1", "X21B.2", "X21B.3", "X21B.4")])
tmt.RIL$M0021.21M <- rowMeans(tmt.RIL[,c("X21M.1", "X21M.2", "X21M.3", "X21M.4")])
tmt.RIL$M0317.B73 <- rowMeans(tmt.RIL[,c("B73.1.4", "B73.2.3", "B73.3.4", "B73.4.4")])
tmt.RIL$M0317.Mo17<- rowMeans(tmt.RIL[,c("Mo17.1.4", "Mo17.2.3", "Mo17.3.4", "Mo17.4.4")])
tmt.RIL$M0317.317 <- rowMeans(tmt.RIL[,c("X317.1", "X317.2", "X317.3", "X317.4")])
tmt.RIL$M0317.317B <- rowMeans(tmt.RIL[,c("X317B.1", "X317B.2", "X317B.3", "X317B.4")])
tmt.RIL$M0317.317M <- rowMeans(tmt.RIL[,c("X317M.1", "X317M.2", "X317M.3", "X317M.4")])
#Calculate hyb/mp ratios
tmt.RIL$l2.B73xMo17.MP <- log2(tmt.RIL$BM.BM/rowMeans(tmt.RIL[,c("BM.B73", "BM.Mo17")]))
tmt.RIL$l2.M0014.14B.MP <- log2(tmt.RIL$M0014.14B/rowMeans(tmt.RIL[,c("M0014.14", "M0014.B73")]))
tmt.RIL$l2.M0014.14M.MP <- log2(tmt.RIL$M0014.14M/rowMeans(tmt.RIL[,c("M0014.14", "M0014.Mo17")]))
tmt.RIL$l2.M0016.16B.MP <- log2(tmt.RIL$M0016.16B/rowMeans(tmt.RIL[,c("M0016.16", "M0016.B73")]))
tmt.RIL$l2.M0016.16M.MP <- log2(tmt.RIL$M0016.16M/rowMeans(tmt.RIL[,c("M0016.16", "M0016.Mo17")]))
tmt.RIL$l2.M0021.21B.MP <- log2(tmt.RIL$M0021.21B/rowMeans(tmt.RIL[,c("M0021.21", "M0021.B73")]))
tmt.RIL$l2.M0021.21M.MP <- log2(tmt.RIL$M0021.21M/rowMeans(tmt.RIL[,c("M0021.21", "M0021.Mo17")]))
tmt.RIL$l2.M0317.317B.MP <- log2(tmt.RIL$M0317.317B/rowMeans(tmt.RIL[,c("M0317.317", "M0317.B73")]))
tmt.RIL$l2.M0317.317M.MP <- log2(tmt.RIL$M0317.317M/rowMeans(tmt.RIL[,c("M0317.317", "M0317.Mo17")]))
#Calculate parent/parent ratios
tmt.RIL$l2.B73.Mo17 <- log2(tmt.RIL$BM.B73/tmt.RIL$BM.Mo17)
tmt.RIL$l2.M0014.14.B73 <- log2(tmt.RIL$M0014.14/tmt.RIL$M0014.B73)
tmt.RIL$l2.M0014.14.Mo17  <- log2(tmt.RIL$M0014.14/tmt.RIL$M0014.Mo17)
tmt.RIL$l2.M0016.16.B73 <- log2(tmt.RIL$M0016.16/tmt.RIL$M0016.B73)
tmt.RIL$l2.M0016.16.Mo17  <- log2(tmt.RIL$M0016.16/tmt.RIL$M0016.Mo17)
tmt.RIL$l2.M0021.21.B73 <- log2(tmt.RIL$M0021.21/tmt.RIL$M0021.B73)
tmt.RIL$l2.M0021.21.Mo17  <- log2(tmt.RIL$M0021.21/tmt.RIL$M0021.Mo17)
tmt.RIL$l2.M0317.317.B73 <- log2(tmt.RIL$M0317.317/tmt.RIL$M0317.B73)
tmt.RIL$l2.M0317.317.Mo17  <- log2(tmt.RIL$M0317.317/tmt.RIL$M0317.Mo17)
```
```{r subset acs mutant experiment}
#Subset acs mutant experiment (acs)
tmt.acs <- select(tmt,c(Group:Gene, contains("acs")))
#Remove proteins not detected in one or more samples
tmt.acs <- na.omit(tmt.acs)
#Calculate averages of bio reps - acs
tmt.acs$B73avg <- rowMeans(select(tmt.acs,contains("B73.r")))
tmt.acs$ACSavg <- rowMeans(select(tmt.acs,contains("acs.r")))
#calculate log2 ratio
tmt.acs$l2.ACS.B73 <- log2(tmt.acs$ACSavg/tmt.acs$B73avg)
#calculate p-value
tmt.acs$ttest <- ttestFun(tmt.acs, 14:18, 19:23)
tmt.acs$l10.pvalue <- -log10(tmt.acs$ttest)
```
#Save protein dataframes
```{r save Protein dfs}
saveRDS(tmt.SL, "tmt.SL")
saveRDS(tmt.LB, "tmt.LB")
saveRDS(tmt.6H, "tmt.6H")
saveRDS(tmt.RIL, "tmt.RIL")
saveRDS(tmt.acs, "tmt.acs")
```
```{r read Protein dfs}
tmt.SL <- readRDS("tmt.SL")
tmt.LB <- readRDS("tmt.LB")
tmt.6H <- readRDS("tmt.6H")
tmt.RIL <- readRDS("tmt.RIL")
tmt.acs <- readRDS("tmt.acs")
```
#Import RNA data
```{r import RNA data}
#Original experiment
cpm.OG.data <- read.table(paste0(PathToData, "cpm.tsv"), sep = '\t', header = TRUE)
#6-hybrid experiment
cpm.6H <- read.csv(paste0(PathToData, "SupplementalTable7.csv"), stringsAsFactors = F)
#RIL experiment
cpm.RIL <- read.csv(paste0(PathToData, "SupplementalTable8.csv"), stringsAsFactors = F)
```
#Subset RNA experiments
```{r subset BxM seedling leaf}
#Original experiment seedling leaf BxM
cpm.SL <- as.data.frame(cpm.OG.data[,c("gid","BR193","BR194","BR195","BR196","BR197","BR198","BR199", "BR200","BR201")])
colnames(cpm.SL) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "B73xMo17.1", "B73xMo17.2", "B73xMo17.3")
#Remove genes not detected in one or more rep
cpm.SL[cpm.SL==0] <- NA
cpm.SL <- na.omit(cpm.SL)
#Calculate averages of bio reps
cpm.SL$B73 <- rowMeans(cpm.SL[,c("B73.1", "B73.2", "B73.3")])
cpm.SL$Mo17 <- rowMeans(cpm.SL[,c("Mo17.1", "Mo17.2", "Mo17.3")])
cpm.SL$B73xMo17 <- rowMeans(cpm.SL[,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")])
#calculate ratios
cpm.SL$l2.B73xMo17.MP <- log2(cpm.SL$B73xMo17/rowMeans(cpm.SL[,c("B73", "Mo17")]))
cpm.SL$l2.B73.Mo17 <-log2(cpm.SL$B73/cpm.SL$Mo17)
#Calculate hyb/MP significance
cpm.SL$MP1 <- rowMeans(cpm.SL[,c("B73.1","Mo17.1")])
cpm.SL$MP2 <- rowMeans(cpm.SL[,c("B73.2","Mo17.2")])
cpm.SL$MP3 <- rowMeans(cpm.SL[,c("B73.3","Mo17.3")])
cpm.SL$ttest.MP <- ttestFun(cpm.SL, c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3"), c("MP1", "MP2", "MP3"))
cpm.SL$l10.ttest.MP <- -(log10(cpm.SL$ttest.MP))
#Define high-parent(HP)
cpm.SL$HP1 <- pmax(cpm.SL$B73.1, cpm.SL$Mo17.1)
cpm.SL$HP2 <- pmax(cpm.SL$B73.2, cpm.SL$Mo17.2)
cpm.SL$HP3 <- pmax(cpm.SL$B73.3, cpm.SL$Mo17.3)
#Define low-parent(LP)
cpm.SL$LP1 <- pmin(cpm.SL$B73.1, cpm.SL$Mo17.1)
cpm.SL$LP2 <- pmin(cpm.SL$B73.2, cpm.SL$Mo17.2)
cpm.SL$LP3 <- pmin(cpm.SL$B73.3, cpm.SL$Mo17.3)
#Calculate hyb/HP
cpm.SL$HP <- rowMeans(cpm.SL[,c("HP1", "HP2", "HP3")])
cpm.SL$l2.B73xMo17.HP <- log2(cpm.SL$B73xMo17/cpm.SL$HP)
#Calculate hyb/LP
cpm.SL$LP <- rowMeans(cpm.SL[,c("LP1", "LP2", "LP3")])
cpm.SL$l2.B73xMo17.LP <- log2(cpm.SL$B73xMo17/cpm.SL$LP)
#Calculate hyb/HP significance
cpm.SL$ttest.HP <- ttestFun(cpm.SL, c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3"), c("HP1", "HP2", "HP3"))
cpm.SL$l10.ttest.HP <- -(log10(cpm.SL$ttest.HP))
#Calculate hyb/LP significance
cpm.SL$ttest.LP <- ttestFun(cpm.SL, c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3"), c("LP1", "LP2", "LP3"))
cpm.SL$l10.ttest.LP <- -(log10(cpm.SL$ttest.LP))
#Calculate parent/parent significance
cpm.SL$ttest.P <- ttestFun(cpm.SL, c("B73.1", "B73.2", "B73.3"), c("Mo17.1", "Mo17.2", "Mo17.3"))
cpm.SL$l10.ttest.P <- -(log10(cpm.SL$ttest.P))
##subset to genes detected as proteins
cpm.SL.protein <- cpm.SL[cpm.SL$Gene%in%tmt.SL$Gene,]
```
```{r subset BxM leaf blade}
#Original experiment seedling leaf BxM
cpm.LB <- as.data.frame(cpm.OG.data[,c("gid","BR001", "BR002", "BR004", "BR003", "BR005", "BR007", "BR006", "BR008", "BR009")])
colnames(cpm.LB) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "B73xMo17.1", "B73xMo17.2", "B73xMo17.3")
#Remove genes not detected in one or more rep
cpm.LB[cpm.LB==0] <- NA
cpm.LB <- na.omit(cpm.LB)
#Calculate averages of bio reps
cpm.LB$B73 <- rowMeans(cpm.LB[,c("B73.1", "B73.2", "B73.3")])
cpm.LB$Mo17 <- rowMeans(cpm.LB[,c("Mo17.1", "Mo17.2", "Mo17.3")])
cpm.LB$B73xMo17 <- rowMeans(cpm.LB[,c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3")])
#calculate ratios
cpm.LB$l2.B73xMo17.MP <- log2(cpm.LB$B73xMo17/rowMeans(cpm.LB[,c("B73", "Mo17")]))
cpm.LB$l2.B73.Mo17 <-log2(cpm.LB$B73/cpm.LB$Mo17)
#Calculate hyb/MP significance
cpm.LB$MP1 <- rowMeans(cpm.LB[,c("B73.1","Mo17.1")])
cpm.LB$MP2 <- rowMeans(cpm.LB[,c("B73.2","Mo17.2")])
cpm.LB$MP3 <- rowMeans(cpm.LB[,c("B73.3","Mo17.3")])
cpm.LB$ttest.MP <- ttestFun(cpm.LB, c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3"), c("MP1", "MP2", "MP3"))
cpm.LB$l10.ttest.MP <- -(log10(cpm.LB$ttest.MP))
#Define high-parent(HP)
cpm.LB$HP1 <- pmax(cpm.LB$B73.1, cpm.LB$Mo17.1)
cpm.LB$HP2 <- pmax(cpm.LB$B73.2, cpm.LB$Mo17.2)
cpm.LB$HP3 <- pmax(cpm.LB$B73.3, cpm.LB$Mo17.3)
#Calculate hyb/HP
cpm.LB$HP <- rowMeans(cpm.LB[,c("HP1", "HP2", "HP3")])
cpm.LB$l2.B73xMo17.HP <- log2(cpm.LB$B73xMo17/cpm.LB$HP)
#Calculate hyb/HP significance
cpm.LB$ttest.HP <- ttestFun(cpm.LB, c("B73xMo17.1", "B73xMo17.2", "B73xMo17.3"), c("HP1", "HP2", "HP3"))
cpm.LB$l10.ttest.HP <- -(log10(cpm.LB$ttest.HP))
#Calculate parent/parent significance
cpm.LB$ttest.P <- ttestFun(cpm.LB, c("B73.1", "B73.2", "B73.3"), c("Mo17.1", "Mo17.2", "Mo17.3"))
cpm.LB$l10.ttest.P <- -(log10(cpm.LB$ttest.P))
#subset to genes detected as protein
cpm.LB.protein <- cpm.LB[cpm.LB$Gene%in%tmt.LB$Gene,]
```
```{r subset 6 hybrid RNA data}
#Remove genes not detected in one or mor rep
cpm.6H[cpm.6H==0] <- NA
cpm.6H <- na.omit(cpm.6H)
#Calculate averages
cpm.6H$B73 <- rowMeans(cpm.6H[,c("B73.2", "B73.3", "B73.4")])
cpm.6H$B84 <- rowMeans(cpm.6H[,c("B84.2", "B84.3", "B84.4")])
cpm.6H$Mo17 <- rowMeans(cpm.6H[,c("Mo17.2", "Mo17.3", "Mo17.4")])
cpm.6H$A682 <- rowMeans(cpm.6H[,c("A682.2", "A682.3", "A682.4")])
cpm.6H$B73xMo17 <- rowMeans(cpm.6H[,c("B73xMo17.2", "B73xMo17.3", "B73xMo17.4")])
cpm.6H$Mo17xB73 <- rowMeans(cpm.6H[,c("Mo17xB73.2", "Mo17xB73.3", "Mo17xB73.4")])
cpm.6H$B84xB73 <- rowMeans(cpm.6H[,c("B84xB73.2", "B84xB73.3", "B84xB73.4")])
cpm.6H$B84xMo17 <- rowMeans(cpm.6H[,c("B84xMo17.2", "B84xMo17.3", "B84xMo17.4")])
cpm.6H$A682xB73 <- rowMeans(cpm.6H[,c("A682xB73.2", "A682xB73.3", "A682xB73.4")])
cpm.6H$A682xMo17 <- rowMeans(cpm.6H[,c("A682xMo17.2", "A682xMo17.3", "A682xMo17.4")])
#calculate ratio of hybrid to midparent
cpm.6H$l2.B73xMo17.MP <- log2(cpm.6H$B73xMo17/rowMeans(cpm.6H[,c("B73","Mo17")]))
cpm.6H$l2.Mo17xB73.MP <- log2(cpm.6H$Mo17xB73/rowMeans(cpm.6H[,c("B73","Mo17")]))
cpm.6H$l2.B84xB73.MP <- log2(cpm.6H$B84xB73/rowMeans(cpm.6H[,c("B84","B73")]))
cpm.6H$l2.B84xMo17.MP <- log2(cpm.6H$B84xMo17/rowMeans(cpm.6H[,c("B84","Mo17")]))
cpm.6H$l2.A682xB73.MP <- log2(cpm.6H$A682xB73/rowMeans(cpm.6H[,c("A682","B73")]))
cpm.6H$l2.A682xMo17.MP <- log2(cpm.6H$A682xMo17/rowMeans(cpm.6H[,c("A682","Mo17")]))
#Calculate ratio of parent to parent
cpm.6H$l2.B73.Mo17 <- log2(cpm.6H$B73/cpm.6H$Mo17)
cpm.6H$l2.Mo17.B73 <- log2(cpm.6H$Mo17/cpm.6H$B73)
cpm.6H$l2.B84.B73 <- log2(cpm.6H$B84/cpm.6H$B73)
cpm.6H$l2.B84.Mo17 <- log2(cpm.6H$B84/cpm.6H$Mo17)
cpm.6H$l2.A682.B73 <- log2(cpm.6H$A682/cpm.6H$B73)
cpm.6H$l2.A682.Mo17 <- log2(cpm.6H$A682/cpm.6H$Mo17)
#Subset to genes with detected protein
cpm.6H.protein <- cpm.6H[cpm.6H$Gene%in%tmt.6H$Gene,]
```
```{r subset RIL RNA data}
#Remove genes not detected in one or mor rep
cpm.RIL[cpm.RIL==0] <- NA
cpm.RIL <- na.omit(cpm.RIL)
#Calculate averages
cpm.RIL$B73 <- rowMeans(cpm.RIL[,c("B73.1", "B73.2", "B73.3", "B73.4")])
cpm.RIL$Mo17 <- rowMeans(cpm.RIL[,c("Mo17.1", "Mo17.2", "Mo17.3", "Mo17.4")])
cpm.RIL$B73xMo17 <- rowMeans(cpm.RIL[,c("BxM.1", "BxM.2", "BxM.3", "BxM.4")])
cpm.RIL$M0014 <- rowMeans(cpm.RIL[,c("M0014.1", "M0014.2", "M0014.3", "M0014.4")])
cpm.RIL$M0014xB73 <- rowMeans(cpm.RIL[,c("M0014xB73.1", "M0014xB73.2", "M0014xB73.3", "M0014xB73.4")])
cpm.RIL$M0014xMo17 <- rowMeans(cpm.RIL[,c("M0014xMo17.1", "M0014xMo17.2", "M0014xMo17.3", "M0014xMo17.4")])
cpm.RIL$M0016 <- rowMeans(cpm.RIL[,c("M0016.1", "M0016.2", "M0016.3", "M0016.4")])
cpm.RIL$M0016xB73 <- rowMeans(cpm.RIL[,c("M0016xB73.1", "M0016xB73.2", "M0016xB73.3")])
cpm.RIL$M0016xMo17 <- rowMeans(cpm.RIL[,c("M0016xMo17.1", "M0016xMo17.2", "M0016xMo17.3", "M0016xMo17.4")])
cpm.RIL$M0021 <- rowMeans(cpm.RIL[,c("M0021.1", "M0021.2", "M0021.3", "M0021.4")])
cpm.RIL$M0021xB73 <- rowMeans(cpm.RIL[,c("M0021xB73.1", "M0021xB73.2", "M0021xB73.3", "M0021xB73.4")])
cpm.RIL$M0021xMo17 <- rowMeans(cpm.RIL[,c("M0021xMo17.1", "M0021xMo17.2", "M0021xMo17.3", "M0021xMo17.4")])
cpm.RIL$M0317 <- rowMeans(cpm.RIL[,c("M0317.1", "M0317.2", "M0317.3", "M0317.4")])
cpm.RIL$M0317xB73 <- rowMeans(cpm.RIL[,c("M0317xB73.1", "M0317xB73.2", "M0317xB73.3", "M0317xB73.4")])
cpm.RIL$M0317xMo17 <- rowMeans(cpm.RIL[,c("M0317xMo17.1", "M0317xMo17.2", "M0317xMo17.3", "M0317xMo17.4")])
#calculate ratio of hybrid to midparent
cpm.RIL$l2.B73xMo17.MP <- log2(cpm.RIL$B73xMo17/rowMeans(cpm.RIL[,c("B73","Mo17")]))
cpm.RIL$l2.M0014xB73.MP <- log2(cpm.RIL$M0014xB73/rowMeans(cpm.RIL[,c("M0014", "B73")]))
cpm.RIL$l2.M0014xMo17.MP <- log2(cpm.RIL$M0014xMo17/rowMeans(cpm.RIL[,c("M0014", "Mo17")]))
cpm.RIL$l2.M0016xB73.MP <- log2(cpm.RIL$M0016xB73/rowMeans(cpm.RIL[,c("M0016", "B73")]))
cpm.RIL$l2.M0016xMo17.MP <- log2(cpm.RIL$M0016xMo17/rowMeans(cpm.RIL[,c("M0016", "Mo17")]))
cpm.RIL$l2.M0021xB73.MP <- log2(cpm.RIL$M0021xB73/rowMeans(cpm.RIL[,c("M0021", "B73")]))
cpm.RIL$l2.M0021xMo17.MP <- log2(cpm.RIL$M0021xMo17/rowMeans(cpm.RIL[,c("M0021", "Mo17")]))
cpm.RIL$l2.M0317xB73.MP <- log2(cpm.RIL$M0317xB73/rowMeans(cpm.RIL[,c("M0317", "B73")]))
cpm.RIL$l2.M0317xMo17.MP <- log2(cpm.RIL$M0317xMo17/rowMeans(cpm.RIL[,c("M0317", "Mo17")]))
#Calculate ratio of parent to parent
cpm.RIL$l2.B73.Mo17 <- log2(cpm.RIL$B73/cpm.RIL$Mo17)
cpm.RIL$l2.M0014.B73 <- log2(cpm.RIL$M0014/cpm.RIL$B73)
cpm.RIL$l2.M0014.Mo17 <- log2(cpm.RIL$M0014/cpm.RIL$Mo17)
cpm.RIL$l2.M0016.B73 <- log2(cpm.RIL$M0016/cpm.RIL$B73)
cpm.RIL$l2.M0016.Mo17 <- log2(cpm.RIL$M0016/cpm.RIL$Mo17)
cpm.RIL$l2.M0021.B73 <- log2(cpm.RIL$M0021/cpm.RIL$B73)
cpm.RIL$l2.M0021.Mo17 <- log2(cpm.RIL$M0021/cpm.RIL$Mo17)
cpm.RIL$l2.M0317.B73 <- log2(cpm.RIL$M0317/cpm.RIL$B73)
cpm.RIL$l2.M0317.Mo17 <- log2(cpm.RIL$M0317/cpm.RIL$Mo17)
#Subset to genes with detected protein
cpm.RIL.protein <- cpm.RIL[cpm.RIL$Gene%in%tmt.RIL$Gene,]
```
#Save RNA dataframes
```{r save RNA dfs}
saveRDS(cpm.SL, "cpm.SL")
saveRDS(cpm.LB, "cpm.LB")
saveRDS(cpm.6H, "cpm.6H")
saveRDS(cpm.RIL, "cpm.RIL")
```
```{r read RNA dfs}
cpm.SL <- readRDS("cpm.SL")
cpm.LB <- readRDS("cpm.LB")
cpm.6H <- readRDS("cpm.6H")
cpm.RIL <- readRDS("cpm.RIL")
```


#Plant Height data
```{r PlantHeights}
PlantHeights <- read.csv(paste0(PathToData, "SupplementalTable4.csv"), stringsAsFactors = F)
#Make vector of 6H hyb/MP ratios
PlantHeights.6h.MP <- PlantHeights$Hyb.MP[PlantHeights$Experiment=="6H"]
#Make vector of RIL hyb/MP ratios
PlantHeights.ril.MP <- PlantHeights$Hyb.MP[PlantHeights$Experiment=="RIL"]
```
```{r}
saveRDS(PlantHeights.6h.MP, "PlantHeights.6h.MP")
saveRDS(PlantHeights.ril.MP,"PlantHeights.ril.MP")
```


#Define GOIs
##Accession conversion
```{r mgdb}
#Downloaded from MaizeGDB 2020/05/18, 07:11pm
mgdb <- read.csv(paste0(PathToData,"mgdb.csv"), stringsAsFactors = F, header = F)
colnames(mgdb) <- mgdb[5,]
mgdb <- mgdb[-c(1:5),]
```


```{r organelleEntrez}
organelleEntrez <- tmt[grep("Zema|Zeam", tmt$Accession),]
organelleEntrez <- data.frame("Accession"=organelleEntrez$Accession, "Entrez"=NA)
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp003.1"] <- "845178"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp004.1"] <- "845232"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp008.1"] <- "845201"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp013.1"] <- "845225"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp014.1"] <- "845226"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp016.1"] <- "845236"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp017.1"] <- "845174"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp018.1"] <- "845173"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp019.1"] <- "845172"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp021.1"] <- "845229"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp022.1"] <- "845196"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp023.1"] <- "845195"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp024.1"] <- "1466361"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp026.1"] <- "845238"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp027.1"] <- "845189"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp028.1"] <- "845190"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp029.1"] <- "845182"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp030.1"] <- "845171"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp032.1"] <- "845212"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp035.1"] <- "845175"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp036.1"] <- "845191"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp039.1"] <- "845204"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp040.1"] <- "845203"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp043.1"] <- "845198"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp044.1"] <- "845222"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp045.1"] <- "845233"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp046.1"] <- "845217"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp047.1"] <- "845176"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp048.1"] <- "845200"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp050.1"] <- "845211"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp051.1"] <- "845205"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp052.1"] <- "845192"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp053.1"] <- "845193"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp054.1"] <- "845224"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp055.1"] <- "845228"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp056.1"] <- "845223"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp057.1"] <- "845177"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp058.1"] <- "845241"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp059.1"] <- "845213"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp060.1"] <- "845214"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp061.1"] <- "845237"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp062.1"] <- "845218"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp063.1"] <- "845235"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp065.1"] <- "845216"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp067.1"] <- "845220"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp075.1"] <- "845181"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp082.1"] <- "845230"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp083.1"] <- "845185"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp084.1"] <- "845221"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp085.1"] <- "1466378"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp086.1"] <- "845183"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp087.1"] <- "1466379"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp088.1"] <- "845184"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp090.1"] <- "845188"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp091.1"] <- "845179"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZemaCp092.1"] <- "845187"

organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp008.1"] <- "4055864"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp024.1"] <- "4055860"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp032.1"] <- "4055888"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp033.1"] <- "4055938"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp036.1"] <- "4055894"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp050.1"] <- "4055956"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp051.1"] <- "4055881"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp057.1"] <- "4055973"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp081.1"] <- "4055890"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp092.1"] <- "4055837"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp097.1"] <- "4055857"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp098.1"] <- "4055896"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp113.1"] <- "4055802"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp119.1"] <- "4055928"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp129.1"] <- "4055878"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp136.1"] <- "4056004"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp159.1"] <- "4055794"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp167.1"] <- "4056007"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp179.1"] <- "4056006"
organelleEntrez$Entrez[organelleEntrez$Accession=="ZeamMp185.1"] <- "4055892"
```
```{r Entrez conversion}
Entrez <- mgdb[,c(1,12)]
colnames(Entrez) <- c("Accession", "Entrez")
Entrez <- rbind(Entrez, organelleEntrez)
```
##Plastid-encoded
```{r PE}
#Plastid-encoded (PE) proteins
PE<-tmt[grep("ZemaCp", tmt$Accession),]
#define vector of PE photosynthesis-associated proteins
PEI <- PE$Gene[grep("psa", PE$Protein)]
PEII <- PE$Gene[grep("psb", PE$Protein)]
PE.NADH <- PE$Gene[grep("ndh", PE$Protein)]
PE.b6f <- PE$Gene[grep("pet", PE$Protein)]
PE.ATP <- PE$Gene[grep("atp", PE$Protein)]
PE.rubisco <- PE$Gene[grep("rbc", PE$Protein)]
PhAPGs <- c(PEI,PEII,PE.NADH,PE.b6f,PE.ATP,PE.rubisco)
#define vector of PE ribosomal proteins
PE.ribo <- PE$Gene[grep("rps|rpl", PE$Protein)]
#define vactor of PE RNA Polymerases
PE.rpo <- PE$Gene[grep("rpo", PE$Protein)]
#define vactor of PE Other
PE.other <- PE$Gene[!PE$Gene %in% c(PhAPGs, PE.ribo, PE.rpo)]
```


##PhANGs
###KEGG - zma00195
```{r KEGG.zma00195.digenomic}
#Retrieved from KEGG on 02/26/2020
KEGG.zma00195 <- read.csv(paste0(PathToData, "KEGGzma00195.csv"), stringsAsFactors = F)
#Get v4 accessions
KEGG.zma00195.digenomic <- join(KEGG.zma00195, Entrez)
```


###KEGG - zma00196
```{r KEGG.zma00196}
#Retrieved from KEGG on 02/26/2020
KEGG.zma00196 <- read.csv(paste0(PathToData, "KEGGzma00196.csv"), stringsAsFactors = F)
#Get v4 accessions
KEGG.zma00196 <- join(KEGG.zma00196, Entrez)
```

###CornCyc - Light Reactions
```{r CornCycLightRxns.digenomic}
#Retrieved from CornCyc on 02/26/2020
CornCycLightRxns.digenomic <- read.csv(paste0(PathToData, "CornCycLightRxns.csv"), stringsAsFactors = F)
```

###NE Rubisco
```{r NE.Rubisco}
#Defined by MaizeGDB
NE.Rubisco <- c("Zm00001d052595", "Zm00001d004894")
```

###Combine groups
```{r PhANGs}
PhA <- unique(na.omit(c(KEGG.zma00195.digenomic$Accession, KEGG.zma00196$Accession, CornCycLightRxns.digenomic$Gene.Accession, NE.Rubisco)))
#Trim to nuclear-encoded
PhANGs <- PhA[grep("Zm", PhA)]
```
##Nuclear-encoded plastid ribosome
```{r NE.PtRibo}
#grep plastid ribosomal proteins from NCBI protein name column
ribosom <- tmt[(grep("ribosom", tmt$Protein)),]
ribosom.pt <- ribosom[(grep("30S|50S", ribosom$Protein)),]
# Read in Maize GAMER GO dataset, which lists maize V4 accessions with all associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim(paste0(PathToData, "maize.B73.AGPv4.aggregate.gaf"), sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$GOterm <- AnnotationDbi::Term(as.character(gamer$term_accession))
#merge gamer GO terms
ribosom.pt <- merge.data.frame(ribosom.pt[,c(176,183)], gamer[,c(2,18)], by.x = "Gene", by.y = "db_object_id", all.x = T, all.y = F)
#Convert duplicated rows into multiple columns for each GO term
ribosom.pt.w <- dcast(setDT(ribosom.pt), Accession~rowid(Accession, prefix="GOterm"), value.var="GOterm")
ribosom.pt <- merge.data.frame(ribosom.pt, ribosom.pt.w, by.x = "Accession", by.y = "Accession")
ribosom.pt <- distinct(ribosom.pt)
#Combine columns of each pathway into single column
ribosom.pt <- unite(ribosom.pt, "GOterm", GOterm1:GOterm9, sep = " // ")
ribosom.pt$GOterm <- gsub(ribosom.pt$GOterm, pattern = " // NA", replacement = "")
ribosom.pt <- distinct(ribosom.pt)
#Define NE plastid ribosomal proteins as those having chloroplast or plastid cellular compartment GO terms
NE.PtRibo <- ribosom.pt$Gene[grep("chloropl|plastid", ribosom.pt$GOterm)]
```

##Nuclear-encoded plastid RNA polymerase
```{r NEP}
NEP <- "Zm00001d046835"
```

##Plastid background
```{r plastid}
#Import supplemental Table I from Friso et al. 2010
friso <- read.csv(paste0(PathToData,"Friso2010_SupTab1.csv"), stringsAsFactors = F)
colnames(friso) <- as.character(friso[1,])
friso <- friso[-1,]
friso <- friso[friso$`best gene model (selected)`=="yes",]
friso$Gene <- substr(friso$Accession, start = 1, stop = 13)
#convert to v4 accessions
friso <- merge.data.frame(friso, mgdb[,c(1,11)], by.x = "Gene", by.y = "v3_gene_model")
friso.v4 <- friso$v4_gene_model
#grep for "chloroplastic" in NCBI annotation
grep.chloroplastic <- tmt$Gene[grep("chloroplastic", tmt$Protein)]
grep.plastid <- tmt$Gene[grep("plastid", tmt$Protein)]
#add previously called photosynthesis digenomic proteins
plastid <- as.character(unique(na.omit(c(friso.v4, PE, PhA, NE.Rubisco, NE.PtRibo, PE.ribo, PhAPGs, grep.chloroplastic, grep.plastid))))
```






##Cytosolic ribosome
```{r CytoRibo}
#Cytosolic ribosomal proteins (NCBI Annotation)
CytoRibo<-tmt[(grep("40S|60S", tmt$Protein)),]
#remove the mitochondrial and biogenesis proteins
CytoRibo<- CytoRibo$Gene[-(grep("mito|Mito|biogen|Biogen", CytoRibo$Protein))]
```
##Etyhlene Biosynthesis
```{r CornCycEtBio}
#Retrieved from CornCyc on 05/11/2020
CornCycEtBio <- read.csv(paste0(PathToData, "CornCycEtBio.csv"), stringsAsFactors = F)
EtBio <- CornCycEtBio$Gene.Accession
SAMS <- CornCycEtBio$Gene.Accession[CornCycEtBio$Reaction.EC=="2.5.1.6"]
ACS <- CornCycEtBio$Gene.Accession[CornCycEtBio$Reaction.EC=="4.4.1.14"]
ACO <- CornCycEtBio$Gene.Accession[CornCycEtBio$Reaction.EC=="1.14.17.4"]
```
```{r acs2 and acs6}
acs2 <- "Zm00001d002592"
acs6 <- "Zm00001d033862"
```
```{r homologs}
AtACO1hom <- "Zm00001d036955"
AtACO2hom <- "Zm00001d020686"
AtSAM1AtSAM2hom <- "Zm00001d010920"
```


##Carbon fixation
```{r CarbonFixation}
CarbonFixationKEGG <- read.csv(paste0(PathToData, "CarbonFixationKEGG.csv"), stringsAsFactors = F)
CarbonFixation <- join(CarbonFixationKEGG, Entrez)
CarbonFixation <- na.omit(CarbonFixation$Accession)
```


##TPR
```{r TPR}
TPR <- tmt$Gene[grep("TPR", tmt$Protein)]
```
##PTAC
```{r PTAC}
PTAC <- tmt$Gene[grep("plastid transcriptionally", tmt$Protein)]
```

##Oxidoreductase
```{r Oxidoreductase}
Oxidoreductase <- gamer$db_object_id[gamer$term_accession=="GO:0016491"]
```

##Biosynthesis of Secondary Metabolites
```{r BiosynSecMet}
BiosynSecMetKEGG <- read.delim(paste0(PathToData, "BiosynSecMetKEGG.txt"))
colnames(BiosynSecMetKEGG)[4] <- "Entrez"
BiosynSecMetKEGG$Entrez <- str_replace(BiosynSecMetKEGG$Entrez, ",", "")
BiosynSecMetKEGG$Entrez <- as.numeric(BiosynSecMetKEGG$Entrez)
BiosynSecMet <- join(BiosynSecMetKEGG, Entrez)
BiosynSecMet <- na.omit(BiosynSecMet$Accession)
length(BiosynSecMet)
```

##Protease
```{r Protease}
Protease <- gamer$db_object_id[gamer$term_accession=="GO:0008233"]
```
##Alpha-linoleic acid metabolism
```{r AlphaLinAcidMet}
AlphaLinAcidMetKEGG <- read.delim(paste0(PathToData, "AlphaLinAcidMet.txt"))
colnames(AlphaLinAcidMetKEGG)[4] <- "Entrez"
AlphaLinAcidMetKEGG$Entrez <- str_replace(AlphaLinAcidMetKEGG$Entrez, ",", "")
AlphaLinAcidMetKEGG$Entrez <- as.numeric(AlphaLinAcidMetKEGG$Entrez)
AlphaLinAcidMet <- join(AlphaLinAcidMetKEGG, Entrez)
AlphaLinAcidMet <- na.omit(AlphaLinAcidMet$Accession)
```

##Protein biosynthesis
```{r ProteinBioSyn}
elongationF <- tmt$Gene[grep("elongation factor", tmt$Protein)]
initiationF <- tmt$Gene[grep("initiation factor", tmt$Protein)]
ProteinBioSyn <- c(elongationF, initiationF)
```

##SigmaFactors
```{r sigmafactors}
sigmafactors <- c("Zm00001d024507", "Zm00001d049494", "Zm00001d049160", "Zm00001d028585", "Zm00001d012810", "Zm00001d039194")
```
##GOI list
```{r make GOIs list}
GOIs <- list(PE.ribo, PhANGs, PhAPGs, plastid, NE.PtRibo, CytoRibo, EtBio, acs2, acs6, AtACO1hom, AtACO2hom, AtSAM1AtSAM2hom, SAMS, ACS, ACO, CarbonFixation, TPR, PTAC, Oxidoreductase, BiosynSecMet, Protease, AlphaLinAcidMet, ProteinBioSyn, sigmafactors, PE.rpo, PE.other, NEP)
names(GOIs) <- c("PE.ribo", "PhANGs", "PhAPGs", "plastid", "NE.PtRibo", "CytoRibo", "EtBio", "acs2",  "acs6", "AtACO1hom", "AtACO2hom", "AtSAM1AtSAM2hom", "SAMS", "ACS", "ACO", "CarbonFixation", "TPR", "PTAC", "Oxidoreductase", "BiosynSecMet", "Protease", "AlphaLinAcidMet", "ProteinBioSyn", "SigmaFactor", "PE.rpo", "PE.other", "NEP")
saveRDS(GOIs, "GOIs")
```


```{r read GOI list}
GOIs <- readRDS("GOIs")
```
#Define settings and functions for plotting
```{r CorFun}
#using dataframe and vector as input
CorFun <- function(x, xcols, y){
  for(i in 1:nrow(x)){
    v <- as.numeric(x[i,xcols])
    c <- cor(v, y, method = "pearson")
    x[i,"cor"]<- c
  }
  x
}
```
```{r get_legend}
get_legend<-function(myggplot){
  tmp <- ggplot_gtable(ggplot_build(myggplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}
```
```{r plotting colors}
colors.p <- c("Other"="#D8D8D8", "Plastid-localized"="black", "Cytosolic ribosome"="#B50000", "PE plastid ribosome"="#6A00EA", "NE plastid ribosome"="#E5A3FF", "PhAPGs"="#008275", "PhANGs"="#81E401", "Ethylene biosynthesis"="#EF8F6D")

colors.ea <- c("Other"="#D8D8D8", "SAMS"="black", "ACS"="black", "ACO"="black", "ZmACS2"="#24ff24", "ZmACS6"="#ff6db6", "AtACO1 homolog"="#006ddb", "AtACO2 homolog"="#b66dff", "AtSAM1/AtSAM2 homolog"="#db6d00")
```



#Figure 1
```{r}
#make list of dfs
tmt.SL.fig1 <- list("Hyb2HP"=tmt.SL[,c("Gene", "l2.B73xMo17.HP", "l10.ttest.HP")],
                    "Hyb2MP"=tmt.SL[,c("Gene", "l2.B73xMo17.MP", "l10.ttest.MP")],
                    "Par2Par"=tmt.SL[,c("Gene", "l2.B73.Mo17", "l10.ttest.P")])
#name plot panels
tmt.SL.fig1$Hyb2HP$plot <- "A"
tmt.SL.fig1$Hyb2MP$plot <- "B"
tmt.SL.fig1$Par2Par$plot <- "C"
#rename column to match
tmt.SL.fig1 <- lapply(tmt.SL.fig1, setNames, c("Gene","l2.ratio", "l10.ttest","plot"))
#bind
tmt.SL.fig1 <- do.call("rbind",tmt.SL.fig1)
```
```{r figure 1 statistics}
#subset GOIs
GOIs.fig1 <- GOIs[c("PhANGs","PhAPGs","NE.PtRibo","PE.ribo")]
names(GOIs.fig1) <- c("PhANGs","PhAPGs","NEplastidRibosome","PEplastidRibosome")

ztestfun <- function(df, name){
  df.l <- list()
  for(i in 1:length(GOIs.fig1)){
      #subset to GOI
      df.x = df$l2.ratio[df$Gene %in% GOIs.fig1[[i]]]
      #subset background
      df.y = df$l2.ratio[!df$Gene %in% GOIs.fig1[[i]]]
      #Calculate sigma.x
      sigx = sd(df.x)
      #Calculate sigma.y
      sigy = sd(df.y)
      #Calculate Z test
      z = z.test(x = df.x, y = df.y, sigma.x = sigx, sigma.y = sigy)
      #convert to dataframe
      z <- z%>%
       unlist()%>%
       as.data.frame()
      colnames(z) = "value"
      z$stat <- rownames(z)
      z$variable <- paste0(name, "_", names(GOIs.fig1[i]))
      df.l[[i]] <- z[1:9,]
  }
  names(df.l) <- names(GOIs.fig1)
  df.l
}

#calculate z test for each figure panel
tmt.SL.Hyb2HP.ztest <- ztestfun(tmt.SL.Hyb2HP, "Hyb/HP")
tmt.SL.Hyb2MP.ztest <- ztestfun(tmt.SL.Hyb2MP, "Hyb/MP")
tmt.SL.Par2Par.ztest <- ztestfun(tmt.SL.Par2Par, "B73/Mo17")

#combine statistics into single dataframe
tmt.SL.Hyb2HP.ztest <- do.call("rbind", tmt.SL.Hyb2HP.ztest)
tmt.SL.Hyb2MP.ztest <- do.call("rbind", tmt.SL.Hyb2MP.ztest)
tmt.SL.Par2Par.ztest <- do.call("rbind", tmt.SL.Par2Par.ztest)

tmt.SL.Ztest <- do.call("rbind", list(tmt.SL.Hyb2HP.ztest, tmt.SL.Hyb2MP.ztest, tmt.SL.Par2Par.ztest))
tmt.SL.Ztest <- tmt.SL.Ztest[order(tmt.SL.Ztest$variable, tmt.SL.Ztest$stat),c("variable","stat","value")]
tmt.SL.Ztest$figure <- "Figure 1"
rownames(tmt.SL.Ztest) <- NULL
```
```{r plot fig1}
 plot<-unique(tmt.SL.fig1[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.SL.fig1[,"plot"])
  for(i in 1:length(plot)){
    f1<-tmt.SL.fig1[which(tmt.SL.fig1[,"plot"]==plot[i]),]
###Set limits
f1$l2.ratio[f1$l2.ratio <= -1] =-1
f1$l2.ratio[f1$l2.ratio >= 1] =1
###Set GOIs
f1$Category ="Other"
f1[(f1$Gene %in% GOIs$plastid),"Category"] ="Plastid-localized"
f1[(f1$Gene %in% GOIs$PhANGs),"Category"] ="PhANGs"
f1[(f1$Gene %in% GOIs$PhAPGs),"Category"] ="PhAPGs"
f1[(f1$Gene %in% GOIs$NE.PtRibo),"Category"] ="NE plastid ribosome"
f1[(f1$Gene %in% GOIs$PE.ribo),"Category"] ="PE plastid ribosome"
###factor
f1$Category =factor(f1$Category, levels = c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
f1 =f1[(order(f1$Category)),]
    #scatter plot with marginal density curves
    f1.scat =ggplot(f1, mapping=aes(x=l2.ratio, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1.5)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome")], name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19,19))+
    theme_classic()+
      xlim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray40", size=0.5)+
      geom_hline(yintercept= 1.30103, linetype="dashed", color = "gray40", size=0.5)+
      ylab("-log10(p-value)")+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text =element_text(size=10), legend.text = element_text(size=10), axis.title = element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3, hjust = -0.2), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(ncol = 2))
        if(i==1){
          f1.scat =f1.scat+ xlab("log2 BxM/HP")
        }
        else if(i==2){
          f1.scat =f1.scat+ xlab("log2 BxM/MP")
        }
        else if(i==3){
          f1.scat =f1.scat+ xlab("log2 B73/Mo17")
        }
  
legend =get_legend(f1.scat)
    f1.scat =f1.scat+theme(legend.position = "none")
###Add density curves
    densityplotX<-axis_canvas(f1.scat, axis="x")+
      geom_density(aes(f1[,"l2.ratio"], color=f1[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome")], name="Category")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(f1.scat, axis="y", coord_flip = T)+
      geom_density(aes(f1[,"l10.ttest"], color=f1[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #put two plots together
      q =insert_xaxis_grob(f1.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q =insert_yaxis_grob(q, densityplotY, grid::unit((0.275), "null"), position = "right")
    if(i==1){
      plotList<-list()
      plotList[[1]]<-q
    }
    else{
      plotList =c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  #setup layout of figure panels
       des <- "A
               B
               C
               D"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], heights = c(1,1,1,0.5), nrow = 4, design = des)

ggsave(plot=r, filename =paste0(PathToPlots, "Figure1.pdf"), dpi=300, width=3.4252, height=8)  
```

#Figure 2
```{r setup fig2 data}
#select hyb/HP cols in order of plant height heterosis levels
tmt.6H.f2.hyb2HP <- tmt.6H[,c("Gene","l2.A682xB73.HP","l2.B84xMo17.HP","l2.Mo17xB73.HP","l2.B73xMo17.HP","l2.B84xB73.HP","l2.A682xMo17.HP")]
#select par/par cols in same order
tmt.6H.f2.par2par <- tmt.6H[,c("Gene","l2.A682.B73","l2.B84.Mo17","l2.Mo17.B73","l2.B73.Mo17","l2.B84.B73","l2.A682.Mo17")]
#rename cols to match figure panels
colnames(tmt.6H.f2.hyb2HP) <- c("Gene", LETTERS[seq(from=1, to=6)])
colnames(tmt.6H.f2.par2par) <- c("Gene", LETTERS[seq(from=1, to=6)])
#combine into single df
tmt.6H.f2.hyb2HP <- melt(tmt.6H.f2.hyb2HP)
tmt.6H.f2.par2par <- melt(tmt.6H.f2.par2par)
colnames(tmt.6H.f2.hyb2HP) <- c("Gene","plot","l2.HYB.HP")
colnames(tmt.6H.f2.par2par) <- c("Gene","plot","l2.parent.parent")
tmt.6H.f2 <- tmt.6H.f2.hyb2HP %>% inner_join(tmt.6H.f2.par2par, by=c('Gene','plot'))
```
```{r plot Fig2}
  plot<-unique(tmt.6H.f2[,"plot"])
  plotList<-vector(mode = "list", length = length(plot))
  plotLab<-unique(tmt.6H.f2[,"plot"])
  for(i in 1:length(plot)){
    f2=tmt.6H.f2[which(tmt.6H.f2[,"plot"]==plot[i]),]
    ###Set limits
f2$l2.HYB.HP[f2$l2.HYB.HP <= -0.75] = -0.75
f2$l2.HYB.HP[f2$l2.HYB.HP >= 0.75] = 0.75

f2$l2.parent.parent[f2$l2.parent.parent <= -1] = -1
f2$l2.parent.parent[f2$l2.parent.parent >= 1] = 1
###Set GOIs
f2 = f2
f2$Category = "Other"
f2[(f2$Gene %in% GOIs$plastid),"Category"] = "Plastid-localized"
f2[(f2$Gene %in% GOIs$PhANGs),"Category"] = "PhANGs"
f2[(f2$Gene %in% GOIs$PhAPGs),"Category"] = "PhAPGs"
f2[(f2$Gene %in% GOIs$NE.PtRibo),"Category"] = "NE plastid ribosome"
f2[(f2$Gene %in% GOIs$PE.ribo),"Category"] = "PE plastid ribosome"
###factor
f2$Category = factor(f2$Category, levels = c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome"))
f2 = f2[(order(f2$Category)),]
    #scatter plot with marginal density curves
    f2.scat = ggplot(f2, mapping=aes(x=l2.HYB.HP, y=l2.parent.parent, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome")], name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19,19), guide="none")+
    theme_classic()+
      xlim(c(-0.75,0.75))+
      ylim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      ggtitle(plotLab[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text=element_text(size=8), legend.text=element_text(size=8),axis.title = element_text(size = 8),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3.5, hjust = -0.485), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(ncol = 2))
legend = get_legend(f2.scat)
    f2.scat = f2.scat+theme(legend.position = "none")
    if(i==1){
      f2.scat = f2.scat+ xlab("log2 A682xB73/HP")+ ylab("log2 A682/B73")
    }
    else if(i==2){
      f2.scat = f2.scat+ xlab("log2 B84xMo17/HP")+ ylab("log2 B84/Mo17")
    }
    else if(i==3){
      f2.scat = f2.scat+ xlab("log2 Mo17xB73/HP")+ ylab("log2 Mo17/B73")
    }
    else if(i==4){
      f2.scat = f2.scat+ xlab("log2 B73xMo17/HP")+ ylab("log2 B73/Mo17")
    }
    else if(i==5){
      f2.scat = f2.scat+ xlab("log2 B84xB73/HP")+ ylab("log2 B84/B73")
    }
    else if(i==6){
      f2.scat = f2.scat+ xlab("log2 A682xMo17/HP")+ ylab("log2 A682/Mo17")
    }
###Add density curves
    densityplotX=axis_canvas(f2.scat, axis="x")+
      geom_density(aes(f2[,"l2.HYB.HP"], color=f2[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome")], name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY=axis_canvas(f2.scat, axis="y", coord_flip = T)+
      geom_density(aes(f2[,"l2.parent.parent"], color=f2[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      theme(plot.margin=grid::unit(c(-2,-2,0,-2), "mm"))
    #put two plots together
      q = insert_xaxis_grob(f2.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q = insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    if(i==1){
      plotList=list()
      plotList[[1]]=q
    }
    else{
      plotList = c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  #setup layout of figure panels
  des <- "AB
          CD
          EG
          HH"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], E=plotList[[5]], G=plotList[[6]], H=plotList[[7]], nrow = 4, heights = c(1,1,1,0.5), design = des)
  
ggsave(plot=r, filename =paste0(PathToPlots, "Figure2.pdf"), dpi=300, width=3.4252, height=5.5)  
```

#Figure 3
```{r setup dataframe fig3 data}
#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(tmt.RIL[c("Accession","l2.B73xMo17.MP","l2.M0014.14B.MP","l2.M0014.14M.MP","l2.M0016.16B.MP","l2.M0016.16M.MP","l2.M0021.21B.MP","l2.M0021.21M.MP","l2.M0317.317B.MP","l2.M0317.317M.MP")], tmt.6H[,c("Accession","l2.B73xMo17.MP","l2.Mo17xB73.MP","l2.B84xB73.MP","l2.B84xMo17.MP","l2.A682xB73.MP","l2.A682xB73.MP","l2.A682xMo17.MP")], by.x = "Accession", by.y = "Accession", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#run correlation function
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#Combine data for plotting
f3 <- tmt.RIL.6H.Hyb2MP.cor[,c("Gene", "cor")]
#Set GOIs
f3$Category <- "Other"
f3[(f3$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
f3[(f3$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
f3[(f3$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
f3[(f3$Gene %in% GOIs$TPR),"Category"] <- "TPR"
f3[(f3$Gene %in% GOIs$PTAC),"Category"] <- "PTAC"
f3[(f3$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
f3[(f3$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
f3[(f3$Gene %in% GOIs$ProteinBioSyn),"Category"] <- "Protein biosynthesis"
f3[(f3$Gene %in% GOIs$CarbonFixation),"Category"] <- "Carbon fixation"
f3[(f3$Gene %in% GOIs$Oxidoreductase),"Category"] <- "Oxidoreductase"
f3[(f3$Gene %in% GOIs$Protease),"Category"] <- "Protease"
f3[(f3$Gene %in% GOIs$BiosynSecMet),"Category"] <- "Biosynthesis of secondary metabolites"
f3[(f3$Gene %in% GOIs$AlphaLinAcidMet),"Category"] <- "Alpha-linoleic acid metabolism"

f3$Category <- factor(f3$Category, levels = c("Other", "PE plastid ribosome", "NE plastid ribosome","Cytosolic ribosome", "TPR", "Protein biosynthesis", "PTAC", "PhANGs", "PhAPGs", "Carbon fixation","Oxidoreductase","Protease","Biosynthesis of secondary metabolites","Alpha-linoleic acid metabolism"))
```
```{r Fig3 Statistics}
GOIs.cor <- GOIs[c("AlphaLinAcidMet","BiosynSecMet","Protease","Oxidoreductase","CarbonFixation","PhAPGs","PhANGs","PTAC","ProteinBioSyn","TPR","CytoRibo","NE.PtRibo","PE.ribo")]
GOIs.cor.names <- c("Alpha-linoleic acid metabolism", "Biosynthesis of secondary metabolites", "Protease","Oxidoreductase","Carbon fixation","PhAPGs", "PhANGs","PTAC", "Protein biosynthesis", "TPR","Cytosolic ribosome", "NE plastid ribosome",  "PE plastid ribosome")
tmt.RIL.6H.Hyb2MP.cor.Ztest <- list()
for(i in 1:13){
  tmt.RIL.6H.Hyb2MP.cor.x <- sd(tmt.RIL.6H.Hyb2MP.cor$cor[tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs.cor[[i]]])
  tmt.RIL.6H.Hyb2MP.cor.y <- sd(tmt.RIL.6H.Hyb2MP.cor$cor[!tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs.cor[[i]]])
  tmt.RIL.6H.Hyb2MP.cor.Ztest[[i]] <- as.data.frame(unlist(z.test(x = tmt.RIL.6H.Hyb2MP.cor$cor[tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs.cor[[i]]], y = tmt.RIL.6H.Hyb2MP.cor$cor[!tmt.RIL.6H.Hyb2MP.cor$Gene %in% GOIs.cor[[i]]], sigma.x = tmt.RIL.6H.Hyb2MP.cor.x, sigma.y = tmt.RIL.6H.Hyb2MP.cor.y)))
  colnames(tmt.RIL.6H.Hyb2MP.cor.Ztest[[i]]) <- GOIs.cor.names[i]
  tmt.RIL.6H.Hyb2MP.cor.Ztest[[i]]$stat <- rownames(tmt.RIL.6H.Hyb2MP.cor.Ztest[[i]])
}

tmt.RIL.6H.Hyb2MP.cor.Ztest <- Reduce(function(x, y, ...) merge(x, y, all = TRUE, by.x = "stat", by.y = "stat"),tmt.RIL.6H.Hyb2MP.cor.Ztest)
tmt.RIL.6H.Hyb2MP.cor.Ztest <- melt(tmt.RIL.6H.Hyb2MP.cor.Ztest, id.vars = "stat")
tmt.RIL.6H.Hyb2MP.cor.Ztest <- tmt.RIL.6H.Hyb2MP.cor.Ztest[,c("variable", "stat", "value")]
tmt.RIL.6H.Hyb2MP.cor.Ztest <- tmt.RIL.6H.Hyb2MP.cor.Ztest[order(tmt.RIL.6H.Hyb2MP.cor.Ztest$variable, tmt.RIL.6H.Hyb2MP.cor.Ztest$stat),]
tmt.RIL.6H.Hyb2MP.cor.Ztest$figure <- "Figure 3"
```
```{r plot Fig3}
ggplot(f3, mapping=aes(y=Category, x=cor, fill = stat(x)))+
  geom_density_ridges_gradient(alpha=0.7) +
  scale_fill_viridis_c(option = "B")+
  theme_ridges() +
  theme(axis.text =element_text(size=8), axis.title = element_text(size=8), legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=8))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)
ggsave(filename =paste0(PathToPlots, "Figure3.pdf"), dpi=300, width=3.4252, height=4)
```


#Figure 4
```{r setup fig4 A data}
#define proteins significantly above mid-parent (AMP) and below mid-parent (BMP) in A682xB73 hybrid
tmt.6H.A682B.Sig.AMP <- tmt.6H[tmt.6H$l2.A682xB73.MP > 0 & tmt.6H$ttest.A682xB73.MP < 0.05,]
tmt.6H.A682B.Sig.BMP <- tmt.6H[tmt.6H$l2.A682xB73.MP < 0 & tmt.6H$ttest.A682xB73.MP < 0.05,]
#make column on acs data indicating whether the protein is AMP or BMP in A682xB73
f4a <- tmt.acs[,c("Gene", "l2.ACS.B73","l10.pvalue")]
f4a$A682B <- "MP"
f4a$A682B[f4a$Gene %in% tmt.6H.A682B.Sig.AMP$Gene] <- "AMP"
f4a$A682B[f4a$Gene %in% tmt.6H.A682B.Sig.BMP$Gene] <- "BMP"
f4a$A682B <- factor(f4a$A682B, levels = c("MP","AMP", "BMP"))
f4a <- f4a[order(f4a$A682B),]
#set plotting axis limits
f4a$l2.ACS.B73[f4a$l2.ACS.B73 < -2] <- -2
f4a$l2.ACS.B73[f4a$l2.ACS.B73 > 2] <- 2
f4a$l10.pvalue[f4a$l10.pvalue > 7.5] <- 7.5
```
```{r setup fig4 B data}
#merge acs with A682xB73 data
f4b <- merge.data.frame(tmt.acs[,c("Accession", "l2.ACS.B73")], tmt.6H[,c("Accession", "l2.A682xB73.MP")], by.x = "Accession", by.y = "Accession", all=F)
#convert accession to gene
f4b$Gene <- substr(f4b$Accession, start = 1, stop = 14)
#Set plotting axis limits
f4b$l2.A682xB73.MP[f4b$l2.A682xB73.MP <= -.75] <- -.75
f4b$l2.A682xB73.MP[f4b$l2.A682xB73.MP >= .75] <- .75
f4b$l2.ACS.B73[f4b$l2.ACS.B73 <= -.75] <- -.75
f4b$l2.ACS.B73[f4b$l2.ACS.B73 >= .75] <- .75
#Set GOIs
f4b <- f4b
f4b$Category <- "Other"
f4b[(f4b$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
f4b[(f4b$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
f4b[(f4b$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
f4b[(f4b$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
f4b[(f4b$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
f4b[(f4b$Gene %in% GOIs$EtBio),"Category"] <- "Ethylene biosynthesis"
#factor
f4b$Category <- factor(f4b$Category, levels = c("Other", "Cytosolic ribosome", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome", "Ethylene biosynthesis"))
f4b <- f4b[(order(f4b$Category)),]
```
```{r fig4B statistics}
GOIs.acs <- GOIs[c("CytoRibo","PhANGs","PhAPGs","NE.PtRibo","PE.ribo","EtBio")]
GOIs.acs.names <- c("Cytosolic ribosome", "PhANGs", "PhAPGs", "NE plastid ribosome",  "PE plastid ribosome", "Ethylene biosynthesis")
tmt.acs.Ztest <- list()
for(i in 1:6){
  tmt.acs.x <- sd(tmt.acs$l2.ACS.B73[tmt.acs$Gene %in% GOIs.acs[[i]]])
  tmt.acs.y <- sd(tmt.acs$l2.ACS.B73[!tmt.acs$Gene %in% GOIs.acs[[i]]])
  tmt.acs.Ztest[[i]] <- as.data.frame(unlist(z.test(x = tmt.acs$l2.ACS.B73[tmt.acs$Gene %in% GOIs.acs[[i]]], y = tmt.acs$l2.ACS.B73[!tmt.acs$Gene %in% GOIs.acs[[i]]], sigma.x = tmt.acs.x, sigma.y = tmt.acs.y)))
  colnames(tmt.acs.Ztest[[i]]) <- GOIs.acs.names[i]
  tmt.acs.Ztest[[i]]$stat <- rownames(tmt.acs.Ztest[[i]])
}

tmt.acs.Ztest <- Reduce(function(x, y, ...) merge(x, y, all = TRUE, by.x = "stat", by.y = "stat"),tmt.acs.Ztest)
tmt.acs.Ztest <- melt(tmt.acs.Ztest, id.vars = "stat")
tmt.acs.Ztest <- tmt.acs.Ztest[,c("variable", "stat", "value")]
tmt.acs.Ztest <- tmt.acs.Ztest[order(tmt.acs.Ztest$variable, tmt.acs.Ztest$stat),]
tmt.acs.Ztest$figure <- "Figure 4"
```
```{r plot Fig4}
#Plot A panel
f4a.vol <- ggplot(f4a, mapping = aes(x=l2.ACS.B73, y=l10.pvalue, color=A682B))+
  geom_point(size=1)+
  scale_color_manual(values = c("MP"="grey", "AMP"="red", "BMP"="blue"), name="")+
  theme_bw()+
  xlab("log2 acs/B73")+
  ylab("-log10 p-value")+
  ggtitle("A")+
  geom_vline(xintercept = 0)+
  geom_hline(yintercept = -log10(0.05))+
  theme(aspect.ratio=1, legend.position = "bottom", axis.title = element_text(size=10), axis.text=element_text(size=10), legend.text = element_text(size=10), plot.title = element_text(hjust = -.05, vjust = -2))+
  coord_cartesian(expand = F)
f4a.vol.l <- get_legend(f4a.vol)
f4a.vol <- f4a.vol+theme(legend.position = "none")

#Plot B panel
f4b.scat <- ggplot(f4b, mapping=aes(x=l2.A682xB73.MP, y=l2.ACS.B73, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p[c("Other", "Cytosolic ribosome", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome", "Ethylene biosynthesis")], name="Category")+
      scale_shape_manual(values=c(1,19,19,19,19,19,19))+
    theme_classic()+
      xlim(c(-.75,.75))+
      ylim(c(-.75,.75))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      geom_hline(yintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      ylab("log2 acs/B73")+
      xlab("log2 A682xB73/MP")+
      ggtitle("B")+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(axis.text=element_text(size=10), legend.text=element_text(size=10),axis.title = element_text(size = 10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(hjust = -.55, vjust = -2))+
      guides(col=guide_legend(ncol = 2))
    f4b.scat.l <- get_legend(f4b.scat)
    f4b.scat <- f4b.scat+theme(legend.position = "none")

###Add density curves
    densityplotX<-axis_canvas(f4b.scat, axis="x")+
      geom_density(aes(f4b[,"l2.A682xB73.MP"], color=f4b[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p[c("Other", "Cytosolic ribosome", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome", "Ethylene biosynthesis")], name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(f4b.scat, axis="y", coord_flip = T)+
      geom_density(aes(f4b[,"l2.ACS.B73"], color=f4b[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p[c("Other", "Cytosolic ribosome", "PhANGs", "PhAPGs", "NE plastid ribosome", "PE plastid ribosome", "Ethylene biosynthesis")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray60", size=0.5)+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(f4b.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
#define panel layout
  des <- "A
          B
          C
          D"
  r<-patchwork::wrap_plots(A=f4a.vol, B=f4a.vol.l, C=q, D=f4b.scat.l,nrow = 4, heights = c(2.5,0.3,3,1.5), ncol = 1, design = des)

ggsave(plot=r, filename =paste0(PathToPlots, "Figure4.pdf"), dpi=300, width = 3.4252, height = 7)  
```



#Figure S1
```{r setup fig S1 data}
#Retrieve entrez ID
Entrez2 <- Entrez
colnames(Entrez2)[1] <- "Gene"
fs1 <- join(tmt.SL[,c("Gene", "l2.B73xMo17.MP")], Entrez2)
#remove proteins with no entrez or duplicate entrez
fs1.nodups <- fs1[!is.na(fs1$Entrez),]
fs1.nodups <- fs1.nodups[!fs1.nodups$Entrez=="",]
fs1.nodups <- fs1.nodups[!duplicated(fs1.nodups$Entrez),]
#make new dataframe with just l2(BM/MP) values and Entrez as rownames
fs1.nodups.trim <- data.frame("l2.B73xMo17.MP"=fs1.nodups$l2.B73xMo17.MP)
rownames(fs1.nodups.trim) <- fs1.nodups$Entrez
```
```{r plot FigS1}
fs1.nodups.trim.pv <- pathview(gene.data = fs1.nodups.trim, pathway.id = "zma00710", species = "zma",  bins = 20, limit = 1, out.suffix="tmt.SL.BxM")
```
#Figure S2
```{r setup fig S2 data}
#combine seedling leaf (SL) and leaf blade (LB) protein data
fs2 <- merge.data.frame(tmt.LB[,c("Accession", "l2.B73xMo17.MP", "l2.B73.Mo17")], tmt.SL[,c("Accession", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Accession", by.y = "Accession", all=F)
```
```{r plot FigS2}
#Set plot axis limits
fs2[fs2 <= -1] <- -1
fs2[fs2 >= 1] <- 1
#Set GOIs
fs2$Gene <- substr(fs2$Accession, start = 1, stop = 14)
fs2$Category <- "Other"
fs2[(fs2$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
fs2[(fs2$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
fs2[(fs2$Gene %in% GOIs$PE.ribo),"Category"] <- "PE plastid ribosome"
fs2[(fs2$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
fs2[(fs2$Gene %in% GOIs$PhAPGs),"Category"] <- "PhAPGs"
#factor
fs2$Category <- factor(fs2$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs"))
fs2 <- fs2[(order(fs2$Category)),]
#Plot Fig S2 A
fs2a.scat <- ggplot(fs2, mapping=aes(x=l2.B73xMo17.MP.x, y=l2.B73xMo17.MP.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs")], name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=8), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.4), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    ggtitle("A")+
    xlab("log2 BxM/MP LB")+
    ylab("log2 BxM/MP SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
fs2a.scat <- fs2a.scat+theme(legend.position = "none")
##Add density curves
###set up marginal density plot X axis
    densityplotX<-axis_canvas(fs2a.scat, axis="x")+
      geom_density(aes(fs2[,"l2.B73xMo17.MP.x"], color=fs2[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs")], name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
###set up marginal density plot Y axis
    densityplotY<-axis_canvas(fs2a.scat, axis="y", coord_flip = T)+
      geom_density(aes(fs2[,"l2.B73xMo17.MP.y"], color=fs2[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
###put two plots together
      A <- insert_xaxis_grob(fs2a.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      A <- insert_yaxis_grob(A, densityplotY, grid::unit(0.275, "null"), position = "right")
#Plot Fig S2 B
fs2b.scat <- ggplot(fs2, mapping=aes(x=l2.B73.Mo17.x, y=l2.B73.Mo17.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs")], name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=8), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.4), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=3))+
    ggtitle("B")+
    xlab("log2 B73/Mo17 LB")+
    ylab("log2 B73/Mo17 SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(fs2b.scat)
fs2b.scat <- fs2b.scat+theme(legend.position = "none")
##Add density curves
###set up marginal density plot X axis
    densityplotX<-axis_canvas(fs2b.scat, axis="x")+
      geom_density(aes(fs2[,"l2.B73.Mo17.x"], color=fs2[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs")], name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
###set up marginal density plot Y axis
    densityplotY<-axis_canvas(fs2b.scat, axis="y", coord_flip = T)+
      geom_density(aes(fs2[,"l2.B73.Mo17.y"], color=fs2[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PE plastid ribosome", "PhANGs", "PhAPGs")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
###put two plots together
      B <- insert_xaxis_grob(fs2b.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      B <- insert_yaxis_grob(B, densityplotY, grid::unit(0.275, "null"), position = "right")
#define panel layout
fs2.design <- "AB
               CC"
fs2ab<-patchwork::wrap_plots(A=A, B=B, C=legend,  nrow = 2, heights = c(3,1.2), design = fs2.design)

ggsave(plot=fs2ab, filename =paste0(PathToPlots, "FigureS2.pdf"), dpi=300, width=4.488189, height=3)
```

#Figure S3
```{r Setup data for Fig S3}
cpm.SL.LB <- merge.data.frame(cpm.SL, cpm.LB, by.x = "Gene", by.y = "Gene", all=F)

fs3v <- c(
  summary(lm(B73.1~Mo17.1, data = cpm.SL))$adj.r.squared,
  summary(lm(B73xMo17.1~B73.1, data = cpm.SL))$adj.r.squared,
  summary(lm(B73xMo17.1~Mo17.1, data = cpm.SL))$adj.r.squared,
  summary(lm(B73xMo17.1~B73xMo17.2, data = cpm.SL))$adj.r.squared,
  summary(lm(B73.1~Mo17.1, data = cpm.LB))$adj.r.squared,
  summary(lm(B73xMo17.1~B73.1, data = cpm.LB))$adj.r.squared,
  summary(lm(B73xMo17.1~Mo17.1, data = cpm.LB))$adj.r.squared,
  summary(lm(B73xMo17.1~B73xMo17.2, data = cpm.LB))$adj.r.squared,
  summary(lm(B73.1.x~B73.1.y, data = cpm.SL.LB))$adj.r.squared,
  summary(lm(Mo17.1.x~Mo17.1.y, data = cpm.SL.LB))$adj.r.squared,
  summary(lm(B73xMo17.1.x~B73xMo17.1.y, data = cpm.SL.LB))$adj.r.squared
)

fs3df <- data.frame("Rsquared"=fs3v, "comparison"=c("SL B73.1 vs SL Mo17.1", "SL BxM.1 vs SL B73.1", "SL BxM.1 vs SL Mo17.1", "SL.BxM.1 vs SL BxM.2", "LB B73.1 vs LB Mo17.1", "LB BxM.1 vs LB B73.1", "LB BxM.1 vs LB Mo17.1", "LB.BxM.1 vs LB BxM.2", "SL B73.1 vs LB B73.1", "SL Mo17.1 vs LB Mo17.1", "SL BxM.1 vs LB BxM.1"))

fs3df$comparison <- factor(fs3df$comparison, levels = fs3df$comparison)
```
```{r Plot S3}
ggplot(fs3df, mapping = aes(x=comparison, y=Rsquared))+
  geom_col()+
  theme_bw()+
  ylab("R squared")+
  xlab("")+
  theme(axis.text.x = element_text(angle=45, hjust = 1, size = 8))
ggsave(paste0(PathToPlots,"FigureS3.pdf"), width = 7.007874, height = 6)
```
#Figure S4
```{r setup data for figS4}
#make list of dfs
cpm.SL.figs4 <- list("Hyb2HP"=cpm.SL[,c("Gene", "l2.B73xMo17.HP", "l10.ttest.HP")],
                     "Hyb2MP"=cpm.SL[,c("Gene", "l2.B73xMo17.MP", "l10.ttest.MP")],
                     "Par2Par"=cpm.SL[,c("Gene", "l2.B73.Mo17", "l10.ttest.P")])
#name plot panels
cpm.SL.figs4$Hyb2HP$plot <- "A"
cpm.SL.figs4$Hyb2MP$plot <- "B"
cpm.SL.figs4$Par2Par$plot <- "C"
#rename column to match
cpm.SL.figs4 <- lapply(cpm.SL.figs4, setNames, c("Gene","l2.ratio", "l10.ttest","plot"))
#bind
cpm.SL.figs4 <- do.call("rbind",cpm.SL.figs4)
```
```{r plot figs4}
plot<-unique(cpm.SL.figs4[,"plot"])
plotList<-vector(mode = "list", length = length(plot))
  for(i in 1:length(plot)){
    figS4=cpm.SL.figs4[which(cpm.SL.figs4[,"plot"]==plot[i]),]
    ###Set plot axis limits
    figS4 = df
    figS4$l2.ratio[figS4$l2.ratio <= -1] = -1
    figS4$l2.ratio[figS4$l2.ratio >= 1] = 1
    ###Set GOIs
    figS4 = figS4
    figS4$Category = "Other"
    figS4[(figS4$Gene %in% GOIs$plastid),"Category"] = "Plastid-localized"
    figS4[(figS4$Gene %in% GOIs$PhANGs),"Category"] = "PhANGs"
    figS4[(figS4$Gene %in% GOIs$NE.PtRibo),"Category"] = "NE plastid ribosome"
    ###factor
    figS4$Category = factor(figS4$Category, levels = c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome"))
    figS4 = figS4[(order(figS4$Category)),]
    #scatter plot with marginal density curves
    figS4.scat = ggplot(figS4, mapping=aes(x=l2.ratio, y=l10.ttest, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1.5)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome")], name="Category")+
      scale_shape_manual(values=c(1,19,19,19))+
      theme_classic()+
      xlim(c(-1,1))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray40", size=0.5)+
      geom_hline(yintercept= 1.30103, linetype="dashed", color = "gray40", size=0.5)+
      ylab("-log10(p-value)")+
      ggtitle(plot[i])+
      theme(aspect.ratio=1)+
      coord_fixed(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
      theme(axis.text =element_text(size=10), legend.text = element_text(size=10), axis.title = element_text(size=10),
        legend.position="bottom", legend.title = element_blank(), plot.title = element_text(size=10, vjust = -3, hjust = -0.2), legend.spacing.x = unit(0.1, 'cm'), legend.spacing.y = unit(-3, 'cm'))+
      guides(col=guide_legend(ncol = 2))
        if(i==1){
          figS4.scat = figS4.scat+ xlab("log2 BxM/HP")
        }
        else if(i==2){
          figS4.scat = figS4.scat+ xlab("log2 BxM/MP")
        }
        else if(i==3){
          figS4.scat = figS4.scat+ xlab("log2 B73/Mo17")
        }
  
    legend = get_legend(figS4.scat)
    figS4.scat = figS4.scat+theme(legend.position = "none")
    ###Add density curves
    densityplotX=axis_canvas(figS4.scat, axis="x")+
      geom_density(aes(figS4[,"l2.ratio"], color=figS4[,"Category"]), size=0.5)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome")], name="Category")+
      theme_void()+
      theme(legend.title = element_blank())+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #set up marginal density plot Y axis
    densityplotY=axis_canvas(figS4.scat, axis="y", coord_flip = T)+
      geom_density(aes(figS4[,"l10.ttest"], color=figS4[,"Category"]), size=0.5)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "PhANGs", "NE plastid ribosome")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip(expand = F)+
      theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))
    #put two plots together
      q = insert_xaxis_grob(figS4.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q = insert_yaxis_grob(q, densityplotY, grid::unit((0.275), "null"), position = "right")
    if(i==1){
      plotList=list()
      plotList[[1]]=q
    }
    else{
      plotList = c(plotList, list(q))
    }
  }
  plotList <- c(plotList, list(legend))
  
#setup layout for panels
       des <- "A
               B
               C
               D"
  r<-patchwork::wrap_plots(A=plotList[[1]], B=plotList[[2]], C=plotList[[3]], D=plotList[[4]], heights = c(1,1,1,0.5), nrow = 4, design = des)

ggsave(plot=r, filename =paste0(PathToPlots, "FigureS4.pdf"), dpi=300, width=3.4252, height=8)  
```

#Figure S5
```{r setup data for Fig S5}
fs5 <- merge.data.frame(cpm.LB[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], cpm.SL[,c("Gene", "l2.B73xMo17.MP", "l2.B73.Mo17")], by.x = "Gene", by.y = "Gene", all=F)
###Set GOIs
fs5$Category <- "Other"
fs5[(fs5$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
fs5[(fs5$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
fs5[(fs5$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
###factor
fs5$Category <- factor(fs5$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs"))
fs5 <- fs5[(order(fs5$Category)),]
```
```{r plot FigS5}
#Panel A
##Set plot axis limits
fs5a <- fs5
fs5a[fs5a <= -2] <- -2
fs5a[fs5a >= 2] <- 2
##Scatterplot
fs5a.scat <- ggplot(fs5a, mapping=aes(x=l2.B73xMo17.MP.x, y=l2.B73xMo17.MP.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs")], name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=8), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="none", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.4), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    ggtitle("A")+
    xlab("log2 BxM/MP LB")+
    ylab("log2 BxM/MP SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
fs5a.scat <- fs5a.scat+theme(legend.position = "none")
##Add density curves
    ###set up marginal density plot X axis
    densityplotX<-axis_canvas(fs5a.scat, axis="x")+
      geom_density(aes(fs5[,"l2.B73xMo17.MP.x"], color=fs5[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs")], name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    ###set up marginal density plot Y axis
    densityplotY<-axis_canvas(fs5a.scat, axis="y", coord_flip = T)+
      geom_density(aes(fs5[,"l2.B73xMo17.MP.y"], color=fs5[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    ###put two plots together
      A <- insert_xaxis_grob(fs5a.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      A <- insert_yaxis_grob(A, densityplotY, grid::unit(0.275, "null"), position = "right")
      
#Panel B
##Set limits
fs5b <- fs5
fs5b[fs5b <= -1] <- -1
fs5b[fs5b >= 1] <- 1
##Set GOIs
fs5b$Category <- "Other"
fs5b[(fs5b$Gene %in% GOIs$plastid),"Category"] <- "Plastid-localized"
fs5b[(fs5b$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
fs5b[(fs5b$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
##factor
fs5b$Category <- factor(fs5b$Category, levels = c("Other", "Plastid-localized", "NE plastid ribosome", "PhANGs"))
fs5b <- fs5b[(order(fs5b$Category)),]
##Scatterplot
fs5b.scat <- ggplot(fs5b, mapping=aes(x=l2.B73.Mo17.x, y=l2.B73.Mo17.y, color=Category, shape=Category))+
    geom_point(aes(color=Category), size=1)+
  scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs")], name="Category")+
  scale_shape_manual(values=c(1,19,19,19,19))+
  theme(plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=8), legend.text=element_text(size=10), axis.title = element_text(size=10),
                legend.position="bottom", legend.title = element_blank(),  plot.title = element_text(size=10, vjust = -3, hjust = -0.4), plot.margin=grid::unit(c(-3,-3,0,-3), "mm"))+
          guides(col=guide_legend(ncol=3))+
    ggtitle("B")+
    xlab("log2 B73/Mo17 LB")+
    ylab("log2 B73/Mo17 SL")+
    theme(aspect.ratio=1)+
    coord_fixed(expand = F)
legend <- get_legend(fs5b.scat)
fs5b.scat <- fs5b.scat+theme(legend.position = "none")
###Add density curves
    ####set up marginal density plot X axis
    densityplotX<-axis_canvas(fs5b.scat, axis="x")+
      geom_density(aes(fs5b[,"l2.B73.Mo17.x"], color=fs5b[,"Category"]), size=0.7)+
    scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs")], name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    ####set up marginal density plot Y axis
    densityplotY<-axis_canvas(fs5b.scat, axis="y", coord_flip = T)+
      geom_density(aes(fs5b[,"l2.B73.Mo17.y"], color=fs5b[,"Category"]), size=0.7)+
      scale_color_manual(values=colors.p[c("Other", "Plastid-localized", "NE plastid ribosome","PhANGs")], name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    ####put two plots together
      B <- insert_xaxis_grob(fs5b.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      B <- insert_yaxis_grob(B, densityplotY, grid::unit(0.275, "null"), position = "right")
      
fs5.des <- "AB
            CC"
fs5 <-patchwork::wrap_plots(A=A, B=B, C=legend,  nrow = 2, heights = c(3,1.2), design = fs5.des)

ggsave(plot=fs5, filename ="fig5.pdf", dpi=300, width=4.488189, height=3)
#ggsave(plot=fs5, filename =paste0(PathToPlots, "FigureS5.pdf"), dpi=300, width=4.488189, height=3)
```








#Figure S6
```{r setup data for Fig S6}
#combine RIL and 6 hybrids Protein
fs6.df <- merge.data.frame(select(cpm.RIL, c(Gene, contains(".MP"))), select(cpm.6H, c(Gene, contains(".MP"))), by.x = "Gene", by.y = "Gene", all = F)
#run correlation function
fs6.df <- CorFun(fs6.df, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#Combine data for plotting
fs6.df$data <- "Hybrid/MP"
fs6.df <- fs6.df[,c("Gene", "cor", "data")]
#Set GOIs
fs6.df$Category <- "Other"
fs6.df[(fs6.df$Gene %in% GOIs$CytoRibo),"Category"] <- "Cytosolic ribosome"
fs6.df[(fs6.df$Gene %in% GOIs$NE.PtRibo),"Category"] <- "NE plastid ribosome"
fs6.df[(fs6.df$Gene %in% GOIs$TPR),"Category"] <- "TPR"
fs6.df[(fs6.df$Gene %in% GOIs$PTAC),"Category"] <- "PTAC"
fs6.df[(fs6.df$Gene %in% GOIs$PhANGs),"Category"] <- "PhANGs"
fs6.df[(fs6.df$Gene %in% GOIs$ProteinBioSyn),"Category"] <- "Protein biosynthesis"
fs6.df[(fs6.df$Gene %in% GOIs$CarbonFixation),"Category"] <- "Carbon fixation"
fs6.df[(fs6.df$Gene %in% GOIs$Oxidoreductase),"Category"] <- "Oxidoreductase"
fs6.df[(fs6.df$Gene %in% GOIs$Protease),"Category"] <- "Protease"
fs6.df[(fs6.df$Gene %in% GOIs$BiosynSecMet),"Category"] <- "Biosynthesis of secondary metabolites"
fs6.df[(fs6.df$Gene %in% GOIs$lphaLinAcidMet),"Category"] <- "Alpha-linoleic acid metabolism"

fs6.df$Category <- factor(fs6.df$Category, levels = c("Other", "NE plastid ribosome","Cytosolic ribosome", "TPR","Protein biosynthesis", "PTAC", "PhANGs", "Carbon fixation","Oxidoreductase","Protease","Biosynthesis of secondary metabolites","Alpha-linoleic acid metabolism"))
```
```{r plot FigS6}
ggplot(fs6.df, mapping=aes(y=Category, x=cor, fill = stat(x)))+
  geom_density_ridges_gradient(alpha=0.7) +
   scale_fill_viridis_c(option = "B")+
  theme_ridges() +
  theme(axis.text =element_text(size=8), axis.title = element_text(size=8), legend.position = "none", axis.title.y = element_blank(), axis.title.x = element_text(hjust = 0.5, size=10))+
  coord_cartesian(expand = F)+
  xlab("Pearson correlation")+
  xlim(-1,1)
ggsave(filename =paste0(PathToPlots, "FigureS6.pdf"), dpi=300, width=3.4252, height=4)
```


#Figure S7
```{r setup dataframe cpm.SL.entrez.nodups.trim}
#merge with mdb to get entrez 
cpm.SL.entrez <- join(cpm.SL[,c("Gene", "l2.B73xMo17.MP")], Entrez2)
#remove proteins with no entrez or duplicate entrez
cpm.SL.entrez.nodups <- cpm.SL.entrez[!is.na(cpm.SL.entrez$Entrez),]
cpm.SL.entrez.nodups <- cpm.SL.entrez.nodups[!cpm.SL.entrez.nodups$Entrez=="",]
cpm.SL.entrez.nodups <- cpm.SL.entrez.nodups[!duplicated(cpm.SL.entrez.nodups$Entrez),]
#make new dataframe with just l2(BM/MP) values and Entrez as rownames
cpm.SL.entrez.nodups.trim <- data.frame("l2.B73xMo17.MP"=cpm.SL.entrez.nodups$l2.B73xMo17.MP)
rownames(cpm.SL.entrez.nodups.trim) <- cpm.SL.entrez.nodups$Entrez
```
```{r plot FigS7A}
cpm.SL.entrez.nodups.trim.pv <- pathview(gene.data = cpm.SL.entrez.nodups.trim, pathway.id = "zma04712", species = "zma",  bins = 20, limit = 1, out.suffix="cpm.SL.BxM")
```
```{r setup dataframe cpm.RIL.6H.Hyb2MP.cor.entrez.nodups}
#merge with mdb to get entrez 
cpm.RIL.6H.Hyb2MP.cor.entrez <- merge.data.frame(cpm.RIL.6H.Hyb2MP.cor, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
cpm.RIL.6H.Hyb2MP.cor.entrez.nodups <- cpm.RIL.6H.Hyb2MP.cor.entrez[!is.na(cpm.RIL.6H.Hyb2MP.cor.entrez$Entrez),]
cpm.RIL.6H.Hyb2MP.cor.entrez.nodups <- cpm.RIL.6H.Hyb2MP.cor.entrez.nodups[!cpm.RIL.6H.Hyb2MP.cor.entrez.nodups$Entrez=="",]
cpm.RIL.6H.Hyb2MP.cor.entrez.nodups <- cpm.RIL.6H.Hyb2MP.cor.entrez.nodups[!duplicated(cpm.RIL.6H.Hyb2MP.cor.entrez.nodups$Entrez),]
#make new dataframe with just l2(BM/MP) values and Entrez as rownames
cpm.RIL.6H.Hyb2MP.cor.entrez.nodups.trim <- data.frame("cor"=cpm.RIL.6H.Hyb2MP.cor.entrez.nodups$cor)
rownames(cpm.RIL.6H.Hyb2MP.cor.entrez.nodups.trim) <- cpm.RIL.6H.Hyb2MP.cor.entrez.nodups$Entrez
```
```{r plot FigS7B}
cpm.RIL.6H.Hyb2MP.cor.entrez.nodups.trim.pv <- pathview(gene.data = cpm.RIL.6H.Hyb2MP.cor.entrez.nodups.trim, pathway.id = "zma04712", species = "zma",  bins = 20, limit = 1, out.suffix="cpm.cor", node.sum = "max.abs")
```



#Figure S8
##A
```{r setup dataframe tmt.A682B.entrez.nodups.trim}
#merge with mdb to get entrez 
tmt.A682B.entrez <- merge.data.frame(tmt.6H, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
tmt.A682B.entrez.nodups <- tmt.A682B.entrez[!is.na(tmt.A682B.entrez$Entrez),]
tmt.A682B.entrez.nodups <- tmt.A682B.entrez.nodups[!tmt.A682B.entrez.nodups$Entrez=="",]
tmt.A682B.entrez.nodups <- tmt.A682B.entrez.nodups[!duplicated(tmt.A682B.entrez.nodups$Entrez),]
#make new dataframe with just l2(BM/MP) values and Entrez as rownames
tmt.A682B.entrez.nodups.trim <- data.frame("l2.A682xB73.MP"=tmt.A682B.entrez.nodups$l2.A682xB73.MP)
rownames(tmt.A682B.entrez.nodups.trim) <- tmt.A682B.entrez.nodups$Entrez
```
```{r plot FigS8A}
tmt.A682B.entrez.nodups.trim.pv <- pathview(gene.data = tmt.A682B.entrez.nodups.trim, pathway.id = "zma00592", species = "zma",  bins = 20, limit = 0.5, out.suffix="tmt.A682B")
```
##B
```{r setup dataframe tmt.acs.entrez.nodups.trim}
#merge with mdb to get entrez 
tmt.acs.entrez <- merge.data.frame(tmt.acs, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", all = F)
#remove proteins with no entrez or duplicate entrez
tmt.acs.entrez.nodups <- tmt.acs.entrez[!is.na(tmt.acs.entrez$Entrez),]
tmt.acs.entrez.nodups <- tmt.acs.entrez.nodups[!tmt.acs.entrez.nodups$Entrez=="",]
tmt.acs.entrez.nodups <- tmt.acs.entrez.nodups[!duplicated(tmt.acs.entrez.nodups$Entrez),]
#make new dataframe with just l2(acs/B73) values and Entrez as rownames
tmt.acs.entrez.nodups.trim <- data.frame("l2.ACS.B73"=tmt.acs.entrez.nodups$l2.ACS.B73)
rownames(tmt.acs.entrez.nodups.trim) <- tmt.acs.entrez.nodups$Entrez
```
```{r plot FigS8B}
tmt.acs.entrez.nodups.trim.pv <- pathview(gene.data = tmt.acs.entrez.nodups.trim, pathway.id = "zma00592", species = "zma",  bins = 20, limit = 0.5, out.suffix="tmt.acs")
```

#Supplemental Table 1
```{r ST1}
tmt.SL.ST1 <- tmt.SL[tmt.SL$Gene %in% c(GOIs$PhAPGs, GOIs$PE.ribo, GOIs$SigmaFactor, GOIs$PE.rpo, GOIs$PE.other, GOIs$NE.PtRibo, GOIs$NEP, GOIs$PTAC),c("Gene",  "Protein", "l2.B73xMo17.MP", "ttest.MP")]
cpm.SL.ST1 <- cpm.SL[cpm.SL$Gene %in% c(GOIs$PhAPGs, GOIs$PE.ribo, GOIs$SigmaFactor, GOIs$PE.rpo, GOIs$PE.other, GOIs$NE.PtRibo, GOIs$NEP, GOIs$PTAC),c("Gene", "l2.B73xMo17.MP", "ttest.MP")]

ST1 <- merge.data.frame(tmt.SL.ST1, cpm.SL.ST1, by.x = "Gene", by.y = "Gene", all = T)
colnames(ST1) <- c("Gene",  "Protein", "Protein log2(B73xMo17/MP)", "Protein p-value", "RNA log2(B73xMo17/MP)", "RNA p-value")

#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(
  select(tmt.RIL, c(Accession, l2.B73xMo17.MP:l2.M0317.317M.MP)),
  select(tmt.6H, c(Accession, l2.B73xMo17.MP:l2.A682xMo17.MP)),
  by.x = "Accession", by.y = "Accession", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#run correlation function
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#combine RIL and 6 hybrids Protein
cpm.RIL.6H.Hyb2MP <- merge.data.frame(
  select(cpm.RIL, c(Gene, l2.B73xMo17.MP:l2.M0317xMo17.MP)),
  select(cpm.6H, c(Gene, l2.B73xMo17.MP:l2.A682xMo17.MP)),
  by.x = "Gene", by.y = "Gene", all = F)
#run correlation function
cpm.RIL.6H.Hyb2MP.cor <- CorFun(cpm.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#Combine
ST1 <- merge.data.frame(ST1, tmt.RIL.6H.Hyb2MP.cor[,c("Gene", "cor")], by.x = "Gene", by.y = "Gene", all.x = T, all.y = F)
ST1 <- merge.data.frame(ST1, cpm.RIL.6H.Hyb2MP.cor[,c("Gene", "cor")], by.x = "Gene", by.y = "Gene", all.x = T, all.y = F)
#asign GOIs
ST1$Category <- NA
ST1$Category[ST1$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
ST1$Category[ST1$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
ST1$Category[ST1$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
ST1$Category[ST1$Gene %in% GOIs$SigmaFactor] <- "Sigma factor"
ST1$Category[ST1$Gene %in% GOIs$PE.rpo] <- "PE plastid RNA polymerase"
ST1$Category[ST1$Gene %in% GOIs$NEP] <- "NE plastid RNA polymerase"
ST1$Category[ST1$Gene %in% GOIs$PTAC] <- "PTAC"
ST1$Category[ST1$Gene %in% GOIs$PE.other] <- "PE other"
ST1$Protein[ST1$Gene=="Zm00001d024507"] <- "sig1A"
ST1$Protein[ST1$Gene=="Zm00001d049494"] <- "sig1B"
ST1$Protein[ST1$Gene=="Zm00001d049160"] <- "sig2A"
ST1$Protein[ST1$Gene=="Zm00001d028585"] <- "sig2B"
ST1$Protein[ST1$Gene=="Zm00001d012810"] <- "sig6"
ST1$Protein[ST1$Gene=="Zm00001d039194"] <- "sig8"
ST1$Protein[ST1$Gene=="Zm00001d046835"] <- "rpo2"
colnames(ST1)[7:8] <- c("Protein correlation", "RNA correlation")
#re-order cols
ST1 <- ST1[,c("Gene","Protein","Category","Protein log2(B73xMo17/MP)","Protein p-value","Protein correlation","RNA log2(B73xMo17/MP)","RNA p-value","RNA correlation")]
ST1 <- ST1[order(ST1$Category),]
write.csv(ST1, paste0(PathToTables,"SupplementalTable1.csv"), row.names = F)
```


#Supplemental Dataset 1
```{r SD1}
SD1 <- tmt.SL[,c("Gene", "Accession", "Protein", "B73xMo17.MP", "B73.Mo17")]
colnames(SD1) <- c("Gene", "Accession", "Protein", "BxM.Midparent", "B73.Mo17")
#Add category column
SD1$Category <- "Other"
SD1$Category[SD1$Gene %in% GOIs$plastid] <- "Plastid"
SD1$Category[SD1$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
SD1$Category[SD1$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
SD1$Category[SD1$Gene %in% GOIs$PhANGs] <- "PhANGs"
SD1$Category[SD1$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
SD1$Category[SD1$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
SD1$Category[SD1$Gene %in% GOIs$SAMS] <- "Ethylene biosynthesis - SAM synthase"
SD1$Category[SD1$Gene %in% GOIs$ACS] <- "Ethylene biosynthesis - ACC synthase"
SD1$Category[SD1$Gene %in% GOIs$ACO] <- "Ethylene biosynthesis - ACC oxidase"
SD1$Category[SD1$Gene %in% GOIs$acs2] <- "Ethylene biosynthesis - ACS2"
SD1$Category[SD1$Gene %in% GOIs$acs6] <- "Ethylene biosynthesis - ACS6"
SD1$Category[SD1$Gene %in% GOIs$AtACO1hom] <- "Ethylene biosynthesis - AtACO1 homolog"
SD1$Category[SD1$Gene %in% GOIs$AtACO2hom] <- "Ethylene biosynthesis - AtACO2 homolog"
SD1$Category[SD1$Gene %in% GOIs$AtSAM1AtSAM2hom] <- "Ethylene biosynthesis - AtSAM1/2 homolog"

write.csv(SD1, paste0(PathToDatasets,"SupplementalDataset1.csv"), row.names = F)
```

#Supplemental Dataset 2
```{r SD2}
SD2 <- rbind(tmt.SL.Ztest, tmt.RIL.6H.Hyb2MP.cor.Ztest, tmt.acs.Ztest)
write.csv(SD2, paste0(PathToDatasets, "SupplementalDataset2.csv"), row.names = F)
```


#Supplemental Dataset 5
```{r}
colnames(tmt.RIL)
```
```{r}

```

```{r SD5}
#combine RIL and 6 hybrids Protein
tmt.RIL.6H.Hyb2MP <- merge.data.frame(
  select(tmt.RIL, c(Accession, l2.B73xMo17.MP:l2.M0317.317M.MP)),
  select(tmt.6H, c(Accession, l2.B73xMo17.MP:l2.A682xMo17.MP)),
  by.x = "Accession", by.y = "Accession", all = F)
#combine RIL and 6 hybrids Protein
cpm.RIL.6H.Hyb2MP <- merge.data.frame(
  select(cpm.RIL, c(Gene, l2.B73xMo17.MP:l2.M0317xMo17.MP)),
  select(cpm.6H, c(Gene, l2.B73xMo17.MP:l2.A682xMo17.MP)),
  by.x = "Gene", by.y = "Gene", all = F)
#convert Accession column to gene column
tmt.RIL.6H.Hyb2MP$Gene <- substr(tmt.RIL.6H.Hyb2MP$Accession, start = 1, stop = 14)
#run correlation function
cpm.RIL.6H.Hyb2MP.cor <- CorFun(cpm.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
tmt.RIL.6H.Hyb2MP.cor <- CorFun(tmt.RIL.6H.Hyb2MP, 2:16, c(PlantHeights.ril.MP, PlantHeights.6h.MP))
#merge RIL and 6 hybrids
SupT5 <- merge.data.frame(tmt.RIL.6H.Hyb2MP.cor[,c("Accession","Gene","cor")], cpm.RIL.6H.Hyb2MP.cor[,c("Gene","cor")], by.x = "Gene", by.y = "Gene", all=T)
#Set GOIs
colnames(SupT5) <- c("Gene", "Accession", "Protein_PearsonCor", "RNA_PearsonCor")
SupT5$Category <- "Other"
SupT5$Category[SupT5$Gene %in% GOIs$plastid] <- "Plastid"
SupT5$Category[SupT5$Gene %in% GOIs$CarbonFixation] <- "Carbon fixation"
SupT5$Category[SupT5$Gene %in% GOIs$TPR] <- "TPR"
SupT5$Category[SupT5$Gene %in% GOIs$PTAC] <- "PTAC"
SupT5$Category[SupT5$Gene %in% GOIs$Oxidoreductase] <- "Oxidoreductase"
SupT5$Category[SupT5$Gene %in% GOIs$Protease] <- "Protease"
SupT5$Category[SupT5$Gene %in% GOIs$AlphaLinAcidMet] <- "Alpha-linoleic acid metabolism"
SupT5$Category[SupT5$Gene %in% GOIs$BiosynSecMet] <- "Biosynthesis of secondary metabolites"
SupT5$Category[SupT5$Gene %in% GOIs$PE.ribo] <- "PE plastid ribosome"
SupT5$Category[SupT5$Gene %in% GOIs$NE.PtRibo] <- "NE plastid ribosome"
SupT5$Category[SupT5$Gene %in% GOIs$PhANGs] <- "PhANGs"
SupT5$Category[SupT5$Gene %in% GOIs$PhAPGs] <- "PhAPGs"
SupT5$Category[SupT5$Gene %in% GOIs$CytoRibo] <- "Cytosolic ribosome"
SupT5$Category[SupT5$Gene %in% GOIs$SAMS] <- "Ethylene biosynthesis - SAM synthase"
SupT5$Category[SupT5$Gene %in% GOIs$ACS] <- "Ethylene biosynthesis - ACC synthase"
SupT5$Category[SupT5$Gene %in% GOIs$ACO] <- "Ethylene biosynthesis - ACC oxidase"
SupT5$Category[SupT5$Gene %in% GOIs$acs2] <- "Ethylene biosynthesis - ACS2"
SupT5$Category[SupT5$Gene %in% GOIs$acs6] <- "Ethylene biosynthesis - ACS6"
SupT5$Category[SupT5$Gene %in% GOIs$AtACO1hom] <- "Ethylene biosynthesis - AtACO1 homolog"
SupT5$Category[SupT5$Gene %in% GOIs$AtACO2hom] <- "Ethylene biosynthesis - AtACO2 homolog"
SupT5$Category[SupT5$Gene %in% GOIs$AtSAM1AtSAM2hom] <- "Ethylene biosynthesis - AtSAM1/2 homolog"
#Re-order
SupT5 <- SupT5[(order(-SupT5$Protein_PearsonCor)),]

write.csv(SupT5[,c(1,2,5,3,4)], paste0(PathToDatasets,"SupplementalDataset5.csv"), row.names = F)
```
#Supplemental Dataset 6
```{r SD6}
SD6a <- tmt.SL[tmt.SL$Gene %in% GOIs$EtBio,c("Gene", "Accession", "Protein", "B73xMo17.MP", "ttest.MP")]
colnames(SD6a) <- c("Gene", "Accession", "Protein", "BxM.Midparent", "p-value")
SD6a$Expression <- "Protein"
SD6b <- cpm.SL[cpm.SL$Gene %in% GOIs$EtBio,c("Gene", "B73xMo17.MP", "ttest.MP")]
colnames(SD6b) <- c("Gene", "BxM.Midparent", "p-value")
SD6b$Expression <- "RNA"
SD6b$Accession <- ""
SD6b$Protein <- ""
SD6 <- rbind(SD6a, SD6b)
SD6$Category <- "Other"
SD6[(SD6$Gene %in% GOIs$SAMS),"Category"] <- "SAMS"
SD6[(SD6$Gene %in% GOIs$ACS),"Category"] <- "ACS"
SD6[(SD6$Gene %in% GOIs$ACO),"Category"] <- "ACO"
SD6[(SD6$Gene %in% GOIs$acs2),"Category"] <- "ZmACS2"
SD6[(SD6$Gene %in% GOIs$acs6),"Category"] <- "ZmACS6"
SD6[(SD6$Gene %in% GOIs$AtACO1hom),"Category"] <- "AtACO1 homolog"
SD6[(SD6$Gene %in% GOIs$AtACO2hom),"Category"] <- "AtACO2 homolog"
SD6[(SD6$Gene %in% GOIs$AtSAM1AtSAM2hom),"Category"] <- "AtSAM1/AtSAM2 homolog"
SD6 <- SD6[order(SD6$Category, SD6$Gene, SD6$Expression),]
write.csv(SD6, paste0(PathToDatasets,"SupplementalDataset6.csv"), row.names = F)
```

#Supplemental Dataset 10
```{r SD10}
SD10 <- as.data.frame(unlist(GOIs, use.names = F))
colnames(SD10) <- "Accession"
SD10$Accession <- as.character(SD10$Accession)
SD10$Group <- NA
SD10$Source <- NA
SD10[(SD10$Accession %in% GOIs$plastid),"Group"] <- "Plastid-localized"
SD10[(SD10$Accession %in% GOIs$plastid),"Source"] <- "Friso et al. (2010)"
SD10[(SD10$Accession %in% GOIs$PhANGs),"Group"] <- "PhANGs"
SD10[(SD10$Accession %in% KEGG.zma00195.digenomic$Accession),"Source"] <- "KEGG zma00195"
SD10[(SD10$Accession %in% KEGG.zma00196$Accession),"Source"] <- "KEGG zma00196"
SD10[(SD10$Accession %in% CornCycLightRxns.digenomic$Gene.Accession),"Source"] <- "CornCyc Photosynthesis Light Reactions"
SD10[(SD10$Accession %in% NE.Rubisco),"Source"] <- "MaizeGDB"
SD10[(SD10$Accession %in% GOIs$PhAPGs),"Group"] <- "PhAPGs"
SD10[(SD10$Accession %in% GOIs$PhAPGs),"Source"] <- "NCBI"
SD10[(SD10$Accession %in% GOIs$NE.PtRibo),"Group"] <- "NE plastid ribosome"
SD10[(SD10$Accession %in% GOIs$NE.PtRibo),"Source"] <- "NCBI and maize-GAMER GO"
SD10[(SD10$Accession %in% GOIs$PE.ribo),"Group"] <- "PE plastid ribosome"
SD10[(SD10$Accession %in% GOIs$PE.ribo),"Source"] <- "NCBI"
SD10[(SD10$Accession %in% GOIs$CytoRibo),"Group"] <- "Cytosolic ribosome"
SD10[(SD10$Accession %in% GOIs$CytoRibo),"Source"] <- "NCBI"
SD10[(SD10$Accession %in% GOIs$SAMS),"Group"] <- "SAM synthase"
SD10[(SD10$Accession %in% GOIs$SAMS),"Source"] <- "CornCyc 2.5.1.6"
SD10[(SD10$Accession %in% GOIs$ACS),"Group"] <- "ACC synthase"
SD10[(SD10$Accession %in% GOIs$ACS),"Source"] <- "CornCyc 4.4.1.14"
SD10[(SD10$Accession %in% GOIs$ACO),"Group"] <- "ACC oxidase"
SD10[(SD10$Accession %in% GOIs$ACO),"Source"] <- "CornCyc 1.14.17.4"
SD10[(SD10$Accession %in% GOIs$acs2),"Group"] <- "ACS2"
SD10[(SD10$Accession %in% GOIs$acs2),"Source"] <- "MaizeGDB"
SD10[(SD10$Accession %in% GOIs$acs6),"Group"] <- "ACS6"
SD10[(SD10$Accession %in% GOIs$acs6),"Source"] <- "MaizeGDB"
SD10[(SD10$Accession %in% GOIs$AtACO1hom),"Group"] <- "AtACO1 homolog"
SD10[(SD10$Accession %in% GOIs$AtACO1hom),"Source"] <- "BLAST"
SD10[(SD10$Accession %in% GOIs$AtACO2hom),"Group"] <- "AtACO2 homolog"
SD10[(SD10$Accession %in% GOIs$AtACO2hom),"Source"] <- "BLAST"
SD10[(SD10$Accession %in% GOIs$AtSAM1AtSAM2hom),"Group"] <- "AtSAM1/2 homolog"
SD10[(SD10$Accession %in% GOIs$AtSAM1AtSAM2hom),"Source"] <- "BLAST"
SD10[(SD10$Accession %in% GOIs$CarbonFixation),"Group"] <- "Carbon fixation"
SD10[(SD10$Accession %in% GOIs$CarbonFixation),"Source"] <- "KEGG zma00710"
SD10[(SD10$Accession %in% GOIs$TPR),"Group"] <- "TPR"
SD10[(SD10$Accession %in% GOIs$TPR),"Source"] <- "NCBI"
SD10[(SD10$Accession %in% GOIs$PTAC),"Group"] <- "PTAC"
SD10[(SD10$Accession %in% GOIs$PTAC),"Source"] <- "NCBI"
SD10[(SD10$Accession %in% GOIs$Oxidoreductase),"Group"] <- "Oxidoreductase"
SD10[(SD10$Accession %in% GOIs$Oxidoreductase),"Source"] <- "GAMER GO:0016491"
SD10[(SD10$Accession %in% GOIs$BiosynSecMet),"Group"] <- "Biosynthesis of secondary metabolites"
SD10[(SD10$Accession %in% GOIs$BiosynSecMet),"Source"] <- "KEGG zma01110"
SD10[(SD10$Accession %in% GOIs$Protease),"Group"] <- "Protease"
SD10[(SD10$Accession %in% GOIs$Protease),"Source"] <- "GAMER GO:0008233"
SD10[(SD10$Accession %in% GOIs$AlphaLinAcidMet),"Group"] <- "Alpha-linoleic acid metabolism"
SD10[(SD10$Accession %in% GOIs$AlphaLinAcidMet),"Source"] <- "KEGG zma00592"
SD10[(SD10$Accession %in% GOIs$ProteinBioSyn),"Group"] <- "Protein biosynthesis"
SD10[(SD10$Accession %in% GOIs$ProteinBioSyn),"Source"] <- "NCBI"

SD10 <- SD10[order(SD10$Group),]

write.csv(SD10, paste0(PathToDatasets,"SupplementalDataset10.csv"), row.names = F)
```








